{"metadata":{"kernelspec":{"name":"ir","display_name":"R","language":"R"},"language_info":{"name":"R","codemirror_mode":"r","pygments_lexer":"r","mimetype":"text/x-r-source","file_extension":".r","version":"4.4.0"},"kaggle":{"accelerator":"none","dataSources":[],"dockerImageVersionId":30749,"isInternetEnabled":true,"language":"r","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"# This R environment comes with many helpful analytics packages installed\n# It is defined by the kaggle/rstats Docker image: https://github.com/kaggle/docker-rstats\n# For example, here's a helpful package to load\n\nlibrary(tidyverse) # metapackage of all tidyverse packages\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nlist.files(path = \"../input\")\n\n# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","metadata":{"_uuid":"051d70d956493feee0c6d64651c6a088724dca2a","_execution_state":"idle","trusted":false},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\n\nBiocManager::install(\"ALL\")\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T10:48:05.969849Z","iopub.execute_input":"2025-04-13T10:48:05.971556Z","iopub.status.idle":"2025-04-13T10:48:53.260952Z","shell.execute_reply":"2025-04-13T10:48:53.258690Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'ALL'\n\nalso installing the dependency ‘Biobase’\n\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'AID', 'akc', 'ale',\n  'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'antiword', 'anytime', 'ape',\n  'aplot', 'apollo', 'aqp', 'arrow', 'ars', 'ARTool', 'arules', 'AsioHeaders',\n  'askpass', 'ASMap', 'astsa', 'automap', 'baguette', 'BAS', 'base64',\n  'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr', 'beepr',\n  'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD', 'bigstatsr',\n  'bigstep', 'billboarder', 'bindr', 'BiocManager', 'BiodiversityR', 'biomod2',\n  'biotools', 'bipartite', 'bit', 'bit64', 'bitops', 'blme', 'blorr', 'BMA',\n  'BMisc', 'bnlearn', 'bonsai', 'bookdown', 'boot', 'brms', 'broom',\n  'broom.helpers', 'broom.mixed', 'brulee', 'bslib', 'bundle', 'butcher',\n  'C50', 'car', 'caret', 'caretEnsemble', 'caTools', 'cba', 'censored',\n  'changepoint', 'cheapr', 'checkmate', 'chk', 'CholWishart', 'choroplethr',\n  'chromote', 'chron', 'circular', 'class', 'classInt', 'cld2', 'cleaner',\n  'cli', 'clock', 'clue', 'cluster', 'clusterSim', 'cmsafops',\n  'coarseDataTools', 'cobs', 'collapse', 'colorspace', 'commonmark',\n  'compareGroups', 'Compind', 'contentid', 'copula', 'CORElearn', 'coro',\n  'correlation', 'corrplot', 'countrycode', 'cpp11', 'cppdoubles',\n  'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature', 'Cubist',\n  'curl', 'cutpointr', 'cvms', 'data.table', 'datamods', 'datasauRus',\n  'datawizard', 'datetimeutils', 'dbarts', 'dbscan', 'dendextend', 'DEoptimR',\n  'Deriv', 'DescTools', 'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'digest',\n  'dismo', 'distances', 'distillery', 'distr', 'distrEx', 'distributional',\n  'DistributionUtils', 'dlm', 'dm', 'doBy', 'docopt', 'doFuture', 'doRNG',\n  'DoseFinding', 'dotCall64', 'dotwhisker', 'DoubleML', 'downloader', 'dplR',\n  'drat', 'DRDID', 'dtwclust', 'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat',\n  'ecp', 'effectsize', 'eha', 'embed', 'emmeans', 'energy', 'entropy',\n  'EnvStats', 'Epi', 'epiR', 'erer', 'ergm', 'ergm.multi', 'esquisse',\n  'estimatr', 'etm', 'evaluate', 'evd', 'Exact', 'explore', 'expm', 'extRemes',\n  'fable', 'fabletools', 'faraway', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastverse', 'fBasics', 'fda', 'fda.usc', 'fdapace', 'fdrtool',\n  'feasts', 'ff', 'fftw', 'FielDHub', 'fields', 'filehashSQLite', 'finalfit',\n  'fitdistrplus', 'FKF', 'flare', 'flexclust', 'flexmix', 'flexsurv',\n  'flextable', 'float', 'FNN', 'fontawesome', 'forecast', 'foreign',\n  'forestplot', 'fpc', 'fpp3', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa',\n  'future', 'future.apply', 'gam', 'gamboostLSS', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'genridge', 'GenSA', 'geobr', 'geodist', 'geojsonR',\n  'geometry', 'geomorph', 'geomtextpath', 'geosphere', 'gert', 'GetBCBData',\n  'ggdag', 'ggeasy', 'ggeffects', 'ggfun', 'ggh4x', 'ggiraph', 'ggmap',\n  'ggnewscale', 'ggpath', 'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc',\n  'ggrepel', 'ggstats', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmtoolbox', 'glue', 'gmapsdistance', 'gmp',\n  'GoFKernel', 'golem', 'googlePolylines', 'gower', 'GPArotation', 'GPfit',\n  'gplots', 'grafify', 'gRain', 'graphlayouts', 'gRbase', 'greybox', 'grf',\n  'gridpattern', 'grImport2', 'groupdata2', 'grpreg', 'gsheet', 'gss', 'gstat',\n  'gsw', 'gt', 'gtable', 'gtsummary', 'GWalkR', 'GWmodel', 'h3lib', 'h3r',\n  'hardhat', 'harmony', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'htmlTable', 'hts', 'httr2', 'hunspell', 'huxtable',\n  'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph', 'ijtiff', 'imager', 'iml',\n  'inline', 'insight', 'inspectdf', 'interactions', 'intervals', 'ipred',\n  'isotree', 'ISwR', 'ivreg', 'janitor', 'jiebaR', 'JMbayes2', 'jmvconnect',\n  'jmvcore', 'jmvReadWrite', 'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity',\n  'keras3', 'kernelshap', 'kernlab', 'KernSmooth', 'khroma', 'kit', 'knitr',\n  'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman', 'lares', 'later',\n  'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave', 'lcmm', 'leafgl',\n  'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR', 'lightgbm', 'limma', 'lintr',\n  'littler', 'lme4', 'lmodel2', 'lmom', 'locfit', 'logger', 'longitudinalData',\n  'loo', 'lordif', 'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa',\n  'lubridate', 'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'matlib', 'Matrix', 'MatrixModels',\n  'matrixStats', 'MazamaCoreUtils', 'MazamaSpatialUtils', 'MBA', 'mboost',\n  'MCMCpack', 'mda', 'mdsr', 'memisc', 'meta', 'metacoder', 'metadat',\n  'metafor', 'meteospain', 'MetricsWeighted', 'mets', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'minqa', 'minty', 'mirt',\n  'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench', 'mlflow',\n  'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data', 'mlr3filters',\n  'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures', 'mlr3misc',\n  'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse', 'mlr3viz',\n  'modelbased', 'modelenv', 'modelsummary', 'modeltime', 'moderndive',\n  'mongolite', 'monitoR', 'msigdbr', 'msm', 'mstate', 'mtsdi', 'multcomp',\n  'MuMIn', 'mvtnorm', 'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson',\n  'ndtv', 'network', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'OpenMx', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ore', 'osmextract',\n  'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown', 'pammtools',\n  'pander', 'parabar', 'paradox', 'parallelly', 'parameters', 'ParamHelpers',\n  'paran', 'parsnip', 'party', 'partykit', 'patchwork', 'pbdZMQ', 'pcaPP',\n  'pdftools', 'pdp', 'pdR', 'PearsonDS', 'performance', 'PerformanceAnalytics',\n  'pgmm', 'phangorn', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools', 'pls',\n  'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip', 'popdemo',\n  'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'PreProcess', 'prettyGraphs', 'priceR', 'processx', 'profvis',\n  'progressr', 'PROJ', 'proj4', 'promises', 'prospectr', 'protolite', 'PRROC',\n  'ps', 'pspline', 'PSweight', 'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf',\n  'qs', 'qtl', 'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr',\n  'QuickJSR', 'qvcalc', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'reactR',\n  'readODS', 'readxl', 'rearrr', 'rebmix', 'recipes', 'refund', 'renv',\n  'report', 'reprex', 'reticulate', 'Rfast', 'rfishbase', 'rgbif', 'rgenoud',\n  'rgl', 'rio', 'rjson', 'RJSONIO', 'rlang', 'RMariaDB', 'rmarkdown', 'rminer',\n  'rms', 'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpostgis', 'RPostgres', 'RPostgreSQL',\n  'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample', 'rsconnect',\n  'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite', 'Rssa',\n  'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rugarch', 'Rvcg', 'rvg',\n  's2', 'sandwich', 'sass', 'SCtools', 'sda', 'sdm', 'seasonal',\n  'seasonalview', 'see', 'seededlda', 'segmented', 'sem', 'semTools',\n  'sensemakr', 'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinyWidgets', 'SimDesign',\n  'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot', 'skmeans', 'slam',\n  'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2', 'sna', 'sodium',\n  'soilDB', 'sommer', 'sp', 'spacefillr', 'spacetime', 'spam', 'sparklyr',\n  'SparseM', 'spatialreg', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'strucchange', 'summarytools',\n  'SuppDists', 'survival', 'survminer', 'svd', 'sys', 'systemfonts', 'tables',\n  'tau', 'taxize', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract',\n  'testthat', 'texreg', 'text', 'textcat', 'textrecipes', 'textshaping',\n  'textutils', 'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'timeDate', 'timeplyr', 'timereg', 'timeSeries', 'tinytable',\n  'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui', 'topicmodels',\n  'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal', 'tsbox', 'tsDyn',\n  'tseries', 'tsibble', 'tuber', 'tune', 'tzdb', 'units', 'univariateML',\n  'unpivotr', 'usethis', 'usmapdata', 'uuid', 'uwot', 'V8', 'vcd', 'vegan',\n  'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR', 'wbacon',\n  'WDI', 'websocket', 'webutils', 'WeightSVM', 'withr', 'wk', 'wooldridge',\n  'workflows', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick',\n  'yulab.utils', 'zip', 'zoo', 'spatial'\n\n","output_type":"stream"}],"execution_count":3},{"cell_type":"code","source":"library(ALL)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:22:22.206214Z","iopub.execute_input":"2025-04-13T13:22:22.207967Z","iopub.status.idle":"2025-04-13T13:22:22.220526Z","shell.execute_reply":"2025-04-13T13:22:22.218817Z"}},"outputs":[],"execution_count":91},{"cell_type":"code","source":"data(ALL)\nALL","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:22:23.310881Z","iopub.execute_input":"2025-04-13T13:22:23.312506Z","iopub.status.idle":"2025-04-13T13:22:24.491605Z","shell.execute_reply":"2025-04-13T13:22:24.489641Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"ExpressionSet (storageMode: lockedEnvironment)\nassayData: 12625 features, 128 samples \n  element names: exprs \nprotocolData: none\nphenoData\n  sampleNames: 01005 01010 ... LAL4 (128 total)\n  varLabels: cod diagnosis ... date last seen (21 total)\n  varMetadata: labelDescription\nfeatureData: none\nexperimentData: use 'experimentData(object)'\n  pubMedIds: 14684422 16243790 \nAnnotation: hgu95av2 "},"metadata":{}}],"execution_count":92},{"cell_type":"code","source":"dim(exprs(ALL))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:22:30.301700Z","iopub.execute_input":"2025-04-13T13:22:30.303750Z","iopub.status.idle":"2025-04-13T13:22:30.321557Z","shell.execute_reply":"2025-04-13T13:22:30.319739Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>12625</li><li>128</li></ol>\n","text/markdown":"1. 12625\n2. 128\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 12625\n\\item 128\n\\end{enumerate*}\n","text/plain":"[1] 12625   128"},"metadata":{}}],"execution_count":93},{"cell_type":"code","source":"expr_matrix = exprs(ALL)\nexpr_matrix","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:24:44.218350Z","iopub.execute_input":"2025-04-13T13:24:44.220070Z","iopub.status.idle":"2025-04-13T13:24:44.536883Z","shell.execute_reply":"2025-04-13T13:24:44.533559Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A matrix: 12625 × 128 of type dbl</caption>\n<thead>\n\t<tr><th></th><th scope=col>01005</th><th scope=col>01010</th><th scope=col>03002</th><th scope=col>04006</th><th scope=col>04007</th><th scope=col>04008</th><th scope=col>04010</th><th scope=col>04016</th><th scope=col>06002</th><th scope=col>08001</th><th scope=col>⋯</th><th scope=col>37001</th><th scope=col>43006</th><th scope=col>43015</th><th scope=col>44001</th><th scope=col>49004</th><th scope=col>56007</th><th scope=col>64005</th><th scope=col>65003</th><th scope=col>83001</th><th scope=col>LAL4</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1000_at</th><td>7.597323</td><td> 7.479445</td><td>7.567593</td><td>7.384684</td><td>7.905312</td><td> 7.065914</td><td> 7.474537</td><td> 7.536119</td><td>7.183331</td><td> 7.735545</td><td>⋯</td><td> 7.849455</td><td>7.960842</td><td>8.188617</td><td>7.399999</td><td>7.813474</td><td>7.816922</td><td> 7.913249</td><td>7.800199</td><td>8.030047</td><td>7.702217</td></tr>\n\t<tr><th scope=row>1001_at</th><td>5.046194</td><td> 4.932537</td><td>4.799294</td><td>4.922627</td><td>4.844565</td><td> 5.147762</td><td> 5.122518</td><td> 5.016132</td><td>5.288943</td><td> 4.633217</td><td>⋯</td><td> 4.959450</td><td>4.537677</td><td>5.154500</td><td>5.071885</td><td>4.874525</td><td>4.788699</td><td> 5.403640</td><td>5.443827</td><td>5.178633</td><td>5.029670</td></tr>\n\t<tr><th scope=row>1002_f_at</th><td>3.900466</td><td> 4.208155</td><td>3.886169</td><td>4.206798</td><td>3.416923</td><td> 3.945869</td><td> 4.150506</td><td> 3.576360</td><td>3.900935</td><td> 3.630190</td><td>⋯</td><td> 4.047428</td><td>3.694877</td><td>3.949546</td><td>3.837517</td><td>3.767672</td><td>4.248075</td><td> 4.119252</td><td>3.957468</td><td>4.053458</td><td>4.002306</td></tr>\n\t<tr><th scope=row>1003_s_at</th><td>5.903856</td><td> 6.169024</td><td>5.860459</td><td>6.116890</td><td>5.687997</td><td> 6.208061</td><td> 6.292713</td><td> 5.665991</td><td>5.842326</td><td> 5.875375</td><td>⋯</td><td> 6.193790</td><td>5.973162</td><td>6.033704</td><td>5.560886</td><td>5.930329</td><td>5.878533</td><td> 6.193376</td><td>6.006443</td><td>6.050129</td><td>6.297549</td></tr>\n\t<tr><th scope=row>1004_at</th><td>5.925260</td><td> 5.912780</td><td>5.893209</td><td>6.170245</td><td>5.615210</td><td> 5.923487</td><td> 6.046607</td><td> 5.738218</td><td>5.994515</td><td> 5.748350</td><td>⋯</td><td> 5.824930</td><td>5.841095</td><td>5.870163</td><td>5.686555</td><td>5.793403</td><td>5.913503</td><td> 6.053053</td><td>5.549087</td><td>5.932412</td><td>5.827131</td></tr>\n\t<tr><th scope=row>1005_at</th><td>8.570990</td><td>10.428299</td><td>9.616713</td><td>9.937155</td><td>9.983809</td><td>10.063484</td><td>10.662059</td><td>11.269115</td><td>8.812869</td><td>10.165159</td><td>⋯</td><td>10.408645</td><td>7.524865</td><td>8.381740</td><td>8.055760</td><td>7.234242</td><td>8.083834</td><td>10.163355</td><td>8.184442</td><td>8.798521</td><td>7.687370</td></tr>\n\t<tr><th scope=row>1006_at</th><td>3.656143</td><td> 3.853979</td><td>3.646808</td><td>3.874289</td><td>3.547361</td><td> 3.771648</td><td> 3.747963</td><td> 3.318315</td><td>3.874648</td><td> 3.525401</td><td>⋯</td><td> 3.670568</td><td>3.652459</td><td>3.707005</td><td>3.852594</td><td>3.527330</td><td>3.429451</td><td> 3.756402</td><td>3.571963</td><td>3.585416</td><td>3.674371</td></tr>\n\t<tr><th scope=row>1007_s_at</th><td>7.623562</td><td> 7.543604</td><td>7.916954</td><td>6.816397</td><td>7.516981</td><td> 7.726716</td><td> 7.288960</td><td> 7.724153</td><td>6.763490</td><td> 7.543907</td><td>⋯</td><td> 6.609299</td><td>6.482215</td><td>6.651151</td><td>6.463802</td><td>7.083294</td><td>7.107712</td><td> 6.895677</td><td>6.926471</td><td>6.861161</td><td>6.731454</td></tr>\n\t<tr><th scope=row>1008_f_at</th><td>8.903547</td><td> 9.903953</td><td>8.494499</td><td>9.533983</td><td>8.871669</td><td> 9.424092</td><td> 8.999938</td><td> 8.006848</td><td>9.190553</td><td> 8.408789</td><td>⋯</td><td> 8.561828</td><td>8.765609</td><td>8.459983</td><td>8.611438</td><td>8.911405</td><td>9.043327</td><td> 8.284549</td><td>8.569874</td><td>8.474002</td><td>8.539480</td></tr>\n\t<tr><th scope=row>1009_at</th><td>9.371888</td><td> 9.322177</td><td>9.304982</td><td>9.135370</td><td>9.627175</td><td> 9.189420</td><td> 9.312164</td><td> 9.946430</td><td>9.112255</td><td> 9.577024</td><td>⋯</td><td> 9.492835</td><td>9.237562</td><td>9.183180</td><td>9.250913</td><td>9.609640</td><td>9.474761</td><td> 9.702681</td><td>9.110240</td><td>9.884863</td><td>9.575421</td></tr>\n\t<tr><th scope=row>100_g_at</th><td>7.985255</td><td> 8.015146</td><td>7.730108</td><td>7.827044</td><td>7.834880</td><td> 7.984250</td><td> 8.120029</td><td> 8.084865</td><td>7.697962</td><td> 8.302688</td><td>⋯</td><td> 8.003010</td><td>7.836479</td><td>8.007742</td><td>7.825349</td><td>7.867227</td><td>8.009918</td><td> 8.271588</td><td>8.169891</td><td>7.897170</td><td>7.965106</td></tr>\n\t<tr><th scope=row>1010_at</th><td>5.399275</td><td> 5.620731</td><td>5.172665</td><td>5.967741</td><td>5.185482</td><td> 5.484939</td><td> 5.840837</td><td> 5.067395</td><td>5.237119</td><td> 5.273373</td><td>⋯</td><td> 5.058770</td><td>4.848082</td><td>5.330210</td><td>4.881275</td><td>5.180372</td><td>5.310220</td><td> 5.237757</td><td>4.993497</td><td>5.385707</td><td>4.982707</td></tr>\n\t<tr><th scope=row>1011_s_at</th><td>5.698984</td><td> 6.172831</td><td>6.298505</td><td>6.680342</td><td>6.658723</td><td> 6.372048</td><td> 7.252254</td><td> 6.674473</td><td>6.883788</td><td> 5.714778</td><td>⋯</td><td> 6.434650</td><td>5.573140</td><td>5.360469</td><td>6.067903</td><td>5.246391</td><td>4.942719</td><td> 5.109609</td><td>6.490558</td><td>6.023058</td><td>5.809493</td></tr>\n\t<tr><th scope=row>1012_at</th><td>3.432633</td><td> 3.921244</td><td>3.649822</td><td>3.784770</td><td>3.833315</td><td> 3.948326</td><td> 3.676359</td><td> 3.499718</td><td>3.439177</td><td> 3.883105</td><td>⋯</td><td> 3.563572</td><td>3.877966</td><td>3.393784</td><td>3.173837</td><td>3.760740</td><td>4.148892</td><td> 4.065844</td><td>3.762194</td><td>3.735753</td><td>3.664340</td></tr>\n\t<tr><th scope=row>1013_at</th><td>3.442516</td><td> 3.002145</td><td>3.314890</td><td>3.474333</td><td>3.851746</td><td> 3.851362</td><td> 3.565626</td><td> 2.851111</td><td>3.070907</td><td> 2.969788</td><td>⋯</td><td> 2.799112</td><td>3.335830</td><td>2.823894</td><td>3.162021</td><td>2.899012</td><td>3.008075</td><td> 3.153419</td><td>2.902942</td><td>3.018644</td><td>3.127662</td></tr>\n\t<tr><th scope=row>1014_at</th><td>7.318896</td><td> 7.653855</td><td>7.678002</td><td>6.971806</td><td>7.446160</td><td> 7.475994</td><td> 7.527383</td><td> 7.550745</td><td>7.119259</td><td> 7.594602</td><td>⋯</td><td> 7.247758</td><td>7.442379</td><td>7.125079</td><td>7.846200</td><td>7.571595</td><td>7.931793</td><td> 7.654051</td><td>7.414572</td><td>7.505681</td><td>7.532083</td></tr>\n\t<tr><th scope=row>1015_s_at</th><td>5.479742</td><td> 6.110416</td><td>5.679977</td><td>5.962049</td><td>5.154362</td><td> 5.696765</td><td> 5.917831</td><td> 5.535112</td><td>5.723400</td><td> 5.569632</td><td>⋯</td><td> 5.733726</td><td>5.487096</td><td>5.628726</td><td>5.473225</td><td>5.499078</td><td>5.398977</td><td> 5.688522</td><td>5.753532</td><td>5.555595</td><td>5.721859</td></tr>\n\t<tr><th scope=row>1016_s_at</th><td>3.384669</td><td> 3.361198</td><td>3.174670</td><td>3.020869</td><td>3.177107</td><td> 3.207662</td><td> 3.229092</td><td> 3.210501</td><td>3.353474</td><td> 3.176632</td><td>⋯</td><td> 3.498991</td><td>3.250506</td><td>3.176352</td><td>3.924138</td><td>3.245476</td><td>3.476038</td><td> 3.285540</td><td>3.392091</td><td>3.274649</td><td>3.653479</td></tr>\n\t<tr><th scope=row>1017_at</th><td>4.921528</td><td> 4.115588</td><td>4.743844</td><td>4.691007</td><td>5.406613</td><td> 4.431900</td><td> 3.931028</td><td> 5.129296</td><td>3.956742</td><td> 5.178051</td><td>⋯</td><td> 4.483604</td><td>5.133695</td><td>4.789429</td><td>4.478413</td><td>5.773424</td><td>5.111104</td><td> 4.185028</td><td>4.268057</td><td>4.030064</td><td>3.826977</td></tr>\n\t<tr><th scope=row>1018_at</th><td>6.281061</td><td> 6.147906</td><td>5.941267</td><td>6.260564</td><td>5.782521</td><td> 6.032982</td><td> 6.095990</td><td> 5.903504</td><td>6.231557</td><td> 5.861168</td><td>⋯</td><td> 6.519862</td><td>6.104468</td><td>6.013004</td><td>6.493142</td><td>6.333817</td><td>6.164413</td><td> 6.674320</td><td>6.566802</td><td>6.095361</td><td>6.425974</td></tr>\n\t<tr><th scope=row>1019_g_at</th><td>6.546212</td><td> 6.754309</td><td>6.704541</td><td>6.898213</td><td>6.265019</td><td> 6.370988</td><td> 6.262760</td><td> 6.373234</td><td>6.648472</td><td> 6.306895</td><td>⋯</td><td> 6.673348</td><td>6.658479</td><td>6.402795</td><td>6.766007</td><td>6.851714</td><td>6.696270</td><td> 7.033197</td><td>7.395795</td><td>6.356240</td><td>6.963269</td></tr>\n\t<tr><th scope=row>101_at</th><td>6.123411</td><td> 6.163813</td><td>6.016756</td><td>5.930699</td><td>5.848487</td><td> 5.820837</td><td> 6.319743</td><td> 5.794677</td><td>6.079454</td><td> 6.109053</td><td>⋯</td><td> 6.953927</td><td>6.442707</td><td>6.347830</td><td>6.134284</td><td>6.043279</td><td>6.572743</td><td> 6.666000</td><td>6.729782</td><td>6.750178</td><td>6.789686</td></tr>\n\t<tr><th scope=row>1020_s_at</th><td>7.440408</td><td> 8.225343</td><td>6.916695</td><td>7.921413</td><td>7.489289</td><td> 7.520384</td><td> 7.608336</td><td> 7.262121</td><td>8.059713</td><td> 7.466821</td><td>⋯</td><td> 8.241378</td><td>7.422417</td><td>7.080051</td><td>8.689273</td><td>7.759317</td><td>7.530735</td><td> 8.074391</td><td>8.082602</td><td>8.028259</td><td>7.878384</td></tr>\n\t<tr><th scope=row>1021_at</th><td>4.635150</td><td> 5.182118</td><td>5.485365</td><td>4.848357</td><td>4.547636</td><td> 4.433935</td><td> 4.690830</td><td> 4.394778</td><td>4.567002</td><td> 6.617756</td><td>⋯</td><td> 5.006477</td><td>4.117340</td><td>4.883207</td><td>4.955042</td><td>4.716777</td><td>4.750484</td><td> 4.891067</td><td>4.548111</td><td>4.481377</td><td>4.397078</td></tr>\n\t<tr><th scope=row>1022_f_at</th><td>3.818130</td><td> 3.697340</td><td>3.756676</td><td>3.688513</td><td>3.525746</td><td> 3.647198</td><td> 3.372396</td><td> 3.561139</td><td>3.991485</td><td> 3.716743</td><td>⋯</td><td> 4.163488</td><td>3.741596</td><td>3.567559</td><td>4.022900</td><td>3.789003</td><td>3.831971</td><td> 4.111677</td><td>3.939289</td><td>3.707847</td><td>4.329154</td></tr>\n\t<tr><th scope=row>1023_at</th><td>5.724247</td><td> 6.102434</td><td>5.535088</td><td>6.308213</td><td>5.564462</td><td> 5.773668</td><td> 5.873469</td><td> 5.490342</td><td>5.908396</td><td> 5.505490</td><td>⋯</td><td> 5.829125</td><td>5.414204</td><td>5.776841</td><td>5.269463</td><td>5.700922</td><td>5.956093</td><td> 5.774114</td><td>5.858850</td><td>5.637539</td><td>5.408432</td></tr>\n\t<tr><th scope=row>1024_at</th><td>5.754872</td><td> 5.887494</td><td>5.605735</td><td>6.336704</td><td>5.773976</td><td> 5.846395</td><td> 6.022922</td><td> 5.760028</td><td>5.644638</td><td> 5.556812</td><td>⋯</td><td> 5.741901</td><td>5.466854</td><td>5.805061</td><td>5.698824</td><td>5.723203</td><td>5.847403</td><td> 5.870010</td><td>5.765627</td><td>5.834069</td><td>5.874103</td></tr>\n\t<tr><th scope=row>1025_g_at</th><td>4.554326</td><td> 4.599976</td><td>4.366561</td><td>4.729267</td><td>4.177634</td><td> 4.400837</td><td> 4.441394</td><td> 4.134673</td><td>4.423239</td><td> 4.329983</td><td>⋯</td><td> 4.797889</td><td>4.449505</td><td>4.716811</td><td>4.479158</td><td>4.614610</td><td>4.620035</td><td> 4.757783</td><td>4.474032</td><td>4.657209</td><td>4.869916</td></tr>\n\t<tr><th scope=row>1026_s_at</th><td>6.384481</td><td> 6.541548</td><td>6.169922</td><td>6.752466</td><td>6.162612</td><td> 6.918602</td><td> 6.920326</td><td> 6.554345</td><td>6.482749</td><td> 6.386876</td><td>⋯</td><td> 6.504184</td><td>6.252522</td><td>6.457230</td><td>6.678333</td><td>6.413700</td><td>6.595883</td><td> 6.723828</td><td>6.753657</td><td>6.671640</td><td>6.627980</td></tr>\n\t<tr><th scope=row>1027_at</th><td>5.468026</td><td> 5.546560</td><td>5.460557</td><td>5.562977</td><td>5.293422</td><td> 5.269626</td><td> 5.492094</td><td> 4.989486</td><td>5.590867</td><td> 5.481647</td><td>⋯</td><td> 5.288263</td><td>5.305203</td><td>5.329512</td><td>5.301670</td><td>5.369972</td><td>5.494767</td><td> 5.680495</td><td>5.259128</td><td>5.640146</td><td>5.544918</td></tr>\n\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n\t<tr><th scope=row>AFFX-HUMISGF3A/M97935_MB_at</th><td>5.082692</td><td> 7.611857</td><td>4.974700</td><td>5.165775</td><td>5.806239</td><td>6.720742</td><td>5.732256</td><td>5.522468</td><td>5.372603</td><td>5.180712</td><td>⋯</td><td>5.130053</td><td>4.951393</td><td>4.903532</td><td>6.024029</td><td>5.088734</td><td>5.227949</td><td>4.962942</td><td>5.180621</td><td>4.803131</td><td>6.137956</td></tr>\n\t<tr><th scope=row>AFFX-HUMRGE/M10098_3_at</th><td>5.743438</td><td> 5.423036</td><td>5.866900</td><td>7.767792</td><td>5.268211</td><td>5.463824</td><td>6.631424</td><td>4.940925</td><td>5.170254</td><td>5.346651</td><td>⋯</td><td>5.322311</td><td>5.001757</td><td>5.678911</td><td>4.983021</td><td>5.542517</td><td>5.852424</td><td>5.379777</td><td>5.232322</td><td>5.091796</td><td>5.179363</td></tr>\n\t<tr><th scope=row>AFFX-HUMRGE/M10098_5_at</th><td>4.922025</td><td> 4.468572</td><td>4.698738</td><td>7.542934</td><td>4.109162</td><td>4.768099</td><td>5.878260</td><td>4.273090</td><td>4.480622</td><td>4.463458</td><td>⋯</td><td>4.578618</td><td>4.281334</td><td>4.793171</td><td>4.632098</td><td>4.498728</td><td>4.831378</td><td>4.546390</td><td>4.436946</td><td>4.290367</td><td>4.529884</td></tr>\n\t<tr><th scope=row>AFFX-HUMRGE/M10098_M_at</th><td>4.725279</td><td> 4.160386</td><td>4.598819</td><td>6.734705</td><td>4.024907</td><td>3.889413</td><td>5.009262</td><td>4.300358</td><td>4.096791</td><td>4.356831</td><td>⋯</td><td>3.903118</td><td>3.785432</td><td>4.225459</td><td>4.089147</td><td>4.153007</td><td>4.442808</td><td>3.981419</td><td>3.888398</td><td>3.656073</td><td>3.876826</td></tr>\n\t<tr><th scope=row>AFFX-HUMTFRR/M11507_3_at</th><td>4.875464</td><td> 4.810450</td><td>4.853883</td><td>5.213899</td><td>5.384624</td><td>5.446088</td><td>5.385702</td><td>4.829268</td><td>4.887564</td><td>4.991675</td><td>⋯</td><td>5.873119</td><td>5.411731</td><td>5.477222</td><td>4.733239</td><td>5.522148</td><td>5.399388</td><td>5.445041</td><td>4.801405</td><td>5.720164</td><td>5.498475</td></tr>\n\t<tr><th scope=row>AFFX-HUMTFRR/M11507_5_at</th><td>3.768308</td><td> 4.178835</td><td>3.755861</td><td>4.776499</td><td>3.911115</td><td>4.279529</td><td>4.921189</td><td>3.904052</td><td>3.927687</td><td>3.587829</td><td>⋯</td><td>4.334639</td><td>3.741585</td><td>3.954517</td><td>3.947313</td><td>3.975845</td><td>4.093276</td><td>4.004460</td><td>3.772255</td><td>3.974060</td><td>3.898813</td></tr>\n\t<tr><th scope=row>AFFX-HUMTFRR/M11507_M_at</th><td>4.163886</td><td> 4.641657</td><td>4.040983</td><td>4.885080</td><td>4.350342</td><td>4.929152</td><td>5.109563</td><td>4.284558</td><td>4.130364</td><td>4.384626</td><td>⋯</td><td>4.673340</td><td>4.308101</td><td>4.520912</td><td>4.291467</td><td>4.670833</td><td>4.634246</td><td>4.507426</td><td>4.387551</td><td>4.546331</td><td>4.331859</td></tr>\n\t<tr><th scope=row>AFFX-LysX-3_at</th><td>3.311952</td><td> 2.850462</td><td>3.084060</td><td>3.126579</td><td>2.871642</td><td>2.817815</td><td>2.833962</td><td>2.778187</td><td>3.074082</td><td>3.102063</td><td>⋯</td><td>3.190449</td><td>2.961977</td><td>4.000543</td><td>3.625823</td><td>2.987288</td><td>3.396457</td><td>2.801641</td><td>2.890539</td><td>2.908193</td><td>3.182116</td></tr>\n\t<tr><th scope=row>AFFX-LysX-5_at</th><td>2.828138</td><td> 2.834018</td><td>2.892196</td><td>2.789425</td><td>2.830019</td><td>2.782114</td><td>2.889998</td><td>2.701625</td><td>3.077016</td><td>2.917622</td><td>⋯</td><td>2.919766</td><td>2.870377</td><td>4.197757</td><td>3.199422</td><td>2.789884</td><td>3.258625</td><td>2.811507</td><td>3.042925</td><td>2.930631</td><td>3.087345</td></tr>\n\t<tr><th scope=row>AFFX-LysX-M_at</th><td>4.721098</td><td> 4.625145</td><td>4.575322</td><td>4.902402</td><td>4.749065</td><td>4.562960</td><td>4.803757</td><td>4.608499</td><td>4.800184</td><td>4.294252</td><td>⋯</td><td>4.721945</td><td>4.430331</td><td>5.599868</td><td>5.334023</td><td>4.450242</td><td>4.750099</td><td>4.506028</td><td>4.637085</td><td>4.539593</td><td>4.936381</td></tr>\n\t<tr><th scope=row>AFFX-M27830_3_at</th><td>7.463929</td><td> 7.438116</td><td>7.247500</td><td>7.655706</td><td>7.049913</td><td>7.277769</td><td>7.500706</td><td>7.350631</td><td>7.515679</td><td>6.974698</td><td>⋯</td><td>7.535924</td><td>6.951264</td><td>7.118890</td><td>7.893386</td><td>7.567082</td><td>7.190904</td><td>7.450166</td><td>7.587643</td><td>6.957847</td><td>7.522223</td></tr>\n\t<tr><th scope=row>AFFX-M27830_5_at</th><td>7.047142</td><td> 8.772183</td><td>7.012569</td><td>7.912946</td><td>5.026355</td><td>5.683872</td><td>7.351909</td><td>5.496411</td><td>4.986038</td><td>5.621757</td><td>⋯</td><td>5.134773</td><td>4.677474</td><td>6.280724</td><td>5.883414</td><td>5.861596</td><td>5.551034</td><td>5.602937</td><td>4.815696</td><td>5.128495</td><td>5.615701</td></tr>\n\t<tr><th scope=row>AFFX-M27830_M_at</th><td>8.116965</td><td>10.008773</td><td>8.103920</td><td>8.584401</td><td>7.309622</td><td>7.933779</td><td>8.444469</td><td>7.517100</td><td>7.451483</td><td>7.695028</td><td>⋯</td><td>8.124331</td><td>7.435762</td><td>8.133971</td><td>7.856379</td><td>7.873945</td><td>8.304769</td><td>8.277190</td><td>7.933101</td><td>7.774132</td><td>7.876990</td></tr>\n\t<tr><th scope=row>AFFX-MurFAS_at</th><td>3.536136</td><td> 3.213659</td><td>3.331824</td><td>3.431021</td><td>3.461410</td><td>3.483763</td><td>3.323103</td><td>3.446710</td><td>3.440076</td><td>3.273991</td><td>⋯</td><td>3.537291</td><td>3.254407</td><td>3.875097</td><td>3.444693</td><td>3.743439</td><td>3.709902</td><td>3.233520</td><td>3.352985</td><td>3.165264</td><td>3.268459</td></tr>\n\t<tr><th scope=row>AFFX-MurIL10_at</th><td>4.341857</td><td> 4.639913</td><td>4.114760</td><td>4.775549</td><td>3.991641</td><td>4.627252</td><td>4.530767</td><td>4.127038</td><td>4.370285</td><td>3.835914</td><td>⋯</td><td>4.321451</td><td>3.974078</td><td>4.718328</td><td>4.095881</td><td>4.001305</td><td>4.763165</td><td>4.161259</td><td>3.959749</td><td>3.975969</td><td>4.193781</td></tr>\n\t<tr><th scope=row>AFFX-MurIL2_at</th><td>5.402049</td><td> 4.970754</td><td>5.136504</td><td>5.328741</td><td>4.934468</td><td>5.036166</td><td>4.914641</td><td>4.886392</td><td>5.372676</td><td>5.018007</td><td>⋯</td><td>5.358090</td><td>4.822789</td><td>5.320897</td><td>5.489717</td><td>5.181606</td><td>5.365144</td><td>5.194208</td><td>5.160640</td><td>4.844565</td><td>4.967381</td></tr>\n\t<tr><th scope=row>AFFX-MurIL4_at</th><td>3.088365</td><td> 2.912783</td><td>2.941528</td><td>3.123581</td><td>3.259221</td><td>3.037260</td><td>3.003451</td><td>2.973277</td><td>3.320805</td><td>2.784965</td><td>⋯</td><td>3.338618</td><td>2.875561</td><td>3.795054</td><td>3.426576</td><td>3.092319</td><td>3.577267</td><td>3.058879</td><td>3.168797</td><td>3.028462</td><td>3.014841</td></tr>\n\t<tr><th scope=row>AFFX-PheX-3_at</th><td>5.073080</td><td> 5.539796</td><td>5.222797</td><td>5.668133</td><td>5.199665</td><td>5.470518</td><td>5.506018</td><td>4.931751</td><td>5.180675</td><td>5.256852</td><td>⋯</td><td>5.056564</td><td>4.865700</td><td>5.419945</td><td>5.860424</td><td>4.921566</td><td>5.486537</td><td>5.354748</td><td>5.261869</td><td>5.249353</td><td>5.188049</td></tr>\n\t<tr><th scope=row>AFFX-PheX-5_at</th><td>3.050993</td><td> 3.116976</td><td>3.301066</td><td>3.366936</td><td>3.008996</td><td>3.125731</td><td>2.902621</td><td>3.074890</td><td>3.364359</td><td>3.217555</td><td>⋯</td><td>3.293791</td><td>3.222073</td><td>4.072204</td><td>3.646259</td><td>3.018502</td><td>3.860386</td><td>3.073433</td><td>3.408542</td><td>3.086195</td><td>3.374656</td></tr>\n\t<tr><th scope=row>AFFX-PheX-M_at</th><td>3.344718</td><td> 3.249449</td><td>3.179357</td><td>3.343461</td><td>2.962293</td><td>3.045536</td><td>3.029851</td><td>3.068849</td><td>3.355382</td><td>3.301770</td><td>⋯</td><td>3.215089</td><td>3.157067</td><td>3.836710</td><td>3.774872</td><td>2.950347</td><td>3.884731</td><td>3.195110</td><td>3.741395</td><td>3.254390</td><td>3.418817</td></tr>\n\t<tr><th scope=row>AFFX-ThrX-3_at</th><td>4.365777</td><td> 4.559372</td><td>4.309676</td><td>4.808575</td><td>4.120906</td><td>4.423895</td><td>4.166162</td><td>4.301804</td><td>4.439963</td><td>4.102016</td><td>⋯</td><td>4.302587</td><td>4.304613</td><td>4.927189</td><td>4.667565</td><td>4.370202</td><td>5.101768</td><td>4.440123</td><td>4.675419</td><td>4.094304</td><td>4.474966</td></tr>\n\t<tr><th scope=row>AFFX-ThrX-5_at</th><td>4.127825</td><td> 4.452035</td><td>4.157649</td><td>4.205917</td><td>3.899734</td><td>4.104524</td><td>3.994176</td><td>4.076177</td><td>4.530618</td><td>4.142404</td><td>⋯</td><td>4.553950</td><td>4.168382</td><td>4.797841</td><td>4.485632</td><td>4.261370</td><td>4.831275</td><td>4.366302</td><td>4.738456</td><td>4.194136</td><td>4.555346</td></tr>\n\t<tr><th scope=row>AFFX-ThrX-M_at</th><td>3.469720</td><td> 3.243264</td><td>3.228969</td><td>3.399563</td><td>3.027191</td><td>3.295530</td><td>3.257909</td><td>2.994594</td><td>3.525741</td><td>3.232638</td><td>⋯</td><td>3.458331</td><td>3.348805</td><td>4.113437</td><td>3.747689</td><td>3.395225</td><td>4.110112</td><td>3.436514</td><td>3.479872</td><td>3.160448</td><td>3.500401</td></tr>\n\t<tr><th scope=row>AFFX-TrpnX-3_at</th><td>2.977916</td><td> 2.650942</td><td>2.751669</td><td>2.960073</td><td>2.806713</td><td>2.962976</td><td>3.152471</td><td>2.695692</td><td>2.920628</td><td>2.844983</td><td>⋯</td><td>2.937670</td><td>2.850850</td><td>3.582664</td><td>3.059061</td><td>2.876187</td><td>3.408258</td><td>2.809911</td><td>2.771741</td><td>2.775778</td><td>2.780948</td></tr>\n\t<tr><th scope=row>AFFX-TrpnX-5_at</th><td>3.623308</td><td> 3.621928</td><td>3.497549</td><td>3.834567</td><td>3.485573</td><td>3.832459</td><td>3.657211</td><td>3.685598</td><td>3.863668</td><td>3.474924</td><td>⋯</td><td>3.830677</td><td>3.416041</td><td>4.315469</td><td>3.965705</td><td>3.807634</td><td>4.254020</td><td>3.711537</td><td>3.701861</td><td>3.753496</td><td>3.701939</td></tr>\n\t<tr><th scope=row>AFFX-TrpnX-M_at</th><td>3.416213</td><td> 3.344428</td><td>3.250417</td><td>3.589504</td><td>3.140334</td><td>3.397393</td><td>3.272373</td><td>3.290285</td><td>3.638814</td><td>3.343970</td><td>⋯</td><td>3.117524</td><td>3.253874</td><td>3.971773</td><td>3.438820</td><td>3.380928</td><td>3.753142</td><td>3.462627</td><td>3.551272</td><td>3.114569</td><td>3.253312</td></tr>\n\t<tr><th scope=row>AFFX-YEL002c/WBP1_at</th><td>3.281261</td><td> 2.997496</td><td>2.888443</td><td>3.050276</td><td>3.037369</td><td>2.904478</td><td>3.065646</td><td>3.357285</td><td>2.989665</td><td>2.933019</td><td>⋯</td><td>3.048893</td><td>3.005741</td><td>3.767146</td><td>3.256931</td><td>3.065277</td><td>3.120296</td><td>2.958656</td><td>3.001848</td><td>2.802044</td><td>3.009121</td></tr>\n\t<tr><th scope=row>AFFX-YEL018w/_at</th><td>3.216799</td><td> 2.978343</td><td>3.145237</td><td>3.098193</td><td>3.292282</td><td>3.204104</td><td>3.361988</td><td>2.930769</td><td>3.213673</td><td>3.039339</td><td>⋯</td><td>3.209500</td><td>2.973227</td><td>3.825889</td><td>3.129444</td><td>3.223136</td><td>3.586048</td><td>3.417289</td><td>3.273183</td><td>3.248411</td><td>3.095670</td></tr>\n\t<tr><th scope=row>AFFX-YEL021w/URA3_at</th><td>3.395100</td><td> 3.154582</td><td>3.076809</td><td>3.134195</td><td>5.943431</td><td>3.084158</td><td>3.246510</td><td>3.092988</td><td>3.386872</td><td>4.499956</td><td>⋯</td><td>3.776172</td><td>3.063341</td><td>5.634363</td><td>3.716873</td><td>3.038208</td><td>3.380032</td><td>5.568760</td><td>3.175193</td><td>4.309682</td><td>3.342961</td></tr>\n\t<tr><th scope=row>AFFX-YEL024w/RIP1_at</th><td>3.805801</td><td> 3.092056</td><td>3.428454</td><td>3.336388</td><td>4.043344</td><td>3.470484</td><td>3.375992</td><td>3.341414</td><td>3.785948</td><td>3.618562</td><td>⋯</td><td>3.944137</td><td>3.503261</td><td>4.276009</td><td>3.813406</td><td>3.537594</td><td>3.610839</td><td>4.388177</td><td>3.779012</td><td>3.631034</td><td>3.842535</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA matrix: 12625 × 128 of type dbl\n\n| <!--/--> | 01005 | 01010 | 03002 | 04006 | 04007 | 04008 | 04010 | 04016 | 06002 | 08001 | ⋯ | 37001 | 43006 | 43015 | 44001 | 49004 | 56007 | 64005 | 65003 | 83001 | LAL4 |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 1000_at | 7.597323 |  7.479445 | 7.567593 | 7.384684 | 7.905312 |  7.065914 |  7.474537 |  7.536119 | 7.183331 |  7.735545 | ⋯ |  7.849455 | 7.960842 | 8.188617 | 7.399999 | 7.813474 | 7.816922 |  7.913249 | 7.800199 | 8.030047 | 7.702217 |\n| 1001_at | 5.046194 |  4.932537 | 4.799294 | 4.922627 | 4.844565 |  5.147762 |  5.122518 |  5.016132 | 5.288943 |  4.633217 | ⋯ |  4.959450 | 4.537677 | 5.154500 | 5.071885 | 4.874525 | 4.788699 |  5.403640 | 5.443827 | 5.178633 | 5.029670 |\n| 1002_f_at | 3.900466 |  4.208155 | 3.886169 | 4.206798 | 3.416923 |  3.945869 |  4.150506 |  3.576360 | 3.900935 |  3.630190 | ⋯ |  4.047428 | 3.694877 | 3.949546 | 3.837517 | 3.767672 | 4.248075 |  4.119252 | 3.957468 | 4.053458 | 4.002306 |\n| 1003_s_at | 5.903856 |  6.169024 | 5.860459 | 6.116890 | 5.687997 |  6.208061 |  6.292713 |  5.665991 | 5.842326 |  5.875375 | ⋯ |  6.193790 | 5.973162 | 6.033704 | 5.560886 | 5.930329 | 5.878533 |  6.193376 | 6.006443 | 6.050129 | 6.297549 |\n| 1004_at | 5.925260 |  5.912780 | 5.893209 | 6.170245 | 5.615210 |  5.923487 |  6.046607 |  5.738218 | 5.994515 |  5.748350 | ⋯ |  5.824930 | 5.841095 | 5.870163 | 5.686555 | 5.793403 | 5.913503 |  6.053053 | 5.549087 | 5.932412 | 5.827131 |\n| 1005_at | 8.570990 | 10.428299 | 9.616713 | 9.937155 | 9.983809 | 10.063484 | 10.662059 | 11.269115 | 8.812869 | 10.165159 | ⋯ | 10.408645 | 7.524865 | 8.381740 | 8.055760 | 7.234242 | 8.083834 | 10.163355 | 8.184442 | 8.798521 | 7.687370 |\n| 1006_at | 3.656143 |  3.853979 | 3.646808 | 3.874289 | 3.547361 |  3.771648 |  3.747963 |  3.318315 | 3.874648 |  3.525401 | ⋯ |  3.670568 | 3.652459 | 3.707005 | 3.852594 | 3.527330 | 3.429451 |  3.756402 | 3.571963 | 3.585416 | 3.674371 |\n| 1007_s_at | 7.623562 |  7.543604 | 7.916954 | 6.816397 | 7.516981 |  7.726716 |  7.288960 |  7.724153 | 6.763490 |  7.543907 | ⋯ |  6.609299 | 6.482215 | 6.651151 | 6.463802 | 7.083294 | 7.107712 |  6.895677 | 6.926471 | 6.861161 | 6.731454 |\n| 1008_f_at | 8.903547 |  9.903953 | 8.494499 | 9.533983 | 8.871669 |  9.424092 |  8.999938 |  8.006848 | 9.190553 |  8.408789 | ⋯ |  8.561828 | 8.765609 | 8.459983 | 8.611438 | 8.911405 | 9.043327 |  8.284549 | 8.569874 | 8.474002 | 8.539480 |\n| 1009_at | 9.371888 |  9.322177 | 9.304982 | 9.135370 | 9.627175 |  9.189420 |  9.312164 |  9.946430 | 9.112255 |  9.577024 | ⋯ |  9.492835 | 9.237562 | 9.183180 | 9.250913 | 9.609640 | 9.474761 |  9.702681 | 9.110240 | 9.884863 | 9.575421 |\n| 100_g_at | 7.985255 |  8.015146 | 7.730108 | 7.827044 | 7.834880 |  7.984250 |  8.120029 |  8.084865 | 7.697962 |  8.302688 | ⋯ |  8.003010 | 7.836479 | 8.007742 | 7.825349 | 7.867227 | 8.009918 |  8.271588 | 8.169891 | 7.897170 | 7.965106 |\n| 1010_at | 5.399275 |  5.620731 | 5.172665 | 5.967741 | 5.185482 |  5.484939 |  5.840837 |  5.067395 | 5.237119 |  5.273373 | ⋯ |  5.058770 | 4.848082 | 5.330210 | 4.881275 | 5.180372 | 5.310220 |  5.237757 | 4.993497 | 5.385707 | 4.982707 |\n| 1011_s_at | 5.698984 |  6.172831 | 6.298505 | 6.680342 | 6.658723 |  6.372048 |  7.252254 |  6.674473 | 6.883788 |  5.714778 | ⋯ |  6.434650 | 5.573140 | 5.360469 | 6.067903 | 5.246391 | 4.942719 |  5.109609 | 6.490558 | 6.023058 | 5.809493 |\n| 1012_at | 3.432633 |  3.921244 | 3.649822 | 3.784770 | 3.833315 |  3.948326 |  3.676359 |  3.499718 | 3.439177 |  3.883105 | ⋯ |  3.563572 | 3.877966 | 3.393784 | 3.173837 | 3.760740 | 4.148892 |  4.065844 | 3.762194 | 3.735753 | 3.664340 |\n| 1013_at | 3.442516 |  3.002145 | 3.314890 | 3.474333 | 3.851746 |  3.851362 |  3.565626 |  2.851111 | 3.070907 |  2.969788 | ⋯ |  2.799112 | 3.335830 | 2.823894 | 3.162021 | 2.899012 | 3.008075 |  3.153419 | 2.902942 | 3.018644 | 3.127662 |\n| 1014_at | 7.318896 |  7.653855 | 7.678002 | 6.971806 | 7.446160 |  7.475994 |  7.527383 |  7.550745 | 7.119259 |  7.594602 | ⋯ |  7.247758 | 7.442379 | 7.125079 | 7.846200 | 7.571595 | 7.931793 |  7.654051 | 7.414572 | 7.505681 | 7.532083 |\n| 1015_s_at | 5.479742 |  6.110416 | 5.679977 | 5.962049 | 5.154362 |  5.696765 |  5.917831 |  5.535112 | 5.723400 |  5.569632 | ⋯ |  5.733726 | 5.487096 | 5.628726 | 5.473225 | 5.499078 | 5.398977 |  5.688522 | 5.753532 | 5.555595 | 5.721859 |\n| 1016_s_at | 3.384669 |  3.361198 | 3.174670 | 3.020869 | 3.177107 |  3.207662 |  3.229092 |  3.210501 | 3.353474 |  3.176632 | ⋯ |  3.498991 | 3.250506 | 3.176352 | 3.924138 | 3.245476 | 3.476038 |  3.285540 | 3.392091 | 3.274649 | 3.653479 |\n| 1017_at | 4.921528 |  4.115588 | 4.743844 | 4.691007 | 5.406613 |  4.431900 |  3.931028 |  5.129296 | 3.956742 |  5.178051 | ⋯ |  4.483604 | 5.133695 | 4.789429 | 4.478413 | 5.773424 | 5.111104 |  4.185028 | 4.268057 | 4.030064 | 3.826977 |\n| 1018_at | 6.281061 |  6.147906 | 5.941267 | 6.260564 | 5.782521 |  6.032982 |  6.095990 |  5.903504 | 6.231557 |  5.861168 | ⋯ |  6.519862 | 6.104468 | 6.013004 | 6.493142 | 6.333817 | 6.164413 |  6.674320 | 6.566802 | 6.095361 | 6.425974 |\n| 1019_g_at | 6.546212 |  6.754309 | 6.704541 | 6.898213 | 6.265019 |  6.370988 |  6.262760 |  6.373234 | 6.648472 |  6.306895 | ⋯ |  6.673348 | 6.658479 | 6.402795 | 6.766007 | 6.851714 | 6.696270 |  7.033197 | 7.395795 | 6.356240 | 6.963269 |\n| 101_at | 6.123411 |  6.163813 | 6.016756 | 5.930699 | 5.848487 |  5.820837 |  6.319743 |  5.794677 | 6.079454 |  6.109053 | ⋯ |  6.953927 | 6.442707 | 6.347830 | 6.134284 | 6.043279 | 6.572743 |  6.666000 | 6.729782 | 6.750178 | 6.789686 |\n| 1020_s_at | 7.440408 |  8.225343 | 6.916695 | 7.921413 | 7.489289 |  7.520384 |  7.608336 |  7.262121 | 8.059713 |  7.466821 | ⋯ |  8.241378 | 7.422417 | 7.080051 | 8.689273 | 7.759317 | 7.530735 |  8.074391 | 8.082602 | 8.028259 | 7.878384 |\n| 1021_at | 4.635150 |  5.182118 | 5.485365 | 4.848357 | 4.547636 |  4.433935 |  4.690830 |  4.394778 | 4.567002 |  6.617756 | ⋯ |  5.006477 | 4.117340 | 4.883207 | 4.955042 | 4.716777 | 4.750484 |  4.891067 | 4.548111 | 4.481377 | 4.397078 |\n| 1022_f_at | 3.818130 |  3.697340 | 3.756676 | 3.688513 | 3.525746 |  3.647198 |  3.372396 |  3.561139 | 3.991485 |  3.716743 | ⋯ |  4.163488 | 3.741596 | 3.567559 | 4.022900 | 3.789003 | 3.831971 |  4.111677 | 3.939289 | 3.707847 | 4.329154 |\n| 1023_at | 5.724247 |  6.102434 | 5.535088 | 6.308213 | 5.564462 |  5.773668 |  5.873469 |  5.490342 | 5.908396 |  5.505490 | ⋯ |  5.829125 | 5.414204 | 5.776841 | 5.269463 | 5.700922 | 5.956093 |  5.774114 | 5.858850 | 5.637539 | 5.408432 |\n| 1024_at | 5.754872 |  5.887494 | 5.605735 | 6.336704 | 5.773976 |  5.846395 |  6.022922 |  5.760028 | 5.644638 |  5.556812 | ⋯ |  5.741901 | 5.466854 | 5.805061 | 5.698824 | 5.723203 | 5.847403 |  5.870010 | 5.765627 | 5.834069 | 5.874103 |\n| 1025_g_at | 4.554326 |  4.599976 | 4.366561 | 4.729267 | 4.177634 |  4.400837 |  4.441394 |  4.134673 | 4.423239 |  4.329983 | ⋯ |  4.797889 | 4.449505 | 4.716811 | 4.479158 | 4.614610 | 4.620035 |  4.757783 | 4.474032 | 4.657209 | 4.869916 |\n| 1026_s_at | 6.384481 |  6.541548 | 6.169922 | 6.752466 | 6.162612 |  6.918602 |  6.920326 |  6.554345 | 6.482749 |  6.386876 | ⋯ |  6.504184 | 6.252522 | 6.457230 | 6.678333 | 6.413700 | 6.595883 |  6.723828 | 6.753657 | 6.671640 | 6.627980 |\n| 1027_at | 5.468026 |  5.546560 | 5.460557 | 5.562977 | 5.293422 |  5.269626 |  5.492094 |  4.989486 | 5.590867 |  5.481647 | ⋯ |  5.288263 | 5.305203 | 5.329512 | 5.301670 | 5.369972 | 5.494767 |  5.680495 | 5.259128 | 5.640146 | 5.544918 |\n| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n| AFFX-HUMISGF3A/M97935_MB_at | 5.082692 |  7.611857 | 4.974700 | 5.165775 | 5.806239 | 6.720742 | 5.732256 | 5.522468 | 5.372603 | 5.180712 | ⋯ | 5.130053 | 4.951393 | 4.903532 | 6.024029 | 5.088734 | 5.227949 | 4.962942 | 5.180621 | 4.803131 | 6.137956 |\n| AFFX-HUMRGE/M10098_3_at | 5.743438 |  5.423036 | 5.866900 | 7.767792 | 5.268211 | 5.463824 | 6.631424 | 4.940925 | 5.170254 | 5.346651 | ⋯ | 5.322311 | 5.001757 | 5.678911 | 4.983021 | 5.542517 | 5.852424 | 5.379777 | 5.232322 | 5.091796 | 5.179363 |\n| AFFX-HUMRGE/M10098_5_at | 4.922025 |  4.468572 | 4.698738 | 7.542934 | 4.109162 | 4.768099 | 5.878260 | 4.273090 | 4.480622 | 4.463458 | ⋯ | 4.578618 | 4.281334 | 4.793171 | 4.632098 | 4.498728 | 4.831378 | 4.546390 | 4.436946 | 4.290367 | 4.529884 |\n| AFFX-HUMRGE/M10098_M_at | 4.725279 |  4.160386 | 4.598819 | 6.734705 | 4.024907 | 3.889413 | 5.009262 | 4.300358 | 4.096791 | 4.356831 | ⋯ | 3.903118 | 3.785432 | 4.225459 | 4.089147 | 4.153007 | 4.442808 | 3.981419 | 3.888398 | 3.656073 | 3.876826 |\n| AFFX-HUMTFRR/M11507_3_at | 4.875464 |  4.810450 | 4.853883 | 5.213899 | 5.384624 | 5.446088 | 5.385702 | 4.829268 | 4.887564 | 4.991675 | ⋯ | 5.873119 | 5.411731 | 5.477222 | 4.733239 | 5.522148 | 5.399388 | 5.445041 | 4.801405 | 5.720164 | 5.498475 |\n| AFFX-HUMTFRR/M11507_5_at | 3.768308 |  4.178835 | 3.755861 | 4.776499 | 3.911115 | 4.279529 | 4.921189 | 3.904052 | 3.927687 | 3.587829 | ⋯ | 4.334639 | 3.741585 | 3.954517 | 3.947313 | 3.975845 | 4.093276 | 4.004460 | 3.772255 | 3.974060 | 3.898813 |\n| AFFX-HUMTFRR/M11507_M_at | 4.163886 |  4.641657 | 4.040983 | 4.885080 | 4.350342 | 4.929152 | 5.109563 | 4.284558 | 4.130364 | 4.384626 | ⋯ | 4.673340 | 4.308101 | 4.520912 | 4.291467 | 4.670833 | 4.634246 | 4.507426 | 4.387551 | 4.546331 | 4.331859 |\n| AFFX-LysX-3_at | 3.311952 |  2.850462 | 3.084060 | 3.126579 | 2.871642 | 2.817815 | 2.833962 | 2.778187 | 3.074082 | 3.102063 | ⋯ | 3.190449 | 2.961977 | 4.000543 | 3.625823 | 2.987288 | 3.396457 | 2.801641 | 2.890539 | 2.908193 | 3.182116 |\n| AFFX-LysX-5_at | 2.828138 |  2.834018 | 2.892196 | 2.789425 | 2.830019 | 2.782114 | 2.889998 | 2.701625 | 3.077016 | 2.917622 | ⋯ | 2.919766 | 2.870377 | 4.197757 | 3.199422 | 2.789884 | 3.258625 | 2.811507 | 3.042925 | 2.930631 | 3.087345 |\n| AFFX-LysX-M_at | 4.721098 |  4.625145 | 4.575322 | 4.902402 | 4.749065 | 4.562960 | 4.803757 | 4.608499 | 4.800184 | 4.294252 | ⋯ | 4.721945 | 4.430331 | 5.599868 | 5.334023 | 4.450242 | 4.750099 | 4.506028 | 4.637085 | 4.539593 | 4.936381 |\n| AFFX-M27830_3_at | 7.463929 |  7.438116 | 7.247500 | 7.655706 | 7.049913 | 7.277769 | 7.500706 | 7.350631 | 7.515679 | 6.974698 | ⋯ | 7.535924 | 6.951264 | 7.118890 | 7.893386 | 7.567082 | 7.190904 | 7.450166 | 7.587643 | 6.957847 | 7.522223 |\n| AFFX-M27830_5_at | 7.047142 |  8.772183 | 7.012569 | 7.912946 | 5.026355 | 5.683872 | 7.351909 | 5.496411 | 4.986038 | 5.621757 | ⋯ | 5.134773 | 4.677474 | 6.280724 | 5.883414 | 5.861596 | 5.551034 | 5.602937 | 4.815696 | 5.128495 | 5.615701 |\n| AFFX-M27830_M_at | 8.116965 | 10.008773 | 8.103920 | 8.584401 | 7.309622 | 7.933779 | 8.444469 | 7.517100 | 7.451483 | 7.695028 | ⋯ | 8.124331 | 7.435762 | 8.133971 | 7.856379 | 7.873945 | 8.304769 | 8.277190 | 7.933101 | 7.774132 | 7.876990 |\n| AFFX-MurFAS_at | 3.536136 |  3.213659 | 3.331824 | 3.431021 | 3.461410 | 3.483763 | 3.323103 | 3.446710 | 3.440076 | 3.273991 | ⋯ | 3.537291 | 3.254407 | 3.875097 | 3.444693 | 3.743439 | 3.709902 | 3.233520 | 3.352985 | 3.165264 | 3.268459 |\n| AFFX-MurIL10_at | 4.341857 |  4.639913 | 4.114760 | 4.775549 | 3.991641 | 4.627252 | 4.530767 | 4.127038 | 4.370285 | 3.835914 | ⋯ | 4.321451 | 3.974078 | 4.718328 | 4.095881 | 4.001305 | 4.763165 | 4.161259 | 3.959749 | 3.975969 | 4.193781 |\n| AFFX-MurIL2_at | 5.402049 |  4.970754 | 5.136504 | 5.328741 | 4.934468 | 5.036166 | 4.914641 | 4.886392 | 5.372676 | 5.018007 | ⋯ | 5.358090 | 4.822789 | 5.320897 | 5.489717 | 5.181606 | 5.365144 | 5.194208 | 5.160640 | 4.844565 | 4.967381 |\n| AFFX-MurIL4_at | 3.088365 |  2.912783 | 2.941528 | 3.123581 | 3.259221 | 3.037260 | 3.003451 | 2.973277 | 3.320805 | 2.784965 | ⋯ | 3.338618 | 2.875561 | 3.795054 | 3.426576 | 3.092319 | 3.577267 | 3.058879 | 3.168797 | 3.028462 | 3.014841 |\n| AFFX-PheX-3_at | 5.073080 |  5.539796 | 5.222797 | 5.668133 | 5.199665 | 5.470518 | 5.506018 | 4.931751 | 5.180675 | 5.256852 | ⋯ | 5.056564 | 4.865700 | 5.419945 | 5.860424 | 4.921566 | 5.486537 | 5.354748 | 5.261869 | 5.249353 | 5.188049 |\n| AFFX-PheX-5_at | 3.050993 |  3.116976 | 3.301066 | 3.366936 | 3.008996 | 3.125731 | 2.902621 | 3.074890 | 3.364359 | 3.217555 | ⋯ | 3.293791 | 3.222073 | 4.072204 | 3.646259 | 3.018502 | 3.860386 | 3.073433 | 3.408542 | 3.086195 | 3.374656 |\n| AFFX-PheX-M_at | 3.344718 |  3.249449 | 3.179357 | 3.343461 | 2.962293 | 3.045536 | 3.029851 | 3.068849 | 3.355382 | 3.301770 | ⋯ | 3.215089 | 3.157067 | 3.836710 | 3.774872 | 2.950347 | 3.884731 | 3.195110 | 3.741395 | 3.254390 | 3.418817 |\n| AFFX-ThrX-3_at | 4.365777 |  4.559372 | 4.309676 | 4.808575 | 4.120906 | 4.423895 | 4.166162 | 4.301804 | 4.439963 | 4.102016 | ⋯ | 4.302587 | 4.304613 | 4.927189 | 4.667565 | 4.370202 | 5.101768 | 4.440123 | 4.675419 | 4.094304 | 4.474966 |\n| AFFX-ThrX-5_at | 4.127825 |  4.452035 | 4.157649 | 4.205917 | 3.899734 | 4.104524 | 3.994176 | 4.076177 | 4.530618 | 4.142404 | ⋯ | 4.553950 | 4.168382 | 4.797841 | 4.485632 | 4.261370 | 4.831275 | 4.366302 | 4.738456 | 4.194136 | 4.555346 |\n| AFFX-ThrX-M_at | 3.469720 |  3.243264 | 3.228969 | 3.399563 | 3.027191 | 3.295530 | 3.257909 | 2.994594 | 3.525741 | 3.232638 | ⋯ | 3.458331 | 3.348805 | 4.113437 | 3.747689 | 3.395225 | 4.110112 | 3.436514 | 3.479872 | 3.160448 | 3.500401 |\n| AFFX-TrpnX-3_at | 2.977916 |  2.650942 | 2.751669 | 2.960073 | 2.806713 | 2.962976 | 3.152471 | 2.695692 | 2.920628 | 2.844983 | ⋯ | 2.937670 | 2.850850 | 3.582664 | 3.059061 | 2.876187 | 3.408258 | 2.809911 | 2.771741 | 2.775778 | 2.780948 |\n| AFFX-TrpnX-5_at | 3.623308 |  3.621928 | 3.497549 | 3.834567 | 3.485573 | 3.832459 | 3.657211 | 3.685598 | 3.863668 | 3.474924 | ⋯ | 3.830677 | 3.416041 | 4.315469 | 3.965705 | 3.807634 | 4.254020 | 3.711537 | 3.701861 | 3.753496 | 3.701939 |\n| AFFX-TrpnX-M_at | 3.416213 |  3.344428 | 3.250417 | 3.589504 | 3.140334 | 3.397393 | 3.272373 | 3.290285 | 3.638814 | 3.343970 | ⋯ | 3.117524 | 3.253874 | 3.971773 | 3.438820 | 3.380928 | 3.753142 | 3.462627 | 3.551272 | 3.114569 | 3.253312 |\n| AFFX-YEL002c/WBP1_at | 3.281261 |  2.997496 | 2.888443 | 3.050276 | 3.037369 | 2.904478 | 3.065646 | 3.357285 | 2.989665 | 2.933019 | ⋯ | 3.048893 | 3.005741 | 3.767146 | 3.256931 | 3.065277 | 3.120296 | 2.958656 | 3.001848 | 2.802044 | 3.009121 |\n| AFFX-YEL018w/_at | 3.216799 |  2.978343 | 3.145237 | 3.098193 | 3.292282 | 3.204104 | 3.361988 | 2.930769 | 3.213673 | 3.039339 | ⋯ | 3.209500 | 2.973227 | 3.825889 | 3.129444 | 3.223136 | 3.586048 | 3.417289 | 3.273183 | 3.248411 | 3.095670 |\n| AFFX-YEL021w/URA3_at | 3.395100 |  3.154582 | 3.076809 | 3.134195 | 5.943431 | 3.084158 | 3.246510 | 3.092988 | 3.386872 | 4.499956 | ⋯ | 3.776172 | 3.063341 | 5.634363 | 3.716873 | 3.038208 | 3.380032 | 5.568760 | 3.175193 | 4.309682 | 3.342961 |\n| AFFX-YEL024w/RIP1_at | 3.805801 |  3.092056 | 3.428454 | 3.336388 | 4.043344 | 3.470484 | 3.375992 | 3.341414 | 3.785948 | 3.618562 | ⋯ | 3.944137 | 3.503261 | 4.276009 | 3.813406 | 3.537594 | 3.610839 | 4.388177 | 3.779012 | 3.631034 | 3.842535 |\n\n","text/latex":"A matrix: 12625 × 128 of type dbl\n\\begin{tabular}{r|lllllllllllllllllllll}\n  & 01005 & 01010 & 03002 & 04006 & 04007 & 04008 & 04010 & 04016 & 06002 & 08001 & ⋯ & 37001 & 43006 & 43015 & 44001 & 49004 & 56007 & 64005 & 65003 & 83001 & LAL4\\\\\n\\hline\n\t1000\\_at & 7.597323 &  7.479445 & 7.567593 & 7.384684 & 7.905312 &  7.065914 &  7.474537 &  7.536119 & 7.183331 &  7.735545 & ⋯ &  7.849455 & 7.960842 & 8.188617 & 7.399999 & 7.813474 & 7.816922 &  7.913249 & 7.800199 & 8.030047 & 7.702217\\\\\n\t1001\\_at & 5.046194 &  4.932537 & 4.799294 & 4.922627 & 4.844565 &  5.147762 &  5.122518 &  5.016132 & 5.288943 &  4.633217 & ⋯ &  4.959450 & 4.537677 & 5.154500 & 5.071885 & 4.874525 & 4.788699 &  5.403640 & 5.443827 & 5.178633 & 5.029670\\\\\n\t1002\\_f\\_at & 3.900466 &  4.208155 & 3.886169 & 4.206798 & 3.416923 &  3.945869 &  4.150506 &  3.576360 & 3.900935 &  3.630190 & ⋯ &  4.047428 & 3.694877 & 3.949546 & 3.837517 & 3.767672 & 4.248075 &  4.119252 & 3.957468 & 4.053458 & 4.002306\\\\\n\t1003\\_s\\_at & 5.903856 &  6.169024 & 5.860459 & 6.116890 & 5.687997 &  6.208061 &  6.292713 &  5.665991 & 5.842326 &  5.875375 & ⋯ &  6.193790 & 5.973162 & 6.033704 & 5.560886 & 5.930329 & 5.878533 &  6.193376 & 6.006443 & 6.050129 & 6.297549\\\\\n\t1004\\_at & 5.925260 &  5.912780 & 5.893209 & 6.170245 & 5.615210 &  5.923487 &  6.046607 &  5.738218 & 5.994515 &  5.748350 & ⋯ &  5.824930 & 5.841095 & 5.870163 & 5.686555 & 5.793403 & 5.913503 &  6.053053 & 5.549087 & 5.932412 & 5.827131\\\\\n\t1005\\_at & 8.570990 & 10.428299 & 9.616713 & 9.937155 & 9.983809 & 10.063484 & 10.662059 & 11.269115 & 8.812869 & 10.165159 & ⋯ & 10.408645 & 7.524865 & 8.381740 & 8.055760 & 7.234242 & 8.083834 & 10.163355 & 8.184442 & 8.798521 & 7.687370\\\\\n\t1006\\_at & 3.656143 &  3.853979 & 3.646808 & 3.874289 & 3.547361 &  3.771648 &  3.747963 &  3.318315 & 3.874648 &  3.525401 & ⋯ &  3.670568 & 3.652459 & 3.707005 & 3.852594 & 3.527330 & 3.429451 &  3.756402 & 3.571963 & 3.585416 & 3.674371\\\\\n\t1007\\_s\\_at & 7.623562 &  7.543604 & 7.916954 & 6.816397 & 7.516981 &  7.726716 &  7.288960 &  7.724153 & 6.763490 &  7.543907 & ⋯ &  6.609299 & 6.482215 & 6.651151 & 6.463802 & 7.083294 & 7.107712 &  6.895677 & 6.926471 & 6.861161 & 6.731454\\\\\n\t1008\\_f\\_at & 8.903547 &  9.903953 & 8.494499 & 9.533983 & 8.871669 &  9.424092 &  8.999938 &  8.006848 & 9.190553 &  8.408789 & ⋯ &  8.561828 & 8.765609 & 8.459983 & 8.611438 & 8.911405 & 9.043327 &  8.284549 & 8.569874 & 8.474002 & 8.539480\\\\\n\t1009\\_at & 9.371888 &  9.322177 & 9.304982 & 9.135370 & 9.627175 &  9.189420 &  9.312164 &  9.946430 & 9.112255 &  9.577024 & ⋯ &  9.492835 & 9.237562 & 9.183180 & 9.250913 & 9.609640 & 9.474761 &  9.702681 & 9.110240 & 9.884863 & 9.575421\\\\\n\t100\\_g\\_at & 7.985255 &  8.015146 & 7.730108 & 7.827044 & 7.834880 &  7.984250 &  8.120029 &  8.084865 & 7.697962 &  8.302688 & ⋯ &  8.003010 & 7.836479 & 8.007742 & 7.825349 & 7.867227 & 8.009918 &  8.271588 & 8.169891 & 7.897170 & 7.965106\\\\\n\t1010\\_at & 5.399275 &  5.620731 & 5.172665 & 5.967741 & 5.185482 &  5.484939 &  5.840837 &  5.067395 & 5.237119 &  5.273373 & ⋯ &  5.058770 & 4.848082 & 5.330210 & 4.881275 & 5.180372 & 5.310220 &  5.237757 & 4.993497 & 5.385707 & 4.982707\\\\\n\t1011\\_s\\_at & 5.698984 &  6.172831 & 6.298505 & 6.680342 & 6.658723 &  6.372048 &  7.252254 &  6.674473 & 6.883788 &  5.714778 & ⋯ &  6.434650 & 5.573140 & 5.360469 & 6.067903 & 5.246391 & 4.942719 &  5.109609 & 6.490558 & 6.023058 & 5.809493\\\\\n\t1012\\_at & 3.432633 &  3.921244 & 3.649822 & 3.784770 & 3.833315 &  3.948326 &  3.676359 &  3.499718 & 3.439177 &  3.883105 & ⋯ &  3.563572 & 3.877966 & 3.393784 & 3.173837 & 3.760740 & 4.148892 &  4.065844 & 3.762194 & 3.735753 & 3.664340\\\\\n\t1013\\_at & 3.442516 &  3.002145 & 3.314890 & 3.474333 & 3.851746 &  3.851362 &  3.565626 &  2.851111 & 3.070907 &  2.969788 & ⋯ &  2.799112 & 3.335830 & 2.823894 & 3.162021 & 2.899012 & 3.008075 &  3.153419 & 2.902942 & 3.018644 & 3.127662\\\\\n\t1014\\_at & 7.318896 &  7.653855 & 7.678002 & 6.971806 & 7.446160 &  7.475994 &  7.527383 &  7.550745 & 7.119259 &  7.594602 & ⋯ &  7.247758 & 7.442379 & 7.125079 & 7.846200 & 7.571595 & 7.931793 &  7.654051 & 7.414572 & 7.505681 & 7.532083\\\\\n\t1015\\_s\\_at & 5.479742 &  6.110416 & 5.679977 & 5.962049 & 5.154362 &  5.696765 &  5.917831 &  5.535112 & 5.723400 &  5.569632 & ⋯ &  5.733726 & 5.487096 & 5.628726 & 5.473225 & 5.499078 & 5.398977 &  5.688522 & 5.753532 & 5.555595 & 5.721859\\\\\n\t1016\\_s\\_at & 3.384669 &  3.361198 & 3.174670 & 3.020869 & 3.177107 &  3.207662 &  3.229092 &  3.210501 & 3.353474 &  3.176632 & ⋯ &  3.498991 & 3.250506 & 3.176352 & 3.924138 & 3.245476 & 3.476038 &  3.285540 & 3.392091 & 3.274649 & 3.653479\\\\\n\t1017\\_at & 4.921528 &  4.115588 & 4.743844 & 4.691007 & 5.406613 &  4.431900 &  3.931028 &  5.129296 & 3.956742 &  5.178051 & ⋯ &  4.483604 & 5.133695 & 4.789429 & 4.478413 & 5.773424 & 5.111104 &  4.185028 & 4.268057 & 4.030064 & 3.826977\\\\\n\t1018\\_at & 6.281061 &  6.147906 & 5.941267 & 6.260564 & 5.782521 &  6.032982 &  6.095990 &  5.903504 & 6.231557 &  5.861168 & ⋯ &  6.519862 & 6.104468 & 6.013004 & 6.493142 & 6.333817 & 6.164413 &  6.674320 & 6.566802 & 6.095361 & 6.425974\\\\\n\t1019\\_g\\_at & 6.546212 &  6.754309 & 6.704541 & 6.898213 & 6.265019 &  6.370988 &  6.262760 &  6.373234 & 6.648472 &  6.306895 & ⋯ &  6.673348 & 6.658479 & 6.402795 & 6.766007 & 6.851714 & 6.696270 &  7.033197 & 7.395795 & 6.356240 & 6.963269\\\\\n\t101\\_at & 6.123411 &  6.163813 & 6.016756 & 5.930699 & 5.848487 &  5.820837 &  6.319743 &  5.794677 & 6.079454 &  6.109053 & ⋯ &  6.953927 & 6.442707 & 6.347830 & 6.134284 & 6.043279 & 6.572743 &  6.666000 & 6.729782 & 6.750178 & 6.789686\\\\\n\t1020\\_s\\_at & 7.440408 &  8.225343 & 6.916695 & 7.921413 & 7.489289 &  7.520384 &  7.608336 &  7.262121 & 8.059713 &  7.466821 & ⋯ &  8.241378 & 7.422417 & 7.080051 & 8.689273 & 7.759317 & 7.530735 &  8.074391 & 8.082602 & 8.028259 & 7.878384\\\\\n\t1021\\_at & 4.635150 &  5.182118 & 5.485365 & 4.848357 & 4.547636 &  4.433935 &  4.690830 &  4.394778 & 4.567002 &  6.617756 & ⋯ &  5.006477 & 4.117340 & 4.883207 & 4.955042 & 4.716777 & 4.750484 &  4.891067 & 4.548111 & 4.481377 & 4.397078\\\\\n\t1022\\_f\\_at & 3.818130 &  3.697340 & 3.756676 & 3.688513 & 3.525746 &  3.647198 &  3.372396 &  3.561139 & 3.991485 &  3.716743 & ⋯ &  4.163488 & 3.741596 & 3.567559 & 4.022900 & 3.789003 & 3.831971 &  4.111677 & 3.939289 & 3.707847 & 4.329154\\\\\n\t1023\\_at & 5.724247 &  6.102434 & 5.535088 & 6.308213 & 5.564462 &  5.773668 &  5.873469 &  5.490342 & 5.908396 &  5.505490 & ⋯ &  5.829125 & 5.414204 & 5.776841 & 5.269463 & 5.700922 & 5.956093 &  5.774114 & 5.858850 & 5.637539 & 5.408432\\\\\n\t1024\\_at & 5.754872 &  5.887494 & 5.605735 & 6.336704 & 5.773976 &  5.846395 &  6.022922 &  5.760028 & 5.644638 &  5.556812 & ⋯ &  5.741901 & 5.466854 & 5.805061 & 5.698824 & 5.723203 & 5.847403 &  5.870010 & 5.765627 & 5.834069 & 5.874103\\\\\n\t1025\\_g\\_at & 4.554326 &  4.599976 & 4.366561 & 4.729267 & 4.177634 &  4.400837 &  4.441394 &  4.134673 & 4.423239 &  4.329983 & ⋯ &  4.797889 & 4.449505 & 4.716811 & 4.479158 & 4.614610 & 4.620035 &  4.757783 & 4.474032 & 4.657209 & 4.869916\\\\\n\t1026\\_s\\_at & 6.384481 &  6.541548 & 6.169922 & 6.752466 & 6.162612 &  6.918602 &  6.920326 &  6.554345 & 6.482749 &  6.386876 & ⋯ &  6.504184 & 6.252522 & 6.457230 & 6.678333 & 6.413700 & 6.595883 &  6.723828 & 6.753657 & 6.671640 & 6.627980\\\\\n\t1027\\_at & 5.468026 &  5.546560 & 5.460557 & 5.562977 & 5.293422 &  5.269626 &  5.492094 &  4.989486 & 5.590867 &  5.481647 & ⋯ &  5.288263 & 5.305203 & 5.329512 & 5.301670 & 5.369972 & 5.494767 &  5.680495 & 5.259128 & 5.640146 & 5.544918\\\\\n\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n\tAFFX-HUMISGF3A/M97935\\_MB\\_at & 5.082692 &  7.611857 & 4.974700 & 5.165775 & 5.806239 & 6.720742 & 5.732256 & 5.522468 & 5.372603 & 5.180712 & ⋯ & 5.130053 & 4.951393 & 4.903532 & 6.024029 & 5.088734 & 5.227949 & 4.962942 & 5.180621 & 4.803131 & 6.137956\\\\\n\tAFFX-HUMRGE/M10098\\_3\\_at & 5.743438 &  5.423036 & 5.866900 & 7.767792 & 5.268211 & 5.463824 & 6.631424 & 4.940925 & 5.170254 & 5.346651 & ⋯ & 5.322311 & 5.001757 & 5.678911 & 4.983021 & 5.542517 & 5.852424 & 5.379777 & 5.232322 & 5.091796 & 5.179363\\\\\n\tAFFX-HUMRGE/M10098\\_5\\_at & 4.922025 &  4.468572 & 4.698738 & 7.542934 & 4.109162 & 4.768099 & 5.878260 & 4.273090 & 4.480622 & 4.463458 & ⋯ & 4.578618 & 4.281334 & 4.793171 & 4.632098 & 4.498728 & 4.831378 & 4.546390 & 4.436946 & 4.290367 & 4.529884\\\\\n\tAFFX-HUMRGE/M10098\\_M\\_at & 4.725279 &  4.160386 & 4.598819 & 6.734705 & 4.024907 & 3.889413 & 5.009262 & 4.300358 & 4.096791 & 4.356831 & ⋯ & 3.903118 & 3.785432 & 4.225459 & 4.089147 & 4.153007 & 4.442808 & 3.981419 & 3.888398 & 3.656073 & 3.876826\\\\\n\tAFFX-HUMTFRR/M11507\\_3\\_at & 4.875464 &  4.810450 & 4.853883 & 5.213899 & 5.384624 & 5.446088 & 5.385702 & 4.829268 & 4.887564 & 4.991675 & ⋯ & 5.873119 & 5.411731 & 5.477222 & 4.733239 & 5.522148 & 5.399388 & 5.445041 & 4.801405 & 5.720164 & 5.498475\\\\\n\tAFFX-HUMTFRR/M11507\\_5\\_at & 3.768308 &  4.178835 & 3.755861 & 4.776499 & 3.911115 & 4.279529 & 4.921189 & 3.904052 & 3.927687 & 3.587829 & ⋯ & 4.334639 & 3.741585 & 3.954517 & 3.947313 & 3.975845 & 4.093276 & 4.004460 & 3.772255 & 3.974060 & 3.898813\\\\\n\tAFFX-HUMTFRR/M11507\\_M\\_at & 4.163886 &  4.641657 & 4.040983 & 4.885080 & 4.350342 & 4.929152 & 5.109563 & 4.284558 & 4.130364 & 4.384626 & ⋯ & 4.673340 & 4.308101 & 4.520912 & 4.291467 & 4.670833 & 4.634246 & 4.507426 & 4.387551 & 4.546331 & 4.331859\\\\\n\tAFFX-LysX-3\\_at & 3.311952 &  2.850462 & 3.084060 & 3.126579 & 2.871642 & 2.817815 & 2.833962 & 2.778187 & 3.074082 & 3.102063 & ⋯ & 3.190449 & 2.961977 & 4.000543 & 3.625823 & 2.987288 & 3.396457 & 2.801641 & 2.890539 & 2.908193 & 3.182116\\\\\n\tAFFX-LysX-5\\_at & 2.828138 &  2.834018 & 2.892196 & 2.789425 & 2.830019 & 2.782114 & 2.889998 & 2.701625 & 3.077016 & 2.917622 & ⋯ & 2.919766 & 2.870377 & 4.197757 & 3.199422 & 2.789884 & 3.258625 & 2.811507 & 3.042925 & 2.930631 & 3.087345\\\\\n\tAFFX-LysX-M\\_at & 4.721098 &  4.625145 & 4.575322 & 4.902402 & 4.749065 & 4.562960 & 4.803757 & 4.608499 & 4.800184 & 4.294252 & ⋯ & 4.721945 & 4.430331 & 5.599868 & 5.334023 & 4.450242 & 4.750099 & 4.506028 & 4.637085 & 4.539593 & 4.936381\\\\\n\tAFFX-M27830\\_3\\_at & 7.463929 &  7.438116 & 7.247500 & 7.655706 & 7.049913 & 7.277769 & 7.500706 & 7.350631 & 7.515679 & 6.974698 & ⋯ & 7.535924 & 6.951264 & 7.118890 & 7.893386 & 7.567082 & 7.190904 & 7.450166 & 7.587643 & 6.957847 & 7.522223\\\\\n\tAFFX-M27830\\_5\\_at & 7.047142 &  8.772183 & 7.012569 & 7.912946 & 5.026355 & 5.683872 & 7.351909 & 5.496411 & 4.986038 & 5.621757 & ⋯ & 5.134773 & 4.677474 & 6.280724 & 5.883414 & 5.861596 & 5.551034 & 5.602937 & 4.815696 & 5.128495 & 5.615701\\\\\n\tAFFX-M27830\\_M\\_at & 8.116965 & 10.008773 & 8.103920 & 8.584401 & 7.309622 & 7.933779 & 8.444469 & 7.517100 & 7.451483 & 7.695028 & ⋯ & 8.124331 & 7.435762 & 8.133971 & 7.856379 & 7.873945 & 8.304769 & 8.277190 & 7.933101 & 7.774132 & 7.876990\\\\\n\tAFFX-MurFAS\\_at & 3.536136 &  3.213659 & 3.331824 & 3.431021 & 3.461410 & 3.483763 & 3.323103 & 3.446710 & 3.440076 & 3.273991 & ⋯ & 3.537291 & 3.254407 & 3.875097 & 3.444693 & 3.743439 & 3.709902 & 3.233520 & 3.352985 & 3.165264 & 3.268459\\\\\n\tAFFX-MurIL10\\_at & 4.341857 &  4.639913 & 4.114760 & 4.775549 & 3.991641 & 4.627252 & 4.530767 & 4.127038 & 4.370285 & 3.835914 & ⋯ & 4.321451 & 3.974078 & 4.718328 & 4.095881 & 4.001305 & 4.763165 & 4.161259 & 3.959749 & 3.975969 & 4.193781\\\\\n\tAFFX-MurIL2\\_at & 5.402049 &  4.970754 & 5.136504 & 5.328741 & 4.934468 & 5.036166 & 4.914641 & 4.886392 & 5.372676 & 5.018007 & ⋯ & 5.358090 & 4.822789 & 5.320897 & 5.489717 & 5.181606 & 5.365144 & 5.194208 & 5.160640 & 4.844565 & 4.967381\\\\\n\tAFFX-MurIL4\\_at & 3.088365 &  2.912783 & 2.941528 & 3.123581 & 3.259221 & 3.037260 & 3.003451 & 2.973277 & 3.320805 & 2.784965 & ⋯ & 3.338618 & 2.875561 & 3.795054 & 3.426576 & 3.092319 & 3.577267 & 3.058879 & 3.168797 & 3.028462 & 3.014841\\\\\n\tAFFX-PheX-3\\_at & 5.073080 &  5.539796 & 5.222797 & 5.668133 & 5.199665 & 5.470518 & 5.506018 & 4.931751 & 5.180675 & 5.256852 & ⋯ & 5.056564 & 4.865700 & 5.419945 & 5.860424 & 4.921566 & 5.486537 & 5.354748 & 5.261869 & 5.249353 & 5.188049\\\\\n\tAFFX-PheX-5\\_at & 3.050993 &  3.116976 & 3.301066 & 3.366936 & 3.008996 & 3.125731 & 2.902621 & 3.074890 & 3.364359 & 3.217555 & ⋯ & 3.293791 & 3.222073 & 4.072204 & 3.646259 & 3.018502 & 3.860386 & 3.073433 & 3.408542 & 3.086195 & 3.374656\\\\\n\tAFFX-PheX-M\\_at & 3.344718 &  3.249449 & 3.179357 & 3.343461 & 2.962293 & 3.045536 & 3.029851 & 3.068849 & 3.355382 & 3.301770 & ⋯ & 3.215089 & 3.157067 & 3.836710 & 3.774872 & 2.950347 & 3.884731 & 3.195110 & 3.741395 & 3.254390 & 3.418817\\\\\n\tAFFX-ThrX-3\\_at & 4.365777 &  4.559372 & 4.309676 & 4.808575 & 4.120906 & 4.423895 & 4.166162 & 4.301804 & 4.439963 & 4.102016 & ⋯ & 4.302587 & 4.304613 & 4.927189 & 4.667565 & 4.370202 & 5.101768 & 4.440123 & 4.675419 & 4.094304 & 4.474966\\\\\n\tAFFX-ThrX-5\\_at & 4.127825 &  4.452035 & 4.157649 & 4.205917 & 3.899734 & 4.104524 & 3.994176 & 4.076177 & 4.530618 & 4.142404 & ⋯ & 4.553950 & 4.168382 & 4.797841 & 4.485632 & 4.261370 & 4.831275 & 4.366302 & 4.738456 & 4.194136 & 4.555346\\\\\n\tAFFX-ThrX-M\\_at & 3.469720 &  3.243264 & 3.228969 & 3.399563 & 3.027191 & 3.295530 & 3.257909 & 2.994594 & 3.525741 & 3.232638 & ⋯ & 3.458331 & 3.348805 & 4.113437 & 3.747689 & 3.395225 & 4.110112 & 3.436514 & 3.479872 & 3.160448 & 3.500401\\\\\n\tAFFX-TrpnX-3\\_at & 2.977916 &  2.650942 & 2.751669 & 2.960073 & 2.806713 & 2.962976 & 3.152471 & 2.695692 & 2.920628 & 2.844983 & ⋯ & 2.937670 & 2.850850 & 3.582664 & 3.059061 & 2.876187 & 3.408258 & 2.809911 & 2.771741 & 2.775778 & 2.780948\\\\\n\tAFFX-TrpnX-5\\_at & 3.623308 &  3.621928 & 3.497549 & 3.834567 & 3.485573 & 3.832459 & 3.657211 & 3.685598 & 3.863668 & 3.474924 & ⋯ & 3.830677 & 3.416041 & 4.315469 & 3.965705 & 3.807634 & 4.254020 & 3.711537 & 3.701861 & 3.753496 & 3.701939\\\\\n\tAFFX-TrpnX-M\\_at & 3.416213 &  3.344428 & 3.250417 & 3.589504 & 3.140334 & 3.397393 & 3.272373 & 3.290285 & 3.638814 & 3.343970 & ⋯ & 3.117524 & 3.253874 & 3.971773 & 3.438820 & 3.380928 & 3.753142 & 3.462627 & 3.551272 & 3.114569 & 3.253312\\\\\n\tAFFX-YEL002c/WBP1\\_at & 3.281261 &  2.997496 & 2.888443 & 3.050276 & 3.037369 & 2.904478 & 3.065646 & 3.357285 & 2.989665 & 2.933019 & ⋯ & 3.048893 & 3.005741 & 3.767146 & 3.256931 & 3.065277 & 3.120296 & 2.958656 & 3.001848 & 2.802044 & 3.009121\\\\\n\tAFFX-YEL018w/\\_at & 3.216799 &  2.978343 & 3.145237 & 3.098193 & 3.292282 & 3.204104 & 3.361988 & 2.930769 & 3.213673 & 3.039339 & ⋯ & 3.209500 & 2.973227 & 3.825889 & 3.129444 & 3.223136 & 3.586048 & 3.417289 & 3.273183 & 3.248411 & 3.095670\\\\\n\tAFFX-YEL021w/URA3\\_at & 3.395100 &  3.154582 & 3.076809 & 3.134195 & 5.943431 & 3.084158 & 3.246510 & 3.092988 & 3.386872 & 4.499956 & ⋯ & 3.776172 & 3.063341 & 5.634363 & 3.716873 & 3.038208 & 3.380032 & 5.568760 & 3.175193 & 4.309682 & 3.342961\\\\\n\tAFFX-YEL024w/RIP1\\_at & 3.805801 &  3.092056 & 3.428454 & 3.336388 & 4.043344 & 3.470484 & 3.375992 & 3.341414 & 3.785948 & 3.618562 & ⋯ & 3.944137 & 3.503261 & 4.276009 & 3.813406 & 3.537594 & 3.610839 & 4.388177 & 3.779012 & 3.631034 & 3.842535\\\\\n\\end{tabular}\n","text/plain":"                            01005    01010     03002    04006    04007   \n1000_at                     7.597323  7.479445 7.567593 7.384684 7.905312\n1001_at                     5.046194  4.932537 4.799294 4.922627 4.844565\n1002_f_at                   3.900466  4.208155 3.886169 4.206798 3.416923\n1003_s_at                   5.903856  6.169024 5.860459 6.116890 5.687997\n1004_at                     5.925260  5.912780 5.893209 6.170245 5.615210\n1005_at                     8.570990 10.428299 9.616713 9.937155 9.983809\n1006_at                     3.656143  3.853979 3.646808 3.874289 3.547361\n1007_s_at                   7.623562  7.543604 7.916954 6.816397 7.516981\n1008_f_at                   8.903547  9.903953 8.494499 9.533983 8.871669\n1009_at                     9.371888  9.322177 9.304982 9.135370 9.627175\n100_g_at                    7.985255  8.015146 7.730108 7.827044 7.834880\n1010_at                     5.399275  5.620731 5.172665 5.967741 5.185482\n1011_s_at                   5.698984  6.172831 6.298505 6.680342 6.658723\n1012_at                     3.432633  3.921244 3.649822 3.784770 3.833315\n1013_at                     3.442516  3.002145 3.314890 3.474333 3.851746\n1014_at                     7.318896  7.653855 7.678002 6.971806 7.446160\n1015_s_at                   5.479742  6.110416 5.679977 5.962049 5.154362\n1016_s_at                   3.384669  3.361198 3.174670 3.020869 3.177107\n1017_at                     4.921528  4.115588 4.743844 4.691007 5.406613\n1018_at                     6.281061  6.147906 5.941267 6.260564 5.782521\n1019_g_at                   6.546212  6.754309 6.704541 6.898213 6.265019\n101_at                      6.123411  6.163813 6.016756 5.930699 5.848487\n1020_s_at                   7.440408  8.225343 6.916695 7.921413 7.489289\n1021_at                     4.635150  5.182118 5.485365 4.848357 4.547636\n1022_f_at                   3.818130  3.697340 3.756676 3.688513 3.525746\n1023_at                     5.724247  6.102434 5.535088 6.308213 5.564462\n1024_at                     5.754872  5.887494 5.605735 6.336704 5.773976\n1025_g_at                   4.554326  4.599976 4.366561 4.729267 4.177634\n1026_s_at                   6.384481  6.541548 6.169922 6.752466 6.162612\n1027_at                     5.468026  5.546560 5.460557 5.562977 5.293422\n⋮                           ⋮        ⋮         ⋮        ⋮        ⋮       \nAFFX-HUMISGF3A/M97935_MB_at 5.082692  7.611857 4.974700 5.165775 5.806239\nAFFX-HUMRGE/M10098_3_at     5.743438  5.423036 5.866900 7.767792 5.268211\nAFFX-HUMRGE/M10098_5_at     4.922025  4.468572 4.698738 7.542934 4.109162\nAFFX-HUMRGE/M10098_M_at     4.725279  4.160386 4.598819 6.734705 4.024907\nAFFX-HUMTFRR/M11507_3_at    4.875464  4.810450 4.853883 5.213899 5.384624\nAFFX-HUMTFRR/M11507_5_at    3.768308  4.178835 3.755861 4.776499 3.911115\nAFFX-HUMTFRR/M11507_M_at    4.163886  4.641657 4.040983 4.885080 4.350342\nAFFX-LysX-3_at              3.311952  2.850462 3.084060 3.126579 2.871642\nAFFX-LysX-5_at              2.828138  2.834018 2.892196 2.789425 2.830019\nAFFX-LysX-M_at              4.721098  4.625145 4.575322 4.902402 4.749065\nAFFX-M27830_3_at            7.463929  7.438116 7.247500 7.655706 7.049913\nAFFX-M27830_5_at            7.047142  8.772183 7.012569 7.912946 5.026355\nAFFX-M27830_M_at            8.116965 10.008773 8.103920 8.584401 7.309622\nAFFX-MurFAS_at              3.536136  3.213659 3.331824 3.431021 3.461410\nAFFX-MurIL10_at             4.341857  4.639913 4.114760 4.775549 3.991641\nAFFX-MurIL2_at              5.402049  4.970754 5.136504 5.328741 4.934468\nAFFX-MurIL4_at              3.088365  2.912783 2.941528 3.123581 3.259221\nAFFX-PheX-3_at              5.073080  5.539796 5.222797 5.668133 5.199665\nAFFX-PheX-5_at              3.050993  3.116976 3.301066 3.366936 3.008996\nAFFX-PheX-M_at              3.344718  3.249449 3.179357 3.343461 2.962293\nAFFX-ThrX-3_at              4.365777  4.559372 4.309676 4.808575 4.120906\nAFFX-ThrX-5_at              4.127825  4.452035 4.157649 4.205917 3.899734\nAFFX-ThrX-M_at              3.469720  3.243264 3.228969 3.399563 3.027191\nAFFX-TrpnX-3_at             2.977916  2.650942 2.751669 2.960073 2.806713\nAFFX-TrpnX-5_at             3.623308  3.621928 3.497549 3.834567 3.485573\nAFFX-TrpnX-M_at             3.416213  3.344428 3.250417 3.589504 3.140334\nAFFX-YEL002c/WBP1_at        3.281261  2.997496 2.888443 3.050276 3.037369\nAFFX-YEL018w/_at            3.216799  2.978343 3.145237 3.098193 3.292282\nAFFX-YEL021w/URA3_at        3.395100  3.154582 3.076809 3.134195 5.943431\nAFFX-YEL024w/RIP1_at        3.805801  3.092056 3.428454 3.336388 4.043344\n                            04008     04010     04016     06002    08001     ⋯\n1000_at                      7.065914  7.474537  7.536119 7.183331  7.735545 ⋯\n1001_at                      5.147762  5.122518  5.016132 5.288943  4.633217 ⋯\n1002_f_at                    3.945869  4.150506  3.576360 3.900935  3.630190 ⋯\n1003_s_at                    6.208061  6.292713  5.665991 5.842326  5.875375 ⋯\n1004_at                      5.923487  6.046607  5.738218 5.994515  5.748350 ⋯\n1005_at                     10.063484 10.662059 11.269115 8.812869 10.165159 ⋯\n1006_at                      3.771648  3.747963  3.318315 3.874648  3.525401 ⋯\n1007_s_at                    7.726716  7.288960  7.724153 6.763490  7.543907 ⋯\n1008_f_at                    9.424092  8.999938  8.006848 9.190553  8.408789 ⋯\n1009_at                      9.189420  9.312164  9.946430 9.112255  9.577024 ⋯\n100_g_at                     7.984250  8.120029  8.084865 7.697962  8.302688 ⋯\n1010_at                      5.484939  5.840837  5.067395 5.237119  5.273373 ⋯\n1011_s_at                    6.372048  7.252254  6.674473 6.883788  5.714778 ⋯\n1012_at                      3.948326  3.676359  3.499718 3.439177  3.883105 ⋯\n1013_at                      3.851362  3.565626  2.851111 3.070907  2.969788 ⋯\n1014_at                      7.475994  7.527383  7.550745 7.119259  7.594602 ⋯\n1015_s_at                    5.696765  5.917831  5.535112 5.723400  5.569632 ⋯\n1016_s_at                    3.207662  3.229092  3.210501 3.353474  3.176632 ⋯\n1017_at                      4.431900  3.931028  5.129296 3.956742  5.178051 ⋯\n1018_at                      6.032982  6.095990  5.903504 6.231557  5.861168 ⋯\n1019_g_at                    6.370988  6.262760  6.373234 6.648472  6.306895 ⋯\n101_at                       5.820837  6.319743  5.794677 6.079454  6.109053 ⋯\n1020_s_at                    7.520384  7.608336  7.262121 8.059713  7.466821 ⋯\n1021_at                      4.433935  4.690830  4.394778 4.567002  6.617756 ⋯\n1022_f_at                    3.647198  3.372396  3.561139 3.991485  3.716743 ⋯\n1023_at                      5.773668  5.873469  5.490342 5.908396  5.505490 ⋯\n1024_at                      5.846395  6.022922  5.760028 5.644638  5.556812 ⋯\n1025_g_at                    4.400837  4.441394  4.134673 4.423239  4.329983 ⋯\n1026_s_at                    6.918602  6.920326  6.554345 6.482749  6.386876 ⋯\n1027_at                      5.269626  5.492094  4.989486 5.590867  5.481647 ⋯\n⋮                           ⋮         ⋮         ⋮         ⋮        ⋮         ⋱\nAFFX-HUMISGF3A/M97935_MB_at 6.720742  5.732256  5.522468  5.372603 5.180712  ⋯\nAFFX-HUMRGE/M10098_3_at     5.463824  6.631424  4.940925  5.170254 5.346651  ⋯\nAFFX-HUMRGE/M10098_5_at     4.768099  5.878260  4.273090  4.480622 4.463458  ⋯\nAFFX-HUMRGE/M10098_M_at     3.889413  5.009262  4.300358  4.096791 4.356831  ⋯\nAFFX-HUMTFRR/M11507_3_at    5.446088  5.385702  4.829268  4.887564 4.991675  ⋯\nAFFX-HUMTFRR/M11507_5_at    4.279529  4.921189  3.904052  3.927687 3.587829  ⋯\nAFFX-HUMTFRR/M11507_M_at    4.929152  5.109563  4.284558  4.130364 4.384626  ⋯\nAFFX-LysX-3_at              2.817815  2.833962  2.778187  3.074082 3.102063  ⋯\nAFFX-LysX-5_at              2.782114  2.889998  2.701625  3.077016 2.917622  ⋯\nAFFX-LysX-M_at              4.562960  4.803757  4.608499  4.800184 4.294252  ⋯\nAFFX-M27830_3_at            7.277769  7.500706  7.350631  7.515679 6.974698  ⋯\nAFFX-M27830_5_at            5.683872  7.351909  5.496411  4.986038 5.621757  ⋯\nAFFX-M27830_M_at            7.933779  8.444469  7.517100  7.451483 7.695028  ⋯\nAFFX-MurFAS_at              3.483763  3.323103  3.446710  3.440076 3.273991  ⋯\nAFFX-MurIL10_at             4.627252  4.530767  4.127038  4.370285 3.835914  ⋯\nAFFX-MurIL2_at              5.036166  4.914641  4.886392  5.372676 5.018007  ⋯\nAFFX-MurIL4_at              3.037260  3.003451  2.973277  3.320805 2.784965  ⋯\nAFFX-PheX-3_at              5.470518  5.506018  4.931751  5.180675 5.256852  ⋯\nAFFX-PheX-5_at              3.125731  2.902621  3.074890  3.364359 3.217555  ⋯\nAFFX-PheX-M_at              3.045536  3.029851  3.068849  3.355382 3.301770  ⋯\nAFFX-ThrX-3_at              4.423895  4.166162  4.301804  4.439963 4.102016  ⋯\nAFFX-ThrX-5_at              4.104524  3.994176  4.076177  4.530618 4.142404  ⋯\nAFFX-ThrX-M_at              3.295530  3.257909  2.994594  3.525741 3.232638  ⋯\nAFFX-TrpnX-3_at             2.962976  3.152471  2.695692  2.920628 2.844983  ⋯\nAFFX-TrpnX-5_at             3.832459  3.657211  3.685598  3.863668 3.474924  ⋯\nAFFX-TrpnX-M_at             3.397393  3.272373  3.290285  3.638814 3.343970  ⋯\nAFFX-YEL002c/WBP1_at        2.904478  3.065646  3.357285  2.989665 2.933019  ⋯\nAFFX-YEL018w/_at            3.204104  3.361988  2.930769  3.213673 3.039339  ⋯\nAFFX-YEL021w/URA3_at        3.084158  3.246510  3.092988  3.386872 4.499956  ⋯\nAFFX-YEL024w/RIP1_at        3.470484  3.375992  3.341414  3.785948 3.618562  ⋯\n                            37001     43006    43015    44001    49004   \n1000_at                      7.849455 7.960842 8.188617 7.399999 7.813474\n1001_at                      4.959450 4.537677 5.154500 5.071885 4.874525\n1002_f_at                    4.047428 3.694877 3.949546 3.837517 3.767672\n1003_s_at                    6.193790 5.973162 6.033704 5.560886 5.930329\n1004_at                      5.824930 5.841095 5.870163 5.686555 5.793403\n1005_at                     10.408645 7.524865 8.381740 8.055760 7.234242\n1006_at                      3.670568 3.652459 3.707005 3.852594 3.527330\n1007_s_at                    6.609299 6.482215 6.651151 6.463802 7.083294\n1008_f_at                    8.561828 8.765609 8.459983 8.611438 8.911405\n1009_at                      9.492835 9.237562 9.183180 9.250913 9.609640\n100_g_at                     8.003010 7.836479 8.007742 7.825349 7.867227\n1010_at                      5.058770 4.848082 5.330210 4.881275 5.180372\n1011_s_at                    6.434650 5.573140 5.360469 6.067903 5.246391\n1012_at                      3.563572 3.877966 3.393784 3.173837 3.760740\n1013_at                      2.799112 3.335830 2.823894 3.162021 2.899012\n1014_at                      7.247758 7.442379 7.125079 7.846200 7.571595\n1015_s_at                    5.733726 5.487096 5.628726 5.473225 5.499078\n1016_s_at                    3.498991 3.250506 3.176352 3.924138 3.245476\n1017_at                      4.483604 5.133695 4.789429 4.478413 5.773424\n1018_at                      6.519862 6.104468 6.013004 6.493142 6.333817\n1019_g_at                    6.673348 6.658479 6.402795 6.766007 6.851714\n101_at                       6.953927 6.442707 6.347830 6.134284 6.043279\n1020_s_at                    8.241378 7.422417 7.080051 8.689273 7.759317\n1021_at                      5.006477 4.117340 4.883207 4.955042 4.716777\n1022_f_at                    4.163488 3.741596 3.567559 4.022900 3.789003\n1023_at                      5.829125 5.414204 5.776841 5.269463 5.700922\n1024_at                      5.741901 5.466854 5.805061 5.698824 5.723203\n1025_g_at                    4.797889 4.449505 4.716811 4.479158 4.614610\n1026_s_at                    6.504184 6.252522 6.457230 6.678333 6.413700\n1027_at                      5.288263 5.305203 5.329512 5.301670 5.369972\n⋮                           ⋮         ⋮        ⋮        ⋮        ⋮       \nAFFX-HUMISGF3A/M97935_MB_at 5.130053  4.951393 4.903532 6.024029 5.088734\nAFFX-HUMRGE/M10098_3_at     5.322311  5.001757 5.678911 4.983021 5.542517\nAFFX-HUMRGE/M10098_5_at     4.578618  4.281334 4.793171 4.632098 4.498728\nAFFX-HUMRGE/M10098_M_at     3.903118  3.785432 4.225459 4.089147 4.153007\nAFFX-HUMTFRR/M11507_3_at    5.873119  5.411731 5.477222 4.733239 5.522148\nAFFX-HUMTFRR/M11507_5_at    4.334639  3.741585 3.954517 3.947313 3.975845\nAFFX-HUMTFRR/M11507_M_at    4.673340  4.308101 4.520912 4.291467 4.670833\nAFFX-LysX-3_at              3.190449  2.961977 4.000543 3.625823 2.987288\nAFFX-LysX-5_at              2.919766  2.870377 4.197757 3.199422 2.789884\nAFFX-LysX-M_at              4.721945  4.430331 5.599868 5.334023 4.450242\nAFFX-M27830_3_at            7.535924  6.951264 7.118890 7.893386 7.567082\nAFFX-M27830_5_at            5.134773  4.677474 6.280724 5.883414 5.861596\nAFFX-M27830_M_at            8.124331  7.435762 8.133971 7.856379 7.873945\nAFFX-MurFAS_at              3.537291  3.254407 3.875097 3.444693 3.743439\nAFFX-MurIL10_at             4.321451  3.974078 4.718328 4.095881 4.001305\nAFFX-MurIL2_at              5.358090  4.822789 5.320897 5.489717 5.181606\nAFFX-MurIL4_at              3.338618  2.875561 3.795054 3.426576 3.092319\nAFFX-PheX-3_at              5.056564  4.865700 5.419945 5.860424 4.921566\nAFFX-PheX-5_at              3.293791  3.222073 4.072204 3.646259 3.018502\nAFFX-PheX-M_at              3.215089  3.157067 3.836710 3.774872 2.950347\nAFFX-ThrX-3_at              4.302587  4.304613 4.927189 4.667565 4.370202\nAFFX-ThrX-5_at              4.553950  4.168382 4.797841 4.485632 4.261370\nAFFX-ThrX-M_at              3.458331  3.348805 4.113437 3.747689 3.395225\nAFFX-TrpnX-3_at             2.937670  2.850850 3.582664 3.059061 2.876187\nAFFX-TrpnX-5_at             3.830677  3.416041 4.315469 3.965705 3.807634\nAFFX-TrpnX-M_at             3.117524  3.253874 3.971773 3.438820 3.380928\nAFFX-YEL002c/WBP1_at        3.048893  3.005741 3.767146 3.256931 3.065277\nAFFX-YEL018w/_at            3.209500  2.973227 3.825889 3.129444 3.223136\nAFFX-YEL021w/URA3_at        3.776172  3.063341 5.634363 3.716873 3.038208\nAFFX-YEL024w/RIP1_at        3.944137  3.503261 4.276009 3.813406 3.537594\n                            56007    64005     65003    83001    LAL4    \n1000_at                     7.816922  7.913249 7.800199 8.030047 7.702217\n1001_at                     4.788699  5.403640 5.443827 5.178633 5.029670\n1002_f_at                   4.248075  4.119252 3.957468 4.053458 4.002306\n1003_s_at                   5.878533  6.193376 6.006443 6.050129 6.297549\n1004_at                     5.913503  6.053053 5.549087 5.932412 5.827131\n1005_at                     8.083834 10.163355 8.184442 8.798521 7.687370\n1006_at                     3.429451  3.756402 3.571963 3.585416 3.674371\n1007_s_at                   7.107712  6.895677 6.926471 6.861161 6.731454\n1008_f_at                   9.043327  8.284549 8.569874 8.474002 8.539480\n1009_at                     9.474761  9.702681 9.110240 9.884863 9.575421\n100_g_at                    8.009918  8.271588 8.169891 7.897170 7.965106\n1010_at                     5.310220  5.237757 4.993497 5.385707 4.982707\n1011_s_at                   4.942719  5.109609 6.490558 6.023058 5.809493\n1012_at                     4.148892  4.065844 3.762194 3.735753 3.664340\n1013_at                     3.008075  3.153419 2.902942 3.018644 3.127662\n1014_at                     7.931793  7.654051 7.414572 7.505681 7.532083\n1015_s_at                   5.398977  5.688522 5.753532 5.555595 5.721859\n1016_s_at                   3.476038  3.285540 3.392091 3.274649 3.653479\n1017_at                     5.111104  4.185028 4.268057 4.030064 3.826977\n1018_at                     6.164413  6.674320 6.566802 6.095361 6.425974\n1019_g_at                   6.696270  7.033197 7.395795 6.356240 6.963269\n101_at                      6.572743  6.666000 6.729782 6.750178 6.789686\n1020_s_at                   7.530735  8.074391 8.082602 8.028259 7.878384\n1021_at                     4.750484  4.891067 4.548111 4.481377 4.397078\n1022_f_at                   3.831971  4.111677 3.939289 3.707847 4.329154\n1023_at                     5.956093  5.774114 5.858850 5.637539 5.408432\n1024_at                     5.847403  5.870010 5.765627 5.834069 5.874103\n1025_g_at                   4.620035  4.757783 4.474032 4.657209 4.869916\n1026_s_at                   6.595883  6.723828 6.753657 6.671640 6.627980\n1027_at                     5.494767  5.680495 5.259128 5.640146 5.544918\n⋮                           ⋮        ⋮         ⋮        ⋮        ⋮       \nAFFX-HUMISGF3A/M97935_MB_at 5.227949 4.962942  5.180621 4.803131 6.137956\nAFFX-HUMRGE/M10098_3_at     5.852424 5.379777  5.232322 5.091796 5.179363\nAFFX-HUMRGE/M10098_5_at     4.831378 4.546390  4.436946 4.290367 4.529884\nAFFX-HUMRGE/M10098_M_at     4.442808 3.981419  3.888398 3.656073 3.876826\nAFFX-HUMTFRR/M11507_3_at    5.399388 5.445041  4.801405 5.720164 5.498475\nAFFX-HUMTFRR/M11507_5_at    4.093276 4.004460  3.772255 3.974060 3.898813\nAFFX-HUMTFRR/M11507_M_at    4.634246 4.507426  4.387551 4.546331 4.331859\nAFFX-LysX-3_at              3.396457 2.801641  2.890539 2.908193 3.182116\nAFFX-LysX-5_at              3.258625 2.811507  3.042925 2.930631 3.087345\nAFFX-LysX-M_at              4.750099 4.506028  4.637085 4.539593 4.936381\nAFFX-M27830_3_at            7.190904 7.450166  7.587643 6.957847 7.522223\nAFFX-M27830_5_at            5.551034 5.602937  4.815696 5.128495 5.615701\nAFFX-M27830_M_at            8.304769 8.277190  7.933101 7.774132 7.876990\nAFFX-MurFAS_at              3.709902 3.233520  3.352985 3.165264 3.268459\nAFFX-MurIL10_at             4.763165 4.161259  3.959749 3.975969 4.193781\nAFFX-MurIL2_at              5.365144 5.194208  5.160640 4.844565 4.967381\nAFFX-MurIL4_at              3.577267 3.058879  3.168797 3.028462 3.014841\nAFFX-PheX-3_at              5.486537 5.354748  5.261869 5.249353 5.188049\nAFFX-PheX-5_at              3.860386 3.073433  3.408542 3.086195 3.374656\nAFFX-PheX-M_at              3.884731 3.195110  3.741395 3.254390 3.418817\nAFFX-ThrX-3_at              5.101768 4.440123  4.675419 4.094304 4.474966\nAFFX-ThrX-5_at              4.831275 4.366302  4.738456 4.194136 4.555346\nAFFX-ThrX-M_at              4.110112 3.436514  3.479872 3.160448 3.500401\nAFFX-TrpnX-3_at             3.408258 2.809911  2.771741 2.775778 2.780948\nAFFX-TrpnX-5_at             4.254020 3.711537  3.701861 3.753496 3.701939\nAFFX-TrpnX-M_at             3.753142 3.462627  3.551272 3.114569 3.253312\nAFFX-YEL002c/WBP1_at        3.120296 2.958656  3.001848 2.802044 3.009121\nAFFX-YEL018w/_at            3.586048 3.417289  3.273183 3.248411 3.095670\nAFFX-YEL021w/URA3_at        3.380032 5.568760  3.175193 4.309682 3.342961\nAFFX-YEL024w/RIP1_at        3.610839 4.388177  3.779012 3.631034 3.842535"},"metadata":{}}],"execution_count":94},{"cell_type":"code","source":"# Compute mean expression for sample 5 (i.e., column 5)\nmean_sample_5 = mean(expr_matrix[, 5], na.rm = TRUE)\n\n# Show result\nmean_sample_5","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:25:21.303656Z","iopub.execute_input":"2025-04-13T13:25:21.305270Z","iopub.status.idle":"2025-04-13T13:25:21.328650Z","shell.execute_reply":"2025-04-13T13:25:21.326133Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"5.62962704512381","text/markdown":"5.62962704512381","text/latex":"5.62962704512381","text/plain":"[1] 5.629627"},"metadata":{}}],"execution_count":95},{"cell_type":"code","source":"# Extract probe IDs (Affymetrix IDs)\nprobes = featureNames(ALL)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:28:52.681406Z","iopub.execute_input":"2025-04-13T13:28:52.683049Z","iopub.status.idle":"2025-04-13T13:28:52.696127Z","shell.execute_reply":"2025-04-13T13:28:52.694386Z"}},"outputs":[],"execution_count":96},{"cell_type":"code","source":"# Connect to Ensembl 75 (hg19 build)\nensembl75 = useMart(\n  host = \"feb2014.archive.ensembl.org\",\n  biomart = \"ENSEMBL_MART_ENSEMBL\"\n)\n\n# Choose the correct dataset for humans\nensembl75 = useDataset(\"hsapiens_gene_ensembl\", mart = ensembl75)\n\n# Annotate: Map Affymetrix probe IDs to Ensembl Gene IDs\nannot = getBM(\n  attributes = c(\"affy_hg_u95av2\", \"ensembl_gene_id\"),\n  filters = \"affy_hg_u95av2\",\n  values = probes,\n  mart = ensembl75\n)\n\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:29:42.309000Z","iopub.execute_input":"2025-04-13T13:29:42.310579Z","iopub.status.idle":"2025-04-13T13:29:48.946630Z","shell.execute_reply":"2025-04-13T13:29:48.944685Z"}},"outputs":[{"name":"stderr","text":"Warning message:\n“Ensembl will soon enforce the use of https.\nEnsure the 'host' argument includes \"https://\"”\n","output_type":"stream"}],"execution_count":97},{"cell_type":"code","source":"# Remove rows with missing annotations\nannot <- na.omit(annot)\n\n# Count how many probes map to more than one Ensembl gene\nmulti_map_count <- sum(duplicated(annot))\n\n# Or more clearly:\ntable_counts <- table(annot$affy_hg_u95av2)\nmulti_gene_probes <- sum(table_counts > 1)\n\n# Output the result\ncat(\"Number of probesets annotated with more than one Ensembl gene ID:\", multi_gene_probes, \"\\n\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:30:14.013046Z","iopub.execute_input":"2025-04-13T13:30:14.015012Z","iopub.status.idle":"2025-04-13T13:30:14.324668Z","shell.execute_reply":"2025-04-13T13:30:14.322628Z"}},"outputs":[{"name":"stdout","text":"Number of probesets annotated with more than one Ensembl gene ID: 1045 \n","output_type":"stream"}],"execution_count":98},{"cell_type":"code","source":"# Choose human gene dataset\nensembl75 <- useDataset(\"hsapiens_gene_ensembl\", mart = ensembl75)\n\n# Retrieve annotations including chromosome\nannot <- getBM(\n  attributes = c(\"affy_hg_u95av2\", \"ensembl_gene_id\", \"chromosome_name\"),\n  filters = \"affy_hg_u95av2\",\n  values = probes,\n  mart = ensembl75\n)\n\n# Filter only for autosomal chromosomes 1 through 22\nautosomes <- as.character(1:22)\nannot_autosomes <- annot[annot$chromosome_name %in% autosomes, ]\n\n# Count unique probesets with at least one autosomal gene annotation\nunique_probes_autosomes <- unique(annot_autosomes$affy_hg_u95av2)\nnum_probes_autosomes <- length(unique_probes_autosomes)\n\n# Output the result\ncat(\"Number of probesets annotated with one or more genes on autosomes (chr 1–22):\", num_probes_autosomes, \"\\n\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:33:18.378971Z","iopub.execute_input":"2025-04-13T13:33:18.380660Z","iopub.status.idle":"2025-04-13T13:33:25.260388Z","shell.execute_reply":"2025-04-13T13:33:25.257939Z"}},"outputs":[{"name":"stdout","text":"Number of probesets annotated with one or more genes on autosomes (chr 1–22): 11016 \n","output_type":"stream"}],"execution_count":99},{"cell_type":"code","source":"# Install and load minfiData (if not already installed)\nif (!requireNamespace(\"minfiData\", quietly = TRUE)) {\n  BiocManager::install(\"minfiData\")\n}\nlibrary(minfiData)\nlibrary(minfi)\n\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:46:18.219296Z","iopub.execute_input":"2025-04-13T13:46:18.220965Z","iopub.status.idle":"2025-04-13T13:51:48.304214Z","shell.execute_reply":"2025-04-13T13:51:48.302404Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'minfiData'\n\nalso installing the dependencies ‘illuminaio’, ‘HDF5Array’, ‘minfi’, ‘IlluminaHumanMethylation450kmanifest’, ‘IlluminaHumanMethylation450kanno.ilmn12.hg19’\n\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'AID', 'akc', 'ale',\n  'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'antiword', 'anytime', 'ape',\n  'aplot', 'apollo', 'aqp', 'arrow', 'ars', 'ARTool', 'arules', 'AsioHeaders',\n  'askpass', 'ASMap', 'astsa', 'automap', 'baguette', 'BAS', 'base64',\n  'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr', 'beepr',\n  'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD', 'bigstatsr',\n  'bigstep', 'billboarder', 'bindr', 'BiocManager', 'BiodiversityR', 'biomod2',\n  'biotools', 'bipartite', 'bit', 'bit64', 'bitops', 'blme', 'blorr', 'BMA',\n  'BMisc', 'bnlearn', 'bonsai', 'bookdown', 'boot', 'brms', 'broom',\n  'broom.helpers', 'broom.mixed', 'brulee', 'bslib', 'bundle', 'butcher',\n  'C50', 'car', 'caret', 'caretEnsemble', 'caTools', 'cba', 'censored',\n  'changepoint', 'cheapr', 'checkmate', 'chk', 'CholWishart', 'choroplethr',\n  'chromote', 'chron', 'circular', 'class', 'classInt', 'cld2', 'cleaner',\n  'cli', 'clock', 'clue', 'cluster', 'clusterSim', 'cmsafops',\n  'coarseDataTools', 'cobs', 'collapse', 'colorspace', 'commonmark',\n  'compareGroups', 'Compind', 'contentid', 'copula', 'CORElearn', 'coro',\n  'correlation', 'corrplot', 'countrycode', 'cpp11', 'cppdoubles',\n  'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature', 'Cubist',\n  'curl', 'cutpointr', 'cvms', 'data.table', 'datamods', 'datasauRus',\n  'datawizard', 'datetimeutils', 'dbarts', 'dbscan', 'dendextend', 'DEoptimR',\n  'Deriv', 'DescTools', 'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'digest',\n  'dismo', 'distances', 'distillery', 'distr', 'distrEx', 'distributional',\n  'DistributionUtils', 'dlm', 'dm', 'doBy', 'docopt', 'doFuture', 'doRNG',\n  'DoseFinding', 'dotCall64', 'dotwhisker', 'DoubleML', 'downloader', 'dplR',\n  'drat', 'DRDID', 'dtwclust', 'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat',\n  'ecp', 'effectsize', 'eha', 'embed', 'emmeans', 'energy', 'entropy',\n  'EnvStats', 'Epi', 'epiR', 'erer', 'ergm', 'ergm.multi', 'esquisse',\n  'estimatr', 'etm', 'evaluate', 'evd', 'Exact', 'explore', 'expm', 'extRemes',\n  'fable', 'fabletools', 'faraway', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastverse', 'fBasics', 'fda', 'fda.usc', 'fdapace', 'fdrtool',\n  'feasts', 'ff', 'fftw', 'FielDHub', 'fields', 'filehashSQLite', 'finalfit',\n  'fitdistrplus', 'FKF', 'flare', 'flexclust', 'flexmix', 'flexsurv',\n  'flextable', 'float', 'FNN', 'fontawesome', 'forecast', 'foreign',\n  'forestplot', 'fpc', 'fpp3', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa',\n  'future', 'future.apply', 'gam', 'gamboostLSS', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'genridge', 'GenSA', 'geobr', 'geodist', 'geojsonR',\n  'geometry', 'geomorph', 'geomtextpath', 'geosphere', 'gert', 'GetBCBData',\n  'ggdag', 'ggeasy', 'ggeffects', 'ggfun', 'ggh4x', 'ggiraph', 'ggmap',\n  'ggnewscale', 'ggpath', 'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc',\n  'ggrepel', 'ggstats', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmtoolbox', 'glue', 'gmapsdistance', 'gmp',\n  'GoFKernel', 'golem', 'googlePolylines', 'gower', 'GPArotation', 'GPfit',\n  'gplots', 'grafify', 'gRain', 'graphlayouts', 'gRbase', 'greybox', 'grf',\n  'gridpattern', 'grImport2', 'groupdata2', 'grpreg', 'gsheet', 'gss', 'gstat',\n  'gsw', 'gt', 'gtable', 'gtsummary', 'GWalkR', 'GWmodel', 'h3lib', 'h3r',\n  'hardhat', 'harmony', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'htmlTable', 'hts', 'httr2', 'hunspell', 'huxtable',\n  'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph', 'ijtiff', 'imager', 'iml',\n  'inline', 'insight', 'inspectdf', 'interactions', 'intervals', 'ipred',\n  'isotree', 'ISwR', 'ivreg', 'janitor', 'jiebaR', 'JMbayes2', 'jmvconnect',\n  'jmvcore', 'jmvReadWrite', 'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity',\n  'keras3', 'kernelshap', 'kernlab', 'KernSmooth', 'khroma', 'kit', 'knitr',\n  'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman', 'lares', 'later',\n  'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave', 'lcmm', 'leafgl',\n  'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR', 'lightgbm', 'limma', 'lintr',\n  'littler', 'lme4', 'lmodel2', 'lmom', 'locfit', 'logger', 'longitudinalData',\n  'loo', 'lordif', 'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa',\n  'lubridate', 'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'matlib', 'Matrix', 'MatrixModels',\n  'matrixStats', 'MazamaCoreUtils', 'MazamaSpatialUtils', 'MBA', 'mboost',\n  'MCMCpack', 'mda', 'mdsr', 'memisc', 'meta', 'metacoder', 'metadat',\n  'metafor', 'meteospain', 'MetricsWeighted', 'mets', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'minqa', 'minty', 'mirt',\n  'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench', 'mlflow',\n  'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data', 'mlr3filters',\n  'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures', 'mlr3misc',\n  'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse', 'mlr3viz',\n  'modelbased', 'modelenv', 'modelsummary', 'modeltime', 'moderndive',\n  'mongolite', 'monitoR', 'msigdbr', 'msm', 'mstate', 'mtsdi', 'multcomp',\n  'MuMIn', 'mvtnorm', 'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson',\n  'ndtv', 'network', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'OpenMx', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ore', 'osmextract',\n  'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown', 'pammtools',\n  'pander', 'parabar', 'paradox', 'parallelly', 'parameters', 'ParamHelpers',\n  'paran', 'parsnip', 'party', 'partykit', 'patchwork', 'pbdZMQ', 'pcaPP',\n  'pdftools', 'pdp', 'pdR', 'PearsonDS', 'performance', 'PerformanceAnalytics',\n  'pgmm', 'phangorn', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools', 'pls',\n  'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip', 'popdemo',\n  'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'PreProcess', 'prettyGraphs', 'priceR', 'processx', 'profvis',\n  'progressr', 'PROJ', 'proj4', 'promises', 'prospectr', 'protolite', 'PRROC',\n  'ps', 'pspline', 'PSweight', 'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf',\n  'qs', 'qtl', 'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr',\n  'QuickJSR', 'qvcalc', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'reactR',\n  'readODS', 'readxl', 'rearrr', 'rebmix', 'recipes', 'refund', 'renv',\n  'report', 'reprex', 'reticulate', 'Rfast', 'rfishbase', 'rgbif', 'rgenoud',\n  'rgl', 'rio', 'rjson', 'RJSONIO', 'rlang', 'RMariaDB', 'rmarkdown', 'rminer',\n  'rms', 'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpostgis', 'RPostgres', 'RPostgreSQL',\n  'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample', 'rsconnect',\n  'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite', 'Rssa',\n  'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rugarch', 'Rvcg', 'rvg',\n  's2', 'sandwich', 'sass', 'SCtools', 'sda', 'sdm', 'seasonal',\n  'seasonalview', 'see', 'seededlda', 'segmented', 'sem', 'semTools',\n  'sensemakr', 'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinyWidgets', 'SimDesign',\n  'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot', 'skmeans', 'slam',\n  'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2', 'sna', 'sodium',\n  'soilDB', 'sommer', 'sp', 'spacefillr', 'spacetime', 'spam', 'sparklyr',\n  'SparseM', 'spatialreg', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'strucchange', 'summarytools',\n  'SuppDists', 'survival', 'survminer', 'svd', 'sys', 'systemfonts', 'tables',\n  'tau', 'taxize', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract',\n  'testthat', 'texreg', 'text', 'textcat', 'textrecipes', 'textshaping',\n  'textutils', 'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'timeDate', 'timeplyr', 'timereg', 'timeSeries', 'tinytable',\n  'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui', 'topicmodels',\n  'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal', 'tsbox', 'tsDyn',\n  'tseries', 'tsibble', 'tuber', 'tune', 'tzdb', 'units', 'univariateML',\n  'unpivotr', 'usethis', 'usmapdata', 'uuid', 'uwot', 'V8', 'vcd', 'vegan',\n  'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR', 'wbacon',\n  'WDI', 'websocket', 'webutils', 'WeightSVM', 'withr', 'wk', 'wooldridge',\n  'workflows', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick',\n  'yulab.utils', 'zip', 'zoo', 'spatial'\n\nLoading required package: minfi\n\nLoading required package: Biostrings\n\nLoading required package: XVector\n\n\nAttaching package: ‘Biostrings’\n\n\nThe following object is masked from ‘package:base’:\n\n    strsplit\n\n\nLoading required package: bumphunter\n\nLoading required package: foreach\n\nLoading required package: iterators\n\nLoading required package: parallel\n\nLoading required package: locfit\n\nlocfit 1.5-9.10 \t 2024-06-24\n\nLoading required package: IlluminaHumanMethylation450kmanifest\n\nLoading required package: IlluminaHumanMethylation450kanno.ilmn12.hg19\n\n","output_type":"stream"}],"execution_count":101},{"cell_type":"code","source":"# Load the MsetEx dataset\ndata(MsetEx)\n\n# Check sample names to ensure the target sample is present\nsampleNames(MsetEx)\n\n# Extract the methylated signal matrix\nmeth_matrix <- getMeth(MsetEx)\n\n# Confirm target sample exists\n\"5723646052_R04C01\" %in% colnames(meth_matrix)\n\n# Calculate the mean methylation value across all features for the target sample\nmean_methylation <- mean(meth_matrix[, \"5723646052_R04C01\"], na.rm = TRUE)\n\n# Output the result\ncat(\"Mean methylation value for sample 5723646052_R04C01:\", mean_methylation, \"\\n\")\n","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"# Download the GEO dataset (GSE788)\ngse <- getGEO(\"GSE788\", GSEMatrix = TRUE)\n\n","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"# If it returns a list, extract the first element\ngse_data <- gse[[1]]\n\n# View the sample names\nsampleNames(gse_data)\n\n# Extract the expression matrix\nexpr_matrix <- exprs(gse_data)\n\n# Check if sample GSM9024 is present\n\"GSM9024\" %in% colnames(expr_matrix)\n\n# Compute the mean expression level for GSM9024\nmean_expr <- mean(expr_matrix[, \"GSM9024\"], na.rm = TRUE)\n\n# Output the result\ncat(\"Mean expression level of sample GSM9024:\", mean_expr, \"\\n\")","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"experimentData(ALL)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T10:52:01.172580Z","iopub.execute_input":"2025-04-13T10:52:01.174171Z","iopub.status.idle":"2025-04-13T10:52:01.189148Z","shell.execute_reply":"2025-04-13T10:52:01.187359Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"Experiment data\n  Experimenter name: Chiaretti et al. \n  Laboratory: Department of Medical Oncology, Dana-Farber Cancer Institute, Department of Medicine, Brigham and Women's Hospital, Harvard Medical School, Boston, MA 02115, USA. \n  Contact information:  \n  Title: Gene expression profile of adult T-cell acute lymphocytic leukemia identifies distinct subsets of patients with different response to therapy and survival. \n  URL:  \n  PMIDs: 14684422 16243790 \n\n  Abstract: A 187 word abstract is available. Use 'abstract' method."},"metadata":{}}],"execution_count":9},{"cell_type":"code","source":"?ALL ","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T10:52:54.052098Z","iopub.execute_input":"2025-04-13T10:52:54.053827Z","iopub.status.idle":"2025-04-13T10:52:54.246172Z","shell.execute_reply":"2025-04-13T10:52:54.243527Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"output_type":"display_data","data":{"text/html":"\n<table style=\"width: 100%;\"><tr><td>ALL {ALL}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n\n<h2>Acute Lymphoblastic Leukemia Data from the Ritz Laboratory</h2>\n\n<h3>Description</h3>\n\n<p>The data consist of microarrays from 128 different individuals with \nacute lymphoblastic leukemia (ALL). A number of additional covariates\nare available. The data have been normalized (using rma) and it is\nthe jointly normalized data that are available here. The data are\npresented in the form of an <code>exprSet</code> object.\n</p>\n\n\n<h3>Usage</h3>\n\n<pre><code class='language-R'>data(ALL)</code></pre>\n\n\n<h3>Format</h3>\n\n<p>The different covariates are:\n</p>\n\n<ul>\n<li> <p><code>cod</code>: The patient IDs.\n</p>\n</li>\n<li> <p><code>diagnosis</code> The date of diagnosis.\n</p>\n</li>\n<li> <p><code>sex</code>   The sex of the patient, coded as <code>M</code> and <code>F</code>.\n</p>\n</li>\n<li> <p><code>age</code>  The age of the patient in years.\n</p>\n</li>\n<li> <p><code>BT</code>  The type and stage of the disease; <code>B</code> indicates \nB-cell ALL while a <code>T</code> indicates T-cell ALL. \n</p>\n</li>\n<li> <p><code>remission</code> A factor with two levels, either <code>CR</code> indicate \nthat remission was achieved or <code>REF</code> indicating that the patient was\nrefractory and remission was not achieved. \n</p>\n</li>\n<li> <p><code>CR</code>  A vector with the following values: 1: &ldquo;CR&rdquo;, remission achieved; 2: &ldquo;DEATH IN CR&rdquo;, patient died while in remission; 3: &ldquo;DEATH IN INDUCTION&rdquo;, patient died while in induction therapy; 4: &ldquo;REF&rdquo;, patient was refractory to therapy. \n</p>\n</li>\n<li> <p><code>date.cr</code> The date on which remission was achieved.\n</p>\n</li>\n<li> <p><code>t(4;11)</code>  A logical vector indicating whether a t(4;11) \ntranslocation was detected.\n</p>\n</li>\n<li> <p><code>t(9;22)</code>  A logical vector indicating whether a t(9;22) \ntranslocation was detected.\n</p>\n</li>\n<li> <p><code>cyto.normal</code> A logical vector indicating whether the \ncytogenetics were normal.\n</p>\n</li>\n<li> <p><code>citog</code>  A vector indicating the various cytogenetic abnormalities\nthat were detected.\n</p>\n</li>\n<li> <p><code>mol.biol</code> The assigned molecular biology of the cancer (mainly \nfor those with B-cell ALL), BCR\\/ABL, ALL\\/AF4, E2APBX etc. \n</p>\n</li>\n<li> <p><code>fusion protein</code> For those with BCR\\/ABL which of the fusion \nproteins was detected,  <code>p190</code>, <code>p190\\/p210</code>, <code>p210</code>.\n</p>\n</li>\n<li> <p><code>mdr</code>   The patients response to multidrug resistance, either\n<code>NEG</code>, or <code>POS</code>.\n</p>\n</li>\n<li> <p><code>kinet</code>  ploidy, either diploid or hyperd.\n</p>\n</li>\n<li> <p><code>ccr</code>  A vector indicating whether the patient had continuous complete remission nor not. \n</p>\n</li>\n<li> <p><code>relapse</code>  A vector indicating whether the patient had relapse or not.\n</p>\n</li>\n<li> <p><code>transplant</code>  Did the patient receive a bone marrow transplant or\nnot.\n</p>\n</li>\n<li> <p><code>f.u</code> Follow-up data. The possible values are\n1: &ldquo;AUBMT \\/ REL&rdquo;: autologous bone marrow transplant and subsequent\nrelapse; 2: &ldquo;BMT \\/ CCR&rdquo;: allogeneic bone marrow transplant and \nstill in continuous complete remission; 3: &ldquo;BMT \\/ DEATH IN CR&rdquo;:\nafter allogeneic bone marrow transplant patient died without\nrelapsing; 4: &ldquo;BMT \\/ REL&rdquo;: after allogeneic bone marrow transplant patient relapsed; 5: &ldquo;CCR&rdquo;: patient was in \ncontinuous complete remission;  6: &ldquo;CCR \\/ OFF&rdquo;: patient was in \ncontinuous complete remission but off-protocol for some reasons;  \n7: &ldquo;DEATH IN CR&rdquo;: died when in complete remission; 8: &ldquo;MUD \\/ DEATH IN CR&rdquo;: unrelated allogeneic bone marrow transplant and \ndeath without relapsing; 9:  &ldquo;REL&rdquo;: relapse; 10:  &ldquo;REL \\/ SNC&rdquo;: relapse occurred at central nervous system.   \n</p>\n</li>\n<li> <p> date last seen Date the patient was last seen.\n</p>\n</li></ul>\n\n\n\n<h3>Source</h3>\n\n<p>Sabina Chiaretti, Xiaochun Li, Robert Gentleman, Antonella Vitale, Marco Vignetti, Franco Mandelli, Jerome Ritz, and Robin Foa Gene expression profile of adult T-cell acute lymphocytic leukemia identifies distinct subsets of patients with different response to therapy and survival. Blood, 1 April 2004, Vol. 103, No. 7.\n</p>\n\n\n<h3>Examples</h3>\n\n<pre><code class='language-R'>data(ALL)\n</code></pre>\n\n<hr /><div style=\"text-align: center;\">[Package <em>ALL</em> version 1.46.0 ]</div></main>\n\n</div>","text/latex":"\\inputencoding{utf8}\n\\HeaderA{ALL}{Acute Lymphoblastic Leukemia Data from the Ritz Laboratory}{ALL}\n\\keyword{datasets}{ALL}\n%\n\\begin{Description}\nThe data consist of microarrays from 128 different individuals with \nacute lymphoblastic leukemia (ALL). A number of additional covariates\nare available. The data have been normalized (using rma) and it is\nthe jointly normalized data that are available here. The data are\npresented in the form of an \\code{exprSet} object.\n\\end{Description}\n%\n\\begin{Usage}\n\\begin{verbatim}\ndata(ALL)\n\\end{verbatim}\n\\end{Usage}\n%\n\\begin{Format}\nThe different covariates are:\n\\begin{itemize}\n\n\\item{} \\code{cod}: The patient IDs.\n\\item{} \\code{diagnosis} The date of diagnosis.\n\\item{} \\code{sex}   The sex of the patient, coded as \\code{M} and \\code{F}.\n\\item{} \\code{age}  The age of the patient in years.\n\\item{} \\code{BT}  The type and stage of the disease; \\code{B} indicates \nB-cell ALL while a \\code{T} indicates T-cell ALL. \n\\item{} \\code{remission} A factor with two levels, either \\code{CR} indicate \nthat remission was achieved or \\code{REF} indicating that the patient was\nrefractory and remission was not achieved. \n\\item{} \\code{CR}  A vector with the following values: 1: ``CR'', remission achieved; 2: ``DEATH IN CR'', patient died while in remission; 3: ``DEATH IN INDUCTION'', patient died while in induction therapy; 4: ``REF'', patient was refractory to therapy. \n\\item{} \\code{date.cr} The date on which remission was achieved.\n\\item{} \\code{t(4;11)}  A logical vector indicating whether a t(4;11) \ntranslocation was detected.\n\\item{} \\code{t(9;22)}  A logical vector indicating whether a t(9;22) \ntranslocation was detected.\n\\item{} \\code{cyto.normal} A logical vector indicating whether the \ncytogenetics were normal.\n\\item{} \\code{citog}  A vector indicating the various cytogenetic abnormalities\nthat were detected.\n\\item{} \\code{mol.biol} The assigned molecular biology of the cancer (mainly \nfor those with B-cell ALL), BCR\\bsl{}/ABL, ALL\\bsl{}/AF4, E2APBX etc. \n\\item{} \\code{fusion protein} For those with BCR\\bsl{}/ABL which of the fusion \nproteins was detected,  \\code{p190}, \\code{p190\\bsl{}/p210}, \\code{p210}.\n\\item{} \\code{mdr}   The patients response to multidrug resistance, either\n\\code{NEG}, or \\code{POS}.\n\\item{} \\code{kinet}  ploidy, either diploid or hyperd.\n\\item{} \\code{ccr}  A vector indicating whether the patient had continuous complete remission nor not. \n\\item{} \\code{relapse}  A vector indicating whether the patient had relapse or not.\n\\item{} \\code{transplant}  Did the patient receive a bone marrow transplant or\nnot.\n\\item{} \\code{f.u} Follow-up data. The possible values are\n1: ``AUBMT \\bsl{}/ REL'': autologous bone marrow transplant and subsequent\nrelapse; 2: ``BMT \\bsl{}/ CCR'': allogeneic bone marrow transplant and \nstill in continuous complete remission; 3: ``BMT \\bsl{}/ DEATH IN CR'':\nafter allogeneic bone marrow transplant patient died without\nrelapsing; 4: ``BMT \\bsl{}/ REL'': after allogeneic bone marrow transplant patient relapsed; 5: ``CCR'': patient was in \ncontinuous complete remission;  6: ``CCR \\bsl{}/ OFF'': patient was in \ncontinuous complete remission but off-protocol for some reasons;  \n7: ``DEATH IN CR'': died when in complete remission; 8: ``MUD \\bsl{}/ DEATH IN CR'': unrelated allogeneic bone marrow transplant and \ndeath without relapsing; 9:  ``REL'': relapse; 10:  ``REL \\bsl{}/ SNC'': relapse occurred at central nervous system.   \n\\item{}  date last seen Date the patient was last seen.\n\n\\end{itemize}\n\n\\end{Format}\n%\n\\begin{Source}\nSabina Chiaretti, Xiaochun Li, Robert Gentleman, Antonella Vitale, Marco Vignetti, Franco Mandelli, Jerome Ritz, and Robin Foa Gene expression profile of adult T-cell acute lymphocytic leukemia identifies distinct subsets of patients with different response to therapy and survival. Blood, 1 April 2004, Vol. 103, No. 7.\n\\end{Source}\n%\n\\begin{Examples}\n\\begin{ExampleCode}\ndata(ALL)\n\\end{ExampleCode}\n\\end{Examples}","text/plain":"ALL                    package:ALL                     R Documentation\n\n_\bA_\bc_\bu_\bt_\be _\bL_\by_\bm_\bp_\bh_\bo_\bb_\bl_\ba_\bs_\bt_\bi_\bc _\bL_\be_\bu_\bk_\be_\bm_\bi_\ba _\bD_\ba_\bt_\ba _\bf_\br_\bo_\bm _\bt_\bh_\be _\bR_\bi_\bt_\bz _\bL_\ba_\bb_\bo_\br_\ba_\bt_\bo_\br_\by\n\n_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n\n     The data consist of microarrays from 128 different individuals\n     with acute lymphoblastic leukemia (ALL). A number of additional\n     covariates are available. The data have been normalized (using\n     rma) and it is the jointly normalized data that are available\n     here. The data are presented in the form of an ‘exprSet’ object.\n\n_\bU_\bs_\ba_\bg_\be:\n\n     data(ALL)\n     \n_\bF_\bo_\br_\bm_\ba_\bt:\n\n     The different covariates are:\n\n        • ‘cod’: The patient IDs.\n\n        • ‘diagnosis’ The date of diagnosis.\n\n        • ‘sex’ The sex of the patient, coded as ‘M’ and ‘F’.\n\n        • ‘age’ The age of the patient in years.\n\n        • ‘BT’ The type and stage of the disease; ‘B’ indicates B-cell\n          ALL while a ‘T’ indicates T-cell ALL.\n\n        • ‘remission’ A factor with two levels, either ‘CR’ indicate\n          that remission was achieved or ‘REF’ indicating that the\n          patient was refractory and remission was not achieved.\n\n        • ‘CR’ A vector with the following values: 1: “CR”, remission\n          achieved; 2: “DEATH IN CR”, patient died while in remission;\n          3: “DEATH IN INDUCTION”, patient died while in induction\n          therapy; 4: “REF”, patient was refractory to therapy.\n\n        • ‘date.cr’ The date on which remission was achieved.\n\n        • ‘t(4;11)’ A logical vector indicating whether a t(4;11)\n          translocation was detected.\n\n        • ‘t(9;22)’ A logical vector indicating whether a t(9;22)\n          translocation was detected.\n\n        • ‘cyto.normal’ A logical vector indicating whether the\n          cytogenetics were normal.\n\n        • ‘citog’ A vector indicating the various cytogenetic\n          abnormalities that were detected.\n\n        • ‘mol.biol’ The assigned molecular biology of the cancer\n          (mainly for those with B-cell ALL), BCR\\/ABL, ALL\\/AF4,\n          E2APBX etc.\n\n        • ‘fusion protein’ For those with BCR\\/ABL which of the fusion\n          proteins was detected, ‘p190’, ‘p190\\/p210’, ‘p210’.\n\n        • ‘mdr’ The patients response to multidrug resistance, either\n          ‘NEG’, or ‘POS’.\n\n        • ‘kinet’ ploidy, either diploid or hyperd.\n\n        • ‘ccr’ A vector indicating whether the patient had continuous\n          complete remission nor not.\n\n        • ‘relapse’ A vector indicating whether the patient had relapse\n          or not.\n\n        • ‘transplant’ Did the patient receive a bone marrow transplant\n          or not.\n\n        • ‘f.u’ Follow-up data. The possible values are 1: “AUBMT \\/\n          REL”: autologous bone marrow transplant and subsequent\n          relapse; 2: “BMT \\/ CCR”: allogeneic bone marrow transplant\n          and still in continuous complete remission; 3: “BMT \\/ DEATH\n          IN CR”: after allogeneic bone marrow transplant patient died\n          without relapsing; 4: “BMT \\/ REL”: after allogeneic bone\n          marrow transplant patient relapsed; 5: “CCR”: patient was in\n          continuous complete remission; 6: “CCR \\/ OFF”: patient was\n          in continuous complete remission but off-protocol for some\n          reasons; 7: “DEATH IN CR”: died when in complete remission;\n          8: “MUD \\/ DEATH IN CR”: unrelated allogeneic bone marrow\n          transplant and death without relapsing; 9: “REL”: relapse;\n          10: “REL \\/ SNC”: relapse occurred at central nervous system.\n\n        • date last seen Date the patient was last seen.\n\n_\bS_\bo_\bu_\br_\bc_\be:\n\n     Sabina Chiaretti, Xiaochun Li, Robert Gentleman, Antonella Vitale,\n     Marco Vignetti, Franco Mandelli, Jerome Ritz, and Robin Foa Gene\n     expression profile of adult T-cell acute lymphocytic leukemia\n     identifies distinct subsets of patients with different response to\n     therapy and survival. Blood, 1 April 2004, Vol. 103, No. 7.\n\n_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n\n     data(ALL)\n     "},"metadata":{}}],"execution_count":10},{"cell_type":"code","source":"exprs(ALL)[1:4,]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T10:55:39.486263Z","iopub.execute_input":"2025-04-13T10:55:39.488265Z","iopub.status.idle":"2025-04-13T10:55:39.520479Z","shell.execute_reply":"2025-04-13T10:55:39.518425Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A matrix: 4 × 128 of type dbl</caption>\n<thead>\n\t<tr><th></th><th scope=col>01005</th><th scope=col>01010</th><th scope=col>03002</th><th scope=col>04006</th><th scope=col>04007</th><th scope=col>04008</th><th scope=col>04010</th><th scope=col>04016</th><th scope=col>06002</th><th scope=col>08001</th><th scope=col>⋯</th><th scope=col>37001</th><th scope=col>43006</th><th scope=col>43015</th><th scope=col>44001</th><th scope=col>49004</th><th scope=col>56007</th><th scope=col>64005</th><th scope=col>65003</th><th scope=col>83001</th><th scope=col>LAL4</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1000_at</th><td>7.597323</td><td>7.479445</td><td>7.567593</td><td>7.384684</td><td>7.905312</td><td>7.065914</td><td>7.474537</td><td>7.536119</td><td>7.183331</td><td>7.735545</td><td>⋯</td><td>7.849455</td><td>7.960842</td><td>8.188617</td><td>7.399999</td><td>7.813474</td><td>7.816922</td><td>7.913249</td><td>7.800199</td><td>8.030047</td><td>7.702217</td></tr>\n\t<tr><th scope=row>1001_at</th><td>5.046194</td><td>4.932537</td><td>4.799294</td><td>4.922627</td><td>4.844565</td><td>5.147762</td><td>5.122518</td><td>5.016132</td><td>5.288943</td><td>4.633217</td><td>⋯</td><td>4.959450</td><td>4.537677</td><td>5.154500</td><td>5.071885</td><td>4.874525</td><td>4.788699</td><td>5.403640</td><td>5.443827</td><td>5.178633</td><td>5.029670</td></tr>\n\t<tr><th scope=row>1002_f_at</th><td>3.900466</td><td>4.208155</td><td>3.886169</td><td>4.206798</td><td>3.416923</td><td>3.945869</td><td>4.150506</td><td>3.576360</td><td>3.900935</td><td>3.630190</td><td>⋯</td><td>4.047428</td><td>3.694877</td><td>3.949546</td><td>3.837517</td><td>3.767672</td><td>4.248075</td><td>4.119252</td><td>3.957468</td><td>4.053458</td><td>4.002306</td></tr>\n\t<tr><th scope=row>1003_s_at</th><td>5.903856</td><td>6.169024</td><td>5.860459</td><td>6.116890</td><td>5.687997</td><td>6.208061</td><td>6.292713</td><td>5.665991</td><td>5.842326</td><td>5.875375</td><td>⋯</td><td>6.193790</td><td>5.973162</td><td>6.033704</td><td>5.560886</td><td>5.930329</td><td>5.878533</td><td>6.193376</td><td>6.006443</td><td>6.050129</td><td>6.297549</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA matrix: 4 × 128 of type dbl\n\n| <!--/--> | 01005 | 01010 | 03002 | 04006 | 04007 | 04008 | 04010 | 04016 | 06002 | 08001 | ⋯ | 37001 | 43006 | 43015 | 44001 | 49004 | 56007 | 64005 | 65003 | 83001 | LAL4 |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 1000_at | 7.597323 | 7.479445 | 7.567593 | 7.384684 | 7.905312 | 7.065914 | 7.474537 | 7.536119 | 7.183331 | 7.735545 | ⋯ | 7.849455 | 7.960842 | 8.188617 | 7.399999 | 7.813474 | 7.816922 | 7.913249 | 7.800199 | 8.030047 | 7.702217 |\n| 1001_at | 5.046194 | 4.932537 | 4.799294 | 4.922627 | 4.844565 | 5.147762 | 5.122518 | 5.016132 | 5.288943 | 4.633217 | ⋯ | 4.959450 | 4.537677 | 5.154500 | 5.071885 | 4.874525 | 4.788699 | 5.403640 | 5.443827 | 5.178633 | 5.029670 |\n| 1002_f_at | 3.900466 | 4.208155 | 3.886169 | 4.206798 | 3.416923 | 3.945869 | 4.150506 | 3.576360 | 3.900935 | 3.630190 | ⋯ | 4.047428 | 3.694877 | 3.949546 | 3.837517 | 3.767672 | 4.248075 | 4.119252 | 3.957468 | 4.053458 | 4.002306 |\n| 1003_s_at | 5.903856 | 6.169024 | 5.860459 | 6.116890 | 5.687997 | 6.208061 | 6.292713 | 5.665991 | 5.842326 | 5.875375 | ⋯ | 6.193790 | 5.973162 | 6.033704 | 5.560886 | 5.930329 | 5.878533 | 6.193376 | 6.006443 | 6.050129 | 6.297549 |\n\n","text/latex":"A matrix: 4 × 128 of type dbl\n\\begin{tabular}{r|lllllllllllllllllllll}\n  & 01005 & 01010 & 03002 & 04006 & 04007 & 04008 & 04010 & 04016 & 06002 & 08001 & ⋯ & 37001 & 43006 & 43015 & 44001 & 49004 & 56007 & 64005 & 65003 & 83001 & LAL4\\\\\n\\hline\n\t1000\\_at & 7.597323 & 7.479445 & 7.567593 & 7.384684 & 7.905312 & 7.065914 & 7.474537 & 7.536119 & 7.183331 & 7.735545 & ⋯ & 7.849455 & 7.960842 & 8.188617 & 7.399999 & 7.813474 & 7.816922 & 7.913249 & 7.800199 & 8.030047 & 7.702217\\\\\n\t1001\\_at & 5.046194 & 4.932537 & 4.799294 & 4.922627 & 4.844565 & 5.147762 & 5.122518 & 5.016132 & 5.288943 & 4.633217 & ⋯ & 4.959450 & 4.537677 & 5.154500 & 5.071885 & 4.874525 & 4.788699 & 5.403640 & 5.443827 & 5.178633 & 5.029670\\\\\n\t1002\\_f\\_at & 3.900466 & 4.208155 & 3.886169 & 4.206798 & 3.416923 & 3.945869 & 4.150506 & 3.576360 & 3.900935 & 3.630190 & ⋯ & 4.047428 & 3.694877 & 3.949546 & 3.837517 & 3.767672 & 4.248075 & 4.119252 & 3.957468 & 4.053458 & 4.002306\\\\\n\t1003\\_s\\_at & 5.903856 & 6.169024 & 5.860459 & 6.116890 & 5.687997 & 6.208061 & 6.292713 & 5.665991 & 5.842326 & 5.875375 & ⋯ & 6.193790 & 5.973162 & 6.033704 & 5.560886 & 5.930329 & 5.878533 & 6.193376 & 6.006443 & 6.050129 & 6.297549\\\\\n\\end{tabular}\n","text/plain":"          01005    01010    03002    04006    04007    04008    04010   \n1000_at   7.597323 7.479445 7.567593 7.384684 7.905312 7.065914 7.474537\n1001_at   5.046194 4.932537 4.799294 4.922627 4.844565 5.147762 5.122518\n1002_f_at 3.900466 4.208155 3.886169 4.206798 3.416923 3.945869 4.150506\n1003_s_at 5.903856 6.169024 5.860459 6.116890 5.687997 6.208061 6.292713\n          04016    06002    08001    ⋯ 37001    43006    43015    44001   \n1000_at   7.536119 7.183331 7.735545 ⋯ 7.849455 7.960842 8.188617 7.399999\n1001_at   5.016132 5.288943 4.633217 ⋯ 4.959450 4.537677 5.154500 5.071885\n1002_f_at 3.576360 3.900935 3.630190 ⋯ 4.047428 3.694877 3.949546 3.837517\n1003_s_at 5.665991 5.842326 5.875375 ⋯ 6.193790 5.973162 6.033704 5.560886\n          49004    56007    64005    65003    83001    LAL4    \n1000_at   7.813474 7.816922 7.913249 7.800199 8.030047 7.702217\n1001_at   4.874525 4.788699 5.403640 5.443827 5.178633 5.029670\n1002_f_at 3.767672 4.248075 4.119252 3.957468 4.053458 4.002306\n1003_s_at 5.930329 5.878533 6.193376 6.006443 6.050129 6.297549"},"metadata":{}}],"execution_count":12},{"cell_type":"code","source":"head(sampleNames(ALL))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T10:58:19.632121Z","iopub.execute_input":"2025-04-13T10:58:19.633906Z","iopub.status.idle":"2025-04-13T10:58:19.651973Z","shell.execute_reply":"2025-04-13T10:58:19.649975Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'01005'</li><li>'01010'</li><li>'03002'</li><li>'04006'</li><li>'04007'</li><li>'04008'</li></ol>\n","text/markdown":"1. '01005'\n2. '01010'\n3. '03002'\n4. '04006'\n5. '04007'\n6. '04008'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item '01005'\n\\item '01010'\n\\item '03002'\n\\item '04006'\n\\item '04007'\n\\item '04008'\n\\end{enumerate*}\n","text/plain":"[1] \"01005\" \"01010\" \"03002\" \"04006\" \"04007\" \"04008\""},"metadata":{}}],"execution_count":14},{"cell_type":"code","source":"head(featureNames(ALL))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T10:59:02.753529Z","iopub.execute_input":"2025-04-13T10:59:02.755603Z","iopub.status.idle":"2025-04-13T10:59:02.775506Z","shell.execute_reply":"2025-04-13T10:59:02.772972Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'1000_at'</li><li>'1001_at'</li><li>'1002_f_at'</li><li>'1003_s_at'</li><li>'1004_at'</li><li>'1005_at'</li></ol>\n","text/markdown":"1. '1000_at'\n2. '1001_at'\n3. '1002_f_at'\n4. '1003_s_at'\n5. '1004_at'\n6. '1005_at'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item '1000\\_at'\n\\item '1001\\_at'\n\\item '1002\\_f\\_at'\n\\item '1003\\_s\\_at'\n\\item '1004\\_at'\n\\item '1005\\_at'\n\\end{enumerate*}\n","text/plain":"[1] \"1000_at\"   \"1001_at\"   \"1002_f_at\" \"1003_s_at\" \"1004_at\"   \"1005_at\"  "},"metadata":{}}],"execution_count":15},{"cell_type":"code","source":"head(pData(ALL))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:00:01.915335Z","iopub.execute_input":"2025-04-13T11:00:01.917049Z","iopub.status.idle":"2025-04-13T11:00:01.954950Z","shell.execute_reply":"2025-04-13T11:00:01.952877Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 21</caption>\n<thead>\n\t<tr><th></th><th scope=col>cod</th><th scope=col>diagnosis</th><th scope=col>sex</th><th scope=col>age</th><th scope=col>BT</th><th scope=col>remission</th><th scope=col>CR</th><th scope=col>date.cr</th><th scope=col>t(4;11)</th><th scope=col>t(9;22)</th><th scope=col>⋯</th><th scope=col>citog</th><th scope=col>mol.biol</th><th scope=col>fusion protein</th><th scope=col>mdr</th><th scope=col>kinet</th><th scope=col>ccr</th><th scope=col>relapse</th><th scope=col>transplant</th><th scope=col>f.u</th><th scope=col>date last seen</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>01005</th><td>1005</td><td>5/21/1997</td><td>M</td><td>53</td><td>B2</td><td>CR</td><td>CR</td><td>8/6/1997 </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)     </td><td>BCR/ABL </td><td>p210</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td>FALSE</td><td> TRUE</td><td>BMT / DEATH IN CR</td><td>NA        </td></tr>\n\t<tr><th scope=row>01010</th><td>1010</td><td>3/29/2000</td><td>M</td><td>19</td><td>B2</td><td>CR</td><td>CR</td><td>6/27/2000</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG     </td><td>NA  </td><td>POS</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>8/28/2000 </td></tr>\n\t<tr><th scope=row>03002</th><td>3002</td><td>6/24/1998</td><td>F</td><td>52</td><td>B4</td><td>CR</td><td>CR</td><td>8/17/1998</td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>BCR/ABL </td><td>p190</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>10/15/1999</td></tr>\n\t<tr><th scope=row>04006</th><td>4006</td><td>7/17/1997</td><td>M</td><td>38</td><td>B1</td><td>CR</td><td>CR</td><td>9/8/1997 </td><td> TRUE</td><td>FALSE</td><td>⋯</td><td>t(4;11)     </td><td>ALL1/AF4</td><td>NA  </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>1/23/1998 </td></tr>\n\t<tr><th scope=row>04007</th><td>4007</td><td>7/22/1997</td><td>M</td><td>57</td><td>B2</td><td>CR</td><td>CR</td><td>9/17/1997</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(6q)     </td><td>NEG     </td><td>NA  </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>11/4/1997 </td></tr>\n\t<tr><th scope=row>04008</th><td>4008</td><td>7/30/1997</td><td>M</td><td>17</td><td>B1</td><td>CR</td><td>CR</td><td>9/27/1997</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>complex alt.</td><td>NEG     </td><td>NA  </td><td>NEG</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>12/15/1997</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 21\n\n| <!--/--> | cod &lt;chr&gt; | diagnosis &lt;chr&gt; | sex &lt;fct&gt; | age &lt;int&gt; | BT &lt;fct&gt; | remission &lt;fct&gt; | CR &lt;chr&gt; | date.cr &lt;chr&gt; | t(4;11) &lt;lgl&gt; | t(9;22) &lt;lgl&gt; | ⋯ ⋯ | citog &lt;chr&gt; | mol.biol &lt;fct&gt; | fusion protein &lt;fct&gt; | mdr &lt;fct&gt; | kinet &lt;fct&gt; | ccr &lt;lgl&gt; | relapse &lt;lgl&gt; | transplant &lt;lgl&gt; | f.u &lt;chr&gt; | date last seen &lt;chr&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 01005 | 1005 | 5/21/1997 | M | 53 | B2 | CR | CR | 8/6/1997  | FALSE |  TRUE | ⋯ | t(9;22)      | BCR/ABL  | p210 | NEG | dyploid | FALSE | FALSE |  TRUE | BMT / DEATH IN CR | NA         |\n| 01010 | 1010 | 3/29/2000 | M | 19 | B2 | CR | CR | 6/27/2000 | FALSE | FALSE | ⋯ | simple alt.  | NEG      | NA   | POS | dyploid | FALSE |  TRUE | FALSE | REL               | 8/28/2000  |\n| 03002 | 3002 | 6/24/1998 | F | 52 | B4 | CR | CR | 8/17/1998 |    NA |    NA | ⋯ | NA           | BCR/ABL  | p190 | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 10/15/1999 |\n| 04006 | 4006 | 7/17/1997 | M | 38 | B1 | CR | CR | 9/8/1997  |  TRUE | FALSE | ⋯ | t(4;11)      | ALL1/AF4 | NA   | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 1/23/1998  |\n| 04007 | 4007 | 7/22/1997 | M | 57 | B2 | CR | CR | 9/17/1997 | FALSE | FALSE | ⋯ | del(6q)      | NEG      | NA   | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 11/4/1997  |\n| 04008 | 4008 | 7/30/1997 | M | 17 | B1 | CR | CR | 9/27/1997 | FALSE | FALSE | ⋯ | complex alt. | NEG      | NA   | NEG | hyperd. | FALSE |  TRUE | FALSE | REL               | 12/15/1997 |\n\n","text/latex":"A data.frame: 6 × 21\n\\begin{tabular}{r|lllllllllllllllllllll}\n  & cod & diagnosis & sex & age & BT & remission & CR & date.cr & t(4;11) & t(9;22) & ⋯ & citog & mol.biol & fusion protein & mdr & kinet & ccr & relapse & transplant & f.u & date last seen\\\\\n  & <chr> & <chr> & <fct> & <int> & <fct> & <fct> & <chr> & <chr> & <lgl> & <lgl> & ⋯ & <chr> & <fct> & <fct> & <fct> & <fct> & <lgl> & <lgl> & <lgl> & <chr> & <chr>\\\\\n\\hline\n\t01005 & 1005 & 5/21/1997 & M & 53 & B2 & CR & CR & 8/6/1997  & FALSE &  TRUE & ⋯ & t(9;22)      & BCR/ABL  & p210 & NEG & dyploid & FALSE & FALSE &  TRUE & BMT / DEATH IN CR & NA        \\\\\n\t01010 & 1010 & 3/29/2000 & M & 19 & B2 & CR & CR & 6/27/2000 & FALSE & FALSE & ⋯ & simple alt.  & NEG      & NA   & POS & dyploid & FALSE &  TRUE & FALSE & REL               & 8/28/2000 \\\\\n\t03002 & 3002 & 6/24/1998 & F & 52 & B4 & CR & CR & 8/17/1998 &    NA &    NA & ⋯ & NA           & BCR/ABL  & p190 & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 10/15/1999\\\\\n\t04006 & 4006 & 7/17/1997 & M & 38 & B1 & CR & CR & 9/8/1997  &  TRUE & FALSE & ⋯ & t(4;11)      & ALL1/AF4 & NA   & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 1/23/1998 \\\\\n\t04007 & 4007 & 7/22/1997 & M & 57 & B2 & CR & CR & 9/17/1997 & FALSE & FALSE & ⋯ & del(6q)      & NEG      & NA   & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 11/4/1997 \\\\\n\t04008 & 4008 & 7/30/1997 & M & 17 & B1 & CR & CR & 9/27/1997 & FALSE & FALSE & ⋯ & complex alt. & NEG      & NA   & NEG & hyperd. & FALSE &  TRUE & FALSE & REL               & 12/15/1997\\\\\n\\end{tabular}\n","text/plain":"      cod  diagnosis sex age BT remission CR date.cr   t(4;11) t(9;22) ⋯\n01005 1005 5/21/1997 M   53  B2 CR        CR 8/6/1997  FALSE    TRUE   ⋯\n01010 1010 3/29/2000 M   19  B2 CR        CR 6/27/2000 FALSE   FALSE   ⋯\n03002 3002 6/24/1998 F   52  B4 CR        CR 8/17/1998    NA      NA   ⋯\n04006 4006 7/17/1997 M   38  B1 CR        CR 9/8/1997   TRUE   FALSE   ⋯\n04007 4007 7/22/1997 M   57  B2 CR        CR 9/17/1997 FALSE   FALSE   ⋯\n04008 4008 7/30/1997 M   17  B1 CR        CR 9/27/1997 FALSE   FALSE   ⋯\n      citog        mol.biol fusion protein mdr kinet   ccr   relapse transplant\n01005 t(9;22)      BCR/ABL  p210           NEG dyploid FALSE FALSE    TRUE     \n01010 simple alt.  NEG      NA             POS dyploid FALSE  TRUE   FALSE     \n03002 NA           BCR/ABL  p190           NEG dyploid FALSE  TRUE   FALSE     \n04006 t(4;11)      ALL1/AF4 NA             NEG dyploid FALSE  TRUE   FALSE     \n04007 del(6q)      NEG      NA             NEG dyploid FALSE  TRUE   FALSE     \n04008 complex alt. NEG      NA             NEG hyperd. FALSE  TRUE   FALSE     \n      f.u               date last seen\n01005 BMT / DEATH IN CR NA            \n01010 REL               8/28/2000     \n03002 REL               10/15/1999    \n04006 REL               1/23/1998     \n04007 REL               11/4/1997     \n04008 REL               12/15/1997    "},"metadata":{}}],"execution_count":16},{"cell_type":"code","source":"ALL$sex","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:01:27.692779Z","iopub.execute_input":"2025-04-13T11:01:27.694547Z","iopub.status.idle":"2025-04-13T11:01:27.713371Z","shell.execute_reply":"2025-04-13T11:01:27.711488Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>F</li><li>F</li><li>F</li><li>F</li><li>F</li><li>F</li><li>M</li><li>M</li><li>F</li><li>F</li><li>F</li><li>F</li><li>F</li><li>M</li><li>F</li><li>M</li><li>F</li><li>M</li><li>&lt;NA&gt;</li><li>M</li><li>F</li><li>M</li><li>F</li><li>F</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>F</li><li>M</li><li>F</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>F</li><li>F</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>F</li><li>F</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>F</li><li>M</li><li>&lt;NA&gt;</li><li>M</li><li>F</li><li>F</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>F</li><li>F</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>F</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>M</li><li>&lt;NA&gt;</li></ol>\n\n<details>\n\t<summary style=display:list-item;cursor:pointer>\n\t\t<strong>Levels</strong>:\n\t</summary>\n\t<style>\n\t.list-inline {list-style: none; margin:0; padding: 0}\n\t.list-inline>li {display: inline-block}\n\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n\t</style>\n\t<ol class=list-inline><li>'F'</li><li>'M'</li></ol>\n</details>","text/markdown":"1. M\n2. M\n3. F\n4. M\n5. M\n6. M\n7. F\n8. M\n9. M\n10. M\n11. M\n12. M\n13. M\n14. M\n15. M\n16. F\n17. M\n18. M\n19. M\n20. F\n21. M\n22. M\n23. M\n24. M\n25. M\n26. M\n27. F\n28. F\n29. F\n30. F\n31. F\n32. F\n33. M\n34. M\n35. F\n36. F\n37. F\n38. F\n39. F\n40. M\n41. F\n42. M\n43. F\n44. M\n45. &lt;NA&gt;\n46. M\n47. F\n48. M\n49. F\n50. F\n51. F\n52. M\n53. M\n54. M\n55. M\n56. F\n57. M\n58. F\n59. M\n60. F\n61. M\n62. M\n63. F\n64. M\n65. M\n66. M\n67. M\n68. M\n69. M\n70. M\n71. F\n72. M\n73. M\n74. F\n75. F\n76. M\n77. M\n78. M\n79. F\n80. M\n81. M\n82. M\n83. F\n84. F\n85. F\n86. M\n87. M\n88. M\n89. M\n90. F\n91. M\n92. M\n93. F\n94. M\n95. &lt;NA&gt;\n96. M\n97. F\n98. F\n99. M\n100. F\n101. M\n102. M\n103. M\n104. M\n105. M\n106. M\n107. F\n108. F\n109. M\n110. F\n111. M\n112. M\n113. M\n114. F\n115. M\n116. M\n117. F\n118. M\n119. M\n120. M\n121. M\n122. M\n123. M\n124. M\n125. M\n126. M\n127. M\n128. &lt;NA&gt;\n\n\n\n**Levels**: 1. 'F'\n2. 'M'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item F\n\\item F\n\\item F\n\\item F\n\\item F\n\\item F\n\\item M\n\\item M\n\\item F\n\\item F\n\\item F\n\\item F\n\\item F\n\\item M\n\\item F\n\\item M\n\\item F\n\\item M\n\\item <NA>\n\\item M\n\\item F\n\\item M\n\\item F\n\\item F\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item F\n\\item M\n\\item F\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item F\n\\item F\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item F\n\\item F\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item F\n\\item M\n\\item <NA>\n\\item M\n\\item F\n\\item F\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item F\n\\item F\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item F\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item M\n\\item <NA>\n\\end{enumerate*}\n\n\\emph{Levels}: \\begin{enumerate*}\n\\item 'F'\n\\item 'M'\n\\end{enumerate*}\n","text/plain":"  [1] M    M    F    M    M    M    F    M    M    M    M    M    M    M    M   \n [16] F    M    M    M    F    M    M    M    M    M    M    F    F    F    F   \n [31] F    F    M    M    F    F    F    F    F    M    F    M    F    M    <NA>\n [46] M    F    M    F    F    F    M    M    M    M    F    M    F    M    F   \n [61] M    M    F    M    M    M    M    M    M    M    F    M    M    F    F   \n [76] M    M    M    F    M    M    M    F    F    F    M    M    M    M    F   \n [91] M    M    F    M    <NA> M    F    F    M    F    M    M    M    M    M   \n[106] M    F    F    M    F    M    M    M    F    M    M    F    M    M    M   \n[121] M    M    M    M    M    M    M    <NA>\nLevels: F M"},"metadata":{}}],"execution_count":17},{"cell_type":"code","source":"ALL[1:10,1:5]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:03:15.778952Z","iopub.execute_input":"2025-04-13T11:03:15.780629Z","iopub.status.idle":"2025-04-13T11:03:15.842130Z","shell.execute_reply":"2025-04-13T11:03:15.840308Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"ExpressionSet (storageMode: lockedEnvironment)\nassayData: 10 features, 5 samples \n  element names: exprs \nprotocolData: none\nphenoData\n  sampleNames: 01005 01010 ... 04007 (5 total)\n  varLabels: cod diagnosis ... date last seen (21 total)\n  varMetadata: labelDescription\nfeatureData: none\nexperimentData: use 'experimentData(object)'\n  pubMedIds: 14684422 16243790 \nAnnotation: hgu95av2 "},"metadata":{}}],"execution_count":18},{"cell_type":"code","source":"featureData(ALL)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:04:15.740813Z","iopub.execute_input":"2025-04-13T11:04:15.742406Z","iopub.status.idle":"2025-04-13T11:04:15.757427Z","shell.execute_reply":"2025-04-13T11:04:15.755697Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"An object of class 'AnnotatedDataFrame': none"},"metadata":{}}],"execution_count":19},{"cell_type":"code","source":"phenoData(ALL)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:07:42.502182Z","iopub.execute_input":"2025-04-13T11:07:42.503848Z","iopub.status.idle":"2025-04-13T11:07:42.520750Z","shell.execute_reply":"2025-04-13T11:07:42.518969Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"An object of class 'AnnotatedDataFrame'\n  sampleNames: 01005 01010 ... LAL4 (128 total)\n  varLabels: cod diagnosis ... date last seen (21 total)\n  varMetadata: labelDescription"},"metadata":{}}],"execution_count":20},{"cell_type":"code","source":"names(pData(ALL))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:08:41.991041Z","iopub.execute_input":"2025-04-13T11:08:41.992758Z","iopub.status.idle":"2025-04-13T11:08:42.009230Z","shell.execute_reply":"2025-04-13T11:08:42.007476Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'cod'</li><li>'diagnosis'</li><li>'sex'</li><li>'age'</li><li>'BT'</li><li>'remission'</li><li>'CR'</li><li>'date.cr'</li><li>'t(4;11)'</li><li>'t(9;22)'</li><li>'cyto.normal'</li><li>'citog'</li><li>'mol.biol'</li><li>'fusion protein'</li><li>'mdr'</li><li>'kinet'</li><li>'ccr'</li><li>'relapse'</li><li>'transplant'</li><li>'f.u'</li><li>'date last seen'</li></ol>\n","text/markdown":"1. 'cod'\n2. 'diagnosis'\n3. 'sex'\n4. 'age'\n5. 'BT'\n6. 'remission'\n7. 'CR'\n8. 'date.cr'\n9. 't(4;11)'\n10. 't(9;22)'\n11. 'cyto.normal'\n12. 'citog'\n13. 'mol.biol'\n14. 'fusion protein'\n15. 'mdr'\n16. 'kinet'\n17. 'ccr'\n18. 'relapse'\n19. 'transplant'\n20. 'f.u'\n21. 'date last seen'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'cod'\n\\item 'diagnosis'\n\\item 'sex'\n\\item 'age'\n\\item 'BT'\n\\item 'remission'\n\\item 'CR'\n\\item 'date.cr'\n\\item 't(4;11)'\n\\item 't(9;22)'\n\\item 'cyto.normal'\n\\item 'citog'\n\\item 'mol.biol'\n\\item 'fusion protein'\n\\item 'mdr'\n\\item 'kinet'\n\\item 'ccr'\n\\item 'relapse'\n\\item 'transplant'\n\\item 'f.u'\n\\item 'date last seen'\n\\end{enumerate*}\n","text/plain":" [1] \"cod\"            \"diagnosis\"      \"sex\"            \"age\"           \n [5] \"BT\"             \"remission\"      \"CR\"             \"date.cr\"       \n [9] \"t(4;11)\"        \"t(9;22)\"        \"cyto.normal\"    \"citog\"         \n[13] \"mol.biol\"       \"fusion protein\" \"mdr\"            \"kinet\"         \n[17] \"ccr\"            \"relapse\"        \"transplant\"     \"f.u\"           \n[21] \"date last seen\""},"metadata":{}}],"execution_count":21},{"cell_type":"code","source":"varLabels(ALL)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:09:43.496537Z","iopub.execute_input":"2025-04-13T11:09:43.498167Z","iopub.status.idle":"2025-04-13T11:09:43.515449Z","shell.execute_reply":"2025-04-13T11:09:43.513256Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'cod'</li><li>'diagnosis'</li><li>'sex'</li><li>'age'</li><li>'BT'</li><li>'remission'</li><li>'CR'</li><li>'date.cr'</li><li>'t(4;11)'</li><li>'t(9;22)'</li><li>'cyto.normal'</li><li>'citog'</li><li>'mol.biol'</li><li>'fusion protein'</li><li>'mdr'</li><li>'kinet'</li><li>'ccr'</li><li>'relapse'</li><li>'transplant'</li><li>'f.u'</li><li>'date last seen'</li></ol>\n","text/markdown":"1. 'cod'\n2. 'diagnosis'\n3. 'sex'\n4. 'age'\n5. 'BT'\n6. 'remission'\n7. 'CR'\n8. 'date.cr'\n9. 't(4;11)'\n10. 't(9;22)'\n11. 'cyto.normal'\n12. 'citog'\n13. 'mol.biol'\n14. 'fusion protein'\n15. 'mdr'\n16. 'kinet'\n17. 'ccr'\n18. 'relapse'\n19. 'transplant'\n20. 'f.u'\n21. 'date last seen'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'cod'\n\\item 'diagnosis'\n\\item 'sex'\n\\item 'age'\n\\item 'BT'\n\\item 'remission'\n\\item 'CR'\n\\item 'date.cr'\n\\item 't(4;11)'\n\\item 't(9;22)'\n\\item 'cyto.normal'\n\\item 'citog'\n\\item 'mol.biol'\n\\item 'fusion protein'\n\\item 'mdr'\n\\item 'kinet'\n\\item 'ccr'\n\\item 'relapse'\n\\item 'transplant'\n\\item 'f.u'\n\\item 'date last seen'\n\\end{enumerate*}\n","text/plain":" [1] \"cod\"            \"diagnosis\"      \"sex\"            \"age\"           \n [5] \"BT\"             \"remission\"      \"CR\"             \"date.cr\"       \n [9] \"t(4;11)\"        \"t(9;22)\"        \"cyto.normal\"    \"citog\"         \n[13] \"mol.biol\"       \"fusion protein\" \"mdr\"            \"kinet\"         \n[17] \"ccr\"            \"relapse\"        \"transplant\"     \"f.u\"           \n[21] \"date last seen\""},"metadata":{}}],"execution_count":22},{"cell_type":"code","source":"pData(phenoData(ALL))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:10:30.296155Z","iopub.execute_input":"2025-04-13T11:10:30.298023Z","iopub.status.idle":"2025-04-13T11:10:30.502230Z","shell.execute_reply":"2025-04-13T11:10:30.500409Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 128 × 21</caption>\n<thead>\n\t<tr><th></th><th scope=col>cod</th><th scope=col>diagnosis</th><th scope=col>sex</th><th scope=col>age</th><th scope=col>BT</th><th scope=col>remission</th><th scope=col>CR</th><th scope=col>date.cr</th><th scope=col>t(4;11)</th><th scope=col>t(9;22)</th><th scope=col>⋯</th><th scope=col>citog</th><th scope=col>mol.biol</th><th scope=col>fusion protein</th><th scope=col>mdr</th><th scope=col>kinet</th><th scope=col>ccr</th><th scope=col>relapse</th><th scope=col>transplant</th><th scope=col>f.u</th><th scope=col>date last seen</th></tr>\n\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>01005</th><td>1005 </td><td>5/21/1997 </td><td>M</td><td>53</td><td>B2</td><td>CR </td><td>CR         </td><td>8/6/1997  </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)        </td><td>BCR/ABL </td><td>p210     </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td>FALSE</td><td> TRUE</td><td>BMT / DEATH IN CR</td><td>NA        </td></tr>\n\t<tr><th scope=row>01010</th><td>1010 </td><td>3/29/2000 </td><td>M</td><td>19</td><td>B2</td><td>CR </td><td>CR         </td><td>6/27/2000 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt.    </td><td>NEG     </td><td>NA       </td><td>POS</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>8/28/2000 </td></tr>\n\t<tr><th scope=row>03002</th><td>3002 </td><td>6/24/1998 </td><td>F</td><td>52</td><td>B4</td><td>CR </td><td>CR         </td><td>8/17/1998 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA             </td><td>BCR/ABL </td><td>p190     </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>10/15/1999</td></tr>\n\t<tr><th scope=row>04006</th><td>4006 </td><td>7/17/1997 </td><td>M</td><td>38</td><td>B1</td><td>CR </td><td>CR         </td><td>9/8/1997  </td><td> TRUE</td><td>FALSE</td><td>⋯</td><td>t(4;11)        </td><td>ALL1/AF4</td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>1/23/1998 </td></tr>\n\t<tr><th scope=row>04007</th><td>4007 </td><td>7/22/1997 </td><td>M</td><td>57</td><td>B2</td><td>CR </td><td>CR         </td><td>9/17/1997 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(6q)        </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>11/4/1997 </td></tr>\n\t<tr><th scope=row>04008</th><td>4008 </td><td>7/30/1997 </td><td>M</td><td>17</td><td>B1</td><td>CR </td><td>CR         </td><td>9/27/1997 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>complex alt.   </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>12/15/1997</td></tr>\n\t<tr><th scope=row>04010</th><td>4010 </td><td>10/30/1997</td><td>F</td><td>18</td><td>B1</td><td>CR </td><td>CR         </td><td>1/7/1998  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>complex alt.   </td><td>NEG     </td><td>NA       </td><td>POS</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>3/5/1998  </td></tr>\n\t<tr><th scope=row>04016</th><td>4016 </td><td>2/10/2000 </td><td>M</td><td>16</td><td>B1</td><td>CR </td><td>CR         </td><td>4/17/2000 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt.    </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>9/26/2000 </td></tr>\n\t<tr><th scope=row>06002</th><td>6002 </td><td>3/19/1997 </td><td>M</td><td>15</td><td>B2</td><td>CR </td><td>CR         </td><td>6/9/1997  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal         </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>3/18/1998 </td></tr>\n\t<tr><th scope=row>08001</th><td>8001 </td><td>1/15/1997 </td><td>M</td><td>40</td><td>B2</td><td>CR </td><td>CR         </td><td>3/26/1997 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(p15)       </td><td>BCR/ABL </td><td>p190     </td><td>NEG</td><td>NA     </td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>7/11/1997 </td></tr>\n\t<tr><th scope=row>08011</th><td>8011 </td><td>8/21/1998 </td><td>M</td><td>33</td><td>B3</td><td>CR </td><td>CR         </td><td>10/8/1998 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(p15/p16)   </td><td>BCR/ABL </td><td>p190/p210</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td>FALSE</td><td> TRUE</td><td>BMT / DEATH IN CR</td><td>NA        </td></tr>\n\t<tr><th scope=row>08012</th><td>8012 </td><td>10/22/1998</td><td>M</td><td>55</td><td>B3</td><td>CR </td><td>CR         </td><td>1/9/1999  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt.    </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>4/9/1999  </td></tr>\n\t<tr><th scope=row>08018</th><td>8018 </td><td>8/27/1999 </td><td>M</td><td> 5</td><td>B3</td><td>CR </td><td>CR         </td><td>10/18/1999</td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA             </td><td>E2A/PBX1</td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>5/23/2000 </td></tr>\n\t<tr><th scope=row>08024</th><td>8024 </td><td>7/20/2000 </td><td>M</td><td>18</td><td>B2</td><td>CR </td><td>DEATH IN CR</td><td>NA        </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt.    </td><td>NEG     </td><td>NA       </td><td>POS</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA               </td><td>NA        </td></tr>\n\t<tr><th scope=row>09008</th><td>9008 </td><td>12/17/1999</td><td>M</td><td>41</td><td>B3</td><td>CR </td><td>CR         </td><td>2/15/2000 </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)+other  </td><td>BCR/ABL </td><td>p190     </td><td>NEG</td><td>hyperd.</td><td> TRUE</td><td>FALSE</td><td> TRUE</td><td>BMT / CCR        </td><td>00/09/01  </td></tr>\n\t<tr><th scope=row>09017</th><td>9017 </td><td>2/3/2000  </td><td>F</td><td>27</td><td>B </td><td>CR </td><td>CR         </td><td>3/23/2000 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal         </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>9/11/2001 </td></tr>\n\t<tr><th scope=row>11005</th><td>11005</td><td>6/1/1998  </td><td>M</td><td>27</td><td>B2</td><td>CR </td><td>DEATH IN CR</td><td>8/3/1998  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(7q) + altro</td><td>BCR/ABL </td><td>p190     </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>DEATH IN CR      </td><td>NA        </td></tr>\n\t<tr><th scope=row>12006</th><td>12006</td><td>2/20/1997 </td><td>M</td><td>46</td><td>B3</td><td>REF</td><td>REF        </td><td>NA        </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)        </td><td>BCR/ABL </td><td>p210     </td><td>NEG</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA               </td><td>NA        </td></tr>\n\t<tr><th scope=row>12007</th><td>12007</td><td>3/11/1997 </td><td>M</td><td>37</td><td>B2</td><td>CR </td><td>CR         </td><td>5/7/1997  </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)+ del(9)</td><td>BCR/ABL </td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>9/26/1997 </td></tr>\n\t<tr><th scope=row>12012</th><td>12012</td><td>5/21/1997 </td><td>F</td><td>36</td><td>B3</td><td>REF</td><td>REF        </td><td>NA        </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)        </td><td>BCR/ABL </td><td>p190     </td><td>NEG</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA               </td><td>NA        </td></tr>\n\t<tr><th scope=row>12019</th><td>12019</td><td>9/4/1997  </td><td>M</td><td>53</td><td>B2</td><td>CR </td><td>CR         </td><td>11/11/1997</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal         </td><td>NEG     </td><td>NA       </td><td>POS</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR              </td><td>6/6/2002  </td></tr>\n\t<tr><th scope=row>12026</th><td>12026</td><td>5/29/1998 </td><td>M</td><td>39</td><td>B2</td><td>REF</td><td>REF        </td><td>NA        </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)        </td><td>BCR/ABL </td><td>p190/p210</td><td>NA </td><td>dyploid</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>DEATH IN CR      </td><td>NA        </td></tr>\n\t<tr><th scope=row>14016</th><td>14016</td><td>5/27/1999 </td><td>M</td><td>53</td><td>B2</td><td>NA </td><td>NA         </td><td>NA        </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)        </td><td>BCR/ABL </td><td>p210     </td><td>NEG</td><td>NA     </td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA               </td><td>NA        </td></tr>\n\t<tr><th scope=row>15001</th><td>15001</td><td>9/3/1997  </td><td>M</td><td>20</td><td>B1</td><td>CR </td><td>CR         </td><td>11/11/1997</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal         </td><td>NEG     </td><td>NA       </td><td>NEG</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR              </td><td>6/21/2002 </td></tr>\n\t<tr><th scope=row>15004</th><td>15004</td><td>2/10/2000 </td><td>M</td><td>44</td><td>B1</td><td>CR </td><td>CR         </td><td>4/3/2000  </td><td> TRUE</td><td>FALSE</td><td>⋯</td><td>t(4;11)        </td><td>ALL1/AF4</td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>12/19/2000</td></tr>\n\t<tr><th scope=row>15005</th><td>15005</td><td>5/4/2000  </td><td>M</td><td>28</td><td>B2</td><td>CR </td><td>CR         </td><td>6/26/2000 </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)+other  </td><td>BCR/ABL </td><td>p190     </td><td>POS</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>11/15/2000</td></tr>\n\t<tr><th scope=row>16004</th><td>16004</td><td>4/19/1997 </td><td>F</td><td>58</td><td>B1</td><td>CR </td><td>CR         </td><td>7/15/1997 </td><td> TRUE</td><td>FALSE</td><td>⋯</td><td>t(4;11)        </td><td>ALL1/AF4</td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>12/9/1997 </td></tr>\n\t<tr><th scope=row>16009</th><td>16009</td><td>7/11/2000 </td><td>F</td><td>43</td><td>B2</td><td>NA </td><td>NA         </td><td>NA        </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA             </td><td>NEG     </td><td>NA       </td><td>POS</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR / OFF        </td><td>5/23/2002 </td></tr>\n\t<tr><th scope=row>19005</th><td>19005</td><td>11/15/1997</td><td>F</td><td>48</td><td>B1</td><td>CR </td><td>CR         </td><td>2/3/1998  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal         </td><td>ALL1/AF4</td><td>NA       </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>2/4/1998  </td></tr>\n\t<tr><th scope=row>20002</th><td>20002</td><td>5/9/1997  </td><td>F</td><td>58</td><td>B2</td><td>CR </td><td>CR         </td><td>8/19/1997 </td><td>FALSE</td><td> TRUE</td><td>⋯</td><td>t(9;22)+other  </td><td>BCR/ABL </td><td>p190     </td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL              </td><td>12/15/1997</td></tr>\n\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n\t<tr><th scope=row>04018</th><td>4018 </td><td>3/24/2000 </td><td>M </td><td>17</td><td>T2</td><td>CR </td><td>CR                </td><td>5/23/2000 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>5/14/2001 </td></tr>\n\t<tr><th scope=row>09002</th><td>9002 </td><td>5/14/1998 </td><td>F </td><td>40</td><td>T3</td><td>CR </td><td>CR                </td><td>7/21/1998 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL / SNC</td><td>9/14/1999 </td></tr>\n\t<tr><th scope=row>10005</th><td>10005</td><td>9/30/1997 </td><td>M </td><td>22</td><td>T2</td><td>CR </td><td>CR                </td><td>1/26/1998 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>10/10/1999</td></tr>\n\t<tr><th scope=row>11002</th><td>11002</td><td>12/23/1996</td><td>M </td><td>30</td><td>T </td><td>CR </td><td>CR                </td><td>3/18/1997 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal      </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>4/14/1998 </td></tr>\n\t<tr><th scope=row>12008</th><td>12008</td><td>3/22/1997 </td><td>M </td><td>18</td><td>T4</td><td>REF</td><td>REF               </td><td>NA        </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(6q)     </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>15006</th><td>15006</td><td>6/10/1998 </td><td>M </td><td>22</td><td>T2</td><td>CR </td><td>CR                </td><td>8/11/1998 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>complex alt.</td><td>NEG</td><td>NA</td><td>NEG</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>9/15/2000 </td></tr>\n\t<tr><th scope=row>16002</th><td>16002</td><td>4/10/1997 </td><td>M </td><td>50</td><td>T3</td><td>CR </td><td>CR                </td><td>6/10/1997 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>12/7/1999 </td></tr>\n\t<tr><th scope=row>16007</th><td>16007</td><td>11/1/1998 </td><td>M </td><td>41</td><td>T3</td><td>CR </td><td>CR                </td><td>11/5/1998 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>1/8/2002  </td></tr>\n\t<tr><th scope=row>17003</th><td>17003</td><td>4/8/1997  </td><td>F </td><td>40</td><td>T </td><td>REF</td><td>REF               </td><td>NA        </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>POS</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>18001</th><td>18001</td><td>4/23/1997 </td><td>F </td><td>28</td><td>T2</td><td>REF</td><td>REF               </td><td>NA        </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>hyperd.</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>19002</th><td>19002</td><td>1/29/1997 </td><td>M </td><td>25</td><td>T3</td><td>CR </td><td>CR                </td><td>4/10/1997 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG</td><td>NA</td><td>NEG</td><td>hyperd.</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>11/10/1997</td></tr>\n\t<tr><th scope=row>19008</th><td>19008</td><td>4/29/1998 </td><td>F </td><td>16</td><td>T2</td><td>REF</td><td>REF               </td><td>NA        </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal      </td><td>NEG</td><td>NA</td><td>POS</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>19014</th><td>19014</td><td>3/24/1999 </td><td>M </td><td>31</td><td>T2</td><td>CR </td><td>CR                </td><td>5/13/1999 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>4/1/2000  </td></tr>\n\t<tr><th scope=row>19017</th><td>19017</td><td>3/17/2000 </td><td>M </td><td>14</td><td>T2</td><td>CR </td><td>CR                </td><td>6/15/2000 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal      </td><td>NEG</td><td>NA</td><td>POS</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>7/5/2001  </td></tr>\n\t<tr><th scope=row>20005</th><td>20005</td><td>3/15/2000 </td><td>M </td><td>24</td><td>T1</td><td>CR </td><td>CR                </td><td>5/5/2000  </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>3/20/2002 </td></tr>\n\t<tr><th scope=row>24006</th><td>24006</td><td>1/14/1997 </td><td>F </td><td>19</td><td>T4</td><td>CR </td><td>CR                </td><td>not known </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>6/5/2002  </td></tr>\n\t<tr><th scope=row>26009</th><td>26009</td><td>8/26/1999 </td><td>M </td><td>37</td><td>T </td><td>CR </td><td>CR                </td><td>10/18/1999</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal      </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>6/12/2000 </td></tr>\n\t<tr><th scope=row>28008</th><td>28008</td><td>3/27/1997 </td><td>M </td><td>23</td><td>T2</td><td>CR </td><td>CR                </td><td>5/27/1997 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal      </td><td>NEG</td><td>NA</td><td>NEG</td><td>hyperd.</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>4/9/2002  </td></tr>\n\t<tr><th scope=row>28009</th><td>28009</td><td>4/19/1997 </td><td>F </td><td>30</td><td>T3</td><td>CR </td><td>CR                </td><td>6/13/1997 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>6/30/1998 </td></tr>\n\t<tr><th scope=row>31015</th><td>31015</td><td>12/3/1998 </td><td>M </td><td>48</td><td>T2</td><td>NA </td><td>DEATH IN INDUCTION</td><td>NA        </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>POS</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>37001</th><td>37001</td><td>1/30/1997 </td><td>M </td><td>22</td><td>T2</td><td>CR </td><td>CR                </td><td>4/3/1997  </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>1/7/1998  </td></tr>\n\t<tr><th scope=row>43006</th><td>43006</td><td>6/17/1997 </td><td>M </td><td>41</td><td>T2</td><td>REF</td><td>REF               </td><td>NA        </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG</td><td>NA</td><td>POS</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>43015</th><td>43015</td><td>2/29/2000 </td><td>M </td><td>52</td><td>T2</td><td>CR </td><td>CR                </td><td>6/8/2000  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>normal      </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>3/15/2002 </td></tr>\n\t<tr><th scope=row>44001</th><td>44001</td><td>1/15/1997 </td><td>M </td><td>32</td><td>T3</td><td>CR </td><td>CR                </td><td>3/7/1997  </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(6q)     </td><td>NEG</td><td>NA</td><td>NEG</td><td>NA     </td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>11/25/1997</td></tr>\n\t<tr><th scope=row>49004</th><td>49004</td><td>9/18/1997 </td><td>M </td><td>24</td><td>T3</td><td>CR </td><td>CR                </td><td>11/11/1997</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>del(7q)     </td><td>NEG</td><td>NA</td><td>POS</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>6/14/2001 </td></tr>\n\t<tr><th scope=row>56007</th><td>56007</td><td>8/6/1999  </td><td>M </td><td>37</td><td>T3</td><td>CR </td><td>CR                </td><td>9/24/1999 </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>1/26/2001 </td></tr>\n\t<tr><th scope=row>64005</th><td>64005</td><td>10/1/1998 </td><td>M </td><td>19</td><td>T2</td><td>REF</td><td>REF               </td><td>NA        </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n\t<tr><th scope=row>65003</th><td>65003</td><td>3/27/1998 </td><td>M </td><td>30</td><td>T3</td><td>CR </td><td>CR                </td><td>5/27/1998 </td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>simple alt. </td><td>NEG</td><td>NA</td><td>NEG</td><td>dyploid</td><td>FALSE</td><td> TRUE</td><td>FALSE</td><td>REL      </td><td>2/11/1999 </td></tr>\n\t<tr><th scope=row>83001</th><td>83001</td><td>10/23/1998</td><td>M </td><td>29</td><td>T2</td><td>CR </td><td>CR                </td><td>12/21/1998</td><td>FALSE</td><td>FALSE</td><td>⋯</td><td>complex alt.</td><td>NEG</td><td>NA</td><td>NEG</td><td>hyperd.</td><td> TRUE</td><td>FALSE</td><td>FALSE</td><td>CCR      </td><td>5/24/2002 </td></tr>\n\t<tr><th scope=row>LAL4</th><td>LAL4 </td><td>NA        </td><td>NA</td><td>NA</td><td>T </td><td>NA </td><td>NA                </td><td>NA        </td><td>   NA</td><td>   NA</td><td>⋯</td><td>NA          </td><td>NEG</td><td>NA</td><td>NA </td><td>NA     </td><td>   NA</td><td>   NA</td><td>   NA</td><td>NA       </td><td>NA        </td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 128 × 21\n\n| <!--/--> | cod &lt;chr&gt; | diagnosis &lt;chr&gt; | sex &lt;fct&gt; | age &lt;int&gt; | BT &lt;fct&gt; | remission &lt;fct&gt; | CR &lt;chr&gt; | date.cr &lt;chr&gt; | t(4;11) &lt;lgl&gt; | t(9;22) &lt;lgl&gt; | ⋯ ⋯ | citog &lt;chr&gt; | mol.biol &lt;fct&gt; | fusion protein &lt;fct&gt; | mdr &lt;fct&gt; | kinet &lt;fct&gt; | ccr &lt;lgl&gt; | relapse &lt;lgl&gt; | transplant &lt;lgl&gt; | f.u &lt;chr&gt; | date last seen &lt;chr&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| 01005 | 1005  | 5/21/1997  | M | 53 | B2 | CR  | CR          | 8/6/1997   | FALSE |  TRUE | ⋯ | t(9;22)         | BCR/ABL  | p210      | NEG | dyploid | FALSE | FALSE |  TRUE | BMT / DEATH IN CR | NA         |\n| 01010 | 1010  | 3/29/2000  | M | 19 | B2 | CR  | CR          | 6/27/2000  | FALSE | FALSE | ⋯ | simple alt.     | NEG      | NA        | POS | dyploid | FALSE |  TRUE | FALSE | REL               | 8/28/2000  |\n| 03002 | 3002  | 6/24/1998  | F | 52 | B4 | CR  | CR          | 8/17/1998  |    NA |    NA | ⋯ | NA              | BCR/ABL  | p190      | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 10/15/1999 |\n| 04006 | 4006  | 7/17/1997  | M | 38 | B1 | CR  | CR          | 9/8/1997   |  TRUE | FALSE | ⋯ | t(4;11)         | ALL1/AF4 | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 1/23/1998  |\n| 04007 | 4007  | 7/22/1997  | M | 57 | B2 | CR  | CR          | 9/17/1997  | FALSE | FALSE | ⋯ | del(6q)         | NEG      | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 11/4/1997  |\n| 04008 | 4008  | 7/30/1997  | M | 17 | B1 | CR  | CR          | 9/27/1997  | FALSE | FALSE | ⋯ | complex alt.    | NEG      | NA        | NEG | hyperd. | FALSE |  TRUE | FALSE | REL               | 12/15/1997 |\n| 04010 | 4010  | 10/30/1997 | F | 18 | B1 | CR  | CR          | 1/7/1998   | FALSE | FALSE | ⋯ | complex alt.    | NEG      | NA        | POS | hyperd. | FALSE |  TRUE | FALSE | REL               | 3/5/1998   |\n| 04016 | 4016  | 2/10/2000  | M | 16 | B1 | CR  | CR          | 4/17/2000  | FALSE | FALSE | ⋯ | simple alt.     | NEG      | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 9/26/2000  |\n| 06002 | 6002  | 3/19/1997  | M | 15 | B2 | CR  | CR          | 6/9/1997   | FALSE | FALSE | ⋯ | normal          | NEG      | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 3/18/1998  |\n| 08001 | 8001  | 1/15/1997  | M | 40 | B2 | CR  | CR          | 3/26/1997  | FALSE | FALSE | ⋯ | del(p15)        | BCR/ABL  | p190      | NEG | NA      | FALSE |  TRUE | FALSE | REL               | 7/11/1997  |\n| 08011 | 8011  | 8/21/1998  | M | 33 | B3 | CR  | CR          | 10/8/1998  | FALSE | FALSE | ⋯ | del(p15/p16)    | BCR/ABL  | p190/p210 | NEG | dyploid | FALSE | FALSE |  TRUE | BMT / DEATH IN CR | NA         |\n| 08012 | 8012  | 10/22/1998 | M | 55 | B3 | CR  | CR          | 1/9/1999   | FALSE | FALSE | ⋯ | simple alt.     | NEG      | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 4/9/1999   |\n| 08018 | 8018  | 8/27/1999  | M |  5 | B3 | CR  | CR          | 10/18/1999 |    NA |    NA | ⋯ | NA              | E2A/PBX1 | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 5/23/2000  |\n| 08024 | 8024  | 7/20/2000  | M | 18 | B2 | CR  | DEATH IN CR | NA         | FALSE | FALSE | ⋯ | simple alt.     | NEG      | NA        | POS | dyploid |    NA |    NA |    NA | NA                | NA         |\n| 09008 | 9008  | 12/17/1999 | M | 41 | B3 | CR  | CR          | 2/15/2000  | FALSE |  TRUE | ⋯ | t(9;22)+other   | BCR/ABL  | p190      | NEG | hyperd. |  TRUE | FALSE |  TRUE | BMT / CCR         | 00/09/01   |\n| 09017 | 9017  | 2/3/2000   | F | 27 | B  | CR  | CR          | 3/23/2000  | FALSE | FALSE | ⋯ | normal          | NEG      | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 9/11/2001  |\n| 11005 | 11005 | 6/1/1998   | M | 27 | B2 | CR  | DEATH IN CR | 8/3/1998   | FALSE | FALSE | ⋯ | del(7q) + altro | BCR/ABL  | p190      | NEG | dyploid | FALSE | FALSE | FALSE | DEATH IN CR       | NA         |\n| 12006 | 12006 | 2/20/1997  | M | 46 | B3 | REF | REF         | NA         | FALSE |  TRUE | ⋯ | t(9;22)         | BCR/ABL  | p210      | NEG | dyploid |    NA |    NA |    NA | NA                | NA         |\n| 12007 | 12007 | 3/11/1997  | M | 37 | B2 | CR  | CR          | 5/7/1997   | FALSE |  TRUE | ⋯ | t(9;22)+ del(9) | BCR/ABL  | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 9/26/1997  |\n| 12012 | 12012 | 5/21/1997  | F | 36 | B3 | REF | REF         | NA         | FALSE |  TRUE | ⋯ | t(9;22)         | BCR/ABL  | p190      | NEG | dyploid |    NA |    NA |    NA | NA                | NA         |\n| 12019 | 12019 | 9/4/1997   | M | 53 | B2 | CR  | CR          | 11/11/1997 | FALSE | FALSE | ⋯ | normal          | NEG      | NA        | POS | dyploid |  TRUE | FALSE | FALSE | CCR               | 6/6/2002   |\n| 12026 | 12026 | 5/29/1998  | M | 39 | B2 | REF | REF         | NA         | FALSE |  TRUE | ⋯ | t(9;22)         | BCR/ABL  | p190/p210 | NA  | dyploid | FALSE | FALSE | FALSE | DEATH IN CR       | NA         |\n| 14016 | 14016 | 5/27/1999  | M | 53 | B2 | NA  | NA          | NA         | FALSE |  TRUE | ⋯ | t(9;22)         | BCR/ABL  | p210      | NEG | NA      |    NA |    NA |    NA | NA                | NA         |\n| 15001 | 15001 | 9/3/1997   | M | 20 | B1 | CR  | CR          | 11/11/1997 | FALSE | FALSE | ⋯ | normal          | NEG      | NA        | NEG | dyploid |  TRUE | FALSE | FALSE | CCR               | 6/21/2002  |\n| 15004 | 15004 | 2/10/2000  | M | 44 | B1 | CR  | CR          | 4/3/2000   |  TRUE | FALSE | ⋯ | t(4;11)         | ALL1/AF4 | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 12/19/2000 |\n| 15005 | 15005 | 5/4/2000   | M | 28 | B2 | CR  | CR          | 6/26/2000  | FALSE |  TRUE | ⋯ | t(9;22)+other   | BCR/ABL  | p190      | POS | hyperd. | FALSE |  TRUE | FALSE | REL               | 11/15/2000 |\n| 16004 | 16004 | 4/19/1997  | F | 58 | B1 | CR  | CR          | 7/15/1997  |  TRUE | FALSE | ⋯ | t(4;11)         | ALL1/AF4 | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 12/9/1997  |\n| 16009 | 16009 | 7/11/2000  | F | 43 | B2 | NA  | NA          | NA         |    NA |    NA | ⋯ | NA              | NEG      | NA        | POS | dyploid |  TRUE | FALSE | FALSE | CCR / OFF         | 5/23/2002  |\n| 19005 | 19005 | 11/15/1997 | F | 48 | B1 | CR  | CR          | 2/3/1998   | FALSE | FALSE | ⋯ | normal          | ALL1/AF4 | NA        | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 2/4/1998   |\n| 20002 | 20002 | 5/9/1997   | F | 58 | B2 | CR  | CR          | 8/19/1997  | FALSE |  TRUE | ⋯ | t(9;22)+other   | BCR/ABL  | p190      | NEG | dyploid | FALSE |  TRUE | FALSE | REL               | 12/15/1997 |\n| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n| 04018 | 4018  | 3/24/2000  | M  | 17 | T2 | CR  | CR                 | 5/23/2000  | FALSE | FALSE | ⋯ | simple alt.  | NEG | NA | NEG | dyploid |  TRUE | FALSE | FALSE | CCR       | 5/14/2001  |\n| 09002 | 9002  | 5/14/1998  | F  | 40 | T3 | CR  | CR                 | 7/21/1998  |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL / SNC | 9/14/1999  |\n| 10005 | 10005 | 9/30/1997  | M  | 22 | T2 | CR  | CR                 | 1/26/1998  |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 10/10/1999 |\n| 11002 | 11002 | 12/23/1996 | M  | 30 | T  | CR  | CR                 | 3/18/1997  | FALSE | FALSE | ⋯ | normal       | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 4/14/1998  |\n| 12008 | 12008 | 3/22/1997  | M  | 18 | T4 | REF | REF                | NA         | FALSE | FALSE | ⋯ | del(6q)      | NEG | NA | NEG | dyploid |    NA |    NA |    NA | NA        | NA         |\n| 15006 | 15006 | 6/10/1998  | M  | 22 | T2 | CR  | CR                 | 8/11/1998  | FALSE | FALSE | ⋯ | complex alt. | NEG | NA | NEG | hyperd. | FALSE |  TRUE | FALSE | REL       | 9/15/2000  |\n| 16002 | 16002 | 4/10/1997  | M  | 50 | T3 | CR  | CR                 | 6/10/1997  |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | hyperd. | FALSE |  TRUE | FALSE | REL       | 12/7/1999  |\n| 16007 | 16007 | 11/1/1998  | M  | 41 | T3 | CR  | CR                 | 11/5/1998  |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid |  TRUE | FALSE | FALSE | CCR       | 1/8/2002   |\n| 17003 | 17003 | 4/8/1997   | F  | 40 | T  | REF | REF                | NA         |    NA |    NA | ⋯ | NA           | NEG | NA | POS | dyploid |    NA |    NA |    NA | NA        | NA         |\n| 18001 | 18001 | 4/23/1997  | F  | 28 | T2 | REF | REF                | NA         |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | hyperd. |    NA |    NA |    NA | NA        | NA         |\n| 19002 | 19002 | 1/29/1997  | M  | 25 | T3 | CR  | CR                 | 4/10/1997  | FALSE | FALSE | ⋯ | simple alt.  | NEG | NA | NEG | hyperd. | FALSE |  TRUE | FALSE | REL       | 11/10/1997 |\n| 19008 | 19008 | 4/29/1998  | F  | 16 | T2 | REF | REF                | NA         | FALSE | FALSE | ⋯ | normal       | NEG | NA | POS | dyploid |    NA |    NA |    NA | NA        | NA         |\n| 19014 | 19014 | 3/24/1999  | M  | 31 | T2 | CR  | CR                 | 5/13/1999  | FALSE | FALSE | ⋯ | simple alt.  | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 4/1/2000   |\n| 19017 | 19017 | 3/17/2000  | M  | 14 | T2 | CR  | CR                 | 6/15/2000  | FALSE | FALSE | ⋯ | normal       | NEG | NA | POS | dyploid | FALSE |  TRUE | FALSE | REL       | 7/5/2001   |\n| 20005 | 20005 | 3/15/2000  | M  | 24 | T1 | CR  | CR                 | 5/5/2000   |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid |  TRUE | FALSE | FALSE | CCR       | 3/20/2002  |\n| 24006 | 24006 | 1/14/1997  | F  | 19 | T4 | CR  | CR                 | not known  | FALSE | FALSE | ⋯ | simple alt.  | NEG | NA | NEG | dyploid |  TRUE | FALSE | FALSE | CCR       | 6/5/2002   |\n| 26009 | 26009 | 8/26/1999  | M  | 37 | T  | CR  | CR                 | 10/18/1999 | FALSE | FALSE | ⋯ | normal       | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 6/12/2000  |\n| 28008 | 28008 | 3/27/1997  | M  | 23 | T2 | CR  | CR                 | 5/27/1997  | FALSE | FALSE | ⋯ | normal       | NEG | NA | NEG | hyperd. |  TRUE | FALSE | FALSE | CCR       | 4/9/2002   |\n| 28009 | 28009 | 4/19/1997  | F  | 30 | T3 | CR  | CR                 | 6/13/1997  |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 6/30/1998  |\n| 31015 | 31015 | 12/3/1998  | M  | 48 | T2 | NA  | DEATH IN INDUCTION | NA         |    NA |    NA | ⋯ | NA           | NEG | NA | POS | dyploid |    NA |    NA |    NA | NA        | NA         |\n| 37001 | 37001 | 1/30/1997  | M  | 22 | T2 | CR  | CR                 | 4/3/1997   |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 1/7/1998   |\n| 43006 | 43006 | 6/17/1997  | M  | 41 | T2 | REF | REF                | NA         | FALSE | FALSE | ⋯ | simple alt.  | NEG | NA | POS | dyploid |    NA |    NA |    NA | NA        | NA         |\n| 43015 | 43015 | 2/29/2000  | M  | 52 | T2 | CR  | CR                 | 6/8/2000   | FALSE | FALSE | ⋯ | normal       | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 3/15/2002  |\n| 44001 | 44001 | 1/15/1997  | M  | 32 | T3 | CR  | CR                 | 3/7/1997   | FALSE | FALSE | ⋯ | del(6q)      | NEG | NA | NEG | NA      | FALSE |  TRUE | FALSE | REL       | 11/25/1997 |\n| 49004 | 49004 | 9/18/1997  | M  | 24 | T3 | CR  | CR                 | 11/11/1997 | FALSE | FALSE | ⋯ | del(7q)      | NEG | NA | POS | dyploid |  TRUE | FALSE | FALSE | CCR       | 6/14/2001  |\n| 56007 | 56007 | 8/6/1999   | M  | 37 | T3 | CR  | CR                 | 9/24/1999  |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid |  TRUE | FALSE | FALSE | CCR       | 1/26/2001  |\n| 64005 | 64005 | 10/1/1998  | M  | 19 | T2 | REF | REF                | NA         |    NA |    NA | ⋯ | NA           | NEG | NA | NEG | dyploid |    NA |    NA |    NA | NA        | NA         |\n| 65003 | 65003 | 3/27/1998  | M  | 30 | T3 | CR  | CR                 | 5/27/1998  | FALSE | FALSE | ⋯ | simple alt.  | NEG | NA | NEG | dyploid | FALSE |  TRUE | FALSE | REL       | 2/11/1999  |\n| 83001 | 83001 | 10/23/1998 | M  | 29 | T2 | CR  | CR                 | 12/21/1998 | FALSE | FALSE | ⋯ | complex alt. | NEG | NA | NEG | hyperd. |  TRUE | FALSE | FALSE | CCR       | 5/24/2002  |\n| LAL4 | LAL4  | NA         | NA | NA | T  | NA  | NA                 | NA         |    NA |    NA | ⋯ | NA           | NEG | NA | NA  | NA      |    NA |    NA |    NA | NA        | NA         |\n\n","text/latex":"A data.frame: 128 × 21\n\\begin{tabular}{r|lllllllllllllllllllll}\n  & cod & diagnosis & sex & age & BT & remission & CR & date.cr & t(4;11) & t(9;22) & ⋯ & citog & mol.biol & fusion protein & mdr & kinet & ccr & relapse & transplant & f.u & date last seen\\\\\n  & <chr> & <chr> & <fct> & <int> & <fct> & <fct> & <chr> & <chr> & <lgl> & <lgl> & ⋯ & <chr> & <fct> & <fct> & <fct> & <fct> & <lgl> & <lgl> & <lgl> & <chr> & <chr>\\\\\n\\hline\n\t01005 & 1005  & 5/21/1997  & M & 53 & B2 & CR  & CR          & 8/6/1997   & FALSE &  TRUE & ⋯ & t(9;22)         & BCR/ABL  & p210      & NEG & dyploid & FALSE & FALSE &  TRUE & BMT / DEATH IN CR & NA        \\\\\n\t01010 & 1010  & 3/29/2000  & M & 19 & B2 & CR  & CR          & 6/27/2000  & FALSE & FALSE & ⋯ & simple alt.     & NEG      & NA        & POS & dyploid & FALSE &  TRUE & FALSE & REL               & 8/28/2000 \\\\\n\t03002 & 3002  & 6/24/1998  & F & 52 & B4 & CR  & CR          & 8/17/1998  &    NA &    NA & ⋯ & NA              & BCR/ABL  & p190      & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 10/15/1999\\\\\n\t04006 & 4006  & 7/17/1997  & M & 38 & B1 & CR  & CR          & 9/8/1997   &  TRUE & FALSE & ⋯ & t(4;11)         & ALL1/AF4 & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 1/23/1998 \\\\\n\t04007 & 4007  & 7/22/1997  & M & 57 & B2 & CR  & CR          & 9/17/1997  & FALSE & FALSE & ⋯ & del(6q)         & NEG      & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 11/4/1997 \\\\\n\t04008 & 4008  & 7/30/1997  & M & 17 & B1 & CR  & CR          & 9/27/1997  & FALSE & FALSE & ⋯ & complex alt.    & NEG      & NA        & NEG & hyperd. & FALSE &  TRUE & FALSE & REL               & 12/15/1997\\\\\n\t04010 & 4010  & 10/30/1997 & F & 18 & B1 & CR  & CR          & 1/7/1998   & FALSE & FALSE & ⋯ & complex alt.    & NEG      & NA        & POS & hyperd. & FALSE &  TRUE & FALSE & REL               & 3/5/1998  \\\\\n\t04016 & 4016  & 2/10/2000  & M & 16 & B1 & CR  & CR          & 4/17/2000  & FALSE & FALSE & ⋯ & simple alt.     & NEG      & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 9/26/2000 \\\\\n\t06002 & 6002  & 3/19/1997  & M & 15 & B2 & CR  & CR          & 6/9/1997   & FALSE & FALSE & ⋯ & normal          & NEG      & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 3/18/1998 \\\\\n\t08001 & 8001  & 1/15/1997  & M & 40 & B2 & CR  & CR          & 3/26/1997  & FALSE & FALSE & ⋯ & del(p15)        & BCR/ABL  & p190      & NEG & NA      & FALSE &  TRUE & FALSE & REL               & 7/11/1997 \\\\\n\t08011 & 8011  & 8/21/1998  & M & 33 & B3 & CR  & CR          & 10/8/1998  & FALSE & FALSE & ⋯ & del(p15/p16)    & BCR/ABL  & p190/p210 & NEG & dyploid & FALSE & FALSE &  TRUE & BMT / DEATH IN CR & NA        \\\\\n\t08012 & 8012  & 10/22/1998 & M & 55 & B3 & CR  & CR          & 1/9/1999   & FALSE & FALSE & ⋯ & simple alt.     & NEG      & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 4/9/1999  \\\\\n\t08018 & 8018  & 8/27/1999  & M &  5 & B3 & CR  & CR          & 10/18/1999 &    NA &    NA & ⋯ & NA              & E2A/PBX1 & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 5/23/2000 \\\\\n\t08024 & 8024  & 7/20/2000  & M & 18 & B2 & CR  & DEATH IN CR & NA         & FALSE & FALSE & ⋯ & simple alt.     & NEG      & NA        & POS & dyploid &    NA &    NA &    NA & NA                & NA        \\\\\n\t09008 & 9008  & 12/17/1999 & M & 41 & B3 & CR  & CR          & 2/15/2000  & FALSE &  TRUE & ⋯ & t(9;22)+other   & BCR/ABL  & p190      & NEG & hyperd. &  TRUE & FALSE &  TRUE & BMT / CCR         & 00/09/01  \\\\\n\t09017 & 9017  & 2/3/2000   & F & 27 & B  & CR  & CR          & 3/23/2000  & FALSE & FALSE & ⋯ & normal          & NEG      & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 9/11/2001 \\\\\n\t11005 & 11005 & 6/1/1998   & M & 27 & B2 & CR  & DEATH IN CR & 8/3/1998   & FALSE & FALSE & ⋯ & del(7q) + altro & BCR/ABL  & p190      & NEG & dyploid & FALSE & FALSE & FALSE & DEATH IN CR       & NA        \\\\\n\t12006 & 12006 & 2/20/1997  & M & 46 & B3 & REF & REF         & NA         & FALSE &  TRUE & ⋯ & t(9;22)         & BCR/ABL  & p210      & NEG & dyploid &    NA &    NA &    NA & NA                & NA        \\\\\n\t12007 & 12007 & 3/11/1997  & M & 37 & B2 & CR  & CR          & 5/7/1997   & FALSE &  TRUE & ⋯ & t(9;22)+ del(9) & BCR/ABL  & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 9/26/1997 \\\\\n\t12012 & 12012 & 5/21/1997  & F & 36 & B3 & REF & REF         & NA         & FALSE &  TRUE & ⋯ & t(9;22)         & BCR/ABL  & p190      & NEG & dyploid &    NA &    NA &    NA & NA                & NA        \\\\\n\t12019 & 12019 & 9/4/1997   & M & 53 & B2 & CR  & CR          & 11/11/1997 & FALSE & FALSE & ⋯ & normal          & NEG      & NA        & POS & dyploid &  TRUE & FALSE & FALSE & CCR               & 6/6/2002  \\\\\n\t12026 & 12026 & 5/29/1998  & M & 39 & B2 & REF & REF         & NA         & FALSE &  TRUE & ⋯ & t(9;22)         & BCR/ABL  & p190/p210 & NA  & dyploid & FALSE & FALSE & FALSE & DEATH IN CR       & NA        \\\\\n\t14016 & 14016 & 5/27/1999  & M & 53 & B2 & NA  & NA          & NA         & FALSE &  TRUE & ⋯ & t(9;22)         & BCR/ABL  & p210      & NEG & NA      &    NA &    NA &    NA & NA                & NA        \\\\\n\t15001 & 15001 & 9/3/1997   & M & 20 & B1 & CR  & CR          & 11/11/1997 & FALSE & FALSE & ⋯ & normal          & NEG      & NA        & NEG & dyploid &  TRUE & FALSE & FALSE & CCR               & 6/21/2002 \\\\\n\t15004 & 15004 & 2/10/2000  & M & 44 & B1 & CR  & CR          & 4/3/2000   &  TRUE & FALSE & ⋯ & t(4;11)         & ALL1/AF4 & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 12/19/2000\\\\\n\t15005 & 15005 & 5/4/2000   & M & 28 & B2 & CR  & CR          & 6/26/2000  & FALSE &  TRUE & ⋯ & t(9;22)+other   & BCR/ABL  & p190      & POS & hyperd. & FALSE &  TRUE & FALSE & REL               & 11/15/2000\\\\\n\t16004 & 16004 & 4/19/1997  & F & 58 & B1 & CR  & CR          & 7/15/1997  &  TRUE & FALSE & ⋯ & t(4;11)         & ALL1/AF4 & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 12/9/1997 \\\\\n\t16009 & 16009 & 7/11/2000  & F & 43 & B2 & NA  & NA          & NA         &    NA &    NA & ⋯ & NA              & NEG      & NA        & POS & dyploid &  TRUE & FALSE & FALSE & CCR / OFF         & 5/23/2002 \\\\\n\t19005 & 19005 & 11/15/1997 & F & 48 & B1 & CR  & CR          & 2/3/1998   & FALSE & FALSE & ⋯ & normal          & ALL1/AF4 & NA        & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 2/4/1998  \\\\\n\t20002 & 20002 & 5/9/1997   & F & 58 & B2 & CR  & CR          & 8/19/1997  & FALSE &  TRUE & ⋯ & t(9;22)+other   & BCR/ABL  & p190      & NEG & dyploid & FALSE &  TRUE & FALSE & REL               & 12/15/1997\\\\\n\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n\t04018 & 4018  & 3/24/2000  & M  & 17 & T2 & CR  & CR                 & 5/23/2000  & FALSE & FALSE & ⋯ & simple alt.  & NEG & NA & NEG & dyploid &  TRUE & FALSE & FALSE & CCR       & 5/14/2001 \\\\\n\t09002 & 9002  & 5/14/1998  & F  & 40 & T3 & CR  & CR                 & 7/21/1998  &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL / SNC & 9/14/1999 \\\\\n\t10005 & 10005 & 9/30/1997  & M  & 22 & T2 & CR  & CR                 & 1/26/1998  &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 10/10/1999\\\\\n\t11002 & 11002 & 12/23/1996 & M  & 30 & T  & CR  & CR                 & 3/18/1997  & FALSE & FALSE & ⋯ & normal       & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 4/14/1998 \\\\\n\t12008 & 12008 & 3/22/1997  & M  & 18 & T4 & REF & REF                & NA         & FALSE & FALSE & ⋯ & del(6q)      & NEG & NA & NEG & dyploid &    NA &    NA &    NA & NA        & NA        \\\\\n\t15006 & 15006 & 6/10/1998  & M  & 22 & T2 & CR  & CR                 & 8/11/1998  & FALSE & FALSE & ⋯ & complex alt. & NEG & NA & NEG & hyperd. & FALSE &  TRUE & FALSE & REL       & 9/15/2000 \\\\\n\t16002 & 16002 & 4/10/1997  & M  & 50 & T3 & CR  & CR                 & 6/10/1997  &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & hyperd. & FALSE &  TRUE & FALSE & REL       & 12/7/1999 \\\\\n\t16007 & 16007 & 11/1/1998  & M  & 41 & T3 & CR  & CR                 & 11/5/1998  &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid &  TRUE & FALSE & FALSE & CCR       & 1/8/2002  \\\\\n\t17003 & 17003 & 4/8/1997   & F  & 40 & T  & REF & REF                & NA         &    NA &    NA & ⋯ & NA           & NEG & NA & POS & dyploid &    NA &    NA &    NA & NA        & NA        \\\\\n\t18001 & 18001 & 4/23/1997  & F  & 28 & T2 & REF & REF                & NA         &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & hyperd. &    NA &    NA &    NA & NA        & NA        \\\\\n\t19002 & 19002 & 1/29/1997  & M  & 25 & T3 & CR  & CR                 & 4/10/1997  & FALSE & FALSE & ⋯ & simple alt.  & NEG & NA & NEG & hyperd. & FALSE &  TRUE & FALSE & REL       & 11/10/1997\\\\\n\t19008 & 19008 & 4/29/1998  & F  & 16 & T2 & REF & REF                & NA         & FALSE & FALSE & ⋯ & normal       & NEG & NA & POS & dyploid &    NA &    NA &    NA & NA        & NA        \\\\\n\t19014 & 19014 & 3/24/1999  & M  & 31 & T2 & CR  & CR                 & 5/13/1999  & FALSE & FALSE & ⋯ & simple alt.  & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 4/1/2000  \\\\\n\t19017 & 19017 & 3/17/2000  & M  & 14 & T2 & CR  & CR                 & 6/15/2000  & FALSE & FALSE & ⋯ & normal       & NEG & NA & POS & dyploid & FALSE &  TRUE & FALSE & REL       & 7/5/2001  \\\\\n\t20005 & 20005 & 3/15/2000  & M  & 24 & T1 & CR  & CR                 & 5/5/2000   &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid &  TRUE & FALSE & FALSE & CCR       & 3/20/2002 \\\\\n\t24006 & 24006 & 1/14/1997  & F  & 19 & T4 & CR  & CR                 & not known  & FALSE & FALSE & ⋯ & simple alt.  & NEG & NA & NEG & dyploid &  TRUE & FALSE & FALSE & CCR       & 6/5/2002  \\\\\n\t26009 & 26009 & 8/26/1999  & M  & 37 & T  & CR  & CR                 & 10/18/1999 & FALSE & FALSE & ⋯ & normal       & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 6/12/2000 \\\\\n\t28008 & 28008 & 3/27/1997  & M  & 23 & T2 & CR  & CR                 & 5/27/1997  & FALSE & FALSE & ⋯ & normal       & NEG & NA & NEG & hyperd. &  TRUE & FALSE & FALSE & CCR       & 4/9/2002  \\\\\n\t28009 & 28009 & 4/19/1997  & F  & 30 & T3 & CR  & CR                 & 6/13/1997  &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 6/30/1998 \\\\\n\t31015 & 31015 & 12/3/1998  & M  & 48 & T2 & NA  & DEATH IN INDUCTION & NA         &    NA &    NA & ⋯ & NA           & NEG & NA & POS & dyploid &    NA &    NA &    NA & NA        & NA        \\\\\n\t37001 & 37001 & 1/30/1997  & M  & 22 & T2 & CR  & CR                 & 4/3/1997   &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 1/7/1998  \\\\\n\t43006 & 43006 & 6/17/1997  & M  & 41 & T2 & REF & REF                & NA         & FALSE & FALSE & ⋯ & simple alt.  & NEG & NA & POS & dyploid &    NA &    NA &    NA & NA        & NA        \\\\\n\t43015 & 43015 & 2/29/2000  & M  & 52 & T2 & CR  & CR                 & 6/8/2000   & FALSE & FALSE & ⋯ & normal       & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 3/15/2002 \\\\\n\t44001 & 44001 & 1/15/1997  & M  & 32 & T3 & CR  & CR                 & 3/7/1997   & FALSE & FALSE & ⋯ & del(6q)      & NEG & NA & NEG & NA      & FALSE &  TRUE & FALSE & REL       & 11/25/1997\\\\\n\t49004 & 49004 & 9/18/1997  & M  & 24 & T3 & CR  & CR                 & 11/11/1997 & FALSE & FALSE & ⋯ & del(7q)      & NEG & NA & POS & dyploid &  TRUE & FALSE & FALSE & CCR       & 6/14/2001 \\\\\n\t56007 & 56007 & 8/6/1999   & M  & 37 & T3 & CR  & CR                 & 9/24/1999  &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid &  TRUE & FALSE & FALSE & CCR       & 1/26/2001 \\\\\n\t64005 & 64005 & 10/1/1998  & M  & 19 & T2 & REF & REF                & NA         &    NA &    NA & ⋯ & NA           & NEG & NA & NEG & dyploid &    NA &    NA &    NA & NA        & NA        \\\\\n\t65003 & 65003 & 3/27/1998  & M  & 30 & T3 & CR  & CR                 & 5/27/1998  & FALSE & FALSE & ⋯ & simple alt.  & NEG & NA & NEG & dyploid & FALSE &  TRUE & FALSE & REL       & 2/11/1999 \\\\\n\t83001 & 83001 & 10/23/1998 & M  & 29 & T2 & CR  & CR                 & 12/21/1998 & FALSE & FALSE & ⋯ & complex alt. & NEG & NA & NEG & hyperd. &  TRUE & FALSE & FALSE & CCR       & 5/24/2002 \\\\\n\tLAL4 & LAL4  & NA         & NA & NA & T  & NA  & NA                 & NA         &    NA &    NA & ⋯ & NA           & NEG & NA & NA  & NA      &    NA &    NA &    NA & NA        & NA        \\\\\n\\end{tabular}\n","text/plain":"      cod   diagnosis  sex age BT remission CR                 date.cr   \n01005 1005  5/21/1997  M   53  B2 CR        CR                 8/6/1997  \n01010 1010  3/29/2000  M   19  B2 CR        CR                 6/27/2000 \n03002 3002  6/24/1998  F   52  B4 CR        CR                 8/17/1998 \n04006 4006  7/17/1997  M   38  B1 CR        CR                 9/8/1997  \n04007 4007  7/22/1997  M   57  B2 CR        CR                 9/17/1997 \n04008 4008  7/30/1997  M   17  B1 CR        CR                 9/27/1997 \n04010 4010  10/30/1997 F   18  B1 CR        CR                 1/7/1998  \n04016 4016  2/10/2000  M   16  B1 CR        CR                 4/17/2000 \n06002 6002  3/19/1997  M   15  B2 CR        CR                 6/9/1997  \n08001 8001  1/15/1997  M   40  B2 CR        CR                 3/26/1997 \n08011 8011  8/21/1998  M   33  B3 CR        CR                 10/8/1998 \n08012 8012  10/22/1998 M   55  B3 CR        CR                 1/9/1999  \n08018 8018  8/27/1999  M    5  B3 CR        CR                 10/18/1999\n08024 8024  7/20/2000  M   18  B2 CR        DEATH IN CR        NA        \n09008 9008  12/17/1999 M   41  B3 CR        CR                 2/15/2000 \n09017 9017  2/3/2000   F   27  B  CR        CR                 3/23/2000 \n11005 11005 6/1/1998   M   27  B2 CR        DEATH IN CR        8/3/1998  \n12006 12006 2/20/1997  M   46  B3 REF       REF                NA        \n12007 12007 3/11/1997  M   37  B2 CR        CR                 5/7/1997  \n12012 12012 5/21/1997  F   36  B3 REF       REF                NA        \n12019 12019 9/4/1997   M   53  B2 CR        CR                 11/11/1997\n12026 12026 5/29/1998  M   39  B2 REF       REF                NA        \n14016 14016 5/27/1999  M   53  B2 NA        NA                 NA        \n15001 15001 9/3/1997   M   20  B1 CR        CR                 11/11/1997\n15004 15004 2/10/2000  M   44  B1 CR        CR                 4/3/2000  \n15005 15005 5/4/2000   M   28  B2 CR        CR                 6/26/2000 \n16004 16004 4/19/1997  F   58  B1 CR        CR                 7/15/1997 \n16009 16009 7/11/2000  F   43  B2 NA        NA                 NA        \n19005 19005 11/15/1997 F   48  B1 CR        CR                 2/3/1998  \n20002 20002 5/9/1997   F   58  B2 CR        CR                 8/19/1997 \n⋮     ⋮     ⋮          ⋮   ⋮   ⋮  ⋮         ⋮                  ⋮         \n04018 4018  3/24/2000  M   17  T2 CR        CR                 5/23/2000 \n09002 9002  5/14/1998  F   40  T3 CR        CR                 7/21/1998 \n10005 10005 9/30/1997  M   22  T2 CR        CR                 1/26/1998 \n11002 11002 12/23/1996 M   30  T  CR        CR                 3/18/1997 \n12008 12008 3/22/1997  M   18  T4 REF       REF                NA        \n15006 15006 6/10/1998  M   22  T2 CR        CR                 8/11/1998 \n16002 16002 4/10/1997  M   50  T3 CR        CR                 6/10/1997 \n16007 16007 11/1/1998  M   41  T3 CR        CR                 11/5/1998 \n17003 17003 4/8/1997   F   40  T  REF       REF                NA        \n18001 18001 4/23/1997  F   28  T2 REF       REF                NA        \n19002 19002 1/29/1997  M   25  T3 CR        CR                 4/10/1997 \n19008 19008 4/29/1998  F   16  T2 REF       REF                NA        \n19014 19014 3/24/1999  M   31  T2 CR        CR                 5/13/1999 \n19017 19017 3/17/2000  M   14  T2 CR        CR                 6/15/2000 \n20005 20005 3/15/2000  M   24  T1 CR        CR                 5/5/2000  \n24006 24006 1/14/1997  F   19  T4 CR        CR                 not known \n26009 26009 8/26/1999  M   37  T  CR        CR                 10/18/1999\n28008 28008 3/27/1997  M   23  T2 CR        CR                 5/27/1997 \n28009 28009 4/19/1997  F   30  T3 CR        CR                 6/13/1997 \n31015 31015 12/3/1998  M   48  T2 NA        DEATH IN INDUCTION NA        \n37001 37001 1/30/1997  M   22  T2 CR        CR                 4/3/1997  \n43006 43006 6/17/1997  M   41  T2 REF       REF                NA        \n43015 43015 2/29/2000  M   52  T2 CR        CR                 6/8/2000  \n44001 44001 1/15/1997  M   32  T3 CR        CR                 3/7/1997  \n49004 49004 9/18/1997  M   24  T3 CR        CR                 11/11/1997\n56007 56007 8/6/1999   M   37  T3 CR        CR                 9/24/1999 \n64005 64005 10/1/1998  M   19  T2 REF       REF                NA        \n65003 65003 3/27/1998  M   30  T3 CR        CR                 5/27/1998 \n83001 83001 10/23/1998 M   29  T2 CR        CR                 12/21/1998\nLAL4  LAL4  NA         NA  NA  T  NA        NA                 NA        \n      t(4;11) t(9;22) ⋯ citog           mol.biol fusion protein mdr kinet  \n01005 FALSE    TRUE   ⋯ t(9;22)         BCR/ABL  p210           NEG dyploid\n01010 FALSE   FALSE   ⋯ simple alt.     NEG      NA             POS dyploid\n03002    NA      NA   ⋯ NA              BCR/ABL  p190           NEG dyploid\n04006  TRUE   FALSE   ⋯ t(4;11)         ALL1/AF4 NA             NEG dyploid\n04007 FALSE   FALSE   ⋯ del(6q)         NEG      NA             NEG dyploid\n04008 FALSE   FALSE   ⋯ complex alt.    NEG      NA             NEG hyperd.\n04010 FALSE   FALSE   ⋯ complex alt.    NEG      NA             POS hyperd.\n04016 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG dyploid\n06002 FALSE   FALSE   ⋯ normal          NEG      NA             NEG dyploid\n08001 FALSE   FALSE   ⋯ del(p15)        BCR/ABL  p190           NEG NA     \n08011 FALSE   FALSE   ⋯ del(p15/p16)    BCR/ABL  p190/p210      NEG dyploid\n08012 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG dyploid\n08018    NA      NA   ⋯ NA              E2A/PBX1 NA             NEG dyploid\n08024 FALSE   FALSE   ⋯ simple alt.     NEG      NA             POS dyploid\n09008 FALSE    TRUE   ⋯ t(9;22)+other   BCR/ABL  p190           NEG hyperd.\n09017 FALSE   FALSE   ⋯ normal          NEG      NA             NEG dyploid\n11005 FALSE   FALSE   ⋯ del(7q) + altro BCR/ABL  p190           NEG dyploid\n12006 FALSE    TRUE   ⋯ t(9;22)         BCR/ABL  p210           NEG dyploid\n12007 FALSE    TRUE   ⋯ t(9;22)+ del(9) BCR/ABL  NA             NEG dyploid\n12012 FALSE    TRUE   ⋯ t(9;22)         BCR/ABL  p190           NEG dyploid\n12019 FALSE   FALSE   ⋯ normal          NEG      NA             POS dyploid\n12026 FALSE    TRUE   ⋯ t(9;22)         BCR/ABL  p190/p210      NA  dyploid\n14016 FALSE    TRUE   ⋯ t(9;22)         BCR/ABL  p210           NEG NA     \n15001 FALSE   FALSE   ⋯ normal          NEG      NA             NEG dyploid\n15004  TRUE   FALSE   ⋯ t(4;11)         ALL1/AF4 NA             NEG dyploid\n15005 FALSE    TRUE   ⋯ t(9;22)+other   BCR/ABL  p190           POS hyperd.\n16004  TRUE   FALSE   ⋯ t(4;11)         ALL1/AF4 NA             NEG dyploid\n16009    NA      NA   ⋯ NA              NEG      NA             POS dyploid\n19005 FALSE   FALSE   ⋯ normal          ALL1/AF4 NA             NEG dyploid\n20002 FALSE    TRUE   ⋯ t(9;22)+other   BCR/ABL  p190           NEG dyploid\n⋮     ⋮       ⋮       ⋱ ⋮               ⋮        ⋮              ⋮   ⋮      \n04018 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG dyploid\n09002    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n10005    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n11002 FALSE   FALSE   ⋯ normal          NEG      NA             NEG dyploid\n12008 FALSE   FALSE   ⋯ del(6q)         NEG      NA             NEG dyploid\n15006 FALSE   FALSE   ⋯ complex alt.    NEG      NA             NEG hyperd.\n16002    NA      NA   ⋯ NA              NEG      NA             NEG hyperd.\n16007    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n17003    NA      NA   ⋯ NA              NEG      NA             POS dyploid\n18001    NA      NA   ⋯ NA              NEG      NA             NEG hyperd.\n19002 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG hyperd.\n19008 FALSE   FALSE   ⋯ normal          NEG      NA             POS dyploid\n19014 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG dyploid\n19017 FALSE   FALSE   ⋯ normal          NEG      NA             POS dyploid\n20005    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n24006 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG dyploid\n26009 FALSE   FALSE   ⋯ normal          NEG      NA             NEG dyploid\n28008 FALSE   FALSE   ⋯ normal          NEG      NA             NEG hyperd.\n28009    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n31015    NA      NA   ⋯ NA              NEG      NA             POS dyploid\n37001    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n43006 FALSE   FALSE   ⋯ simple alt.     NEG      NA             POS dyploid\n43015 FALSE   FALSE   ⋯ normal          NEG      NA             NEG dyploid\n44001 FALSE   FALSE   ⋯ del(6q)         NEG      NA             NEG NA     \n49004 FALSE   FALSE   ⋯ del(7q)         NEG      NA             POS dyploid\n56007    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n64005    NA      NA   ⋯ NA              NEG      NA             NEG dyploid\n65003 FALSE   FALSE   ⋯ simple alt.     NEG      NA             NEG dyploid\n83001 FALSE   FALSE   ⋯ complex alt.    NEG      NA             NEG hyperd.\nLAL4     NA      NA   ⋯ NA              NEG      NA             NA  NA     \n      ccr   relapse transplant f.u               date last seen\n01005 FALSE FALSE    TRUE      BMT / DEATH IN CR NA            \n01010 FALSE  TRUE   FALSE      REL               8/28/2000     \n03002 FALSE  TRUE   FALSE      REL               10/15/1999    \n04006 FALSE  TRUE   FALSE      REL               1/23/1998     \n04007 FALSE  TRUE   FALSE      REL               11/4/1997     \n04008 FALSE  TRUE   FALSE      REL               12/15/1997    \n04010 FALSE  TRUE   FALSE      REL               3/5/1998      \n04016 FALSE  TRUE   FALSE      REL               9/26/2000     \n06002 FALSE  TRUE   FALSE      REL               3/18/1998     \n08001 FALSE  TRUE   FALSE      REL               7/11/1997     \n08011 FALSE FALSE    TRUE      BMT / DEATH IN CR NA            \n08012 FALSE  TRUE   FALSE      REL               4/9/1999      \n08018 FALSE  TRUE   FALSE      REL               5/23/2000     \n08024    NA    NA      NA      NA                NA            \n09008  TRUE FALSE    TRUE      BMT / CCR         00/09/01      \n09017 FALSE  TRUE   FALSE      REL               9/11/2001     \n11005 FALSE FALSE   FALSE      DEATH IN CR       NA            \n12006    NA    NA      NA      NA                NA            \n12007 FALSE  TRUE   FALSE      REL               9/26/1997     \n12012    NA    NA      NA      NA                NA            \n12019  TRUE FALSE   FALSE      CCR               6/6/2002      \n12026 FALSE FALSE   FALSE      DEATH IN CR       NA            \n14016    NA    NA      NA      NA                NA            \n15001  TRUE FALSE   FALSE      CCR               6/21/2002     \n15004 FALSE  TRUE   FALSE      REL               12/19/2000    \n15005 FALSE  TRUE   FALSE      REL               11/15/2000    \n16004 FALSE  TRUE   FALSE      REL               12/9/1997     \n16009  TRUE FALSE   FALSE      CCR / OFF         5/23/2002     \n19005 FALSE  TRUE   FALSE      REL               2/4/1998      \n20002 FALSE  TRUE   FALSE      REL               12/15/1997    \n⋮     ⋮     ⋮       ⋮          ⋮                 ⋮             \n04018  TRUE FALSE   FALSE      CCR               5/14/2001     \n09002 FALSE  TRUE   FALSE      REL / SNC         9/14/1999     \n10005 FALSE  TRUE   FALSE      REL               10/10/1999    \n11002 FALSE  TRUE   FALSE      REL               4/14/1998     \n12008    NA    NA      NA      NA                NA            \n15006 FALSE  TRUE   FALSE      REL               9/15/2000     \n16002 FALSE  TRUE   FALSE      REL               12/7/1999     \n16007  TRUE FALSE   FALSE      CCR               1/8/2002      \n17003    NA    NA      NA      NA                NA            \n18001    NA    NA      NA      NA                NA            \n19002 FALSE  TRUE   FALSE      REL               11/10/1997    \n19008    NA    NA      NA      NA                NA            \n19014 FALSE  TRUE   FALSE      REL               4/1/2000      \n19017 FALSE  TRUE   FALSE      REL               7/5/2001      \n20005  TRUE FALSE   FALSE      CCR               3/20/2002     \n24006  TRUE FALSE   FALSE      CCR               6/5/2002      \n26009 FALSE  TRUE   FALSE      REL               6/12/2000     \n28008  TRUE FALSE   FALSE      CCR               4/9/2002      \n28009 FALSE  TRUE   FALSE      REL               6/30/1998     \n31015    NA    NA      NA      NA                NA            \n37001 FALSE  TRUE   FALSE      REL               1/7/1998      \n43006    NA    NA      NA      NA                NA            \n43015 FALSE  TRUE   FALSE      REL               3/15/2002     \n44001 FALSE  TRUE   FALSE      REL               11/25/1997    \n49004  TRUE FALSE   FALSE      CCR               6/14/2001     \n56007  TRUE FALSE   FALSE      CCR               1/26/2001     \n64005    NA    NA      NA      NA                NA            \n65003 FALSE  TRUE   FALSE      REL               2/11/1999     \n83001  TRUE FALSE   FALSE      CCR               5/24/2002     \nLAL4     NA    NA      NA      NA                NA            "},"metadata":{}}],"execution_count":23},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\n\nBiocManager::install(\"airway\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:12:36.759865Z","iopub.execute_input":"2025-04-13T11:12:36.761551Z","iopub.status.idle":"2025-04-13T11:18:42.299257Z","shell.execute_reply":"2025-04-13T11:18:42.297444Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'airway'\n\nalso installing the dependencies ‘zlibbioc’, ‘XVector’, ‘UCSC.utils’, ‘GenomeInfoDbData’, ‘SparseArray’, ‘MatrixGenerics’, ‘GenomicRanges’, ‘S4Vectors’, ‘IRanges’, ‘GenomeInfoDb’, ‘S4Arrays’, ‘DelayedArray’, ‘SummarizedExperiment’\n\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'AID', 'akc', 'ale',\n  'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'antiword', 'anytime', 'ape',\n  'aplot', 'apollo', 'aqp', 'arrow', 'ars', 'ARTool', 'arules', 'AsioHeaders',\n  'askpass', 'ASMap', 'astsa', 'automap', 'baguette', 'BAS', 'base64',\n  'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr', 'beepr',\n  'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD', 'bigstatsr',\n  'bigstep', 'billboarder', 'bindr', 'BiocManager', 'BiodiversityR', 'biomod2',\n  'biotools', 'bipartite', 'bit', 'bit64', 'bitops', 'blme', 'blorr', 'BMA',\n  'BMisc', 'bnlearn', 'bonsai', 'bookdown', 'boot', 'brms', 'broom',\n  'broom.helpers', 'broom.mixed', 'brulee', 'bslib', 'bundle', 'butcher',\n  'C50', 'car', 'caret', 'caretEnsemble', 'caTools', 'cba', 'censored',\n  'changepoint', 'cheapr', 'checkmate', 'chk', 'CholWishart', 'choroplethr',\n  'chromote', 'chron', 'circular', 'class', 'classInt', 'cld2', 'cleaner',\n  'cli', 'clock', 'clue', 'cluster', 'clusterSim', 'cmsafops',\n  'coarseDataTools', 'cobs', 'collapse', 'colorspace', 'commonmark',\n  'compareGroups', 'Compind', 'contentid', 'copula', 'CORElearn', 'coro',\n  'correlation', 'corrplot', 'countrycode', 'cpp11', 'cppdoubles',\n  'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature', 'Cubist',\n  'curl', 'cutpointr', 'cvms', 'data.table', 'datamods', 'datasauRus',\n  'datawizard', 'datetimeutils', 'dbarts', 'dbscan', 'dendextend', 'DEoptimR',\n  'Deriv', 'DescTools', 'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'digest',\n  'dismo', 'distances', 'distillery', 'distr', 'distrEx', 'distributional',\n  'DistributionUtils', 'dlm', 'dm', 'doBy', 'docopt', 'doFuture', 'doRNG',\n  'DoseFinding', 'dotCall64', 'dotwhisker', 'DoubleML', 'downloader', 'dplR',\n  'drat', 'DRDID', 'dtwclust', 'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat',\n  'ecp', 'effectsize', 'eha', 'embed', 'emmeans', 'energy', 'entropy',\n  'EnvStats', 'Epi', 'epiR', 'erer', 'ergm', 'ergm.multi', 'esquisse',\n  'estimatr', 'etm', 'evaluate', 'evd', 'Exact', 'explore', 'expm', 'extRemes',\n  'fable', 'fabletools', 'faraway', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastverse', 'fBasics', 'fda', 'fda.usc', 'fdapace', 'fdrtool',\n  'feasts', 'ff', 'fftw', 'FielDHub', 'fields', 'filehashSQLite', 'finalfit',\n  'fitdistrplus', 'FKF', 'flare', 'flexclust', 'flexmix', 'flexsurv',\n  'flextable', 'float', 'FNN', 'fontawesome', 'forecast', 'foreign',\n  'forestplot', 'fpc', 'fpp3', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa',\n  'future', 'future.apply', 'gam', 'gamboostLSS', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'genridge', 'GenSA', 'geobr', 'geodist', 'geojsonR',\n  'geometry', 'geomorph', 'geomtextpath', 'geosphere', 'gert', 'GetBCBData',\n  'ggdag', 'ggeasy', 'ggeffects', 'ggfun', 'ggh4x', 'ggiraph', 'ggmap',\n  'ggnewscale', 'ggpath', 'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc',\n  'ggrepel', 'ggstats', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmtoolbox', 'glue', 'gmapsdistance', 'gmp',\n  'GoFKernel', 'golem', 'googlePolylines', 'gower', 'GPArotation', 'GPfit',\n  'gplots', 'grafify', 'gRain', 'graphlayouts', 'gRbase', 'greybox', 'grf',\n  'gridpattern', 'grImport2', 'groupdata2', 'grpreg', 'gsheet', 'gss', 'gstat',\n  'gsw', 'gt', 'gtable', 'gtsummary', 'GWalkR', 'GWmodel', 'h3lib', 'h3r',\n  'hardhat', 'harmony', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'htmlTable', 'hts', 'httr2', 'hunspell', 'huxtable',\n  'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph', 'ijtiff', 'imager', 'iml',\n  'inline', 'insight', 'inspectdf', 'interactions', 'intervals', 'ipred',\n  'isotree', 'ISwR', 'ivreg', 'janitor', 'jiebaR', 'JMbayes2', 'jmvconnect',\n  'jmvcore', 'jmvReadWrite', 'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity',\n  'keras3', 'kernelshap', 'kernlab', 'KernSmooth', 'khroma', 'kit', 'knitr',\n  'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman', 'lares', 'later',\n  'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave', 'lcmm', 'leafgl',\n  'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR', 'lightgbm', 'limma', 'lintr',\n  'littler', 'lme4', 'lmodel2', 'lmom', 'locfit', 'logger', 'longitudinalData',\n  'loo', 'lordif', 'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa',\n  'lubridate', 'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'matlib', 'Matrix', 'MatrixModels',\n  'matrixStats', 'MazamaCoreUtils', 'MazamaSpatialUtils', 'MBA', 'mboost',\n  'MCMCpack', 'mda', 'mdsr', 'memisc', 'meta', 'metacoder', 'metadat',\n  'metafor', 'meteospain', 'MetricsWeighted', 'mets', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'minqa', 'minty', 'mirt',\n  'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench', 'mlflow',\n  'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data', 'mlr3filters',\n  'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures', 'mlr3misc',\n  'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse', 'mlr3viz',\n  'modelbased', 'modelenv', 'modelsummary', 'modeltime', 'moderndive',\n  'mongolite', 'monitoR', 'msigdbr', 'msm', 'mstate', 'mtsdi', 'multcomp',\n  'MuMIn', 'mvtnorm', 'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson',\n  'ndtv', 'network', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'OpenMx', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ore', 'osmextract',\n  'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown', 'pammtools',\n  'pander', 'parabar', 'paradox', 'parallelly', 'parameters', 'ParamHelpers',\n  'paran', 'parsnip', 'party', 'partykit', 'patchwork', 'pbdZMQ', 'pcaPP',\n  'pdftools', 'pdp', 'pdR', 'PearsonDS', 'performance', 'PerformanceAnalytics',\n  'pgmm', 'phangorn', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools', 'pls',\n  'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip', 'popdemo',\n  'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'PreProcess', 'prettyGraphs', 'priceR', 'processx', 'profvis',\n  'progressr', 'PROJ', 'proj4', 'promises', 'prospectr', 'protolite', 'PRROC',\n  'ps', 'pspline', 'PSweight', 'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf',\n  'qs', 'qtl', 'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr',\n  'QuickJSR', 'qvcalc', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'reactR',\n  'readODS', 'readxl', 'rearrr', 'rebmix', 'recipes', 'refund', 'renv',\n  'report', 'reprex', 'reticulate', 'Rfast', 'rfishbase', 'rgbif', 'rgenoud',\n  'rgl', 'rio', 'rjson', 'RJSONIO', 'rlang', 'RMariaDB', 'rmarkdown', 'rminer',\n  'rms', 'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpostgis', 'RPostgres', 'RPostgreSQL',\n  'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample', 'rsconnect',\n  'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite', 'Rssa',\n  'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rugarch', 'Rvcg', 'rvg',\n  's2', 'sandwich', 'sass', 'SCtools', 'sda', 'sdm', 'seasonal',\n  'seasonalview', 'see', 'seededlda', 'segmented', 'sem', 'semTools',\n  'sensemakr', 'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinyWidgets', 'SimDesign',\n  'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot', 'skmeans', 'slam',\n  'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2', 'sna', 'sodium',\n  'soilDB', 'sommer', 'sp', 'spacefillr', 'spacetime', 'spam', 'sparklyr',\n  'SparseM', 'spatialreg', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'strucchange', 'summarytools',\n  'SuppDists', 'survival', 'survminer', 'svd', 'sys', 'systemfonts', 'tables',\n  'tau', 'taxize', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract',\n  'testthat', 'texreg', 'text', 'textcat', 'textrecipes', 'textshaping',\n  'textutils', 'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'timeDate', 'timeplyr', 'timereg', 'timeSeries', 'tinytable',\n  'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui', 'topicmodels',\n  'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal', 'tsbox', 'tsDyn',\n  'tseries', 'tsibble', 'tuber', 'tune', 'tzdb', 'units', 'univariateML',\n  'unpivotr', 'usethis', 'usmapdata', 'uuid', 'uwot', 'V8', 'vcd', 'vegan',\n  'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR', 'wbacon',\n  'WDI', 'websocket', 'webutils', 'WeightSVM', 'withr', 'wk', 'wooldridge',\n  'workflows', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick',\n  'yulab.utils', 'zip', 'zoo', 'spatial'\n\n","output_type":"stream"}],"execution_count":24},{"cell_type":"code","source":"library(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:20:41.581286Z","iopub.execute_input":"2025-04-13T11:20:41.583017Z","iopub.status.idle":"2025-04-13T11:20:48.154496Z","shell.execute_reply":"2025-04-13T11:20:48.152730Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"Loading required package: SummarizedExperiment\n\nLoading required package: MatrixGenerics\n\nLoading required package: matrixStats\n\n\nAttaching package: ‘matrixStats’\n\n\nThe following objects are masked from ‘package:Biobase’:\n\n    anyMissing, rowMedians\n\n\n\nAttaching package: ‘MatrixGenerics’\n\n\nThe following objects are masked from ‘package:matrixStats’:\n\n    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n    colWeightedMeans, colWeightedMedians, colWeightedSds,\n    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n    rowWeightedSds, rowWeightedVars\n\n\nThe following object is masked from ‘package:Biobase’:\n\n    rowMedians\n\n\nLoading required package: GenomicRanges\n\nLoading required package: stats4\n\nLoading required package: S4Vectors\n\n\nAttaching package: ‘S4Vectors’\n\n\nThe following object is masked from ‘package:utils’:\n\n    findMatches\n\n\nThe following objects are masked from ‘package:base’:\n\n    expand.grid, I, unname\n\n\nLoading required package: IRanges\n\nLoading required package: GenomeInfoDb\n\n","output_type":"stream"}],"execution_count":25},{"cell_type":"code","source":"data(airway)\nairway","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:20:55.967717Z","iopub.execute_input":"2025-04-13T11:20:55.969863Z","iopub.status.idle":"2025-04-13T11:20:57.961474Z","shell.execute_reply":"2025-04-13T11:20:57.959753Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"class: RangedSummarizedExperiment \ndim: 63677 8 \nmetadata(1): ''\nassays(1): counts\nrownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492\n  ENSG00000273493\nrowData names(10): gene_id gene_name ... seq_coord_system symbol\ncolnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521\ncolData names(9): SampleName cell ... Sample BioSample"},"metadata":{}}],"execution_count":26},{"cell_type":"code","source":"colData(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:20:59.669136Z","iopub.execute_input":"2025-04-13T11:20:59.670718Z","iopub.status.idle":"2025-04-13T11:20:59.803283Z","shell.execute_reply":"2025-04-13T11:20:59.801424Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"DataFrame with 8 rows and 9 columns\n           SampleName     cell      dex    albut        Run avgLength\n             <factor> <factor> <factor> <factor>   <factor> <integer>\nSRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126\nSRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126\nSRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126\nSRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87\nSRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120\nSRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126\nSRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101\nSRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98\n           Experiment    Sample    BioSample\n             <factor>  <factor>     <factor>\nSRR1039508  SRX384345 SRS508568 SAMN02422669\nSRR1039509  SRX384346 SRS508567 SAMN02422675\nSRR1039512  SRX384349 SRS508571 SAMN02422678\nSRR1039513  SRX384350 SRS508572 SAMN02422670\nSRR1039516  SRX384353 SRS508575 SAMN02422682\nSRR1039517  SRX384354 SRS508576 SAMN02422673\nSRR1039520  SRX384357 SRS508579 SAMN02422683\nSRR1039521  SRX384358 SRS508580 SAMN02422677"},"metadata":{}}],"execution_count":27},{"cell_type":"code","source":"exprimentData(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:26:08.580748Z","iopub.execute_input":"2025-04-13T11:26:08.582551Z","iopub.status.idle":"2025-04-13T11:26:08.598731Z","shell.execute_reply":"2025-04-13T11:26:08.596661Z"}},"outputs":[{"ename":"ERROR","evalue":"Error in exprimentData(airway): could not find function \"exprimentData\"\n","traceback":["Error in exprimentData(airway): could not find function \"exprimentData\"\nTraceback:\n"],"output_type":"error"}],"execution_count":38},{"cell_type":"code","source":"#as.list(exptdata(airway))\n?airway","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:44:58.161950Z","iopub.execute_input":"2025-04-13T12:44:58.163610Z","iopub.status.idle":"2025-04-13T12:44:58.268422Z","shell.execute_reply":"2025-04-13T12:44:58.266231Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"\n<table style=\"width: 100%;\"><tr><td>airway {airway}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n\n<h2>Airway smooth muscle cells</h2>\n\n<h3>Description</h3>\n\n<p>Read counts per gene for airway smooth muscle cell lines RNA-Seq experiment</p>\n\n\n<h3>Usage</h3>\n\n<pre><code class='language-R'>data(\"airway\")\n</code></pre>\n\n\n<h3>Format</h3>\n\n<p>RangedSummarizedExperiment</p>\n\n\n<h3>Details</h3>\n\n<p>For details on the gene model and read counting procedure, see the\npackage vignette. \n</p>\n<p>The dataset contains four cell lines in two conditions: control and\ntreatment with dexamethasone.\n</p>\n\n\n<h3>Source</h3>\n\n<p>FASTQ files from SRA, phenotypic data from GEO</p>\n\n\n<h3>References</h3>\n\n<p>Himes BE, Jiang X, Wagner P, Hu R, Wang Q, Klanderman B,\nWhitaker RM, Duan Q, Lasky-Su J, Nikolos C, Jester W, Johnson M,\nPanettieri R Jr, Tantisira KG, Weiss ST, Lu Q. 'RNA-Seq Transcriptome\nProfiling Identifies CRISPLD2 as a Glucocorticoid Responsive Gene that\nModulates Cytokine Function in Airway Smooth Muscle Cells.' PLoS\nOne. 2014 Jun 13;9(6):e99625. PMID: 24926665. GEO: GSE52778.\n</p>\n\n<hr /><div style=\"text-align: center;\">[Package <em>airway</em> version 1.24.0 ]</div></main>\n\n</div>","text/latex":"\\inputencoding{utf8}\n\\HeaderA{airway}{Airway smooth muscle cells}{airway}\n\\keyword{datasets}{airway}\n%\n\\begin{Description}\nRead counts per gene for airway smooth muscle cell lines RNA-Seq experiment\n\\end{Description}\n%\n\\begin{Usage}\n\\begin{verbatim}\ndata(\"airway\")\n\\end{verbatim}\n\\end{Usage}\n%\n\\begin{Format}\nRangedSummarizedExperiment\n\\end{Format}\n%\n\\begin{Details}\nFor details on the gene model and read counting procedure, see the\npackage vignette. \n\nThe dataset contains four cell lines in two conditions: control and\ntreatment with dexamethasone.\n\\end{Details}\n%\n\\begin{Source}\nFASTQ files from SRA, phenotypic data from GEO\n\\end{Source}\n%\n\\begin{References}\nHimes BE, Jiang X, Wagner P, Hu R, Wang Q, Klanderman B,\nWhitaker RM, Duan Q, Lasky-Su J, Nikolos C, Jester W, Johnson M,\nPanettieri R Jr, Tantisira KG, Weiss ST, Lu Q. 'RNA-Seq Transcriptome\nProfiling Identifies CRISPLD2 as a Glucocorticoid Responsive Gene that\nModulates Cytokine Function in Airway Smooth Muscle Cells.' PLoS\nOne. 2014 Jun 13;9(6):e99625. PMID: 24926665. GEO: GSE52778.\n\\end{References}","text/plain":"airway                 package:airway                  R Documentation\n\n_\bA_\bi_\br_\bw_\ba_\by _\bs_\bm_\bo_\bo_\bt_\bh _\bm_\bu_\bs_\bc_\bl_\be _\bc_\be_\bl_\bl_\bs\n\n_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n\n     Read counts per gene for airway smooth muscle cell lines RNA-Seq\n     experiment\n\n_\bU_\bs_\ba_\bg_\be:\n\n     data(\"airway\")\n     \n_\bF_\bo_\br_\bm_\ba_\bt:\n\n     RangedSummarizedExperiment\n\n_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n\n     For details on the gene model and read counting procedure, see the\n     package vignette.\n\n     The dataset contains four cell lines in two conditions: control\n     and treatment with dexamethasone.\n\n_\bS_\bo_\bu_\br_\bc_\be:\n\n     FASTQ files from SRA, phenotypic data from GEO\n\n_\bR_\be_\bf_\be_\br_\be_\bn_\bc_\be_\bs:\n\n     Himes BE, Jiang X, Wagner P, Hu R, Wang Q, Klanderman B, Whitaker\n     RM, Duan Q, Lasky-Su J, Nikolos C, Jester W, Johnson M, Panettieri\n     R Jr, Tantisira KG, Weiss ST, Lu Q. 'RNA-Seq Transcriptome\n     Profiling Identifies CRISPLD2 as a Glucocorticoid Responsive Gene\n     that Modulates Cytokine Function in Airway Smooth Muscle Cells.'\n     PLoS One. 2014 Jun 13;9(6):e99625. PMID: 24926665. GEO: GSE52778.\n"},"metadata":{}}],"execution_count":82},{"cell_type":"code","source":"colnames(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:26:53.436698Z","iopub.execute_input":"2025-04-13T11:26:53.439299Z","iopub.status.idle":"2025-04-13T11:26:53.462495Z","shell.execute_reply":"2025-04-13T11:26:53.459768Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'SRR1039508'</li><li>'SRR1039509'</li><li>'SRR1039512'</li><li>'SRR1039513'</li><li>'SRR1039516'</li><li>'SRR1039517'</li><li>'SRR1039520'</li><li>'SRR1039521'</li></ol>\n","text/markdown":"1. 'SRR1039508'\n2. 'SRR1039509'\n3. 'SRR1039512'\n4. 'SRR1039513'\n5. 'SRR1039516'\n6. 'SRR1039517'\n7. 'SRR1039520'\n8. 'SRR1039521'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'SRR1039508'\n\\item 'SRR1039509'\n\\item 'SRR1039512'\n\\item 'SRR1039513'\n\\item 'SRR1039516'\n\\item 'SRR1039517'\n\\item 'SRR1039520'\n\\item 'SRR1039521'\n\\end{enumerate*}\n","text/plain":"[1] \"SRR1039508\" \"SRR1039509\" \"SRR1039512\" \"SRR1039513\" \"SRR1039516\"\n[6] \"SRR1039517\" \"SRR1039520\" \"SRR1039521\""},"metadata":{}}],"execution_count":41},{"cell_type":"code","source":"head(rownames(airway))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:26:59.714198Z","iopub.execute_input":"2025-04-13T11:26:59.716568Z","iopub.status.idle":"2025-04-13T11:26:59.743952Z","shell.execute_reply":"2025-04-13T11:26:59.741442Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'ENSG00000000003'</li><li>'ENSG00000000005'</li><li>'ENSG00000000419'</li><li>'ENSG00000000457'</li><li>'ENSG00000000460'</li><li>'ENSG00000000938'</li></ol>\n","text/markdown":"1. 'ENSG00000000003'\n2. 'ENSG00000000005'\n3. 'ENSG00000000419'\n4. 'ENSG00000000457'\n5. 'ENSG00000000460'\n6. 'ENSG00000000938'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'ENSG00000000003'\n\\item 'ENSG00000000005'\n\\item 'ENSG00000000419'\n\\item 'ENSG00000000457'\n\\item 'ENSG00000000460'\n\\item 'ENSG00000000938'\n\\end{enumerate*}\n","text/plain":"[1] \"ENSG00000000003\" \"ENSG00000000005\" \"ENSG00000000419\" \"ENSG00000000457\"\n[5] \"ENSG00000000460\" \"ENSG00000000938\""},"metadata":{}}],"execution_count":42},{"cell_type":"code","source":"assayNames(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:27:23.276002Z","iopub.execute_input":"2025-04-13T11:27:23.278373Z","iopub.status.idle":"2025-04-13T11:27:23.298021Z","shell.execute_reply":"2025-04-13T11:27:23.295819Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"'counts'","text/markdown":"'counts'","text/latex":"'counts'","text/plain":"[1] \"counts\""},"metadata":{}}],"execution_count":44},{"cell_type":"code","source":"assay(airway,'counts')","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:27:24.695266Z","iopub.execute_input":"2025-04-13T11:27:24.697416Z","iopub.status.idle":"2025-04-13T11:27:25.016742Z","shell.execute_reply":"2025-04-13T11:27:25.014868Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A matrix: 63677 × 8 of type int</caption>\n<thead>\n\t<tr><th></th><th scope=col>SRR1039508</th><th scope=col>SRR1039509</th><th scope=col>SRR1039512</th><th scope=col>SRR1039513</th><th scope=col>SRR1039516</th><th scope=col>SRR1039517</th><th scope=col>SRR1039520</th><th scope=col>SRR1039521</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>ENSG00000000003</th><td> 679</td><td> 448</td><td> 873</td><td> 408</td><td> 1138</td><td> 1047</td><td> 770</td><td> 572</td></tr>\n\t<tr><th scope=row>ENSG00000000005</th><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>    0</td><td>    0</td><td>   0</td><td>   0</td></tr>\n\t<tr><th scope=row>ENSG00000000419</th><td> 467</td><td> 515</td><td> 621</td><td> 365</td><td>  587</td><td>  799</td><td> 417</td><td> 508</td></tr>\n\t<tr><th scope=row>ENSG00000000457</th><td> 260</td><td> 211</td><td> 263</td><td> 164</td><td>  245</td><td>  331</td><td> 233</td><td> 229</td></tr>\n\t<tr><th scope=row>ENSG00000000460</th><td>  60</td><td>  55</td><td>  40</td><td>  35</td><td>   78</td><td>   63</td><td>  76</td><td>  60</td></tr>\n\t<tr><th scope=row>ENSG00000000938</th><td>   0</td><td>   0</td><td>   2</td><td>   0</td><td>    1</td><td>    0</td><td>   0</td><td>   0</td></tr>\n\t<tr><th scope=row>ENSG00000000971</th><td>3251</td><td>3679</td><td>6177</td><td>4252</td><td> 6721</td><td>11027</td><td>5176</td><td>7995</td></tr>\n\t<tr><th scope=row>ENSG00000001036</th><td>1433</td><td>1062</td><td>1733</td><td> 881</td><td> 1424</td><td> 1439</td><td>1359</td><td>1109</td></tr>\n\t<tr><th scope=row>ENSG00000001084</th><td> 519</td><td> 380</td><td> 595</td><td> 493</td><td>  820</td><td>  714</td><td> 696</td><td> 704</td></tr>\n\t<tr><th scope=row>ENSG00000001167</th><td> 394</td><td> 236</td><td> 464</td><td> 175</td><td>  658</td><td>  584</td><td> 360</td><td> 269</td></tr>\n\t<tr><th scope=row>ENSG00000001460</th><td> 172</td><td> 168</td><td> 264</td><td> 118</td><td>  241</td><td>  210</td><td> 155</td><td> 177</td></tr>\n\t<tr><th scope=row>ENSG00000001461</th><td>2112</td><td>1867</td><td>5137</td><td>2657</td><td> 2735</td><td> 2751</td><td>2467</td><td>2905</td></tr>\n\t<tr><th scope=row>ENSG00000001497</th><td> 524</td><td> 488</td><td> 638</td><td> 357</td><td>  676</td><td>  806</td><td> 493</td><td> 475</td></tr>\n\t<tr><th scope=row>ENSG00000001561</th><td>  71</td><td>  51</td><td> 211</td><td> 156</td><td>   23</td><td>   38</td><td> 134</td><td> 172</td></tr>\n\t<tr><th scope=row>ENSG00000001617</th><td> 555</td><td> 394</td><td> 905</td><td> 415</td><td>  727</td><td>  697</td><td> 618</td><td> 599</td></tr>\n\t<tr><th scope=row>ENSG00000001626</th><td>  10</td><td>   2</td><td>   9</td><td>   2</td><td>   10</td><td>    6</td><td>   5</td><td>   5</td></tr>\n\t<tr><th scope=row>ENSG00000001629</th><td>1660</td><td>1251</td><td>2259</td><td>1079</td><td> 2462</td><td> 2514</td><td>1888</td><td>1660</td></tr>\n\t<tr><th scope=row>ENSG00000001630</th><td>  59</td><td>  54</td><td>  66</td><td>  23</td><td>   84</td><td>   87</td><td>  31</td><td>  59</td></tr>\n\t<tr><th scope=row>ENSG00000001631</th><td> 729</td><td> 692</td><td> 943</td><td> 475</td><td> 1034</td><td> 1163</td><td> 731</td><td> 744</td></tr>\n\t<tr><th scope=row>ENSG00000002016</th><td> 201</td><td> 161</td><td> 256</td><td>  99</td><td>  268</td><td>  257</td><td> 160</td><td> 137</td></tr>\n\t<tr><th scope=row>ENSG00000002079</th><td>   3</td><td>   0</td><td>   3</td><td>   1</td><td>    4</td><td>    0</td><td>   0</td><td>   1</td></tr>\n\t<tr><th scope=row>ENSG00000002330</th><td> 206</td><td> 174</td><td> 184</td><td> 111</td><td>  194</td><td>  260</td><td> 156</td><td> 177</td></tr>\n\t<tr><th scope=row>ENSG00000002549</th><td>1459</td><td>1294</td><td>1317</td><td> 998</td><td> 1451</td><td> 1824</td><td> 853</td><td>1031</td></tr>\n\t<tr><th scope=row>ENSG00000002586</th><td>7507</td><td>7203</td><td>9501</td><td>6214</td><td>10973</td><td>12863</td><td>6834</td><td>7225</td></tr>\n\t<tr><th scope=row>ENSG00000002587</th><td>   2</td><td>   0</td><td>   1</td><td>   0</td><td>    0</td><td>    2</td><td>   0</td><td>   0</td></tr>\n\t<tr><th scope=row>ENSG00000002726</th><td>   0</td><td>   0</td><td>   1</td><td>   0</td><td>    0</td><td>    0</td><td>   0</td><td>   0</td></tr>\n\t<tr><th scope=row>ENSG00000002745</th><td>   4</td><td>   6</td><td>  22</td><td>  10</td><td>    2</td><td>    1</td><td>   5</td><td>   3</td></tr>\n\t<tr><th scope=row>ENSG00000002746</th><td> 151</td><td> 139</td><td> 117</td><td>  65</td><td>   90</td><td>  102</td><td>  86</td><td> 119</td></tr>\n\t<tr><th scope=row>ENSG00000002822</th><td> 411</td><td> 303</td><td> 446</td><td> 195</td><td>  445</td><td>  523</td><td> 295</td><td> 300</td></tr>\n\t<tr><th scope=row>ENSG00000002834</th><td>6314</td><td>6364</td><td>7831</td><td>5809</td><td> 6677</td><td>11970</td><td>5766</td><td>7825</td></tr>\n\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n\t<tr><th scope=row>ENSG00000273464</th><td> 0</td><td> 2</td><td> 2</td><td> 0</td><td> 4</td><td> 2</td><td> 1</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273465</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273466</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 1</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273467</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273468</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273469</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273470</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273471</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273472</th><td> 6</td><td> 6</td><td>18</td><td> 9</td><td>19</td><td> 9</td><td>12</td><td>11</td></tr>\n\t<tr><th scope=row>ENSG00000273473</th><td> 0</td><td> 0</td><td> 3</td><td> 0</td><td> 2</td><td> 2</td><td> 2</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273474</th><td> 2</td><td> 1</td><td> 3</td><td> 1</td><td> 2</td><td> 1</td><td> 0</td><td> 3</td></tr>\n\t<tr><th scope=row>ENSG00000273475</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273476</th><td> 1</td><td> 2</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 1</td></tr>\n\t<tr><th scope=row>ENSG00000273477</th><td> 4</td><td> 9</td><td> 6</td><td> 2</td><td> 8</td><td> 9</td><td> 6</td><td>12</td></tr>\n\t<tr><th scope=row>ENSG00000273478</th><td> 9</td><td> 8</td><td> 7</td><td> 5</td><td> 6</td><td> 8</td><td> 5</td><td> 6</td></tr>\n\t<tr><th scope=row>ENSG00000273479</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273480</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273481</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273482</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273483</th><td> 5</td><td> 2</td><td> 0</td><td> 2</td><td> 3</td><td> 2</td><td> 1</td><td> 6</td></tr>\n\t<tr><th scope=row>ENSG00000273484</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273485</th><td> 2</td><td> 3</td><td> 1</td><td> 1</td><td> 1</td><td> 1</td><td> 1</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273486</th><td>14</td><td>11</td><td>25</td><td> 8</td><td>20</td><td>32</td><td>12</td><td>11</td></tr>\n\t<tr><th scope=row>ENSG00000273487</th><td> 5</td><td> 9</td><td> 4</td><td>11</td><td>10</td><td>10</td><td> 4</td><td>10</td></tr>\n\t<tr><th scope=row>ENSG00000273488</th><td> 7</td><td> 5</td><td> 8</td><td> 3</td><td> 8</td><td>16</td><td>11</td><td>14</td></tr>\n\t<tr><th scope=row>ENSG00000273489</th><td> 0</td><td> 0</td><td> 0</td><td> 1</td><td> 0</td><td> 1</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273490</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273491</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273492</th><td> 0</td><td> 0</td><td> 1</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n\t<tr><th scope=row>ENSG00000273493</th><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 1</td><td> 0</td><td> 0</td><td> 0</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA matrix: 63677 × 8 of type int\n\n| <!--/--> | SRR1039508 | SRR1039509 | SRR1039512 | SRR1039513 | SRR1039516 | SRR1039517 | SRR1039520 | SRR1039521 |\n|---|---|---|---|---|---|---|---|---|\n| ENSG00000000003 |  679 |  448 |  873 |  408 |  1138 |  1047 |  770 |  572 |\n| ENSG00000000005 |    0 |    0 |    0 |    0 |     0 |     0 |    0 |    0 |\n| ENSG00000000419 |  467 |  515 |  621 |  365 |   587 |   799 |  417 |  508 |\n| ENSG00000000457 |  260 |  211 |  263 |  164 |   245 |   331 |  233 |  229 |\n| ENSG00000000460 |   60 |   55 |   40 |   35 |    78 |    63 |   76 |   60 |\n| ENSG00000000938 |    0 |    0 |    2 |    0 |     1 |     0 |    0 |    0 |\n| ENSG00000000971 | 3251 | 3679 | 6177 | 4252 |  6721 | 11027 | 5176 | 7995 |\n| ENSG00000001036 | 1433 | 1062 | 1733 |  881 |  1424 |  1439 | 1359 | 1109 |\n| ENSG00000001084 |  519 |  380 |  595 |  493 |   820 |   714 |  696 |  704 |\n| ENSG00000001167 |  394 |  236 |  464 |  175 |   658 |   584 |  360 |  269 |\n| ENSG00000001460 |  172 |  168 |  264 |  118 |   241 |   210 |  155 |  177 |\n| ENSG00000001461 | 2112 | 1867 | 5137 | 2657 |  2735 |  2751 | 2467 | 2905 |\n| ENSG00000001497 |  524 |  488 |  638 |  357 |   676 |   806 |  493 |  475 |\n| ENSG00000001561 |   71 |   51 |  211 |  156 |    23 |    38 |  134 |  172 |\n| ENSG00000001617 |  555 |  394 |  905 |  415 |   727 |   697 |  618 |  599 |\n| ENSG00000001626 |   10 |    2 |    9 |    2 |    10 |     6 |    5 |    5 |\n| ENSG00000001629 | 1660 | 1251 | 2259 | 1079 |  2462 |  2514 | 1888 | 1660 |\n| ENSG00000001630 |   59 |   54 |   66 |   23 |    84 |    87 |   31 |   59 |\n| ENSG00000001631 |  729 |  692 |  943 |  475 |  1034 |  1163 |  731 |  744 |\n| ENSG00000002016 |  201 |  161 |  256 |   99 |   268 |   257 |  160 |  137 |\n| ENSG00000002079 |    3 |    0 |    3 |    1 |     4 |     0 |    0 |    1 |\n| ENSG00000002330 |  206 |  174 |  184 |  111 |   194 |   260 |  156 |  177 |\n| ENSG00000002549 | 1459 | 1294 | 1317 |  998 |  1451 |  1824 |  853 | 1031 |\n| ENSG00000002586 | 7507 | 7203 | 9501 | 6214 | 10973 | 12863 | 6834 | 7225 |\n| ENSG00000002587 |    2 |    0 |    1 |    0 |     0 |     2 |    0 |    0 |\n| ENSG00000002726 |    0 |    0 |    1 |    0 |     0 |     0 |    0 |    0 |\n| ENSG00000002745 |    4 |    6 |   22 |   10 |     2 |     1 |    5 |    3 |\n| ENSG00000002746 |  151 |  139 |  117 |   65 |    90 |   102 |   86 |  119 |\n| ENSG00000002822 |  411 |  303 |  446 |  195 |   445 |   523 |  295 |  300 |\n| ENSG00000002834 | 6314 | 6364 | 7831 | 5809 |  6677 | 11970 | 5766 | 7825 |\n| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n| ENSG00000273464 |  0 |  2 |  2 |  0 |  4 |  2 |  1 |  0 |\n| ENSG00000273465 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273466 |  0 |  0 |  0 |  0 |  0 |  0 |  1 |  0 |\n| ENSG00000273467 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273468 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273469 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273470 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273471 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273472 |  6 |  6 | 18 |  9 | 19 |  9 | 12 | 11 |\n| ENSG00000273473 |  0 |  0 |  3 |  0 |  2 |  2 |  2 |  0 |\n| ENSG00000273474 |  2 |  1 |  3 |  1 |  2 |  1 |  0 |  3 |\n| ENSG00000273475 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273476 |  1 |  2 |  0 |  0 |  0 |  0 |  0 |  1 |\n| ENSG00000273477 |  4 |  9 |  6 |  2 |  8 |  9 |  6 | 12 |\n| ENSG00000273478 |  9 |  8 |  7 |  5 |  6 |  8 |  5 |  6 |\n| ENSG00000273479 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273480 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273481 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273482 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273483 |  5 |  2 |  0 |  2 |  3 |  2 |  1 |  6 |\n| ENSG00000273484 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273485 |  2 |  3 |  1 |  1 |  1 |  1 |  1 |  0 |\n| ENSG00000273486 | 14 | 11 | 25 |  8 | 20 | 32 | 12 | 11 |\n| ENSG00000273487 |  5 |  9 |  4 | 11 | 10 | 10 |  4 | 10 |\n| ENSG00000273488 |  7 |  5 |  8 |  3 |  8 | 16 | 11 | 14 |\n| ENSG00000273489 |  0 |  0 |  0 |  1 |  0 |  1 |  0 |  0 |\n| ENSG00000273490 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273491 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273492 |  0 |  0 |  1 |  0 |  0 |  0 |  0 |  0 |\n| ENSG00000273493 |  0 |  0 |  0 |  0 |  1 |  0 |  0 |  0 |\n\n","text/latex":"A matrix: 63677 × 8 of type int\n\\begin{tabular}{r|llllllll}\n  & SRR1039508 & SRR1039509 & SRR1039512 & SRR1039513 & SRR1039516 & SRR1039517 & SRR1039520 & SRR1039521\\\\\n\\hline\n\tENSG00000000003 &  679 &  448 &  873 &  408 &  1138 &  1047 &  770 &  572\\\\\n\tENSG00000000005 &    0 &    0 &    0 &    0 &     0 &     0 &    0 &    0\\\\\n\tENSG00000000419 &  467 &  515 &  621 &  365 &   587 &   799 &  417 &  508\\\\\n\tENSG00000000457 &  260 &  211 &  263 &  164 &   245 &   331 &  233 &  229\\\\\n\tENSG00000000460 &   60 &   55 &   40 &   35 &    78 &    63 &   76 &   60\\\\\n\tENSG00000000938 &    0 &    0 &    2 &    0 &     1 &     0 &    0 &    0\\\\\n\tENSG00000000971 & 3251 & 3679 & 6177 & 4252 &  6721 & 11027 & 5176 & 7995\\\\\n\tENSG00000001036 & 1433 & 1062 & 1733 &  881 &  1424 &  1439 & 1359 & 1109\\\\\n\tENSG00000001084 &  519 &  380 &  595 &  493 &   820 &   714 &  696 &  704\\\\\n\tENSG00000001167 &  394 &  236 &  464 &  175 &   658 &   584 &  360 &  269\\\\\n\tENSG00000001460 &  172 &  168 &  264 &  118 &   241 &   210 &  155 &  177\\\\\n\tENSG00000001461 & 2112 & 1867 & 5137 & 2657 &  2735 &  2751 & 2467 & 2905\\\\\n\tENSG00000001497 &  524 &  488 &  638 &  357 &   676 &   806 &  493 &  475\\\\\n\tENSG00000001561 &   71 &   51 &  211 &  156 &    23 &    38 &  134 &  172\\\\\n\tENSG00000001617 &  555 &  394 &  905 &  415 &   727 &   697 &  618 &  599\\\\\n\tENSG00000001626 &   10 &    2 &    9 &    2 &    10 &     6 &    5 &    5\\\\\n\tENSG00000001629 & 1660 & 1251 & 2259 & 1079 &  2462 &  2514 & 1888 & 1660\\\\\n\tENSG00000001630 &   59 &   54 &   66 &   23 &    84 &    87 &   31 &   59\\\\\n\tENSG00000001631 &  729 &  692 &  943 &  475 &  1034 &  1163 &  731 &  744\\\\\n\tENSG00000002016 &  201 &  161 &  256 &   99 &   268 &   257 &  160 &  137\\\\\n\tENSG00000002079 &    3 &    0 &    3 &    1 &     4 &     0 &    0 &    1\\\\\n\tENSG00000002330 &  206 &  174 &  184 &  111 &   194 &   260 &  156 &  177\\\\\n\tENSG00000002549 & 1459 & 1294 & 1317 &  998 &  1451 &  1824 &  853 & 1031\\\\\n\tENSG00000002586 & 7507 & 7203 & 9501 & 6214 & 10973 & 12863 & 6834 & 7225\\\\\n\tENSG00000002587 &    2 &    0 &    1 &    0 &     0 &     2 &    0 &    0\\\\\n\tENSG00000002726 &    0 &    0 &    1 &    0 &     0 &     0 &    0 &    0\\\\\n\tENSG00000002745 &    4 &    6 &   22 &   10 &     2 &     1 &    5 &    3\\\\\n\tENSG00000002746 &  151 &  139 &  117 &   65 &    90 &   102 &   86 &  119\\\\\n\tENSG00000002822 &  411 &  303 &  446 &  195 &   445 &   523 &  295 &  300\\\\\n\tENSG00000002834 & 6314 & 6364 & 7831 & 5809 &  6677 & 11970 & 5766 & 7825\\\\\n\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n\tENSG00000273464 &  0 &  2 &  2 &  0 &  4 &  2 &  1 &  0\\\\\n\tENSG00000273465 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273466 &  0 &  0 &  0 &  0 &  0 &  0 &  1 &  0\\\\\n\tENSG00000273467 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273468 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273469 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273470 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273471 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273472 &  6 &  6 & 18 &  9 & 19 &  9 & 12 & 11\\\\\n\tENSG00000273473 &  0 &  0 &  3 &  0 &  2 &  2 &  2 &  0\\\\\n\tENSG00000273474 &  2 &  1 &  3 &  1 &  2 &  1 &  0 &  3\\\\\n\tENSG00000273475 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273476 &  1 &  2 &  0 &  0 &  0 &  0 &  0 &  1\\\\\n\tENSG00000273477 &  4 &  9 &  6 &  2 &  8 &  9 &  6 & 12\\\\\n\tENSG00000273478 &  9 &  8 &  7 &  5 &  6 &  8 &  5 &  6\\\\\n\tENSG00000273479 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273480 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273481 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273482 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273483 &  5 &  2 &  0 &  2 &  3 &  2 &  1 &  6\\\\\n\tENSG00000273484 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273485 &  2 &  3 &  1 &  1 &  1 &  1 &  1 &  0\\\\\n\tENSG00000273486 & 14 & 11 & 25 &  8 & 20 & 32 & 12 & 11\\\\\n\tENSG00000273487 &  5 &  9 &  4 & 11 & 10 & 10 &  4 & 10\\\\\n\tENSG00000273488 &  7 &  5 &  8 &  3 &  8 & 16 & 11 & 14\\\\\n\tENSG00000273489 &  0 &  0 &  0 &  1 &  0 &  1 &  0 &  0\\\\\n\tENSG00000273490 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273491 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273492 &  0 &  0 &  1 &  0 &  0 &  0 &  0 &  0\\\\\n\tENSG00000273493 &  0 &  0 &  0 &  0 &  1 &  0 &  0 &  0\\\\\n\\end{tabular}\n","text/plain":"                SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516\nENSG00000000003  679        448        873        408        1138     \nENSG00000000005    0          0          0          0           0     \nENSG00000000419  467        515        621        365         587     \nENSG00000000457  260        211        263        164         245     \nENSG00000000460   60         55         40         35          78     \nENSG00000000938    0          0          2          0           1     \nENSG00000000971 3251       3679       6177       4252        6721     \nENSG00000001036 1433       1062       1733        881        1424     \nENSG00000001084  519        380        595        493         820     \nENSG00000001167  394        236        464        175         658     \nENSG00000001460  172        168        264        118         241     \nENSG00000001461 2112       1867       5137       2657        2735     \nENSG00000001497  524        488        638        357         676     \nENSG00000001561   71         51        211        156          23     \nENSG00000001617  555        394        905        415         727     \nENSG00000001626   10          2          9          2          10     \nENSG00000001629 1660       1251       2259       1079        2462     \nENSG00000001630   59         54         66         23          84     \nENSG00000001631  729        692        943        475        1034     \nENSG00000002016  201        161        256         99         268     \nENSG00000002079    3          0          3          1           4     \nENSG00000002330  206        174        184        111         194     \nENSG00000002549 1459       1294       1317        998        1451     \nENSG00000002586 7507       7203       9501       6214       10973     \nENSG00000002587    2          0          1          0           0     \nENSG00000002726    0          0          1          0           0     \nENSG00000002745    4          6         22         10           2     \nENSG00000002746  151        139        117         65          90     \nENSG00000002822  411        303        446        195         445     \nENSG00000002834 6314       6364       7831       5809        6677     \n⋮               ⋮          ⋮          ⋮          ⋮          ⋮         \nENSG00000273464  0          2          2          0          4        \nENSG00000273465  0          0          0          0          0        \nENSG00000273466  0          0          0          0          0        \nENSG00000273467  0          0          0          0          0        \nENSG00000273468  0          0          0          0          0        \nENSG00000273469  0          0          0          0          0        \nENSG00000273470  0          0          0          0          0        \nENSG00000273471  0          0          0          0          0        \nENSG00000273472  6          6         18          9         19        \nENSG00000273473  0          0          3          0          2        \nENSG00000273474  2          1          3          1          2        \nENSG00000273475  0          0          0          0          0        \nENSG00000273476  1          2          0          0          0        \nENSG00000273477  4          9          6          2          8        \nENSG00000273478  9          8          7          5          6        \nENSG00000273479  0          0          0          0          0        \nENSG00000273480  0          0          0          0          0        \nENSG00000273481  0          0          0          0          0        \nENSG00000273482  0          0          0          0          0        \nENSG00000273483  5          2          0          2          3        \nENSG00000273484  0          0          0          0          0        \nENSG00000273485  2          3          1          1          1        \nENSG00000273486 14         11         25          8         20        \nENSG00000273487  5          9          4         11         10        \nENSG00000273488  7          5          8          3          8        \nENSG00000273489  0          0          0          1          0        \nENSG00000273490  0          0          0          0          0        \nENSG00000273491  0          0          0          0          0        \nENSG00000273492  0          0          1          0          0        \nENSG00000273493  0          0          0          0          1        \n                SRR1039517 SRR1039520 SRR1039521\nENSG00000000003  1047       770        572      \nENSG00000000005     0         0          0      \nENSG00000000419   799       417        508      \nENSG00000000457   331       233        229      \nENSG00000000460    63        76         60      \nENSG00000000938     0         0          0      \nENSG00000000971 11027      5176       7995      \nENSG00000001036  1439      1359       1109      \nENSG00000001084   714       696        704      \nENSG00000001167   584       360        269      \nENSG00000001460   210       155        177      \nENSG00000001461  2751      2467       2905      \nENSG00000001497   806       493        475      \nENSG00000001561    38       134        172      \nENSG00000001617   697       618        599      \nENSG00000001626     6         5          5      \nENSG00000001629  2514      1888       1660      \nENSG00000001630    87        31         59      \nENSG00000001631  1163       731        744      \nENSG00000002016   257       160        137      \nENSG00000002079     0         0          1      \nENSG00000002330   260       156        177      \nENSG00000002549  1824       853       1031      \nENSG00000002586 12863      6834       7225      \nENSG00000002587     2         0          0      \nENSG00000002726     0         0          0      \nENSG00000002745     1         5          3      \nENSG00000002746   102        86        119      \nENSG00000002822   523       295        300      \nENSG00000002834 11970      5766       7825      \n⋮               ⋮          ⋮          ⋮         \nENSG00000273464  2          1          0        \nENSG00000273465  0          0          0        \nENSG00000273466  0          1          0        \nENSG00000273467  0          0          0        \nENSG00000273468  0          0          0        \nENSG00000273469  0          0          0        \nENSG00000273470  0          0          0        \nENSG00000273471  0          0          0        \nENSG00000273472  9         12         11        \nENSG00000273473  2          2          0        \nENSG00000273474  1          0          3        \nENSG00000273475  0          0          0        \nENSG00000273476  0          0          1        \nENSG00000273477  9          6         12        \nENSG00000273478  8          5          6        \nENSG00000273479  0          0          0        \nENSG00000273480  0          0          0        \nENSG00000273481  0          0          0        \nENSG00000273482  0          0          0        \nENSG00000273483  2          1          6        \nENSG00000273484  0          0          0        \nENSG00000273485  1          1          0        \nENSG00000273486 32         12         11        \nENSG00000273487 10          4         10        \nENSG00000273488 16         11         14        \nENSG00000273489  1          0          0        \nENSG00000273490  0          0          0        \nENSG00000273491  0          0          0        \nENSG00000273492  0          0          0        \nENSG00000273493  0          0          0        "},"metadata":{}}],"execution_count":45},{"cell_type":"code","source":"length(rowRanges(airway))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:27:36.014512Z","iopub.execute_input":"2025-04-13T11:27:36.016183Z","iopub.status.idle":"2025-04-13T11:27:36.039006Z","shell.execute_reply":"2025-04-13T11:27:36.037195Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"63677","text/markdown":"63677","text/latex":"63677","text/plain":"[1] 63677"},"metadata":{}}],"execution_count":46},{"cell_type":"code","source":"rowRanges(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:29:02.957852Z","iopub.execute_input":"2025-04-13T11:29:02.959576Z","iopub.status.idle":"2025-04-13T11:29:03.069169Z","shell.execute_reply":"2025-04-13T11:29:03.067240Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRangesList object of length 63677:\n$ENSG00000000003\nGRanges object with 17 ranges and 2 metadata columns:\n       seqnames            ranges strand |   exon_id       exon_name\n          <Rle>         <IRanges>  <Rle> | <integer>     <character>\n   [1]        X 99883667-99884983      - |    667145 ENSE00001459322\n   [2]        X 99885756-99885863      - |    667146 ENSE00000868868\n   [3]        X 99887482-99887565      - |    667147 ENSE00000401072\n   [4]        X 99887538-99887565      - |    667148 ENSE00001849132\n   [5]        X 99888402-99888536      - |    667149 ENSE00003554016\n   ...      ...               ...    ... .       ...             ...\n  [13]        X 99890555-99890743      - |    667156 ENSE00003512331\n  [14]        X 99891188-99891686      - |    667158 ENSE00001886883\n  [15]        X 99891605-99891803      - |    667159 ENSE00001855382\n  [16]        X 99891790-99892101      - |    667160 ENSE00001863395\n  [17]        X 99894942-99894988      - |    667161 ENSE00001828996\n  -------\n  seqinfo: 722 sequences (1 circular) from an unspecified genome\n\n...\n<63676 more elements>"},"metadata":{}}],"execution_count":47},{"cell_type":"code","source":"sum(elementLengths(rowRanges(airway)))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:32:29.155688Z","iopub.execute_input":"2025-04-13T11:32:29.157407Z","iopub.status.idle":"2025-04-13T11:32:29.171936Z","shell.execute_reply":"2025-04-13T11:32:29.170107Z"}},"outputs":[{"ename":"ERROR","evalue":"Error in elementLengths(rowRanges(airway)): could not find function \"elementLengths\"\n","traceback":["Error in elementLengths(rowRanges(airway)): could not find function \"elementLengths\"\nTraceback:\n"],"output_type":"error"}],"execution_count":49},{"cell_type":"code","source":"start(airway)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:34:10.304534Z","iopub.execute_input":"2025-04-13T11:34:10.306449Z","iopub.status.idle":"2025-04-13T11:34:11.015160Z","shell.execute_reply":"2025-04-13T11:34:11.013365Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"IntegerList of length 63677\n[[\"ENSG00000000003\"]] 99883667 99885756 99887482 ... 99891605 99891790 99894942\n[[\"ENSG00000000005\"]] 99839799 99840228 99848621 ... 99852501 99854013 99854505\n[[\"ENSG00000000419\"]] 49551404 49551404 49551433 ... 49574900 49574900 49574900\n[[\"ENSG00000000457\"]] 169818772 169821804 169822215 ... 169862929 169863148\n[[\"ENSG00000000460\"]] 169631245 169652610 169652897 ... 169821931 169821931\n[[\"ENSG00000000938\"]] 27938575 27938803 27938811 ... 27961576 27961576 27961576\n[[\"ENSG00000000971\"]] 196621008 196621186 196621186 ... 196716241 196716241\n[[\"ENSG00000001036\"]] 143815948 143817763 143818089 ... 143832548 143832548\n[[\"ENSG00000001084\"]] 53362139 53362139 53362940 ... 53428915 53429548 53481684\n[[\"ENSG00000001167\"]] 41040684 41040722 41046768 ... 41062135 41065096 41065096\n...\n<63667 more elements>"},"metadata":{}}],"execution_count":50},{"cell_type":"code","source":"gr=GRanges(\"1\",range=IRanges(start=1,end=10^7))\ngr","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:37:01.138979Z","iopub.execute_input":"2025-04-13T11:37:01.140511Z","iopub.status.idle":"2025-04-13T11:37:01.180391Z","shell.execute_reply":"2025-04-13T11:37:01.178669Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 1 range and 0 metadata columns:\n      seqnames     ranges strand\n         <Rle>  <IRanges>  <Rle>\n  [1]        1 1-10000000      *\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":52},{"cell_type":"code","source":"subsetByOverLaps(airway,gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:37:48.267969Z","iopub.execute_input":"2025-04-13T11:37:48.269561Z","iopub.status.idle":"2025-04-13T11:37:48.283991Z","shell.execute_reply":"2025-04-13T11:37:48.282090Z"}},"outputs":[{"ename":"ERROR","evalue":"Error in subsetByOverLaps(airway, gr): could not find function \"subsetByOverLaps\"\n","traceback":["Error in subsetByOverLaps(airway, gr): could not find function \"subsetByOverLaps\"\nTraceback:\n"],"output_type":"error"}],"execution_count":53},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\n\nBiocManager::install(\"GEOquery\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:40:22.691762Z","iopub.execute_input":"2025-04-13T11:40:22.693502Z","iopub.status.idle":"2025-04-13T11:40:48.539536Z","shell.execute_reply":"2025-04-13T11:40:48.537299Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'GEOquery'\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'AID', 'akc', 'ale',\n  'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'antiword', 'anytime', 'ape',\n  'aplot', 'apollo', 'aqp', 'arrow', 'ars', 'ARTool', 'arules', 'AsioHeaders',\n  'askpass', 'ASMap', 'astsa', 'automap', 'baguette', 'BAS', 'base64',\n  'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr', 'beepr',\n  'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD', 'bigstatsr',\n  'bigstep', 'billboarder', 'bindr', 'BiocManager', 'BiodiversityR', 'biomod2',\n  'biotools', 'bipartite', 'bit', 'bit64', 'bitops', 'blme', 'blorr', 'BMA',\n  'BMisc', 'bnlearn', 'bonsai', 'bookdown', 'boot', 'brms', 'broom',\n  'broom.helpers', 'broom.mixed', 'brulee', 'bslib', 'bundle', 'butcher',\n  'C50', 'car', 'caret', 'caretEnsemble', 'caTools', 'cba', 'censored',\n  'changepoint', 'cheapr', 'checkmate', 'chk', 'CholWishart', 'choroplethr',\n  'chromote', 'chron', 'circular', 'class', 'classInt', 'cld2', 'cleaner',\n  'cli', 'clock', 'clue', 'cluster', 'clusterSim', 'cmsafops',\n  'coarseDataTools', 'cobs', 'collapse', 'colorspace', 'commonmark',\n  'compareGroups', 'Compind', 'contentid', 'copula', 'CORElearn', 'coro',\n  'correlation', 'corrplot', 'countrycode', 'cpp11', 'cppdoubles',\n  'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature', 'Cubist',\n  'curl', 'cutpointr', 'cvms', 'data.table', 'datamods', 'datasauRus',\n  'datawizard', 'datetimeutils', 'dbarts', 'dbscan', 'dendextend', 'DEoptimR',\n  'Deriv', 'DescTools', 'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'digest',\n  'dismo', 'distances', 'distillery', 'distr', 'distrEx', 'distributional',\n  'DistributionUtils', 'dlm', 'dm', 'doBy', 'docopt', 'doFuture', 'doRNG',\n  'DoseFinding', 'dotCall64', 'dotwhisker', 'DoubleML', 'downloader', 'dplR',\n  'drat', 'DRDID', 'dtwclust', 'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat',\n  'ecp', 'effectsize', 'eha', 'embed', 'emmeans', 'energy', 'entropy',\n  'EnvStats', 'Epi', 'epiR', 'erer', 'ergm', 'ergm.multi', 'esquisse',\n  'estimatr', 'etm', 'evaluate', 'evd', 'Exact', 'explore', 'expm', 'extRemes',\n  'fable', 'fabletools', 'faraway', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastverse', 'fBasics', 'fda', 'fda.usc', 'fdapace', 'fdrtool',\n  'feasts', 'ff', 'fftw', 'FielDHub', 'fields', 'filehashSQLite', 'finalfit',\n  'fitdistrplus', 'FKF', 'flare', 'flexclust', 'flexmix', 'flexsurv',\n  'flextable', 'float', 'FNN', 'fontawesome', 'forecast', 'foreign',\n  'forestplot', 'fpc', 'fpp3', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa',\n  'future', 'future.apply', 'gam', 'gamboostLSS', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'genridge', 'GenSA', 'geobr', 'geodist', 'geojsonR',\n  'geometry', 'geomorph', 'geomtextpath', 'geosphere', 'gert', 'GetBCBData',\n  'ggdag', 'ggeasy', 'ggeffects', 'ggfun', 'ggh4x', 'ggiraph', 'ggmap',\n  'ggnewscale', 'ggpath', 'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc',\n  'ggrepel', 'ggstats', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmtoolbox', 'glue', 'gmapsdistance', 'gmp',\n  'GoFKernel', 'golem', 'googlePolylines', 'gower', 'GPArotation', 'GPfit',\n  'gplots', 'grafify', 'gRain', 'graphlayouts', 'gRbase', 'greybox', 'grf',\n  'gridpattern', 'grImport2', 'groupdata2', 'grpreg', 'gsheet', 'gss', 'gstat',\n  'gsw', 'gt', 'gtable', 'gtsummary', 'GWalkR', 'GWmodel', 'h3lib', 'h3r',\n  'hardhat', 'harmony', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'htmlTable', 'hts', 'httr2', 'hunspell', 'huxtable',\n  'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph', 'ijtiff', 'imager', 'iml',\n  'inline', 'insight', 'inspectdf', 'interactions', 'intervals', 'ipred',\n  'isotree', 'ISwR', 'ivreg', 'janitor', 'jiebaR', 'JMbayes2', 'jmvconnect',\n  'jmvcore', 'jmvReadWrite', 'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity',\n  'keras3', 'kernelshap', 'kernlab', 'KernSmooth', 'khroma', 'kit', 'knitr',\n  'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman', 'lares', 'later',\n  'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave', 'lcmm', 'leafgl',\n  'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR', 'lightgbm', 'limma', 'lintr',\n  'littler', 'lme4', 'lmodel2', 'lmom', 'locfit', 'logger', 'longitudinalData',\n  'loo', 'lordif', 'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa',\n  'lubridate', 'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'matlib', 'Matrix', 'MatrixModels',\n  'matrixStats', 'MazamaCoreUtils', 'MazamaSpatialUtils', 'MBA', 'mboost',\n  'MCMCpack', 'mda', 'mdsr', 'memisc', 'meta', 'metacoder', 'metadat',\n  'metafor', 'meteospain', 'MetricsWeighted', 'mets', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'minqa', 'minty', 'mirt',\n  'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench', 'mlflow',\n  'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data', 'mlr3filters',\n  'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures', 'mlr3misc',\n  'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse', 'mlr3viz',\n  'modelbased', 'modelenv', 'modelsummary', 'modeltime', 'moderndive',\n  'mongolite', 'monitoR', 'msigdbr', 'msm', 'mstate', 'mtsdi', 'multcomp',\n  'MuMIn', 'mvtnorm', 'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson',\n  'ndtv', 'network', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'OpenMx', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ore', 'osmextract',\n  'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown', 'pammtools',\n  'pander', 'parabar', 'paradox', 'parallelly', 'parameters', 'ParamHelpers',\n  'paran', 'parsnip', 'party', 'partykit', 'patchwork', 'pbdZMQ', 'pcaPP',\n  'pdftools', 'pdp', 'pdR', 'PearsonDS', 'performance', 'PerformanceAnalytics',\n  'pgmm', 'phangorn', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools', 'pls',\n  'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip', 'popdemo',\n  'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'PreProcess', 'prettyGraphs', 'priceR', 'processx', 'profvis',\n  'progressr', 'PROJ', 'proj4', 'promises', 'prospectr', 'protolite', 'PRROC',\n  'ps', 'pspline', 'PSweight', 'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf',\n  'qs', 'qtl', 'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr',\n  'QuickJSR', 'qvcalc', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'reactR',\n  'readODS', 'readxl', 'rearrr', 'rebmix', 'recipes', 'refund', 'renv',\n  'report', 'reprex', 'reticulate', 'Rfast', 'rfishbase', 'rgbif', 'rgenoud',\n  'rgl', 'rio', 'rjson', 'RJSONIO', 'rlang', 'RMariaDB', 'rmarkdown', 'rminer',\n  'rms', 'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpostgis', 'RPostgres', 'RPostgreSQL',\n  'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample', 'rsconnect',\n  'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite', 'Rssa',\n  'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rugarch', 'Rvcg', 'rvg',\n  's2', 'sandwich', 'sass', 'SCtools', 'sda', 'sdm', 'seasonal',\n  'seasonalview', 'see', 'seededlda', 'segmented', 'sem', 'semTools',\n  'sensemakr', 'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinyWidgets', 'SimDesign',\n  'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot', 'skmeans', 'slam',\n  'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2', 'sna', 'sodium',\n  'soilDB', 'sommer', 'sp', 'spacefillr', 'spacetime', 'spam', 'sparklyr',\n  'SparseM', 'spatialreg', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'strucchange', 'summarytools',\n  'SuppDists', 'survival', 'survminer', 'svd', 'sys', 'systemfonts', 'tables',\n  'tau', 'taxize', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract',\n  'testthat', 'texreg', 'text', 'textcat', 'textrecipes', 'textshaping',\n  'textutils', 'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'timeDate', 'timeplyr', 'timereg', 'timeSeries', 'tinytable',\n  'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui', 'topicmodels',\n  'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal', 'tsbox', 'tsDyn',\n  'tseries', 'tsibble', 'tuber', 'tune', 'tzdb', 'units', 'univariateML',\n  'unpivotr', 'usethis', 'usmapdata', 'uuid', 'uwot', 'V8', 'vcd', 'vegan',\n  'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR', 'wbacon',\n  'WDI', 'websocket', 'webutils', 'WeightSVM', 'withr', 'wk', 'wooldridge',\n  'workflows', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick',\n  'yulab.utils', 'zip', 'zoo', 'spatial'\n\n","output_type":"stream"}],"execution_count":54},{"cell_type":"code","source":"library(GEOquery)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:43:49.737422Z","iopub.execute_input":"2025-04-13T11:43:49.739272Z","iopub.status.idle":"2025-04-13T11:43:50.373791Z","shell.execute_reply":"2025-04-13T11:43:50.372022Z"}},"outputs":[{"name":"stderr","text":"Setting options('download.file.method.GEOquery'='auto')\n\nSetting options('GEOquery.inmemory.gpl'=FALSE)\n\n","output_type":"stream"}],"execution_count":56},{"cell_type":"code","source":"elist=getGEO(\"GSE11675\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:43:56.914041Z","iopub.execute_input":"2025-04-13T11:43:56.915673Z","iopub.status.idle":"2025-04-13T11:44:00.467615Z","shell.execute_reply":"2025-04-13T11:44:00.465895Z"}},"outputs":[{"name":"stderr","text":"Found 1 file(s)\n\nGSE11675_series_matrix.txt.gz\n\n","output_type":"stream"}],"execution_count":57},{"cell_type":"code","source":"length(elist)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:44:48.076930Z","iopub.execute_input":"2025-04-13T11:44:48.078692Z","iopub.status.idle":"2025-04-13T11:44:48.095836Z","shell.execute_reply":"2025-04-13T11:44:48.093545Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"1","text/markdown":"1","text/latex":"1","text/plain":"[1] 1"},"metadata":{}}],"execution_count":58},{"cell_type":"code","source":"edata = elist[[1]]\nedata","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:45:51.485444Z","iopub.execute_input":"2025-04-13T11:45:51.487033Z","iopub.status.idle":"2025-04-13T11:45:51.506417Z","shell.execute_reply":"2025-04-13T11:45:51.504724Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"ExpressionSet (storageMode: lockedEnvironment)\nassayData: 12625 features, 6 samples \n  element names: exprs \nprotocolData: none\nphenoData\n  sampleNames: GSM296630 GSM296635 ... GSM296639 (6 total)\n  varLabels: title geo_accession ... data_row_count (34 total)\n  varMetadata: labelDescription\nfeatureData\n  featureNames: 1000_at 1001_at ... AFFX-YEL024w/RIP1_at (12625 total)\n  fvarLabels: ID GB_ACC ... Gene Ontology Molecular Function (16 total)\n  fvarMetadata: Column Description labelDescription\nexperimentData: use 'experimentData(object)'\n  pubMedIds: 19855080 \nAnnotation: GPL8300 "},"metadata":{}}],"execution_count":60},{"cell_type":"code","source":"names(pData(edata))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:46:45.127575Z","iopub.execute_input":"2025-04-13T11:46:45.129318Z","iopub.status.idle":"2025-04-13T11:46:45.238843Z","shell.execute_reply":"2025-04-13T11:46:45.236776Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'title'</li><li>'geo_accession'</li><li>'status'</li><li>'submission_date'</li><li>'last_update_date'</li><li>'type'</li><li>'channel_count'</li><li>'source_name_ch1'</li><li>'organism_ch1'</li><li>'characteristics_ch1'</li><li>'characteristics_ch1.1'</li><li>'characteristics_ch1.2'</li><li>'molecule_ch1'</li><li>'extract_protocol_ch1'</li><li>'label_ch1'</li><li>'label_protocol_ch1'</li><li>'taxid_ch1'</li><li>'hyb_protocol'</li><li>'scan_protocol'</li><li>'description'</li><li>'data_processing'</li><li>'platform_id'</li><li>'contact_name'</li><li>'contact_email'</li><li>'contact_phone'</li><li>'contact_department'</li><li>'contact_institute'</li><li>'contact_address'</li><li>'contact_city'</li><li>'contact_zip/postal_code'</li><li>'contact_country'</li><li>'supplementary_file'</li><li>'supplementary_file.1'</li><li>'data_row_count'</li></ol>\n","text/markdown":"1. 'title'\n2. 'geo_accession'\n3. 'status'\n4. 'submission_date'\n5. 'last_update_date'\n6. 'type'\n7. 'channel_count'\n8. 'source_name_ch1'\n9. 'organism_ch1'\n10. 'characteristics_ch1'\n11. 'characteristics_ch1.1'\n12. 'characteristics_ch1.2'\n13. 'molecule_ch1'\n14. 'extract_protocol_ch1'\n15. 'label_ch1'\n16. 'label_protocol_ch1'\n17. 'taxid_ch1'\n18. 'hyb_protocol'\n19. 'scan_protocol'\n20. 'description'\n21. 'data_processing'\n22. 'platform_id'\n23. 'contact_name'\n24. 'contact_email'\n25. 'contact_phone'\n26. 'contact_department'\n27. 'contact_institute'\n28. 'contact_address'\n29. 'contact_city'\n30. 'contact_zip/postal_code'\n31. 'contact_country'\n32. 'supplementary_file'\n33. 'supplementary_file.1'\n34. 'data_row_count'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'title'\n\\item 'geo\\_accession'\n\\item 'status'\n\\item 'submission\\_date'\n\\item 'last\\_update\\_date'\n\\item 'type'\n\\item 'channel\\_count'\n\\item 'source\\_name\\_ch1'\n\\item 'organism\\_ch1'\n\\item 'characteristics\\_ch1'\n\\item 'characteristics\\_ch1.1'\n\\item 'characteristics\\_ch1.2'\n\\item 'molecule\\_ch1'\n\\item 'extract\\_protocol\\_ch1'\n\\item 'label\\_ch1'\n\\item 'label\\_protocol\\_ch1'\n\\item 'taxid\\_ch1'\n\\item 'hyb\\_protocol'\n\\item 'scan\\_protocol'\n\\item 'description'\n\\item 'data\\_processing'\n\\item 'platform\\_id'\n\\item 'contact\\_name'\n\\item 'contact\\_email'\n\\item 'contact\\_phone'\n\\item 'contact\\_department'\n\\item 'contact\\_institute'\n\\item 'contact\\_address'\n\\item 'contact\\_city'\n\\item 'contact\\_zip/postal\\_code'\n\\item 'contact\\_country'\n\\item 'supplementary\\_file'\n\\item 'supplementary\\_file.1'\n\\item 'data\\_row\\_count'\n\\end{enumerate*}\n","text/plain":" [1] \"title\"                   \"geo_accession\"          \n [3] \"status\"                  \"submission_date\"        \n [5] \"last_update_date\"        \"type\"                   \n [7] \"channel_count\"           \"source_name_ch1\"        \n [9] \"organism_ch1\"            \"characteristics_ch1\"    \n[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n[13] \"molecule_ch1\"            \"extract_protocol_ch1\"   \n[15] \"label_ch1\"               \"label_protocol_ch1\"     \n[17] \"taxid_ch1\"               \"hyb_protocol\"           \n[19] \"scan_protocol\"           \"description\"            \n[21] \"data_processing\"         \"platform_id\"            \n[23] \"contact_name\"            \"contact_email\"          \n[25] \"contact_phone\"           \"contact_department\"     \n[27] \"contact_institute\"       \"contact_address\"        \n[29] \"contact_city\"            \"contact_zip/postal_code\"\n[31] \"contact_country\"         \"supplementary_file\"     \n[33] \"supplementary_file.1\"    \"data_row_count\"         "},"metadata":{}}],"execution_count":61},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\n\nBiocManager::install(\"biomaRt\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:53:19.883943Z","iopub.execute_input":"2025-04-13T11:53:19.886537Z","iopub.status.idle":"2025-04-13T11:55:22.352358Z","shell.execute_reply":"2025-04-13T11:55:22.350366Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'biomaRt'\n\nalso installing the dependencies ‘Biostrings’, ‘KEGGREST’, ‘AnnotationDbi’, ‘BiocFileCache’\n\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'AID', 'akc', 'ale',\n  'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'antiword', 'anytime', 'ape',\n  'aplot', 'apollo', 'aqp', 'arrow', 'ars', 'ARTool', 'arules', 'AsioHeaders',\n  'askpass', 'ASMap', 'astsa', 'automap', 'baguette', 'BAS', 'base64',\n  'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr', 'beepr',\n  'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD', 'bigstatsr',\n  'bigstep', 'billboarder', 'bindr', 'BiocManager', 'BiodiversityR', 'biomod2',\n  'biotools', 'bipartite', 'bit', 'bit64', 'bitops', 'blme', 'blorr', 'BMA',\n  'BMisc', 'bnlearn', 'bonsai', 'bookdown', 'boot', 'brms', 'broom',\n  'broom.helpers', 'broom.mixed', 'brulee', 'bslib', 'bundle', 'butcher',\n  'C50', 'car', 'caret', 'caretEnsemble', 'caTools', 'cba', 'censored',\n  'changepoint', 'cheapr', 'checkmate', 'chk', 'CholWishart', 'choroplethr',\n  'chromote', 'chron', 'circular', 'class', 'classInt', 'cld2', 'cleaner',\n  'cli', 'clock', 'clue', 'cluster', 'clusterSim', 'cmsafops',\n  'coarseDataTools', 'cobs', 'collapse', 'colorspace', 'commonmark',\n  'compareGroups', 'Compind', 'contentid', 'copula', 'CORElearn', 'coro',\n  'correlation', 'corrplot', 'countrycode', 'cpp11', 'cppdoubles',\n  'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature', 'Cubist',\n  'curl', 'cutpointr', 'cvms', 'data.table', 'datamods', 'datasauRus',\n  'datawizard', 'datetimeutils', 'dbarts', 'dbscan', 'dendextend', 'DEoptimR',\n  'Deriv', 'DescTools', 'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'digest',\n  'dismo', 'distances', 'distillery', 'distr', 'distrEx', 'distributional',\n  'DistributionUtils', 'dlm', 'dm', 'doBy', 'docopt', 'doFuture', 'doRNG',\n  'DoseFinding', 'dotCall64', 'dotwhisker', 'DoubleML', 'downloader', 'dplR',\n  'drat', 'DRDID', 'dtwclust', 'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat',\n  'ecp', 'effectsize', 'eha', 'embed', 'emmeans', 'energy', 'entropy',\n  'EnvStats', 'Epi', 'epiR', 'erer', 'ergm', 'ergm.multi', 'esquisse',\n  'estimatr', 'etm', 'evaluate', 'evd', 'Exact', 'explore', 'expm', 'extRemes',\n  'fable', 'fabletools', 'faraway', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastverse', 'fBasics', 'fda', 'fda.usc', 'fdapace', 'fdrtool',\n  'feasts', 'ff', 'fftw', 'FielDHub', 'fields', 'filehashSQLite', 'finalfit',\n  'fitdistrplus', 'FKF', 'flare', 'flexclust', 'flexmix', 'flexsurv',\n  'flextable', 'float', 'FNN', 'fontawesome', 'forecast', 'foreign',\n  'forestplot', 'fpc', 'fpp3', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa',\n  'future', 'future.apply', 'gam', 'gamboostLSS', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'genridge', 'GenSA', 'geobr', 'geodist', 'geojsonR',\n  'geometry', 'geomorph', 'geomtextpath', 'geosphere', 'gert', 'GetBCBData',\n  'ggdag', 'ggeasy', 'ggeffects', 'ggfun', 'ggh4x', 'ggiraph', 'ggmap',\n  'ggnewscale', 'ggpath', 'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc',\n  'ggrepel', 'ggstats', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmtoolbox', 'glue', 'gmapsdistance', 'gmp',\n  'GoFKernel', 'golem', 'googlePolylines', 'gower', 'GPArotation', 'GPfit',\n  'gplots', 'grafify', 'gRain', 'graphlayouts', 'gRbase', 'greybox', 'grf',\n  'gridpattern', 'grImport2', 'groupdata2', 'grpreg', 'gsheet', 'gss', 'gstat',\n  'gsw', 'gt', 'gtable', 'gtsummary', 'GWalkR', 'GWmodel', 'h3lib', 'h3r',\n  'hardhat', 'harmony', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'htmlTable', 'hts', 'httr2', 'hunspell', 'huxtable',\n  'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph', 'ijtiff', 'imager', 'iml',\n  'inline', 'insight', 'inspectdf', 'interactions', 'intervals', 'ipred',\n  'isotree', 'ISwR', 'ivreg', 'janitor', 'jiebaR', 'JMbayes2', 'jmvconnect',\n  'jmvcore', 'jmvReadWrite', 'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity',\n  'keras3', 'kernelshap', 'kernlab', 'KernSmooth', 'khroma', 'kit', 'knitr',\n  'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman', 'lares', 'later',\n  'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave', 'lcmm', 'leafgl',\n  'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR', 'lightgbm', 'limma', 'lintr',\n  'littler', 'lme4', 'lmodel2', 'lmom', 'locfit', 'logger', 'longitudinalData',\n  'loo', 'lordif', 'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa',\n  'lubridate', 'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'matlib', 'Matrix', 'MatrixModels',\n  'matrixStats', 'MazamaCoreUtils', 'MazamaSpatialUtils', 'MBA', 'mboost',\n  'MCMCpack', 'mda', 'mdsr', 'memisc', 'meta', 'metacoder', 'metadat',\n  'metafor', 'meteospain', 'MetricsWeighted', 'mets', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'minqa', 'minty', 'mirt',\n  'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench', 'mlflow',\n  'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data', 'mlr3filters',\n  'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures', 'mlr3misc',\n  'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse', 'mlr3viz',\n  'modelbased', 'modelenv', 'modelsummary', 'modeltime', 'moderndive',\n  'mongolite', 'monitoR', 'msigdbr', 'msm', 'mstate', 'mtsdi', 'multcomp',\n  'MuMIn', 'mvtnorm', 'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson',\n  'ndtv', 'network', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'OpenMx', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ore', 'osmextract',\n  'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown', 'pammtools',\n  'pander', 'parabar', 'paradox', 'parallelly', 'parameters', 'ParamHelpers',\n  'paran', 'parsnip', 'party', 'partykit', 'patchwork', 'pbdZMQ', 'pcaPP',\n  'pdftools', 'pdp', 'pdR', 'PearsonDS', 'performance', 'PerformanceAnalytics',\n  'pgmm', 'phangorn', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools', 'pls',\n  'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip', 'popdemo',\n  'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'PreProcess', 'prettyGraphs', 'priceR', 'processx', 'profvis',\n  'progressr', 'PROJ', 'proj4', 'promises', 'prospectr', 'protolite', 'PRROC',\n  'ps', 'pspline', 'PSweight', 'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf',\n  'qs', 'qtl', 'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr',\n  'QuickJSR', 'qvcalc', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'reactR',\n  'readODS', 'readxl', 'rearrr', 'rebmix', 'recipes', 'refund', 'renv',\n  'report', 'reprex', 'reticulate', 'Rfast', 'rfishbase', 'rgbif', 'rgenoud',\n  'rgl', 'rio', 'rjson', 'RJSONIO', 'rlang', 'RMariaDB', 'rmarkdown', 'rminer',\n  'rms', 'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpostgis', 'RPostgres', 'RPostgreSQL',\n  'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample', 'rsconnect',\n  'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite', 'Rssa',\n  'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rugarch', 'Rvcg', 'rvg',\n  's2', 'sandwich', 'sass', 'SCtools', 'sda', 'sdm', 'seasonal',\n  'seasonalview', 'see', 'seededlda', 'segmented', 'sem', 'semTools',\n  'sensemakr', 'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinyWidgets', 'SimDesign',\n  'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot', 'skmeans', 'slam',\n  'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2', 'sna', 'sodium',\n  'soilDB', 'sommer', 'sp', 'spacefillr', 'spacetime', 'spam', 'sparklyr',\n  'SparseM', 'spatialreg', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'strucchange', 'summarytools',\n  'SuppDists', 'survival', 'survminer', 'svd', 'sys', 'systemfonts', 'tables',\n  'tau', 'taxize', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract',\n  'testthat', 'texreg', 'text', 'textcat', 'textrecipes', 'textshaping',\n  'textutils', 'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'timeDate', 'timeplyr', 'timereg', 'timeSeries', 'tinytable',\n  'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui', 'topicmodels',\n  'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal', 'tsbox', 'tsDyn',\n  'tseries', 'tsibble', 'tuber', 'tune', 'tzdb', 'units', 'univariateML',\n  'unpivotr', 'usethis', 'usmapdata', 'uuid', 'uwot', 'V8', 'vcd', 'vegan',\n  'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR', 'wbacon',\n  'WDI', 'websocket', 'webutils', 'WeightSVM', 'withr', 'wk', 'wooldridge',\n  'workflows', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick',\n  'yulab.utils', 'zip', 'zoo', 'spatial'\n\n","output_type":"stream"}],"execution_count":64},{"cell_type":"code","source":"library(biomaRt)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:56:31.482520Z","iopub.execute_input":"2025-04-13T11:56:31.484416Z","iopub.status.idle":"2025-04-13T11:56:32.783021Z","shell.execute_reply":"2025-04-13T11:56:32.781094Z"}},"outputs":[],"execution_count":65},{"cell_type":"code","source":"listMarts()","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:57:17.718601Z","iopub.execute_input":"2025-04-13T11:57:17.720998Z","iopub.status.idle":"2025-04-13T11:57:18.603348Z","shell.execute_reply":"2025-04-13T11:57:18.601465Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 4 × 2</caption>\n<thead>\n\t<tr><th scope=col>biomart</th><th scope=col>version</th></tr>\n\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><td>ENSEMBL_MART_ENSEMBL</td><td>Ensembl Genes 113     </td></tr>\n\t<tr><td>ENSEMBL_MART_MOUSE  </td><td>Mouse strains 113     </td></tr>\n\t<tr><td>ENSEMBL_MART_SNP    </td><td>Ensembl Variation 113 </td></tr>\n\t<tr><td>ENSEMBL_MART_FUNCGEN</td><td>Ensembl Regulation 113</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 4 × 2\n\n| biomart &lt;chr&gt; | version &lt;chr&gt; |\n|---|---|\n| ENSEMBL_MART_ENSEMBL | Ensembl Genes 113      |\n| ENSEMBL_MART_MOUSE   | Mouse strains 113      |\n| ENSEMBL_MART_SNP     | Ensembl Variation 113  |\n| ENSEMBL_MART_FUNCGEN | Ensembl Regulation 113 |\n\n","text/latex":"A data.frame: 4 × 2\n\\begin{tabular}{ll}\n biomart & version\\\\\n <chr> & <chr>\\\\\n\\hline\n\t ENSEMBL\\_MART\\_ENSEMBL & Ensembl Genes 113     \\\\\n\t ENSEMBL\\_MART\\_MOUSE   & Mouse strains 113     \\\\\n\t ENSEMBL\\_MART\\_SNP     & Ensembl Variation 113 \\\\\n\t ENSEMBL\\_MART\\_FUNCGEN & Ensembl Regulation 113\\\\\n\\end{tabular}\n","text/plain":"  biomart              version               \n1 ENSEMBL_MART_ENSEMBL Ensembl Genes 113     \n2 ENSEMBL_MART_MOUSE   Mouse strains 113     \n3 ENSEMBL_MART_SNP     Ensembl Variation 113 \n4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 113"},"metadata":{}}],"execution_count":68},{"cell_type":"code","source":"mart = useMart('ENSEMBL_MART_ENSEMBL')\nmart","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:57:38.955181Z","iopub.execute_input":"2025-04-13T11:57:38.956765Z","iopub.status.idle":"2025-04-13T11:57:39.485330Z","shell.execute_reply":"2025-04-13T11:57:39.482858Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"Object of class 'Mart':\n  Using the ENSEMBL_MART_ENSEMBL BioMart database\n  No dataset selected."},"metadata":{}}],"execution_count":69},{"cell_type":"code","source":"head(listDatasets(mart))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T11:57:55.064008Z","iopub.execute_input":"2025-04-13T11:57:55.065871Z","iopub.status.idle":"2025-04-13T11:57:55.673773Z","shell.execute_reply":"2025-04-13T11:57:55.671845Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 6 × 3</caption>\n<thead>\n\t<tr><th></th><th scope=col>dataset</th><th scope=col>description</th><th scope=col>version</th></tr>\n\t<tr><th></th><th scope=col>&lt;I&lt;chr&gt;&gt;</th><th scope=col>&lt;I&lt;chr&gt;&gt;</th><th scope=col>&lt;I&lt;chr&gt;&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><th scope=row>1</th><td>abrachyrhynchus_gene_ensembl</td><td>Pink-footed goose genes (ASM259213v1)</td><td>ASM259213v1</td></tr>\n\t<tr><th scope=row>2</th><td>acalliptera_gene_ensembl    </td><td>Eastern happy genes (fAstCal1.3)     </td><td>fAstCal1.3 </td></tr>\n\t<tr><th scope=row>3</th><td>acarolinensis_gene_ensembl  </td><td>Green anole genes (AnoCar2.0v2)      </td><td>AnoCar2.0v2</td></tr>\n\t<tr><th scope=row>4</th><td>acchrysaetos_gene_ensembl   </td><td>Golden eagle genes (bAquChr1.2)      </td><td>bAquChr1.2 </td></tr>\n\t<tr><th scope=row>5</th><td>acitrinellus_gene_ensembl   </td><td>Midas cichlid genes (Midas_v5)       </td><td>Midas_v5   </td></tr>\n\t<tr><th scope=row>6</th><td>amelanoleuca_gene_ensembl   </td><td>Giant panda genes (ASM200744v2)      </td><td>ASM200744v2</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 6 × 3\n\n| <!--/--> | dataset &lt;I&lt;chr&gt;&gt; | description &lt;I&lt;chr&gt;&gt; | version &lt;I&lt;chr&gt;&gt; |\n|---|---|---|---|\n| 1 | abrachyrhynchus_gene_ensembl | Pink-footed goose genes (ASM259213v1) | ASM259213v1 |\n| 2 | acalliptera_gene_ensembl     | Eastern happy genes (fAstCal1.3)      | fAstCal1.3  |\n| 3 | acarolinensis_gene_ensembl   | Green anole genes (AnoCar2.0v2)       | AnoCar2.0v2 |\n| 4 | acchrysaetos_gene_ensembl    | Golden eagle genes (bAquChr1.2)       | bAquChr1.2  |\n| 5 | acitrinellus_gene_ensembl    | Midas cichlid genes (Midas_v5)        | Midas_v5    |\n| 6 | amelanoleuca_gene_ensembl    | Giant panda genes (ASM200744v2)       | ASM200744v2 |\n\n","text/latex":"A data.frame: 6 × 3\n\\begin{tabular}{r|lll}\n  & dataset & description & version\\\\\n  & <I<chr>> & <I<chr>> & <I<chr>>\\\\\n\\hline\n\t1 & abrachyrhynchus\\_gene\\_ensembl & Pink-footed goose genes (ASM259213v1) & ASM259213v1\\\\\n\t2 & acalliptera\\_gene\\_ensembl     & Eastern happy genes (fAstCal1.3)      & fAstCal1.3 \\\\\n\t3 & acarolinensis\\_gene\\_ensembl   & Green anole genes (AnoCar2.0v2)       & AnoCar2.0v2\\\\\n\t4 & acchrysaetos\\_gene\\_ensembl    & Golden eagle genes (bAquChr1.2)       & bAquChr1.2 \\\\\n\t5 & acitrinellus\\_gene\\_ensembl    & Midas cichlid genes (Midas\\_v5)        & Midas\\_v5   \\\\\n\t6 & amelanoleuca\\_gene\\_ensembl    & Giant panda genes (ASM200744v2)       & ASM200744v2\\\\\n\\end{tabular}\n","text/plain":"  dataset                      description                          \n1 abrachyrhynchus_gene_ensembl Pink-footed goose genes (ASM259213v1)\n2 acalliptera_gene_ensembl     Eastern happy genes (fAstCal1.3)     \n3 acarolinensis_gene_ensembl   Green anole genes (AnoCar2.0v2)      \n4 acchrysaetos_gene_ensembl    Golden eagle genes (bAquChr1.2)      \n5 acitrinellus_gene_ensembl    Midas cichlid genes (Midas_v5)       \n6 amelanoleuca_gene_ensembl    Giant panda genes (ASM200744v2)      \n  version    \n1 ASM259213v1\n2 fAstCal1.3 \n3 AnoCar2.0v2\n4 bAquChr1.2 \n5 Midas_v5   \n6 ASM200744v2"},"metadata":{}}],"execution_count":70},{"cell_type":"code","source":"mart = useDataset(\"hsapiens_gene_ensembl\", mart = mart)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:03:27.576953Z","iopub.execute_input":"2025-04-13T12:03:27.578846Z","iopub.status.idle":"2025-04-13T12:03:29.101333Z","shell.execute_reply":"2025-04-13T12:03:29.099296Z"}},"outputs":[],"execution_count":72},{"cell_type":"code","source":"genes = c(\"BRCA1\", \"TP53\", \"EGFR\")\n\nresults = getBM(attributes = c(\"hgnc_symbol\", \"ensembl_gene_id\", \"chromosome_name\", \"start_position\", \"end_position\"),\n                 filters = \"hgnc_symbol\",\n                 values = genes,\n                 mart = mart)\nresults","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:08:45.036887Z","iopub.execute_input":"2025-04-13T12:08:45.038634Z","iopub.status.idle":"2025-04-13T12:08:46.492362Z","shell.execute_reply":"2025-04-13T12:08:46.490388Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 3 × 5</caption>\n<thead>\n\t<tr><th scope=col>hgnc_symbol</th><th scope=col>ensembl_gene_id</th><th scope=col>chromosome_name</th><th scope=col>start_position</th><th scope=col>end_position</th></tr>\n\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><td>BRCA1</td><td>ENSG00000012048</td><td>17</td><td>43044295</td><td>43170245</td></tr>\n\t<tr><td>EGFR </td><td>ENSG00000146648</td><td> 7</td><td>55019017</td><td>55211628</td></tr>\n\t<tr><td>TP53 </td><td>ENSG00000141510</td><td>17</td><td> 7661779</td><td> 7687546</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 3 × 5\n\n| hgnc_symbol &lt;chr&gt; | ensembl_gene_id &lt;chr&gt; | chromosome_name &lt;int&gt; | start_position &lt;int&gt; | end_position &lt;int&gt; |\n|---|---|---|---|---|\n| BRCA1 | ENSG00000012048 | 17 | 43044295 | 43170245 |\n| EGFR  | ENSG00000146648 |  7 | 55019017 | 55211628 |\n| TP53  | ENSG00000141510 | 17 |  7661779 |  7687546 |\n\n","text/latex":"A data.frame: 3 × 5\n\\begin{tabular}{lllll}\n hgnc\\_symbol & ensembl\\_gene\\_id & chromosome\\_name & start\\_position & end\\_position\\\\\n <chr> & <chr> & <int> & <int> & <int>\\\\\n\\hline\n\t BRCA1 & ENSG00000012048 & 17 & 43044295 & 43170245\\\\\n\t EGFR  & ENSG00000146648 &  7 & 55019017 & 55211628\\\\\n\t TP53  & ENSG00000141510 & 17 &  7661779 &  7687546\\\\\n\\end{tabular}\n","text/plain":"  hgnc_symbol ensembl_gene_id chromosome_name start_position end_position\n1 BRCA1       ENSG00000012048 17              43044295       43170245    \n2 EGFR        ENSG00000146648  7              55019017       55211628    \n3 TP53        ENSG00000141510 17               7661779        7687546    "},"metadata":{}}],"execution_count":73},{"cell_type":"code","source":"attributes= listAttributes(mart)\nattributes","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:16:47.809493Z","iopub.execute_input":"2025-04-13T12:16:47.811120Z","iopub.status.idle":"2025-04-13T12:16:47.875262Z","shell.execute_reply":"2025-04-13T12:16:47.873462Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 3163 × 3</caption>\n<thead>\n\t<tr><th scope=col>name</th><th scope=col>description</th><th scope=col>page</th></tr>\n\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><td>ensembl_gene_id                </td><td>Gene stable ID                              </td><td>feature_page</td></tr>\n\t<tr><td>ensembl_gene_id_version        </td><td>Gene stable ID version                      </td><td>feature_page</td></tr>\n\t<tr><td>ensembl_transcript_id          </td><td>Transcript stable ID                        </td><td>feature_page</td></tr>\n\t<tr><td>ensembl_transcript_id_version  </td><td>Transcript stable ID version                </td><td>feature_page</td></tr>\n\t<tr><td>ensembl_peptide_id             </td><td>Protein stable ID                           </td><td>feature_page</td></tr>\n\t<tr><td>ensembl_peptide_id_version     </td><td>Protein stable ID version                   </td><td>feature_page</td></tr>\n\t<tr><td>ensembl_exon_id                </td><td>Exon stable ID                              </td><td>feature_page</td></tr>\n\t<tr><td>description                    </td><td>Gene description                            </td><td>feature_page</td></tr>\n\t<tr><td>chromosome_name                </td><td>Chromosome/scaffold name                    </td><td>feature_page</td></tr>\n\t<tr><td>start_position                 </td><td>Gene start (bp)                             </td><td>feature_page</td></tr>\n\t<tr><td>end_position                   </td><td>Gene end (bp)                               </td><td>feature_page</td></tr>\n\t<tr><td>strand                         </td><td>Strand                                      </td><td>feature_page</td></tr>\n\t<tr><td>band                           </td><td>Karyotype band                              </td><td>feature_page</td></tr>\n\t<tr><td>transcript_start               </td><td>Transcript start (bp)                       </td><td>feature_page</td></tr>\n\t<tr><td>transcript_end                 </td><td>Transcript end (bp)                         </td><td>feature_page</td></tr>\n\t<tr><td>transcription_start_site       </td><td>Transcription start site (TSS)              </td><td>feature_page</td></tr>\n\t<tr><td>transcript_length              </td><td>Transcript length (including UTRs and CDS)  </td><td>feature_page</td></tr>\n\t<tr><td>transcript_tsl                 </td><td>Transcript support level (TSL)              </td><td>feature_page</td></tr>\n\t<tr><td>transcript_gencode_basic       </td><td>GENCODE basic annotation                    </td><td>feature_page</td></tr>\n\t<tr><td>transcript_primary_basic       </td><td>GENCODE primary annotation                  </td><td>feature_page</td></tr>\n\t<tr><td>transcript_appris              </td><td>APPRIS annotation                           </td><td>feature_page</td></tr>\n\t<tr><td>transcript_is_canonical        </td><td>Ensembl Canonical                           </td><td>feature_page</td></tr>\n\t<tr><td>transcript_mane_select         </td><td>RefSeq match transcript (MANE Select)       </td><td>feature_page</td></tr>\n\t<tr><td>transcript_mane_plus_clinical  </td><td>RefSeq match transcript (MANE Plus Clinical)</td><td>feature_page</td></tr>\n\t<tr><td>external_gene_name             </td><td>Gene name                                   </td><td>feature_page</td></tr>\n\t<tr><td>external_gene_source           </td><td>Source of gene name                         </td><td>feature_page</td></tr>\n\t<tr><td>external_transcript_name       </td><td>Transcript name                             </td><td>feature_page</td></tr>\n\t<tr><td>external_transcript_source_name</td><td>Source of transcript name                   </td><td>feature_page</td></tr>\n\t<tr><td>transcript_count               </td><td>Transcript count                            </td><td>feature_page</td></tr>\n\t<tr><td>percentage_gene_gc_content     </td><td>Gene % GC content                           </td><td>feature_page</td></tr>\n\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n\t<tr><td>5_utr_end                    </td><td>5' UTR end                                </td><td>sequences</td></tr>\n\t<tr><td>3_utr_start                  </td><td>3' UTR start                              </td><td>sequences</td></tr>\n\t<tr><td>3_utr_end                    </td><td>3' UTR end                                </td><td>sequences</td></tr>\n\t<tr><td>ensembl_transcript_id        </td><td>Transcript stable ID                      </td><td>sequences</td></tr>\n\t<tr><td>ensembl_transcript_id_version</td><td>Transcript stable ID version              </td><td>sequences</td></tr>\n\t<tr><td>ensembl_peptide_id           </td><td>Protein stable ID                         </td><td>sequences</td></tr>\n\t<tr><td>ensembl_peptide_id_version   </td><td>Protein stable ID version                 </td><td>sequences</td></tr>\n\t<tr><td>transcript_biotype           </td><td>Transcript type                           </td><td>sequences</td></tr>\n\t<tr><td>transcript_version           </td><td>Version (transcript)                      </td><td>sequences</td></tr>\n\t<tr><td>peptide_version              </td><td>Version (protein)                         </td><td>sequences</td></tr>\n\t<tr><td>strand                       </td><td>Strand                                    </td><td>sequences</td></tr>\n\t<tr><td>transcript_start             </td><td>Transcript start (bp)                     </td><td>sequences</td></tr>\n\t<tr><td>transcript_end               </td><td>Transcript end (bp)                       </td><td>sequences</td></tr>\n\t<tr><td>transcription_start_site     </td><td>Transcription start site (TSS)            </td><td>sequences</td></tr>\n\t<tr><td>transcript_length            </td><td>Transcript length (including UTRs and CDS)</td><td>sequences</td></tr>\n\t<tr><td>cds_length                   </td><td>CDS Length                                </td><td>sequences</td></tr>\n\t<tr><td>cds_start                    </td><td>CDS start                                 </td><td>sequences</td></tr>\n\t<tr><td>cds_end                      </td><td>CDS end                                   </td><td>sequences</td></tr>\n\t<tr><td>ensembl_exon_id              </td><td>Exon stable ID                            </td><td>sequences</td></tr>\n\t<tr><td>exon_chrom_start             </td><td>Exon region start (bp)                    </td><td>sequences</td></tr>\n\t<tr><td>exon_chrom_end               </td><td>Exon region end (bp)                      </td><td>sequences</td></tr>\n\t<tr><td>strand                       </td><td>Strand                                    </td><td>sequences</td></tr>\n\t<tr><td>rank                         </td><td>Exon rank in transcript                   </td><td>sequences</td></tr>\n\t<tr><td>phase                        </td><td>Start phase                               </td><td>sequences</td></tr>\n\t<tr><td>end_phase                    </td><td>End phase                                 </td><td>sequences</td></tr>\n\t<tr><td>cdna_coding_start            </td><td>cDNA coding start                         </td><td>sequences</td></tr>\n\t<tr><td>cdna_coding_end              </td><td>cDNA coding end                           </td><td>sequences</td></tr>\n\t<tr><td>genomic_coding_start         </td><td>Genomic coding start                      </td><td>sequences</td></tr>\n\t<tr><td>genomic_coding_end           </td><td>Genomic coding end                        </td><td>sequences</td></tr>\n\t<tr><td>is_constitutive              </td><td>Constitutive exon                         </td><td>sequences</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 3163 × 3\n\n| name &lt;chr&gt; | description &lt;chr&gt; | page &lt;chr&gt; |\n|---|---|---|\n| ensembl_gene_id                 | Gene stable ID                               | feature_page |\n| ensembl_gene_id_version         | Gene stable ID version                       | feature_page |\n| ensembl_transcript_id           | Transcript stable ID                         | feature_page |\n| ensembl_transcript_id_version   | Transcript stable ID version                 | feature_page |\n| ensembl_peptide_id              | Protein stable ID                            | feature_page |\n| ensembl_peptide_id_version      | Protein stable ID version                    | feature_page |\n| ensembl_exon_id                 | Exon stable ID                               | feature_page |\n| description                     | Gene description                             | feature_page |\n| chromosome_name                 | Chromosome/scaffold name                     | feature_page |\n| start_position                  | Gene start (bp)                              | feature_page |\n| end_position                    | Gene end (bp)                                | feature_page |\n| strand                          | Strand                                       | feature_page |\n| band                            | Karyotype band                               | feature_page |\n| transcript_start                | Transcript start (bp)                        | feature_page |\n| transcript_end                  | Transcript end (bp)                          | feature_page |\n| transcription_start_site        | Transcription start site (TSS)               | feature_page |\n| transcript_length               | Transcript length (including UTRs and CDS)   | feature_page |\n| transcript_tsl                  | Transcript support level (TSL)               | feature_page |\n| transcript_gencode_basic        | GENCODE basic annotation                     | feature_page |\n| transcript_primary_basic        | GENCODE primary annotation                   | feature_page |\n| transcript_appris               | APPRIS annotation                            | feature_page |\n| transcript_is_canonical         | Ensembl Canonical                            | feature_page |\n| transcript_mane_select          | RefSeq match transcript (MANE Select)        | feature_page |\n| transcript_mane_plus_clinical   | RefSeq match transcript (MANE Plus Clinical) | feature_page |\n| external_gene_name              | Gene name                                    | feature_page |\n| external_gene_source            | Source of gene name                          | feature_page |\n| external_transcript_name        | Transcript name                              | feature_page |\n| external_transcript_source_name | Source of transcript name                    | feature_page |\n| transcript_count                | Transcript count                             | feature_page |\n| percentage_gene_gc_content      | Gene % GC content                            | feature_page |\n| ⋮ | ⋮ | ⋮ |\n| 5_utr_end                     | 5' UTR end                                 | sequences |\n| 3_utr_start                   | 3' UTR start                               | sequences |\n| 3_utr_end                     | 3' UTR end                                 | sequences |\n| ensembl_transcript_id         | Transcript stable ID                       | sequences |\n| ensembl_transcript_id_version | Transcript stable ID version               | sequences |\n| ensembl_peptide_id            | Protein stable ID                          | sequences |\n| ensembl_peptide_id_version    | Protein stable ID version                  | sequences |\n| transcript_biotype            | Transcript type                            | sequences |\n| transcript_version            | Version (transcript)                       | sequences |\n| peptide_version               | Version (protein)                          | sequences |\n| strand                        | Strand                                     | sequences |\n| transcript_start              | Transcript start (bp)                      | sequences |\n| transcript_end                | Transcript end (bp)                        | sequences |\n| transcription_start_site      | Transcription start site (TSS)             | sequences |\n| transcript_length             | Transcript length (including UTRs and CDS) | sequences |\n| cds_length                    | CDS Length                                 | sequences |\n| cds_start                     | CDS start                                  | sequences |\n| cds_end                       | CDS end                                    | sequences |\n| ensembl_exon_id               | Exon stable ID                             | sequences |\n| exon_chrom_start              | Exon region start (bp)                     | sequences |\n| exon_chrom_end                | Exon region end (bp)                       | sequences |\n| strand                        | Strand                                     | sequences |\n| rank                          | Exon rank in transcript                    | sequences |\n| phase                         | Start phase                                | sequences |\n| end_phase                     | End phase                                  | sequences |\n| cdna_coding_start             | cDNA coding start                          | sequences |\n| cdna_coding_end               | cDNA coding end                            | sequences |\n| genomic_coding_start          | Genomic coding start                       | sequences |\n| genomic_coding_end            | Genomic coding end                         | sequences |\n| is_constitutive               | Constitutive exon                          | sequences |\n\n","text/latex":"A data.frame: 3163 × 3\n\\begin{tabular}{lll}\n name & description & page\\\\\n <chr> & <chr> & <chr>\\\\\n\\hline\n\t ensembl\\_gene\\_id                 & Gene stable ID                               & feature\\_page\\\\\n\t ensembl\\_gene\\_id\\_version         & Gene stable ID version                       & feature\\_page\\\\\n\t ensembl\\_transcript\\_id           & Transcript stable ID                         & feature\\_page\\\\\n\t ensembl\\_transcript\\_id\\_version   & Transcript stable ID version                 & feature\\_page\\\\\n\t ensembl\\_peptide\\_id              & Protein stable ID                            & feature\\_page\\\\\n\t ensembl\\_peptide\\_id\\_version      & Protein stable ID version                    & feature\\_page\\\\\n\t ensembl\\_exon\\_id                 & Exon stable ID                               & feature\\_page\\\\\n\t description                     & Gene description                             & feature\\_page\\\\\n\t chromosome\\_name                 & Chromosome/scaffold name                     & feature\\_page\\\\\n\t start\\_position                  & Gene start (bp)                              & feature\\_page\\\\\n\t end\\_position                    & Gene end (bp)                                & feature\\_page\\\\\n\t strand                          & Strand                                       & feature\\_page\\\\\n\t band                            & Karyotype band                               & feature\\_page\\\\\n\t transcript\\_start                & Transcript start (bp)                        & feature\\_page\\\\\n\t transcript\\_end                  & Transcript end (bp)                          & feature\\_page\\\\\n\t transcription\\_start\\_site        & Transcription start site (TSS)               & feature\\_page\\\\\n\t transcript\\_length               & Transcript length (including UTRs and CDS)   & feature\\_page\\\\\n\t transcript\\_tsl                  & Transcript support level (TSL)               & feature\\_page\\\\\n\t transcript\\_gencode\\_basic        & GENCODE basic annotation                     & feature\\_page\\\\\n\t transcript\\_primary\\_basic        & GENCODE primary annotation                   & feature\\_page\\\\\n\t transcript\\_appris               & APPRIS annotation                            & feature\\_page\\\\\n\t transcript\\_is\\_canonical         & Ensembl Canonical                            & feature\\_page\\\\\n\t transcript\\_mane\\_select          & RefSeq match transcript (MANE Select)        & feature\\_page\\\\\n\t transcript\\_mane\\_plus\\_clinical   & RefSeq match transcript (MANE Plus Clinical) & feature\\_page\\\\\n\t external\\_gene\\_name              & Gene name                                    & feature\\_page\\\\\n\t external\\_gene\\_source            & Source of gene name                          & feature\\_page\\\\\n\t external\\_transcript\\_name        & Transcript name                              & feature\\_page\\\\\n\t external\\_transcript\\_source\\_name & Source of transcript name                    & feature\\_page\\\\\n\t transcript\\_count                & Transcript count                             & feature\\_page\\\\\n\t percentage\\_gene\\_gc\\_content      & Gene \\% GC content                            & feature\\_page\\\\\n\t ⋮ & ⋮ & ⋮\\\\\n\t 5\\_utr\\_end                     & 5' UTR end                                 & sequences\\\\\n\t 3\\_utr\\_start                   & 3' UTR start                               & sequences\\\\\n\t 3\\_utr\\_end                     & 3' UTR end                                 & sequences\\\\\n\t ensembl\\_transcript\\_id         & Transcript stable ID                       & sequences\\\\\n\t ensembl\\_transcript\\_id\\_version & Transcript stable ID version               & sequences\\\\\n\t ensembl\\_peptide\\_id            & Protein stable ID                          & sequences\\\\\n\t ensembl\\_peptide\\_id\\_version    & Protein stable ID version                  & sequences\\\\\n\t transcript\\_biotype            & Transcript type                            & sequences\\\\\n\t transcript\\_version            & Version (transcript)                       & sequences\\\\\n\t peptide\\_version               & Version (protein)                          & sequences\\\\\n\t strand                        & Strand                                     & sequences\\\\\n\t transcript\\_start              & Transcript start (bp)                      & sequences\\\\\n\t transcript\\_end                & Transcript end (bp)                        & sequences\\\\\n\t transcription\\_start\\_site      & Transcription start site (TSS)             & sequences\\\\\n\t transcript\\_length             & Transcript length (including UTRs and CDS) & sequences\\\\\n\t cds\\_length                    & CDS Length                                 & sequences\\\\\n\t cds\\_start                     & CDS start                                  & sequences\\\\\n\t cds\\_end                       & CDS end                                    & sequences\\\\\n\t ensembl\\_exon\\_id               & Exon stable ID                             & sequences\\\\\n\t exon\\_chrom\\_start              & Exon region start (bp)                     & sequences\\\\\n\t exon\\_chrom\\_end                & Exon region end (bp)                       & sequences\\\\\n\t strand                        & Strand                                     & sequences\\\\\n\t rank                          & Exon rank in transcript                    & sequences\\\\\n\t phase                         & Start phase                                & sequences\\\\\n\t end\\_phase                     & End phase                                  & sequences\\\\\n\t cdna\\_coding\\_start             & cDNA coding start                          & sequences\\\\\n\t cdna\\_coding\\_end               & cDNA coding end                            & sequences\\\\\n\t genomic\\_coding\\_start          & Genomic coding start                       & sequences\\\\\n\t genomic\\_coding\\_end            & Genomic coding end                         & sequences\\\\\n\t is\\_constitutive               & Constitutive exon                          & sequences\\\\\n\\end{tabular}\n","text/plain":"     name                           \n1    ensembl_gene_id                \n2    ensembl_gene_id_version        \n3    ensembl_transcript_id          \n4    ensembl_transcript_id_version  \n5    ensembl_peptide_id             \n6    ensembl_peptide_id_version     \n7    ensembl_exon_id                \n8    description                    \n9    chromosome_name                \n10   start_position                 \n11   end_position                   \n12   strand                         \n13   band                           \n14   transcript_start               \n15   transcript_end                 \n16   transcription_start_site       \n17   transcript_length              \n18   transcript_tsl                 \n19   transcript_gencode_basic       \n20   transcript_primary_basic       \n21   transcript_appris              \n22   transcript_is_canonical        \n23   transcript_mane_select         \n24   transcript_mane_plus_clinical  \n25   external_gene_name             \n26   external_gene_source           \n27   external_transcript_name       \n28   external_transcript_source_name\n29   transcript_count               \n30   percentage_gene_gc_content     \n⋮    ⋮                              \n3134 5_utr_end                      \n3135 3_utr_start                    \n3136 3_utr_end                      \n3137 ensembl_transcript_id          \n3138 ensembl_transcript_id_version  \n3139 ensembl_peptide_id             \n3140 ensembl_peptide_id_version     \n3141 transcript_biotype             \n3142 transcript_version             \n3143 peptide_version                \n3144 strand                         \n3145 transcript_start               \n3146 transcript_end                 \n3147 transcription_start_site       \n3148 transcript_length              \n3149 cds_length                     \n3150 cds_start                      \n3151 cds_end                        \n3152 ensembl_exon_id                \n3153 exon_chrom_start               \n3154 exon_chrom_end                 \n3155 strand                         \n3156 rank                           \n3157 phase                          \n3158 end_phase                      \n3159 cdna_coding_start              \n3160 cdna_coding_end                \n3161 genomic_coding_start           \n3162 genomic_coding_end             \n3163 is_constitutive                \n     description                                  page        \n1    Gene stable ID                               feature_page\n2    Gene stable ID version                       feature_page\n3    Transcript stable ID                         feature_page\n4    Transcript stable ID version                 feature_page\n5    Protein stable ID                            feature_page\n6    Protein stable ID version                    feature_page\n7    Exon stable ID                               feature_page\n8    Gene description                             feature_page\n9    Chromosome/scaffold name                     feature_page\n10   Gene start (bp)                              feature_page\n11   Gene end (bp)                                feature_page\n12   Strand                                       feature_page\n13   Karyotype band                               feature_page\n14   Transcript start (bp)                        feature_page\n15   Transcript end (bp)                          feature_page\n16   Transcription start site (TSS)               feature_page\n17   Transcript length (including UTRs and CDS)   feature_page\n18   Transcript support level (TSL)               feature_page\n19   GENCODE basic annotation                     feature_page\n20   GENCODE primary annotation                   feature_page\n21   APPRIS annotation                            feature_page\n22   Ensembl Canonical                            feature_page\n23   RefSeq match transcript (MANE Select)        feature_page\n24   RefSeq match transcript (MANE Plus Clinical) feature_page\n25   Gene name                                    feature_page\n26   Source of gene name                          feature_page\n27   Transcript name                              feature_page\n28   Source of transcript name                    feature_page\n29   Transcript count                             feature_page\n30   Gene % GC content                            feature_page\n⋮    ⋮                                            ⋮           \n3134 5' UTR end                                   sequences   \n3135 3' UTR start                                 sequences   \n3136 3' UTR end                                   sequences   \n3137 Transcript stable ID                         sequences   \n3138 Transcript stable ID version                 sequences   \n3139 Protein stable ID                            sequences   \n3140 Protein stable ID version                    sequences   \n3141 Transcript type                              sequences   \n3142 Version (transcript)                         sequences   \n3143 Version (protein)                            sequences   \n3144 Strand                                       sequences   \n3145 Transcript start (bp)                        sequences   \n3146 Transcript end (bp)                          sequences   \n3147 Transcription start site (TSS)               sequences   \n3148 Transcript length (including UTRs and CDS)   sequences   \n3149 CDS Length                                   sequences   \n3150 CDS start                                    sequences   \n3151 CDS end                                      sequences   \n3152 Exon stable ID                               sequences   \n3153 Exon region start (bp)                       sequences   \n3154 Exon region end (bp)                         sequences   \n3155 Strand                                       sequences   \n3156 Exon rank in transcript                      sequences   \n3157 Start phase                                  sequences   \n3158 End phase                                    sequences   \n3159 cDNA coding start                            sequences   \n3160 cDNA coding end                              sequences   \n3161 Genomic coding start                         sequences   \n3162 Genomic coding end                           sequences   \n3163 Constitutive exon                            sequences   "},"metadata":{}}],"execution_count":76},{"cell_type":"code","source":"attributePages(mart)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:19:26.329467Z","iopub.execute_input":"2025-04-13T12:19:26.331310Z","iopub.status.idle":"2025-04-13T12:19:26.347646Z","shell.execute_reply":"2025-04-13T12:19:26.345900Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>'feature_page'</li><li>'structure'</li><li>'homologs'</li><li>'snp'</li><li>'snp_somatic'</li><li>'sequences'</li></ol>\n","text/markdown":"1. 'feature_page'\n2. 'structure'\n3. 'homologs'\n4. 'snp'\n5. 'snp_somatic'\n6. 'sequences'\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 'feature\\_page'\n\\item 'structure'\n\\item 'homologs'\n\\item 'snp'\n\\item 'snp\\_somatic'\n\\item 'sequences'\n\\end{enumerate*}\n","text/plain":"[1] \"feature_page\" \"structure\"    \"homologs\"     \"snp\"          \"snp_somatic\" \n[6] \"sequences\"   "},"metadata":{}}],"execution_count":77},{"cell_type":"code","source":"library (GenomicRanges)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:58:52.392704Z","iopub.execute_input":"2025-04-13T12:58:52.394325Z","iopub.status.idle":"2025-04-13T12:58:52.409037Z","shell.execute_reply":"2025-04-13T12:58:52.407075Z"}},"outputs":[],"execution_count":86},{"cell_type":"code","source":"?'ExpressionSet-class'","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:29:45.167819Z","iopub.execute_input":"2025-04-13T12:29:45.169355Z","iopub.status.idle":"2025-04-13T12:29:45.517565Z","shell.execute_reply":"2025-04-13T12:29:45.512317Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"output_type":"display_data","data":{"text/html":"\n<table style=\"width: 100%;\"><tr><td>ExpressionSet {Biobase}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n\n<h2>\nClass to Contain and Describe High-Throughput Expression Level Assays.\n</h2>\n\n<h3>Description</h3>\n\n<p>Container for high-throughput assays and experimental\nmetadata. <code>ExpressionSet</code> class is derived from\n<code>eSet</code>, and requires a matrix named <code>exprs</code> as\nassayData member.\n</p>\n\n\n<h3>Usage</h3>\n\n<pre><code class='language-R'>## Instance creation\nExpressionSet(assayData,\n    phenoData=annotatedDataFrameFrom(assayData, byrow=FALSE),\n    featureData=annotatedDataFrameFrom(assayData, byrow=TRUE),\n    experimentData=MIAME(), annotation=character(),\n    protocolData=annotatedDataFrameFrom(assayData, byrow=FALSE),\n    ...)\n\n## Additional methods documented below\n</code></pre>\n\n\n<h3>Arguments</h3>\n\n<table>\n<tr><td><code id=\"assayData\">assayData</code></td>\n<td>\n<p>A <code>matrix</code> of expression values, or an\n<code>environment</code>.\n</p>\n<p>When <code>assayData</code> is a <code>matrix</code>, the rows represent probe\nsets (&lsquo;features&rsquo; in <code>ExpressionSet</code> parlance).  Columns\nrepresent samples. When present, row names identify features and\ncolumn names identify samples. Row and column names must be unique,\nand consistent with row names of <code>featureData</code> and\n<code>phenoData</code>, respectively. The assay data can be retrieved with\n<code>exprs()</code>.\n</p>\n<p>When <code>assayData</code> is an environment, it contains identically\ndimensioned matrices like that described in the previous\nparagraph. One of the elements of the environment must be named\n&lsquo;exprs&rsquo;; this element is returned with <code>exprs()</code>.\n</p>\n</td></tr>\n<tr><td><code id=\"phenoData\">phenoData</code></td>\n<td>\n<p>An optional <code>AnnotatedDataFrame</code> containing\ninformation about each sample. The number of rows in\n<code>phenoData</code> must match the number of columns in\n<code>assayData</code>. Row names of <code>phenoData</code> must match column\nnames of the matrix / matricies in <code>assayData</code>.</p>\n</td></tr>\n<tr><td><code id=\"featureData\">featureData</code></td>\n<td>\n<p>An optional <code>AnnotatedDataFrame</code> containing\ninformation about each feature. The number of rows in\n<code>featureData</code> must match the number of rows in\n<code>assayData</code>. Row names of <code>featureData</code> must match row\nnames of the matrix / matricies in <code>assayData</code>.</p>\n</td></tr>\n<tr><td><code id=\"experimentData\">experimentData</code></td>\n<td>\n<p>An optional <code>MIAME</code> instance with meta-data\n(e.g., the lab and resulting publications from the analysis) about\nthe experiment.</p>\n</td></tr>\n<tr><td><code id=\"annotation\">annotation</code></td>\n<td>\n<p>A <code>character</code> describing the platform on which\nthe samples were assayed. This is often the name of a Bioconductor\nchip annotation package, which facilitated down-stream analysis.</p>\n</td></tr>\n<tr><td><code id=\"protocolData\">protocolData</code></td>\n<td>\n<p>An optional <code>AnnotatedDataFrame</code> containing\nequipment-generated information about protocols. The number of rows\nand row names of <code>protocolData</code> must agree with the dimension\nand column names of <code>assayData</code>.</p>\n</td></tr>\n<tr><td><code id=\"...\">...</code></td>\n<td>\n<p>Additional arguments, passed to\n<code>new(\"ExpressionSet\", ...)</code> and available for classes that extend\n<code>ExpressionSet</code>.\n</p>\n</td></tr>\n</table>\n\n\n<h3>Extends</h3>\n\n<p>Directly extends class <code>eSet</code>.\n</p>\n\n\n<h3>Creating Objects</h3>\n\n<p><code>ExpressionSet</code> instances are usually created through\n<code>ExpressionSet()</code>.\n</p>\n\n\n<h3>Slots</h3>\n\n<p>Inherited from <code>eSet</code>:\n</p>\n\n<dl>\n<dt><code>assayData</code>:</dt><dd><p>Contains matrices with equal\ndimensions, and with column number equal to\n<code>nrow(phenoData)</code>. <code>assayData</code> must contain a matrix\n<code>exprs</code> with rows representing features (e.g., probe sets)\nand columns representing samples. Additional matrices of\nidentical size (e.g., representing measurement errors) may\nalso be included in <code>assayData</code>. Class:<code>AssayData-class</code></p>\n</dd>\n<dt><code>phenoData</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>featureData</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>experimentData</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>annotation</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>protocolData</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n</dl>\n\n\n\n<h3>Methods</h3>\n\n<p>Class-specific methods.\n</p>\n\n<dl>\n<dt><code>as(exprSet,\"ExpressionSet\")</code></dt><dd><p>Coerce objects of\n<code>exprSet-class</code> to <code>ExpressionSet</code></p>\n</dd>\n<dt><code>as(object,\"data.frame\")</code></dt><dd><p>Coerce objects of\n<code>ExpressionSet-class</code> to <code>data.frame</code> by\ntransposing the expression matrix and concatenating <code>phenoData</code></p>\n</dd>\n<dt><code>exprs(ExpressionSet)</code>, <code>exprs(ExpressionSet,matrix)&lt;-</code></dt><dd><p>Access and\nset elements named <code>exprs</code> in the <code>AssayData-class</code>\nslot.</p>\n</dd>\n<dt><code>esApply(ExpressionSet, MARGIN, FUN,\n         ...)</code></dt><dd><p>'apply'-like function to conveniently operate on\n<code>ExpressionSet</code> objects. See <code>esApply</code>.</p>\n</dd>\n<dt><code>write.exprs(ExpressionSet)</code></dt><dd><p>Write expression values to\na text file. It takes the same arguments as\n<code>write.table</code></p>\n</dd>\n</dl>\n\n<p>Derived from <code>eSet</code>:\n</p>\n\n<dl>\n<dt><code>updateObject(object, ..., verbose=FALSE)</code></dt><dd><p>Update instance to current version, if necessary. See <code>updateObject</code> and <code>eSet</code></p>\n</dd>\n<dt><code>isCurrent(object)</code></dt><dd><p>Determine whether version of object is current. See <code>isCurrent</code></p>\n</dd>\n<dt><code>isVersioned(object)</code></dt><dd><p>Determine whether object contains a 'version' string describing its structure . See <code>isVersioned</code></p>\n</dd>\n<dt><code>assayData(ExpressionSet)</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>sampleNames(ExpressionSet)</code> and <code>sampleNames(ExpressionSet)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>featureNames(ExpressionSet)</code>, <code>featureNames(ExpressionSet, value)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>dims(ExpressionSet)</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>phenoData(ExpressionSet)</code>, <code>phenoData(ExpressionSet,value)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>varLabels(ExpressionSet)</code>, <code>varLabels(ExpressionSet, value)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>varMetadata(ExpressionSet)</code>, <code>varMetadata(ExpressionSet,value)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>pData(ExpressionSet)</code>, <code>pData(ExpressionSet,value)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>varMetadata(ExpressionSet)</code>, <code>varMetadata(ExpressionSet,value)</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>experimentData(ExpressionSet)</code>,<code>experimentData(ExpressionSet,value)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>pubMedIds(ExpressionSet)</code>, <code>pubMedIds(ExpressionSet,value)</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>abstract(ExpressionSet)</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>annotation(ExpressionSet)</code>, <code>annotation(ExpressionSet,value)&lt;-</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>protocolData(ExpressionSet)</code>, <code>protocolData(ExpressionSet,value)&lt;-</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>combine(ExpressionSet,ExpressionSet)</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>storageMode(ExpressionSet)</code>, <code>storageMode(ExpressionSet,character)&lt;-</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n</dl>\n\n<p>Standard generic methods:\n</p>\n\n<dl>\n<dt><code>initialize(ExpressionSet)</code>:</dt><dd><p>Object instantiation, used\nby <code>new</code>; not to be called directly by the user.</p>\n</dd>\n<dt><code>updateObject(ExpressionSet)</code>:</dt><dd><p>Update outdated versions\nof <code>ExpressionSet</code> to their current definition. See\n<code>updateObject</code>, <code>Versions-class</code>.</p>\n</dd>\n<dt><code>validObject(ExpressionSet)</code>:</dt><dd><p>Validity-checking method, ensuring\nthat <code>exprs</code> is a member of\n<code>assayData</code>. <code>checkValidity(ExpressionSet)</code> imposes this\nvalidity check, and the validity checks of <code>eSet</code>.</p>\n</dd>\n<dt><code>makeDataPackage(object, author, email, packageName, packageVersion, license, biocViews, filePath, description=paste(abstract(object), collapse=\"\\n\\n\"), ...)</code></dt><dd>\n<p>Create a data package based on an ExpressionSet object. See\n<code>makeDataPackage</code>.</p>\n</dd>\n<dt><code>as(exprSet,ExpressionSet)</code>:</dt><dd><p>Coerce <code>exprSet</code> to <code>ExpressionSet</code>.</p>\n</dd>\n<dt><code>as(eSet,ExpressionSet)</code>:</dt><dd><p>Coerce the <code>eSet</code> portion of an object to <code>ExpressionSet</code>.</p>\n</dd>\n<dt><code>show(ExpressionSet)</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>dim(ExpressionSet)</code>, <code>ncol</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>ExpressionSet[(index)</code>:</dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>ExpressionSet$</code>, <code>ExpressionSet$&lt;-</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n<dt><code>ExpressionSet[[i]]</code>, <code>ExpressionSet[[i]]&lt;-</code></dt><dd><p>See <code>eSet</code></p>\n</dd>\n</dl>\n\n\n\n<h3>Author(s)</h3>\n\n<p>Biocore team</p>\n\n\n<h3>See Also</h3>\n\n<p><code>eSet-class</code>, <code>ExpressionSet-class</code>.\n</p>\n\n\n<h3>Examples</h3>\n\n<pre><code class='language-R'># create an instance of ExpressionSet\nExpressionSet()\n\nExpressionSet(assayData=matrix(runif(1000), nrow=100, ncol=10))\n\n# update an existing ExpressionSet\ndata(sample.ExpressionSet)\nupdateObject(sample.ExpressionSet)\n\n# information about assay and sample data\nfeatureNames(sample.ExpressionSet)[1:10]\nsampleNames(sample.ExpressionSet)[1:5]\nexperimentData(sample.ExpressionSet)\n\n# subset: first 10 genes, samples 2, 4, and 10\nexpressionSet &lt;- sample.ExpressionSet[1:10,c(2,4,10)]\n\n# named features and their expression levels\nsubset &lt;- expressionSet[c(\"AFFX-BioC-3_at\",\"AFFX-BioDn-5_at\"),]\nexprs(subset)\n\n# samples with above-average 'score' in phenoData\nhighScores &lt;- expressionSet$score &gt; mean(expressionSet$score)\nexpressionSet[,highScores]\n\n# (automatically) coerce to data.frame\nlm(score~AFFX.BioDn.5_at + AFFX.BioC.3_at, data=subset)\n</code></pre>\n\n<hr /><div style=\"text-align: center;\">[Package <em>Biobase</em> version 2.64.0 ]</div></main>\n\n</div>","text/latex":"\\inputencoding{utf8}\n\\HeaderA{ExpressionSet}{Class to Contain and Describe High-Throughput Expression Level Assays.}{ExpressionSet}\n\\aliasA{as.data.frame.ExpressionSet}{ExpressionSet}{as.data.frame.ExpressionSet}\n\\aliasA{class:ExpressionSet}{ExpressionSet}{class:ExpressionSet}\n\\aliasA{coerce,eSet,ExpressionSet-method}{ExpressionSet}{coerce,eSet,ExpressionSet.Rdash.method}\n\\aliasA{coerce,ExpressionSet,data.frame-method}{ExpressionSet}{coerce,ExpressionSet,data.frame.Rdash.method}\n\\aliasA{coerce,exprSet,ExpressionSet-method}{ExpressionSet}{coerce,exprSet,ExpressionSet.Rdash.method}\n\\aliasA{esApply,ExpressionSet-method}{ExpressionSet}{esApply,ExpressionSet.Rdash.method}\n\\aliasA{ExpressionSet-class}{ExpressionSet}{ExpressionSet.Rdash.class}\n\\aliasA{ExpressionSet,environment-method}{ExpressionSet}{ExpressionSet,environment.Rdash.method}\n\\aliasA{ExpressionSet,matrix-method}{ExpressionSet}{ExpressionSet,matrix.Rdash.method}\n\\aliasA{ExpressionSet,missing-method}{ExpressionSet}{ExpressionSet,missing.Rdash.method}\n\\aliasA{exprs,ExpressionSet-method}{ExpressionSet}{exprs,ExpressionSet.Rdash.method}\n\\aliasA{exprs<\\Rdash{},ExpressionSet,matrix-method}{ExpressionSet}{exprs<.Rdash.,ExpressionSet,matrix.Rdash.method}\n\\aliasA{initialize,ExpressionSet-method}{ExpressionSet}{initialize,ExpressionSet.Rdash.method}\n\\aliasA{makeDataPackage,ExpressionSet-method}{ExpressionSet}{makeDataPackage,ExpressionSet.Rdash.method}\n\\aliasA{updateObject,ExpressionSet-method}{ExpressionSet}{updateObject,ExpressionSet.Rdash.method}\n\\aliasA{write.exprs}{ExpressionSet}{write.exprs}\n\\aliasA{write.exprs,ExpressionSet-method}{ExpressionSet}{write.exprs,ExpressionSet.Rdash.method}\n\\keyword{classes}{ExpressionSet}\n%\n\\begin{Description}\nContainer for high-throughput assays and experimental\nmetadata. \\code{ExpressionSet} class is derived from\n\\code{\\LinkA{eSet}{eSet}}, and requires a matrix named \\code{exprs} as\nassayData member.\n\\end{Description}\n%\n\\begin{Usage}\n\\begin{verbatim}\n## Instance creation\nExpressionSet(assayData,\n    phenoData=annotatedDataFrameFrom(assayData, byrow=FALSE),\n    featureData=annotatedDataFrameFrom(assayData, byrow=TRUE),\n    experimentData=MIAME(), annotation=character(),\n    protocolData=annotatedDataFrameFrom(assayData, byrow=FALSE),\n    ...)\n\n## Additional methods documented below\n\\end{verbatim}\n\\end{Usage}\n%\n\\begin{Arguments}\n\n\\begin{ldescription}\n\\item[\\code{assayData}] A \\code{matrix} of expression values, or an\n\\code{environment}.\n\nWhen \\code{assayData} is a \\code{matrix}, the rows represent probe\nsets (`features' in \\code{ExpressionSet} parlance).  Columns\nrepresent samples. When present, row names identify features and\ncolumn names identify samples. Row and column names must be unique,\nand consistent with row names of \\code{featureData} and\n\\code{phenoData}, respectively. The assay data can be retrieved with\n\\code{exprs()}.\n\nWhen \\code{assayData} is an environment, it contains identically\ndimensioned matrices like that described in the previous\nparagraph. One of the elements of the environment must be named\n`exprs'; this element is returned with \\code{exprs()}.\n\n\n\n\\item[\\code{phenoData}] An optional \\code{AnnotatedDataFrame} containing\ninformation about each sample. The number of rows in\n\\code{phenoData} must match the number of columns in\n\\code{assayData}. Row names of \\code{phenoData} must match column\nnames of the matrix / matricies in \\code{assayData}.\n\n\\item[\\code{featureData}] An optional \\code{AnnotatedDataFrame} containing\ninformation about each feature. The number of rows in\n\\code{featureData} must match the number of rows in\n\\code{assayData}. Row names of \\code{featureData} must match row\nnames of the matrix / matricies in \\code{assayData}.\n\n\\item[\\code{experimentData}] An optional \\code{MIAME} instance with meta-data\n(e.g., the lab and resulting publications from the analysis) about\nthe experiment.\n\n\\item[\\code{annotation}] A \\code{character} describing the platform on which\nthe samples were assayed. This is often the name of a Bioconductor\nchip annotation package, which facilitated down-stream analysis.\n\n\\item[\\code{protocolData}] An optional \\code{AnnotatedDataFrame} containing\nequipment-generated information about protocols. The number of rows\nand row names of \\code{protocolData} must agree with the dimension\nand column names of \\code{assayData}.\n\n\\item[\\code{...}] Additional arguments, passed to\n\\code{new(\"ExpressionSet\", ...)} and available for classes that extend\n\\code{ExpressionSet}.\n\n\n\\end{ldescription}\n\\end{Arguments}\n%\n\\begin{Section}{Extends}\nDirectly extends class \\code{\\LinkA{eSet}{eSet}}.\n\\end{Section}\n%\n\\begin{Section}{Creating Objects}\n\n\\code{ExpressionSet} instances are usually created through\n\\code{ExpressionSet()}.\n\n\\end{Section}\n%\n\\begin{Section}{Slots}\nInherited from \\code{eSet}:\n\\begin{description}\n\n\\item[\\code{assayData}:] Contains matrices with equal\ndimensions, and with column number equal to\n\\code{nrow(phenoData)}. \\code{assayData} must contain a matrix\n\\code{exprs} with rows representing features (e.g., probe sets)\nand columns representing samples. Additional matrices of\nidentical size (e.g., representing measurement errors) may\nalso be included in \\code{assayData}. Class:\\code{\\LinkA{AssayData-class}{AssayData.Rdash.class}}\n\\item[\\code{phenoData}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{featureData}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{experimentData}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{annotation}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{protocolData}:] See \\code{\\LinkA{eSet}{eSet}}\n\n\\end{description}\n\n\\end{Section}\n%\n\\begin{Section}{Methods}\n\nClass-specific methods.\n\\begin{description}\n\n\\item[\\code{as(exprSet,\"ExpressionSet\")}] Coerce objects of\n\\code{\\LinkA{exprSet-class}{exprSet.Rdash.class}} to \\code{ExpressionSet}\n\\item[\\code{as(object,\"data.frame\")}] Coerce objects of\n\\code{\\LinkA{ExpressionSet-class}{ExpressionSet.Rdash.class}} to \\code{data.frame} by\ntransposing the expression matrix and concatenating \\code{phenoData}\n\\item[\\code{exprs(ExpressionSet)}, \\code{exprs(ExpressionSet,matrix)<-}] Access and\nset elements named \\code{exprs} in the \\code{AssayData-class}\nslot.\n\\item[\\code{esApply(ExpressionSet, MARGIN, FUN,\n         ...)}] 'apply'-like function to conveniently operate on\n\\code{ExpressionSet} objects. See \\code{\\LinkA{esApply}{esApply}}.\n\\item[\\code{write.exprs(ExpressionSet)}] Write expression values to\na text file. It takes the same arguments as\n\\code{write.table}\n\n\\end{description}\n\n\nDerived from \\code{\\LinkA{eSet}{eSet}}:\n\\begin{description}\n\n\\item[\\code{updateObject(object, ..., verbose=FALSE)}] Update instance to current version, if necessary. See \\code{\\LinkA{updateObject}{updateObject}} and \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{isCurrent(object)}] Determine whether version of object is current. See \\code{\\LinkA{isCurrent}{isCurrent}}\n\\item[\\code{isVersioned(object)}] Determine whether object contains a 'version' string describing its structure . See \\code{\\LinkA{isVersioned}{isVersioned}}\n\\item[\\code{assayData(ExpressionSet)}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{sampleNames(ExpressionSet)} and \\code{sampleNames(ExpressionSet)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{featureNames(ExpressionSet)}, \\code{featureNames(ExpressionSet, value)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{dims(ExpressionSet)}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{phenoData(ExpressionSet)}, \\code{phenoData(ExpressionSet,value)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{varLabels(ExpressionSet)}, \\code{varLabels(ExpressionSet, value)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{varMetadata(ExpressionSet)}, \\code{varMetadata(ExpressionSet,value)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{pData(ExpressionSet)}, \\code{pData(ExpressionSet,value)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{varMetadata(ExpressionSet)}, \\code{varMetadata(ExpressionSet,value)}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{experimentData(ExpressionSet)},\\code{experimentData(ExpressionSet,value)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{pubMedIds(ExpressionSet)}, \\code{pubMedIds(ExpressionSet,value)}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{abstract(ExpressionSet)}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{annotation(ExpressionSet)}, \\code{annotation(ExpressionSet,value)<-}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{protocolData(ExpressionSet)}, \\code{protocolData(ExpressionSet,value)<-}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{combine(ExpressionSet,ExpressionSet)}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{storageMode(ExpressionSet)}, \\code{storageMode(ExpressionSet,character)<-}:] See \\code{\\LinkA{eSet}{eSet}}\n\n\\end{description}\n\n\nStandard generic methods:\n\\begin{description}\n\n\\item[\\code{initialize(ExpressionSet)}:] Object instantiation, used\nby \\code{new}; not to be called directly by the user.\n\\item[\\code{updateObject(ExpressionSet)}:] Update outdated versions\nof \\code{ExpressionSet} to their current definition. See\n\\code{\\LinkA{updateObject}{updateObject}}, \\code{\\LinkA{Versions-class}{Versions.Rdash.class}}.\n\\item[\\code{validObject(ExpressionSet)}:] Validity-checking method, ensuring\nthat \\code{exprs} is a member of\n\\code{assayData}. \\code{checkValidity(ExpressionSet)} imposes this\nvalidity check, and the validity checks of \\code{eSet}.\n\\item[\\code{makeDataPackage(object, author, email, packageName, packageVersion, license, biocViews, filePath, description=paste(abstract(object), collapse=\"\\bsl{}n\\bsl{}n\"), ...)}] \nCreate a data package based on an ExpressionSet object. See\n\\code{\\LinkA{makeDataPackage}{makeDataPackage}}.\n\\item[\\code{as(exprSet,ExpressionSet)}:] Coerce \\code{exprSet} to \\code{ExpressionSet}.\n\\item[\\code{as(eSet,ExpressionSet)}:] Coerce the \\code{eSet} portion of an object to \\code{ExpressionSet}.\n\\item[\\code{show(ExpressionSet)}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{dim(ExpressionSet)}, \\code{ncol}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{ExpressionSet[(index)}:] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{ExpressionSet\\$}, \\code{ExpressionSet\\$<-}] See \\code{\\LinkA{eSet}{eSet}}\n\\item[\\code{ExpressionSet[[i]]}, \\code{ExpressionSet[[i]]<-}] See \\code{\\LinkA{eSet}{eSet}}\n\n\\end{description}\n\n\\end{Section}\n%\n\\begin{Author}\nBiocore team\n\\end{Author}\n%\n\\begin{SeeAlso}\n\\code{\\LinkA{eSet-class}{eSet.Rdash.class}}, \\code{\\LinkA{ExpressionSet-class}{ExpressionSet.Rdash.class}}.\n\\end{SeeAlso}\n%\n\\begin{Examples}\n\\begin{ExampleCode}\n# create an instance of ExpressionSet\nExpressionSet()\n\nExpressionSet(assayData=matrix(runif(1000), nrow=100, ncol=10))\n\n# update an existing ExpressionSet\ndata(sample.ExpressionSet)\nupdateObject(sample.ExpressionSet)\n\n# information about assay and sample data\nfeatureNames(sample.ExpressionSet)[1:10]\nsampleNames(sample.ExpressionSet)[1:5]\nexperimentData(sample.ExpressionSet)\n\n# subset: first 10 genes, samples 2, 4, and 10\nexpressionSet <- sample.ExpressionSet[1:10,c(2,4,10)]\n\n# named features and their expression levels\nsubset <- expressionSet[c(\"AFFX-BioC-3_at\",\"AFFX-BioDn-5_at\"),]\nexprs(subset)\n\n# samples with above-average 'score' in phenoData\nhighScores <- expressionSet$score > mean(expressionSet$score)\nexpressionSet[,highScores]\n\n# (automatically) coerce to data.frame\nlm(score~AFFX.BioDn.5_at + AFFX.BioC.3_at, data=subset)\n\\end{ExampleCode}\n\\end{Examples}","text/plain":"ExpressionSet             package:Biobase              R Documentation\n\n_\bC_\bl_\ba_\bs_\bs _\bt_\bo _\bC_\bo_\bn_\bt_\ba_\bi_\bn _\ba_\bn_\bd _\bD_\be_\bs_\bc_\br_\bi_\bb_\be _\bH_\bi_\bg_\bh-_\bT_\bh_\br_\bo_\bu_\bg_\bh_\bp_\bu_\bt _\bE_\bx_\bp_\br_\be_\bs_\bs_\bi_\bo_\bn _\bL_\be_\bv_\be_\bl _\bA_\bs_\bs_\ba_\by_\bs.\n\n_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n\n     Container for high-throughput assays and experimental metadata.\n     ‘ExpressionSet’ class is derived from ‘eSet’, and requires a\n     matrix named ‘exprs’ as assayData member.\n\n_\bU_\bs_\ba_\bg_\be:\n\n     ## Instance creation\n     ExpressionSet(assayData,\n         phenoData=annotatedDataFrameFrom(assayData, byrow=FALSE),\n         featureData=annotatedDataFrameFrom(assayData, byrow=TRUE),\n         experimentData=MIAME(), annotation=character(),\n         protocolData=annotatedDataFrameFrom(assayData, byrow=FALSE),\n         ...)\n     \n     ## Additional methods documented below\n     \n_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n\nassayData: A ‘matrix’ of expression values, or an ‘environment’.\n\n          When ‘assayData’ is a ‘matrix’, the rows represent probe sets\n          (‘features’ in ‘ExpressionSet’ parlance).  Columns represent\n          samples. When present, row names identify features and column\n          names identify samples. Row and column names must be unique,\n          and consistent with row names of ‘featureData’ and\n          ‘phenoData’, respectively. The assay data can be retrieved\n          with ‘exprs()’.\n\n          When ‘assayData’ is an environment, it contains identically\n          dimensioned matrices like that described in the previous\n          paragraph. One of the elements of the environment must be\n          named ‘exprs’; this element is returned with ‘exprs()’.\n\nphenoData: An optional ‘AnnotatedDataFrame’ containing information\n          about each sample. The number of rows in ‘phenoData’ must\n          match the number of columns in ‘assayData’. Row names of\n          ‘phenoData’ must match column names of the matrix / matricies\n          in ‘assayData’.\n\nfeatureData: An optional ‘AnnotatedDataFrame’ containing information\n          about each feature. The number of rows in ‘featureData’ must\n          match the number of rows in ‘assayData’. Row names of\n          ‘featureData’ must match row names of the matrix / matricies\n          in ‘assayData’.\n\nexperimentData: An optional ‘MIAME’ instance with meta-data (e.g., the\n          lab and resulting publications from the analysis) about the\n          experiment.\n\nannotation: A ‘character’ describing the platform on which the samples\n          were assayed. This is often the name of a Bioconductor chip\n          annotation package, which facilitated down-stream analysis.\n\nprotocolData: An optional ‘AnnotatedDataFrame’ containing\n          equipment-generated information about protocols. The number\n          of rows and row names of ‘protocolData’ must agree with the\n          dimension and column names of ‘assayData’.\n\n     ...: Additional arguments, passed to ‘new(\"ExpressionSet\", ...)’\n          and available for classes that extend ‘ExpressionSet’.\n\n_\bE_\bx_\bt_\be_\bn_\bd_\bs:\n\n     Directly extends class ‘eSet’.\n\n_\bC_\br_\be_\ba_\bt_\bi_\bn_\bg _\bO_\bb_\bj_\be_\bc_\bt_\bs:\n\n     ‘ExpressionSet’ instances are usually created through\n     ‘ExpressionSet()’.\n\n_\bS_\bl_\bo_\bt_\bs:\n\n     Inherited from ‘eSet’:\n\n     ‘assayData’: Contains matrices with equal dimensions, and with\n          column number equal to ‘nrow(phenoData)’. ‘assayData’ must\n          contain a matrix ‘exprs’ with rows representing features\n          (e.g., probe sets) and columns representing samples.\n          Additional matrices of identical size (e.g., representing\n          measurement errors) may also be included in ‘assayData’.\n          Class:‘AssayData-class’\n\n     ‘phenoData’: See ‘eSet’\n\n     ‘featureData’: See ‘eSet’\n\n     ‘experimentData’: See ‘eSet’\n\n     ‘annotation’: See ‘eSet’\n\n     ‘protocolData’: See ‘eSet’\n\n_\bM_\be_\bt_\bh_\bo_\bd_\bs:\n\n     Class-specific methods.\n\n     ‘as(exprSet,\"ExpressionSet\")’ Coerce objects of ‘exprSet-class’ to\n          ‘ExpressionSet’\n\n     ‘as(object,\"data.frame\")’ Coerce objects of ‘ExpressionSet-class’\n          to ‘data.frame’ by transposing the expression matrix and\n          concatenating ‘phenoData’\n\n     ‘exprs(ExpressionSet)’, ‘exprs(ExpressionSet,matrix)<-’ Access and\n          set elements named ‘exprs’ in the ‘AssayData-class’ slot.\n\n     ‘esApply(ExpressionSet, MARGIN, FUN, ...)’ 'apply'-like function\n          to conveniently operate on ‘ExpressionSet’ objects. See\n          ‘esApply’.\n\n     ‘write.exprs(ExpressionSet)’ Write expression values to a text\n          file. It takes the same arguments as ‘write.table’\n\n     Derived from ‘eSet’:\n\n     ‘updateObject(object, ..., verbose=FALSE)’ Update instance to\n          current version, if necessary. See ‘updateObject’ and ‘eSet’\n\n     ‘isCurrent(object)’ Determine whether version of object is\n          current. See ‘isCurrent’\n\n     ‘isVersioned(object)’ Determine whether object contains a\n          'version' string describing its structure . See ‘isVersioned’\n\n     ‘assayData(ExpressionSet)’: See ‘eSet’\n\n     ‘sampleNames(ExpressionSet)’ and ‘sampleNames(ExpressionSet)<-’:\n          See ‘eSet’\n\n     ‘featureNames(ExpressionSet)’, ‘featureNames(ExpressionSet,\n          value)<-’: See ‘eSet’\n\n     ‘dims(ExpressionSet)’: See ‘eSet’\n\n     ‘phenoData(ExpressionSet)’, ‘phenoData(ExpressionSet,value)<-’:\n          See ‘eSet’\n\n     ‘varLabels(ExpressionSet)’, ‘varLabels(ExpressionSet, value)<-’:\n          See ‘eSet’\n\n     ‘varMetadata(ExpressionSet)’,\n          ‘varMetadata(ExpressionSet,value)<-’: See ‘eSet’\n\n     ‘pData(ExpressionSet)’, ‘pData(ExpressionSet,value)<-’: See ‘eSet’\n\n     ‘varMetadata(ExpressionSet)’, ‘varMetadata(ExpressionSet,value)’\n          See ‘eSet’\n\n     ‘experimentData(ExpressionSet)’,‘experimentData(ExpressionSet,value)<-’:\n          See ‘eSet’\n\n     ‘pubMedIds(ExpressionSet)’, ‘pubMedIds(ExpressionSet,value)’ See\n          ‘eSet’\n\n     ‘abstract(ExpressionSet)’: See ‘eSet’\n\n     ‘annotation(ExpressionSet)’, ‘annotation(ExpressionSet,value)<-’\n          See ‘eSet’\n\n     ‘protocolData(ExpressionSet)’,\n          ‘protocolData(ExpressionSet,value)<-’ See ‘eSet’\n\n     ‘combine(ExpressionSet,ExpressionSet)’: See ‘eSet’\n\n     ‘storageMode(ExpressionSet)’,\n          ‘storageMode(ExpressionSet,character)<-’: See ‘eSet’\n\n     Standard generic methods:\n\n     ‘initialize(ExpressionSet)’: Object instantiation, used by ‘new’;\n          not to be called directly by the user.\n\n     ‘updateObject(ExpressionSet)’: Update outdated versions of\n          ‘ExpressionSet’ to their current definition. See\n          ‘updateObject’, ‘Versions-class’.\n\n     ‘validObject(ExpressionSet)’: Validity-checking method, ensuring\n          that ‘exprs’ is a member of ‘assayData’.\n          ‘checkValidity(ExpressionSet)’ imposes this validity check,\n          and the validity checks of ‘eSet’.\n\n     ‘makeDataPackage(object, author, email, packageName,\n          packageVersion, license, biocViews, filePath,\n          description=paste(abstract(object), collapse=\"\\n\\n\"), ...)’\n          Create a data package based on an ExpressionSet object. See\n          ‘makeDataPackage’.\n\n     ‘as(exprSet,ExpressionSet)’: Coerce ‘exprSet’ to ‘ExpressionSet’.\n\n     ‘as(eSet,ExpressionSet)’: Coerce the ‘eSet’ portion of an object\n          to ‘ExpressionSet’.\n\n     ‘show(ExpressionSet)’ See ‘eSet’\n\n     ‘dim(ExpressionSet)’, ‘ncol’ See ‘eSet’\n\n     ‘ExpressionSet[(index)’: See ‘eSet’\n\n     ‘ExpressionSet$’, ‘ExpressionSet$<-’ See ‘eSet’\n\n     ‘ExpressionSet[[i]]’, ‘ExpressionSet[[i]]<-’ See ‘eSet’\n\n_\bA_\bu_\bt_\bh_\bo_\br(_\bs):\n\n     Biocore team\n\n_\bS_\be_\be _\bA_\bl_\bs_\bo:\n\n     ‘eSet-class’, ‘ExpressionSet-class’.\n\n_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n\n     # create an instance of ExpressionSet\n     ExpressionSet()\n     \n     ExpressionSet(assayData=matrix(runif(1000), nrow=100, ncol=10))\n     \n     # update an existing ExpressionSet\n     data(sample.ExpressionSet)\n     updateObject(sample.ExpressionSet)\n     \n     # information about assay and sample data\n     featureNames(sample.ExpressionSet)[1:10]\n     sampleNames(sample.ExpressionSet)[1:5]\n     experimentData(sample.ExpressionSet)\n     \n     # subset: first 10 genes, samples 2, 4, and 10\n     expressionSet <- sample.ExpressionSet[1:10,c(2,4,10)]\n     \n     # named features and their expression levels\n     subset <- expressionSet[c(\"AFFX-BioC-3_at\",\"AFFX-BioDn-5_at\"),]\n     exprs(subset)\n     \n     # samples with above-average 'score' in phenoData\n     highScores <- expressionSet$score > mean(expressionSet$score)\n     expressionSet[,highScores]\n     \n     # (automatically) coerce to data.frame\n     lm(score~AFFX.BioDn.5_at + AFFX.BioC.3_at, data=subset)\n     "},"metadata":{}}],"execution_count":80},{"cell_type":"code","source":"# Define an S4 class called \"Person\"\nsetClass(\"Person\",\n         slots = list(name = \"character\", age = \"numeric\"))\n\n# Create an instance of the Person class\njohn <- new(\"Person\", name = \"John Doe\", age = 30)\n\n# Access the slots\njohn@name  # Output: \"John Doe\"\njohn@age   # Output: 30","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:51:09.487971Z","iopub.execute_input":"2025-04-13T12:51:09.489928Z","iopub.status.idle":"2025-04-13T12:51:09.521844Z","shell.execute_reply":"2025-04-13T12:51:09.519902Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"'John Doe'","text/markdown":"'John Doe'","text/latex":"'John Doe'","text/plain":"[1] \"John Doe\""},"metadata":{}},{"output_type":"display_data","data":{"text/html":"30","text/markdown":"30","text/latex":"30","text/plain":"[1] 30"},"metadata":{}}],"execution_count":83},{"cell_type":"code","source":"# Define an S4 class with a validity check\nsetClass(\"Employee\",\n         slots = list(name = \"character\", salary = \"numeric\"),\n         validity = function(object) {\n           if (object@salary < 0) {\n             return(\"Salary must be non-negative\")\n           }\n           TRUE\n         })\n\n# Create a valid Employee object\nemp1 <- new(\"Employee\", name = \"Alice\", salary = 50000)\n\n# Validate the object\nvalidObject(emp1)  # Output: TRUE\n\n# Create an invalid Employee object\n#emp2 <- new(\"Employee\", name = \"Bob\", salary = -1000)\n\n# Validate the invalid object\n#validObject(emp2)  # Output: Error: Salary must be non-negative","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:52:56.268827Z","iopub.execute_input":"2025-04-13T12:52:56.270940Z","iopub.status.idle":"2025-04-13T12:52:56.300447Z","shell.execute_reply":"2025-04-13T12:52:56.298574Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"TRUE","text/markdown":"TRUE","text/latex":"TRUE","text/plain":"[1] TRUE"},"metadata":{}}],"execution_count":85},{"cell_type":"code","source":"as.data.frame","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T12:59:05.663602Z","iopub.execute_input":"2025-04-13T12:59:05.665283Z","iopub.status.idle":"2025-04-13T12:59:05.697664Z","shell.execute_reply":"2025-04-13T12:59:05.694978Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<pre class=language-r><code>new(\"standardGeneric\", .Data = function (x, row.names = NULL, \n<span style=white-space:pre-wrap>    optional = FALSE, ...) </span>\nstandardGeneric(\"as.data.frame\"), generic = structure(\"as.data.frame\", package = \"BiocGenerics\"), \n<span style=white-space:pre-wrap>    package = \"BiocGenerics\", group = list(), valueClass = character(0), </span>\n<span style=white-space:pre-wrap>    signature = \"x\", default = new(\"derivedDefaultMethod\", .Data = function (x, </span>\n<span style=white-space:pre-wrap>        row.names = NULL, optional = FALSE, ...) </span>\n<span style=white-space:pre-wrap>    {</span>\n<span style=white-space:pre-wrap>        if (is.null(x)) </span>\n<span style=white-space:pre-wrap>            return(as.data.frame(list()))</span>\n<span style=white-space:pre-wrap>        UseMethod(\"as.data.frame\")</span>\n<span style=white-space:pre-wrap>    }, target = new(\"signature\", .Data = \"ANY\", names = \"x\", </span>\n<span style=white-space:pre-wrap>        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", </span>\n<span style=white-space:pre-wrap>        names = \"x\", package = \"methods\"), generic = structure(\"as.data.frame\", package = \"base\")), </span>\n<span style=white-space:pre-wrap>    skeleton = (new(\"derivedDefaultMethod\", .Data = function (x, </span>\n<span style=white-space:pre-wrap>        row.names = NULL, optional = FALSE, ...) </span>\n<span style=white-space:pre-wrap>    {</span>\n<span style=white-space:pre-wrap>        if (is.null(x)) </span>\n<span style=white-space:pre-wrap>            return(as.data.frame(list()))</span>\n<span style=white-space:pre-wrap>        UseMethod(\"as.data.frame\")</span>\n<span style=white-space:pre-wrap>    }, target = new(\"signature\", .Data = \"ANY\", names = \"x\", </span>\n<span style=white-space:pre-wrap>        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", </span>\n<span style=white-space:pre-wrap>        names = \"x\", package = \"methods\"), generic = structure(\"as.data.frame\", package = \"base\")))(x, </span>\n<span style=white-space:pre-wrap>        row.names, optional, ...))</span></code></pre>","text/markdown":"```r\nnew(\"standardGeneric\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \nstandardGeneric(\"as.data.frame\"), generic = structure(\"as.data.frame\", package = \"BiocGenerics\"), \n    package = \"BiocGenerics\", group = list(), valueClass = character(0), \n    signature = \"x\", default = new(\"derivedDefaultMethod\", .Data = function (x, \n        row.names = NULL, optional = FALSE, ...) \n    {\n        if (is.null(x)) \n            return(as.data.frame(list()))\n        UseMethod(\"as.data.frame\")\n    }, target = new(\"signature\", .Data = \"ANY\", names = \"x\", \n        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", \n        names = \"x\", package = \"methods\"), generic = structure(\"as.data.frame\", package = \"base\")), \n    skeleton = (new(\"derivedDefaultMethod\", .Data = function (x, \n        row.names = NULL, optional = FALSE, ...) \n    {\n        if (is.null(x)) \n            return(as.data.frame(list()))\n        UseMethod(\"as.data.frame\")\n    }, target = new(\"signature\", .Data = \"ANY\", names = \"x\", \n        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", \n        names = \"x\", package = \"methods\"), generic = structure(\"as.data.frame\", package = \"base\")))(x, \n        row.names, optional, ...))\n```","text/latex":"\\begin{minted}{r}\nnew(\"standardGeneric\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \nstandardGeneric(\"as.data.frame\"), generic = structure(\"as.data.frame\", package = \"BiocGenerics\"), \n    package = \"BiocGenerics\", group = list(), valueClass = character(0), \n    signature = \"x\", default = new(\"derivedDefaultMethod\", .Data = function (x, \n        row.names = NULL, optional = FALSE, ...) \n    \\{\n        if (is.null(x)) \n            return(as.data.frame(list()))\n        UseMethod(\"as.data.frame\")\n    \\}, target = new(\"signature\", .Data = \"ANY\", names = \"x\", \n        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", \n        names = \"x\", package = \"methods\"), generic = structure(\"as.data.frame\", package = \"base\")), \n    skeleton = (new(\"derivedDefaultMethod\", .Data = function (x, \n        row.names = NULL, optional = FALSE, ...) \n    \\{\n        if (is.null(x)) \n            return(as.data.frame(list()))\n        UseMethod(\"as.data.frame\")\n    \\}, target = new(\"signature\", .Data = \"ANY\", names = \"x\", \n        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", \n        names = \"x\", package = \"methods\"), generic = structure(\"as.data.frame\", package = \"base\")))(x, \n        row.names, optional, ...))\n\\end{minted}","text/plain":"new(\"standardGeneric\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \nstandardGeneric(\"as.data.frame\"), generic = \"as.data.frame\", \n    package = \"BiocGenerics\", group = list(), valueClass = character(0), \n    signature = \"x\", default = new(\"derivedDefaultMethod\", .Data = function (x, \n        row.names = NULL, optional = FALSE, ...) \n    {\n        if (is.null(x)) \n            return(as.data.frame(list()))\n        UseMethod(\"as.data.frame\")\n    }, target = new(\"signature\", .Data = \"ANY\", names = \"x\", \n        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", \n        names = \"x\", package = \"methods\"), generic = \"as.data.frame\"), \n    skeleton = (new(\"derivedDefaultMethod\", .Data = function (x, \n        row.names = NULL, optional = FALSE, ...) \n    {\n        if (is.null(x)) \n            return(as.data.frame(list()))\n        UseMethod(\"as.data.frame\")\n    }, target = new(\"signature\", .Data = \"ANY\", names = \"x\", \n        package = \"methods\"), defined = new(\"signature\", .Data = \"ANY\", \n        names = \"x\", package = \"methods\"), generic = \"as.data.frame\"))(x, \n        row.names, optional, ...))\n<bytecode: 0x5c338f93ef60>\n<environment: 0x5c33901676b0>\nattr(,\"generic\")\n[1] \"as.data.frame\"\nattr(,\"generic\")attr(,\"package\")\n[1] \"BiocGenerics\"\nattr(,\"package\")\n[1] \"BiocGenerics\"\nattr(,\"group\")\nlist()\nattr(,\"valueClass\")\ncharacter(0)\nattr(,\"signature\")\n[1] \"x\"\nattr(,\"default\")\nMethod Definition (Class \"derivedDefaultMethod\"):\n\nfunction (x, row.names = NULL, optional = FALSE, ...) \n{\n    if (is.null(x)) \n        return(as.data.frame(list()))\n    UseMethod(\"as.data.frame\")\n}\n<bytecode: 0x5c339016a1d8>\n<environment: namespace:base>\n\nSignatures:\n        x    \ntarget  \"ANY\"\ndefined \"ANY\"\nattr(,\"skeleton\")\n(new(\"derivedDefaultMethod\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \n{\n    if (is.null(x)) \n        return(as.data.frame(list()))\n    UseMethod(\"as.data.frame\")\n}, target = new(\"signature\", .Data = \"ANY\", names = \"x\", package = \"methods\"), \n    defined = new(\"signature\", .Data = \"ANY\", names = \"x\", package = \"methods\"), \n    generic = \"as.data.frame\"))(x, row.names, optional, ...)\nattr(,\"class\")\n[1] \"standardGeneric\"\nattr(,\"class\")attr(,\"package\")\n[1] \"methods\""},"metadata":{}}],"execution_count":87},{"cell_type":"code","source":"showMethods(as.data.frame)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:01:30.007078Z","iopub.execute_input":"2025-04-13T13:01:30.008713Z","iopub.status.idle":"2025-04-13T13:01:30.027125Z","shell.execute_reply":"2025-04-13T13:01:30.025084Z"}},"outputs":[{"name":"stdout","text":"Function: as.data.frame (package BiocGenerics)\nx=\"ANY\"\nx=\"Array\"\nx=\"Bimap\"\nx=\"DataFrame\"\nx=\"GenomicRanges\"\nx=\"GPos\"\nx=\"GroupedIRanges\"\nx=\"Hits\"\nx=\"IPos\"\nx=\"IPosRanges\"\nx=\"List\"\nx=\"Pairs\"\nx=\"Rle\"\nx=\"Seqinfo\"\nx=\"Vector\"\nx=\"XStringSet\"\nx=\"XStringViews\"\n\n","output_type":"stream"}],"execution_count":88},{"cell_type":"code","source":"getMethod('as.data.frame',signature(x = 'GenomicRanges'))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:04:17.129544Z","iopub.execute_input":"2025-04-13T13:04:17.132078Z","iopub.status.idle":"2025-04-13T13:04:17.158676Z","shell.execute_reply":"2025-04-13T13:04:17.156993Z"}},"outputs":[{"name":"stderr","text":"An object of class “signature”\n\nAn object of class “signature”\n\n","output_type":"stream"},{"output_type":"display_data","data":{"text/html":"<pre class=language-r><code>new(\"MethodDefinition\", .Data = function (x, row.names = NULL, \n<span style=white-space:pre-wrap>    optional = FALSE, ...) </span>\n{\n<span style=white-space:pre-wrap>    if (missing(row.names)) </span>\n<span style=white-space:pre-wrap>        row.names &lt;- names(x)</span>\n<span style=white-space:pre-wrap>    if (!is.null(names(x))) </span>\n<span style=white-space:pre-wrap>        names(x) &lt;- NULL</span>\n<span style=white-space:pre-wrap>    ans &lt;- data.frame(seqnames = as.factor(seqnames(x)), start = start(x), </span>\n<span style=white-space:pre-wrap>        end = end(x), width = width(x), strand = as.factor(strand(x)), </span>\n<span style=white-space:pre-wrap>        row.names = row.names, stringsAsFactors = FALSE)</span>\n<span style=white-space:pre-wrap>    x_mcols &lt;- mcols(x, use.names = FALSE)</span>\n<span style=white-space:pre-wrap>    ans &lt;- cbind(ans, as.data.frame(x_mcols, optional = optional, </span>\n<span style=white-space:pre-wrap>        ...))</span>\n<span style=white-space:pre-wrap>    if (length(extraColumnSlotNames(x)) &gt; 0L) {</span>\n<span style=white-space:pre-wrap>        x_xcols &lt;- extraColumnSlotsAsDF(x)</span>\n<span style=white-space:pre-wrap>        ans &lt;- cbind(ans, as.data.frame(x_xcols, optional = optional, </span>\n<span style=white-space:pre-wrap>            ...))</span>\n<span style=white-space:pre-wrap>    }</span>\n<span style=white-space:pre-wrap>    ans</span>\n}, target = new(\"signature\", .Data = \"GenomicRanges\", names = \"x\", \n<span style=white-space:pre-wrap>    package = \"GenomicRanges\"), defined = new(\"signature\", .Data = \"GenomicRanges\", </span>\n<span style=white-space:pre-wrap>    names = \"x\", package = \"GenomicRanges\"), generic = structure(\"as.data.frame\", package = \"BiocGenerics\"))</span></code></pre>","text/markdown":"```r\nnew(\"MethodDefinition\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \n{\n    if (missing(row.names)) \n        row.names <- names(x)\n    if (!is.null(names(x))) \n        names(x) <- NULL\n    ans <- data.frame(seqnames = as.factor(seqnames(x)), start = start(x), \n        end = end(x), width = width(x), strand = as.factor(strand(x)), \n        row.names = row.names, stringsAsFactors = FALSE)\n    x_mcols <- mcols(x, use.names = FALSE)\n    ans <- cbind(ans, as.data.frame(x_mcols, optional = optional, \n        ...))\n    if (length(extraColumnSlotNames(x)) > 0L) {\n        x_xcols <- extraColumnSlotsAsDF(x)\n        ans <- cbind(ans, as.data.frame(x_xcols, optional = optional, \n            ...))\n    }\n    ans\n}, target = new(\"signature\", .Data = \"GenomicRanges\", names = \"x\", \n    package = \"GenomicRanges\"), defined = new(\"signature\", .Data = \"GenomicRanges\", \n    names = \"x\", package = \"GenomicRanges\"), generic = structure(\"as.data.frame\", package = \"BiocGenerics\"))\n```","text/latex":"\\begin{minted}{r}\nnew(\"MethodDefinition\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \n\\{\n    if (missing(row.names)) \n        row.names <- names(x)\n    if (!is.null(names(x))) \n        names(x) <- NULL\n    ans <- data.frame(seqnames = as.factor(seqnames(x)), start = start(x), \n        end = end(x), width = width(x), strand = as.factor(strand(x)), \n        row.names = row.names, stringsAsFactors = FALSE)\n    x\\_mcols <- mcols(x, use.names = FALSE)\n    ans <- cbind(ans, as.data.frame(x\\_mcols, optional = optional, \n        ...))\n    if (length(extraColumnSlotNames(x)) > 0L) \\{\n        x\\_xcols <- extraColumnSlotsAsDF(x)\n        ans <- cbind(ans, as.data.frame(x\\_xcols, optional = optional, \n            ...))\n    \\}\n    ans\n\\}, target = new(\"signature\", .Data = \"GenomicRanges\", names = \"x\", \n    package = \"GenomicRanges\"), defined = new(\"signature\", .Data = \"GenomicRanges\", \n    names = \"x\", package = \"GenomicRanges\"), generic = structure(\"as.data.frame\", package = \"BiocGenerics\"))\n\\end{minted}","text/plain":"new(\"MethodDefinition\", .Data = function (x, row.names = NULL, \n    optional = FALSE, ...) \n{\n    if (missing(row.names)) \n        row.names <- names(x)\n    if (!is.null(names(x))) \n        names(x) <- NULL\n    ans <- data.frame(seqnames = as.factor(seqnames(x)), start = start(x), \n        end = end(x), width = width(x), strand = as.factor(strand(x)), \n        row.names = row.names, stringsAsFactors = FALSE)\n    x_mcols <- mcols(x, use.names = FALSE)\n    ans <- cbind(ans, as.data.frame(x_mcols, optional = optional, \n        ...))\n    if (length(extraColumnSlotNames(x)) > 0L) {\n        x_xcols <- extraColumnSlotsAsDF(x)\n        ans <- cbind(ans, as.data.frame(x_xcols, optional = optional, \n            ...))\n    }\n    ans\n}, target = new(\"signature\", .Data = \"GenomicRanges\", names = \"x\", \n    package = \"GenomicRanges\"), defined = new(\"signature\", .Data = \"GenomicRanges\", \n    names = \"x\", package = \"GenomicRanges\"), generic = \"as.data.frame\")\n<bytecode: 0x5c339cd9ddc8>\n<environment: namespace:GenomicRanges>\nattr(,\"target\")\n              x \n\"GenomicRanges\" \nattr(,\"defined\")\n              x \n\"GenomicRanges\" \nattr(,\"generic\")\n[1] \"as.data.frame\"\nattr(,\"generic\")attr(,\"package\")\n[1] \"BiocGenerics\"\nattr(,\"class\")\n[1] \"MethodDefinition\"\nattr(,\"class\")attr(,\"package\")\n[1] \"methods\""},"metadata":{}}],"execution_count":90},{"cell_type":"code","source":"library(ShortRead)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-04-13T13:56:16.470882Z","iopub.execute_input":"2025-04-13T13:56:16.481045Z","iopub.status.idle":"2025-04-13T13:56:16.522786Z","shell.execute_reply":"2025-04-13T13:56:16.519833Z"}},"outputs":[{"ename":"ERROR","evalue":"Error in library(ShortRead): there is no package called ‘ShortRead’\n","traceback":["Error in library(ShortRead): there is no package called ‘ShortRead’\nTraceback:\n","1. library(ShortRead)"],"output_type":"error"}],"execution_count":102},{"cell_type":"code","source":"library (Rsamtools)","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"bampath=system.file(\"extdata\",\"ex1.bam\",package=\"Rsamtools\")\nbamfile=BamFile(bampath)\nseqinfo(bamfile)\naln=scanBam(bamfile)\nlength(aln)\nclass(aln)\naln=aln[[1]]\nnames(aln)\nlapply(aln,function(xx),xx[1])\nyieldSize(bamFile)=1\n       bamFile\nopen(bamFile)\nscanBam(bamFile0)[[1]]$seq\nclose(bamFile)\nyieldSize(bamFile)=NA\n\ngr=GRanges(seqnames=\"seq2\",ranges=IRanges(start=c(100,1000))end=c(1500,2000))))\ngr\nparams=scanBamparam(which=gr,what=scanBamWhat())\naln=scanBam(bamfile,param=params)\nnames(aln)\nhead(aln[[1]]$pos)\nquickBamFlafSummary(bamfile)\nbamview=BamViews(bamPath)\naln=scanBam(bamView)\nnames(aln)\n","metadata":{"trusted":true},"outputs":[],"execution_count":null}]}