{"metadata":{"kernelspec":{"name":"ir","display_name":"R","language":"R"},"language_info":{"name":"R","codemirror_mode":"r","pygments_lexer":"r","mimetype":"text/x-r-source","file_extension":".r","version":"4.4.0"},"kaggle":{"accelerator":"none","dataSources":[],"dockerImageVersionId":30749,"isInternetEnabled":true,"language":"r","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"# This R environment comes with many helpful analytics packages installed\n# It is defined by the kaggle/rstats Docker image: https://github.com/kaggle/docker-rstats\n# For example, here's a helpful package to load\n\nlibrary(tidyverse) # metapackage of all tidyverse packages\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nlist.files(path = \"../input\")\n\n# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","metadata":{"_uuid":"051d70d956493feee0c6d64651c6a088724dca2a","_execution_state":"idle","trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:34:56.389610Z","iopub.execute_input":"2025-06-08T09:34:56.391811Z","iopub.status.idle":"2025-06-08T09:34:57.826003Z","shell.execute_reply":"2025-06-08T09:34:57.824231Z"}},"outputs":[{"name":"stderr","text":"── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.5\n\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.1\n\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.1     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.3     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.1\n\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.2     \n── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n","output_type":"stream"},{"output_type":"display_data","data":{"text/html":"","text/markdown":"","text/latex":"","text/plain":"character(0)"},"metadata":{}}],"execution_count":1},{"cell_type":"code","source":"# Install BiocManager\ninstall.packages(\"BiocManager\")\n\n# Install IRanges from Bioconductor\nBiocManager::install(\"IRanges\", update = FALSE, ask = FALSE)\n\n# Load the package\nlibrary(IRanges)\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:35:16.891587Z","iopub.execute_input":"2025-06-08T09:35:16.924352Z","iopub.status.idle":"2025-06-08T09:37:40.198863Z","shell.execute_reply":"2025-06-08T09:37:40.197013Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"Installing package into ‘/usr/local/lib/R/site-library’\n(as ‘lib’ is unspecified)\n\n'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.26), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'IRanges'\n\nalso installing the dependency ‘S4Vectors’\n\n\nLoading required package: BiocGenerics\n\n\nAttaching package: ‘BiocGenerics’\n\n\nThe following objects are masked from ‘package:lubridate’:\n\n    intersect, setdiff, union\n\n\nThe following objects are masked from ‘package:dplyr’:\n\n    combine, intersect, setdiff, union\n\n\nThe following objects are masked from ‘package:stats’:\n\n    IQR, mad, sd, var, xtabs\n\n\nThe following objects are masked from ‘package:base’:\n\n    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n    Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,\n    tapply, union, unique, unsplit, which.max, which.min\n\n\nLoading required package: S4Vectors\n\nLoading required package: stats4\n\n\nAttaching package: ‘S4Vectors’\n\n\nThe following objects are masked from ‘package:lubridate’:\n\n    second, second<-\n\n\nThe following objects are masked from ‘package:dplyr’:\n\n    first, rename\n\n\nThe following object is masked from ‘package:tidyr’:\n\n    expand\n\n\nThe following object is masked from ‘package:utils’:\n\n    findMatches\n\n\nThe following objects are masked from ‘package:base’:\n\n    expand.grid, I, unname\n\n\n\nAttaching package: ‘IRanges’\n\n\nThe following object is masked from ‘package:lubridate’:\n\n    %within%\n\n\nThe following objects are masked from ‘package:dplyr’:\n\n    collapse, desc, slice\n\n\nThe following object is masked from ‘package:purrr’:\n\n    reduce\n\n\n","output_type":"stream"}],"execution_count":2},{"cell_type":"code","source":"first <- IRanges(start=c(1,2,3),width=4)\nsecond <- IRanges(start=c(2,3,4),width=4)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:39:36.743628Z","iopub.execute_input":"2025-06-08T09:39:36.745248Z","iopub.status.idle":"2025-06-08T09:39:36.763096Z","shell.execute_reply":"2025-06-08T09:39:36.761154Z"}},"outputs":[],"execution_count":4},{"cell_type":"code","source":"length(first)\nstart(second)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:40:18.670936Z","iopub.execute_input":"2025-06-08T09:40:18.672658Z","iopub.status.idle":"2025-06-08T09:40:18.694080Z","shell.execute_reply":"2025-06-08T09:40:18.691658Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"3","text/markdown":"3","text/latex":"3","text/plain":"[1] 3"},"metadata":{}},{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>2</li><li>3</li><li>4</li></ol>\n","text/markdown":"1. 2\n2. 3\n3. 4\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 2\n\\item 3\n\\item 4\n\\end{enumerate*}\n","text/plain":"[1] 2 3 4"},"metadata":{}}],"execution_count":6},{"cell_type":"code","source":"names(first) <- paste(\"A\",1:3,sep =\" \")\nfirst","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:43:20.232467Z","iopub.execute_input":"2025-06-08T09:43:20.234751Z","iopub.status.idle":"2025-06-08T09:43:20.255148Z","shell.execute_reply":"2025-06-08T09:43:20.253283Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"IRanges object with 3 ranges and 0 metadata columns:\n          start       end     width\n      <integer> <integer> <integer>\n  A 1         1         4         4\n  A 2         2         5         4\n  A 3         3         6         4"},"metadata":{}}],"execution_count":9},{"cell_type":"code","source":"c(first,second)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:44:58.970764Z","iopub.execute_input":"2025-06-08T09:44:58.972410Z","iopub.status.idle":"2025-06-08T09:44:59.000745Z","shell.execute_reply":"2025-06-08T09:44:58.998801Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"IRanges object with 6 ranges and 0 metadata columns:\n          start       end     width\n      <integer> <integer> <integer>\n  A 1         1         4         4\n  A 2         2         5         4\n  A 3         3         6         4\n              2         5         4\n              3         6         4\n              4         7         4"},"metadata":{}}],"execution_count":14},{"cell_type":"code","source":"resize(second,width=3,fix='start')\nsecond","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:46:26.813821Z","iopub.execute_input":"2025-06-08T09:46:26.815506Z","iopub.status.idle":"2025-06-08T09:46:26.880707Z","shell.execute_reply":"2025-06-08T09:46:26.878794Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"IRanges object with 3 ranges and 0 metadata columns:\n          start       end     width\n      <integer> <integer> <integer>\n  [1]         2         4         3\n  [2]         3         5         3\n  [3]         4         6         3"},"metadata":{}},{"output_type":"display_data","data":{"text/plain":"IRanges object with 3 ranges and 0 metadata columns:\n          start       end     width\n      <integer> <integer> <integer>\n  [1]         2         5         4\n  [2]         3         6         4\n  [3]         4         7         4"},"metadata":{}}],"execution_count":16},{"cell_type":"code","source":"ov=findOverlaps(first,second)\nov","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:57:21.365446Z","iopub.execute_input":"2025-06-08T09:57:21.367128Z","iopub.status.idle":"2025-06-08T09:57:21.390699Z","shell.execute_reply":"2025-06-08T09:57:21.388163Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"Hits object with 9 hits and 0 metadata columns:\n      queryHits subjectHits\n      <integer>   <integer>\n  [1]         1           1\n  [2]         1           2\n  [3]         1           3\n  [4]         2           1\n  [5]         2           2\n  [6]         2           3\n  [7]         3           1\n  [8]         3           2\n  [9]         3           3\n  -------\n  queryLength: 3 / subjectLength: 3"},"metadata":{}}],"execution_count":19},{"cell_type":"code","source":"countOverlaps(first,second)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:56:01.806131Z","iopub.execute_input":"2025-06-08T09:56:01.807709Z","iopub.status.idle":"2025-06-08T09:56:01.824851Z","shell.execute_reply":"2025-06-08T09:56:01.823009Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.dl-inline {width: auto; margin:0; padding: 0}\n.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n.dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n</style><dl class=dl-inline><dt>A 1</dt><dd>3</dd><dt>A 2</dt><dd>3</dd><dt>A 3</dt><dd>3</dd></dl>\n","text/markdown":"A 1\n:   3A 2\n:   3A 3\n:   3\n\n","text/latex":"\\begin{description*}\n\\item[A 1] 3\n\\item[A 2] 3\n\\item[A 3] 3\n\\end{description*}\n","text/plain":"A 1 A 2 A 3 \n  3   3   3 "},"metadata":{}}],"execution_count":18},{"cell_type":"code","source":"unique(queryHits(ov))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T09:58:34.428686Z","iopub.execute_input":"2025-06-08T09:58:34.430416Z","iopub.status.idle":"2025-06-08T09:58:34.447071Z","shell.execute_reply":"2025-06-08T09:58:34.445168Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<style>\n.list-inline {list-style: none; margin:0; padding: 0}\n.list-inline>li {display: inline-block}\n.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n</style>\n<ol class=list-inline><li>1</li><li>2</li><li>3</li></ol>\n","text/markdown":"1. 1\n2. 2\n3. 3\n\n\n","text/latex":"\\begin{enumerate*}\n\\item 1\n\\item 2\n\\item 3\n\\end{enumerate*}\n","text/plain":"[1] 1 2 3"},"metadata":{}}],"execution_count":22},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\nBiocManager::install(\"GenomicRanges\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T09:51:58.526521Z","iopub.execute_input":"2025-06-09T09:51:58.528874Z"}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'GenomicRanges'\n\nalso installing the dependencies ‘UCSC.utils’, ‘GenomeInfoDbData’, ‘zlibbioc’, ‘S4Vectors’, ‘IRanges’, ‘GenomeInfoDb’, ‘XVector’\n\n\n","output_type":"stream"}],"execution_count":null},{"cell_type":"code","source":"library(GenomicRanges)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T18:20:19.220014Z","iopub.execute_input":"2025-06-08T18:20:19.221825Z","iopub.status.idle":"2025-06-08T18:20:19.701802Z","shell.execute_reply":"2025-06-08T18:20:19.699971Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"Loading required package: stats4\n\nLoading required package: S4Vectors\n\n\nAttaching package: ‘S4Vectors’\n\n\nThe following object is masked from ‘package:utils’:\n\n    findMatches\n\n\nThe following objects are masked from ‘package:base’:\n\n    expand.grid, I, unname\n\n\nLoading required package: IRanges\n\nLoading required package: GenomeInfoDb\n\n","output_type":"stream"}],"execution_count":8},{"cell_type":"code","source":"gr=GRanges(seqnames=c('chr1'),strand=c('+',\"+\",\"-\",\"-\"),ranges=IRanges(start=c(2,4,6,7),width=3))\ngr","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T15:15:51.326032Z","iopub.execute_input":"2025-06-08T15:15:51.327779Z","iopub.status.idle":"2025-06-08T15:15:51.367861Z","shell.execute_reply":"2025-06-08T15:15:51.364547Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-4      +\n  [2]     chr1       4-6      +\n  [3]     chr1       6-8      -\n  [4]     chr1       7-9      -\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":10},{"cell_type":"code","source":"flanking_regions <- flank(gr, 5)\n\n# Print the flanking regions\nprint(flanking_regions)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:13:50.221686Z","iopub.execute_input":"2025-06-08T11:13:50.223720Z","iopub.status.idle":"2025-06-08T11:13:50.271550Z","shell.execute_reply":"2025-06-08T11:13:50.268876Z"}},"outputs":[{"name":"stdout","text":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1      -3-1      +\n  [2]     chr1      -1-3      +\n  [3]     chr1      9-13      -\n  [4]     chr1     10-14      -\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths\n","output_type":"stream"}],"execution_count":29},{"cell_type":"code","source":"promoters(gr)\nseqinfo(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:40:41.501411Z","iopub.execute_input":"2025-06-08T11:40:41.505918Z","iopub.status.idle":"2025-06-08T11:40:41.617367Z","shell.execute_reply":"2025-06-08T11:40:41.614005Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1 -1998-201      +\n  [2]     chr1 -1996-203      +\n  [3]     chr1 -191-2008      -\n  [4]     chr1 -190-2009      -\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}},{"output_type":"display_data","data":{"text/plain":"Seqinfo object with 1 sequence from an unspecified genome; no seqlengths:\n  seqnames seqlengths isCircular genome\n  chr1             NA         NA   <NA>"},"metadata":{}}],"execution_count":33},{"cell_type":"code","source":"seqlengths(gr)=c('chr1'=10)\nseqinfo(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:41:27.804445Z","iopub.execute_input":"2025-06-08T11:41:27.806658Z","iopub.status.idle":"2025-06-08T11:41:27.858917Z","shell.execute_reply":"2025-06-08T11:41:27.856075Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"Seqinfo object with 1 sequence from an unspecified genome:\n  seqnames seqlengths isCircular genome\n  chr1             10         NA   <NA>"},"metadata":{}}],"execution_count":34},{"cell_type":"code","source":"seqlevels(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:42:07.513359Z","iopub.execute_input":"2025-06-08T11:42:07.515465Z","iopub.status.idle":"2025-06-08T11:42:07.536815Z","shell.execute_reply":"2025-06-08T11:42:07.534603Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"'chr1'","text/markdown":"'chr1'","text/latex":"'chr1'","text/plain":"[1] \"chr1\""},"metadata":{}}],"execution_count":35},{"cell_type":"code","source":"gaps(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:45:06.068302Z","iopub.execute_input":"2025-06-08T11:45:06.070883Z","iopub.status.idle":"2025-06-08T11:45:06.117735Z","shell.execute_reply":"2025-06-08T11:45:06.115367Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 5 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1         1      +\n  [2]     chr1      7-10      +\n  [3]     chr1       1-5      -\n  [4]     chr1        10      -\n  [5]     chr1      1-10      *\n  -------\n  seqinfo: 1 sequence from an unspecified genome"},"metadata":{}}],"execution_count":36},{"cell_type":"code","source":"seqlevels(gr)=c('chr1','chr2')\nseqnames(gr)=c('chr1',\"chr2\",\"chr1\",\"chr2\")\ngr","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:47:26.846691Z","iopub.execute_input":"2025-06-08T11:47:26.848957Z","iopub.status.idle":"2025-06-08T11:47:26.886062Z","shell.execute_reply":"2025-06-08T11:47:26.884131Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-4      +\n  [2]     chr2       4-6      +\n  [3]     chr1       6-8      -\n  [4]     chr2       7-9      -\n  -------\n  seqinfo: 2 sequences from an unspecified genome"},"metadata":{}}],"execution_count":38},{"cell_type":"code","source":"sort(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T11:47:49.596228Z","iopub.execute_input":"2025-06-08T11:47:49.598076Z","iopub.status.idle":"2025-06-08T11:47:49.630421Z","shell.execute_reply":"2025-06-08T11:47:49.628511Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-4      +\n  [2]     chr1       6-8      -\n  [3]     chr2       4-6      +\n  [4]     chr2       7-9      -\n  -------\n  seqinfo: 2 sequences from an unspecified genome"},"metadata":{}}],"execution_count":39},{"cell_type":"code","source":"values(gr)=DataFrame(score=rnorm(3))\ngr","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T12:21:21.897830Z","iopub.execute_input":"2025-06-08T12:21:21.904396Z","iopub.status.idle":"2025-06-08T12:21:22.040894Z","shell.execute_reply":"2025-06-08T12:21:22.038853Z"}},"outputs":[{"name":"stderr","text":"Warning message in normarg_mcols(value, class(x), length(x)):\n“You supplied a metadata column of length 3 to set on an object of\n  length 4. However please note that the latter is not a multiple of the\n  former.”\n","output_type":"stream"},{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 1 metadata column:\n      seqnames    ranges strand |     score\n         <Rle> <IRanges>  <Rle> | <numeric>\n  [1]     chr1       2-4      + | -0.165715\n  [2]     chr2       4-6      + |  1.032601\n  [3]     chr1       6-8      - | -0.500092\n  [4]     chr2       7-9      - | -0.165715\n  -------\n  seqinfo: 2 sequences from an unspecified genome"},"metadata":{}}],"execution_count":40},{"cell_type":"code","source":"values(gr)\nmcols(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T12:22:13.374432Z","iopub.execute_input":"2025-06-08T12:22:13.376315Z","iopub.status.idle":"2025-06-08T12:22:13.419179Z","shell.execute_reply":"2025-06-08T12:22:13.415900Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"DataFrame with 4 rows and 1 column\n      score\n  <numeric>\n1 -0.165715\n2  1.032601\n3 -0.500092\n4 -0.165715"},"metadata":{}},{"output_type":"display_data","data":{"text/plain":"DataFrame with 4 rows and 1 column\n      score\n  <numeric>\n1 -0.165715\n2  1.032601\n3 -0.500092\n4 -0.165715"},"metadata":{}}],"execution_count":41},{"cell_type":"code","source":"gr$score2=gr$score/2\ngr","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T12:24:10.015042Z","iopub.execute_input":"2025-06-08T12:24:10.017060Z","iopub.status.idle":"2025-06-08T12:24:10.078736Z","shell.execute_reply":"2025-06-08T12:24:10.075984Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 2 metadata columns:\n      seqnames    ranges strand |     score     score2\n         <Rle> <IRanges>  <Rle> | <numeric>  <numeric>\n  [1]     chr1       2-4      + | -0.165715 -0.0828573\n  [2]     chr2       4-6      + |  1.032601  0.5163004\n  [3]     chr1       6-8      - | -0.500092 -0.2500462\n  [4]     chr2       7-9      - | -0.165715 -0.0828573\n  -------\n  seqinfo: 2 sequences from an unspecified genome"},"metadata":{}}],"execution_count":45},{"cell_type":"code","source":"gr2=GRanges(seqnames=c('chr1','chr1','chr2','chr2'),strand=c('+','-','+','+'),ranges=IRanges(start=c(2,3,5,7),width=4))\ngr2","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T15:25:04.870116Z","iopub.execute_input":"2025-06-08T15:25:04.875044Z","iopub.status.idle":"2025-06-08T15:25:04.935554Z","shell.execute_reply":"2025-06-08T15:25:04.932422Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-5      +\n  [2]     chr1       3-6      -\n  [3]     chr2       5-8      +\n  [4]     chr2      7-10      +\n  -------\n  seqinfo: 2 sequences from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":20},{"cell_type":"code","source":"findOverlaps(gr,gr2)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T12:46:53.166411Z","iopub.execute_input":"2025-06-08T12:46:53.168095Z","iopub.status.idle":"2025-06-08T12:46:53.227297Z","shell.execute_reply":"2025-06-08T12:46:53.224486Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"Hits object with 3 hits and 0 metadata columns:\n      queryHits subjectHits\n      <integer>   <integer>\n  [1]         1           1\n  [2]         2           3\n  [3]         3           2\n  -------\n  queryLength: 4 / subjectLength: 4"},"metadata":{}}],"execution_count":50},{"cell_type":"code","source":"subsetByOverlaps(gr,gr2)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T12:48:06.550184Z","iopub.execute_input":"2025-06-08T12:48:06.552072Z","iopub.status.idle":"2025-06-08T12:48:06.601745Z","shell.execute_reply":"2025-06-08T12:48:06.599733Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 3 ranges and 2 metadata columns:\n      seqnames    ranges strand |     score     score2\n         <Rle> <IRanges>  <Rle> | <numeric>  <numeric>\n  [1]     chr1       2-4      + | -0.165715 -0.0828573\n  [2]     chr2       4-6      + |  1.032601  0.5163004\n  [3]     chr1       6-8      - | -0.500092 -0.2500462\n  -------\n  seqinfo: 2 sequences from an unspecified genome"},"metadata":{}}],"execution_count":51},{"cell_type":"code","source":"df=data.frame(chr='chr1',start=1:3,end=4:6,score=rnorm(3))\ndf","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T14:43:25.351836Z","iopub.execute_input":"2025-06-08T14:43:25.354045Z","iopub.status.idle":"2025-06-08T14:43:25.389766Z","shell.execute_reply":"2025-06-08T14:43:25.387739Z"}},"outputs":[{"output_type":"display_data","data":{"text/html":"<table class=\"dataframe\">\n<caption>A data.frame: 3 × 4</caption>\n<thead>\n\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>score</th></tr>\n\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n</thead>\n<tbody>\n\t<tr><td>chr1</td><td>1</td><td>4</td><td>-1.191115</td></tr>\n\t<tr><td>chr1</td><td>2</td><td>5</td><td>-1.064068</td></tr>\n\t<tr><td>chr1</td><td>3</td><td>6</td><td> 1.284383</td></tr>\n</tbody>\n</table>\n","text/markdown":"\nA data.frame: 3 × 4\n\n| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | score &lt;dbl&gt; |\n|---|---|---|---|\n| chr1 | 1 | 4 | -1.191115 |\n| chr1 | 2 | 5 | -1.064068 |\n| chr1 | 3 | 6 |  1.284383 |\n\n","text/latex":"A data.frame: 3 × 4\n\\begin{tabular}{llll}\n chr & start & end & score\\\\\n <chr> & <int> & <int> & <dbl>\\\\\n\\hline\n\t chr1 & 1 & 4 & -1.191115\\\\\n\t chr1 & 2 & 5 & -1.064068\\\\\n\t chr1 & 3 & 6 &  1.284383\\\\\n\\end{tabular}\n","text/plain":"  chr  start end score    \n1 chr1 1     4   -1.191115\n2 chr1 2     5   -1.064068\n3 chr1 3     6    1.284383"},"metadata":{}}],"execution_count":5},{"cell_type":"code","source":"makeGRangesFromDataFrame(df)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T14:43:30.665264Z","iopub.execute_input":"2025-06-08T14:43:30.666986Z","iopub.status.idle":"2025-06-08T14:43:30.772035Z","shell.execute_reply":"2025-06-08T14:43:30.768750Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 3 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       1-4      *\n  [2]     chr1       2-5      *\n  [3]     chr1       3-6      *\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":6},{"cell_type":"code","source":"dropSeqlevels(gr,'chr2')\ngr","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T15:16:56.291872Z","iopub.execute_input":"2025-06-08T15:16:56.293503Z","iopub.status.idle":"2025-06-08T15:16:56.339961Z","shell.execute_reply":"2025-06-08T15:16:56.336811Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-4      +\n  [2]     chr1       4-6      +\n  [3]     chr1       6-8      -\n  [4]     chr1       7-9      -\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}},{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-4      +\n  [2]     chr1       4-6      +\n  [3]     chr1       6-8      -\n  [4]     chr1       7-9      -\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":12},{"cell_type":"code","source":"keepStandardChromosomes(gr)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T15:19:04.573116Z","iopub.execute_input":"2025-06-08T15:19:04.575261Z","iopub.status.idle":"2025-06-08T15:19:04.763915Z","shell.execute_reply":"2025-06-08T15:19:04.761933Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]     chr1       2-4      +\n  [2]     chr1       4-6      +\n  [3]     chr1       6-8      -\n  [4]     chr1       7-9      -\n  -------\n  seqinfo: 1 sequence from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":15},{"cell_type":"code","source":"newstyle= mapSeqlevels(seqlevels(gr2),'NCBI')\ngr2= renameSeqlevels(gr2,newstyle)\ngr2","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-08T15:26:12.383913Z","iopub.execute_input":"2025-06-08T15:26:12.385468Z","iopub.status.idle":"2025-06-08T15:26:12.426332Z","shell.execute_reply":"2025-06-08T15:26:12.424648Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 4 ranges and 0 metadata columns:\n      seqnames    ranges strand\n         <Rle> <IRanges>  <Rle>\n  [1]        1       2-5      +\n  [2]        1       3-6      -\n  [3]        2       5-8      +\n  [4]        2      7-10      +\n  -------\n  seqinfo: 2 sequences from an unspecified genome; no seqlengths"},"metadata":{}}],"execution_count":23},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\nBiocManager::install(\"AnnotationHub\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:42:30.654987Z","iopub.execute_input":"2025-06-09T11:42:30.686896Z","iopub.status.idle":"2025-06-09T11:42:39.208190Z","shell.execute_reply":"2025-06-09T11:42:39.205808Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nWarning message:\n“package(s) not installed when version(s) same as or greater than current; use\n  `force = TRUE` to re-install: 'AnnotationHub'”\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'aglm', 'AID',\n  'akc', 'ale', 'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'anacor',\n  'antiword', 'anytime', 'ape', 'aplot', 'apollo', 'aqp', 'arrow', 'ars',\n  'ARTool', 'AsioHeaders', 'askpass', 'ASMap', 'astsa', 'automap', 'baguette',\n  'BAS', 'base64', 'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr',\n  'beepr', 'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD',\n  'bigstatsr', 'bigstep', 'billboarder', 'bindr', 'BiocManager',\n  'BiodiversityR', 'biomod2', 'biotools', 'bipartite', 'bit', 'bit64',\n  'bitops', 'blme', 'blorr', 'BMA', 'BMisc', 'bnlearn', 'bonsai', 'bookdown',\n  'boot', 'brms', 'brnn', 'broom', 'broom.helpers', 'broom.mixed', 'brulee',\n  'bslib', 'bundle', 'butcher', 'C50', 'car', 'caret', 'caretEnsemble',\n  'caTools', 'cba', 'censored', 'changepoint', 'cheapr', 'checkmate', 'chk',\n  'CholWishart', 'choroplethr', 'chromote', 'chron', 'circular', 'class',\n  'classInt', 'cld2', 'cleaner', 'cli', 'climatol', 'clock', 'clue', 'cluster',\n  'clusterSim', 'ClustOfVar', 'cmsafops', 'coarseDataTools', 'cobalt', 'cobs',\n  'collapse', 'collections', 'colorspace', 'commonmark', 'compareGroups',\n  'Compind', 'condMVNorm', 'ConsRank', 'contentid', 'copula', 'CORElearn',\n  'coro', 'correlation', 'corrplot', 'COUNT', 'countrycode', 'cpp11',\n  'cppdoubles', 'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature',\n  'Cubist', 'curl', 'cutpointr', 'cvms', 'data.table', 'dataMaid', 'datamods',\n  'datasauRus', 'datawizard', 'datetimeutils', 'dbarts', 'dbscan',\n  'dendextend', 'dendroTools', 'DEoptimR', 'Deriv', 'descriptio', 'DescTools',\n  'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'diffobj', 'digest', 'dimRed',\n  'directlabels', 'dismo', 'distances', 'distillery', 'distr', 'distrEx',\n  'distributional', 'DistributionUtils', 'dlm', 'dlnm', 'dm', 'doBy', 'docopt',\n  'DoE.base', 'doFuture', 'doRNG', 'DoseFinding', 'dotCall64', 'dotwhisker',\n  'DoubleML', 'downloader', 'dplR', 'drat', 'DRDID', 'dreamerr', 'dtwclust',\n  'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat', 'ecp', 'effectsize', 'eha',\n  'embed', 'emmeans', 'energy', 'entropy', 'EnvStats', 'Epi', 'epiR', 'erer',\n  'ergm', 'ergm.multi', 'esquisse', 'estimatr', 'etm', 'evaluate', 'evd',\n  'Exact', 'explore', 'expm', 'ExPosition', 'extRemes', 'fable', 'fabletools',\n  'faraway', 'fastcluster', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastrmodels', 'fastverse', 'fBasics', 'fda', 'fda.usc',\n  'fdapace', 'fdrtool', 'feasts', 'ff', 'fftw', 'FielDHub', 'fields',\n  'filehashSQLite', 'finalfit', 'finetune', 'fitdistrplus', 'FKF', 'flare',\n  'flexclust', 'flexmix', 'flexsurv', 'flextable', 'float', 'FNN',\n  'fontawesome', 'forecast', 'foreign', 'forestplot', 'forestploter', 'fpc',\n  'fpp3', 'FrF2', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa', 'future',\n  'future.apply', 'gam', 'gamboostLSS', 'gamm4', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'GDAtools', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'generics', 'genridge', 'GenSA', 'geobr', 'geodist',\n  'geojsonR', 'geometry', 'geomorph', 'geomtextpath', 'geoR', 'geosphere',\n  'gert', 'GetBCBData', 'ggdag', 'ggdist', 'ggeasy', 'ggeffects', 'ggfun',\n  'ggh4x', 'ggiraph', 'gglasso', 'ggmap', 'GGMridge', 'ggnewscale', 'ggpath',\n  'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc', 'ggprism', 'ggrepel',\n  'ggstats', 'gh', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmnet', 'glmtoolbox', 'globals', 'glue',\n  'gmapsdistance', 'gmp', 'GoFKernel', 'golem', 'googlePolylines', 'gower',\n  'GPArotation', 'GPfit', 'gplots', 'grafify', 'gRain', 'graphlayouts',\n  'gRbase', 'greybox', 'grf', 'gridpattern', 'grImport2', 'groupdata2',\n  'grpreg', 'gsheet', 'gss', 'gstat', 'gsw', 'gt', 'gtable', 'gtExtras',\n  'gtsummary', 'gutenbergr', 'GWalkR', 'GWmodel', 'h3lib', 'h3r', 'hardhat',\n  'harmony', 'haven', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'HSAUR', 'htmlTable', 'hts', 'httpuv', 'httr2',\n  'hunspell', 'huxtable', 'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph',\n  'ijtiff', 'imager', 'iml', 'infer', 'inline', 'insight', 'inspectdf',\n  'interactions', 'intervals', 'ipred', 'ISOcodes', 'isotree', 'ISwR', 'itsmr',\n  'ivreg', 'janitor', 'JMbayes2', 'jmvconnect', 'jmvcore', 'jmvReadWrite',\n  'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity', 'keras3', 'kernelshap',\n  'kernlab', 'KernSmooth', 'keyring', 'KFAS', 'khroma', 'kit', 'kknn',\n  'KMsurv', 'knitr', 'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman',\n  'lares', 'later', 'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave',\n  'lcmm', 'leafem', 'leafgl', 'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR',\n  'lightgbm', 'limma', 'linelist', 'lintr', 'littler', 'lme4', 'lmodel2',\n  'lmom', 'locfit', 'logger', 'logistf', 'longitudinalData', 'loo', 'lordif',\n  'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa', 'lubridate',\n  'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend', 'mapproj',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'mathjaxr', 'matlib', 'Matrix',\n  'MatrixModels', 'matrixStats', 'maxstat', 'MazamaCoreUtils',\n  'MazamaSpatialUtils', 'MBA', 'mboost', 'MCMCpack', 'mda', 'mdsr',\n  'mediation', 'memisc', 'meta', 'metacoder', 'metadat', 'metafor',\n  'meteospain', 'MetricsWeighted', 'mets', 'mfx', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'miniUI', 'minqa', 'minty',\n  'mirt', 'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench',\n  'mlflow', 'mlogit', 'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data',\n  'mlr3filters', 'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures',\n  'mlr3misc', 'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse',\n  'mlr3viz', 'modelbased', 'modelenv', 'modelsummary', 'modeltime',\n  'modeltools', 'moderndive', 'mongolite', 'monitoR', 'msigdbr', 'msm', 'msme',\n  'mstate', 'mtsdi', 'multcomp', 'MuMIn', 'MVN', 'mvnormalTest', 'mvtnorm',\n  'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson', 'ndtv', 'network',\n  'networkD3', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'openeo', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ordinalNet', 'ore',\n  'osmextract', 'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown',\n  'pammtools', 'pander', 'parabar', 'paradox', 'parallelly', 'parameters',\n  'ParamHelpers', 'paran', 'parsnip', 'partitions', 'party', 'partykit',\n  'patchwork', 'pbdZMQ', 'pbkrtest', 'pcaPP', 'pdftools', 'pdp', 'pdR',\n  'PearsonDS', 'performance', 'PerformanceAnalytics', 'pgmm', 'phangorn',\n  'pheatmap', 'phia', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pivottabler', 'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools',\n  'pls', 'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip',\n  'popdemo', 'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'precrec', 'PreProcess', 'prettyGraphs', 'priceR', 'probably',\n  'processx', 'prodlim', 'profvis', 'progressr', 'PROJ', 'proj4', 'promises',\n  'prospectr', 'protolite', 'proxyC', 'PRROC', 'ps', 'pspline', 'PSweight',\n  'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf', 'qs', 'qtl', 'qtlcharts',\n  'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr', 'QuickJSR',\n  'qvcalc', 'R.cache', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'Rdsdp',\n  'reactR', 'readODS', 'readstata13', 'readxl', 'rearrr', 'rebmix', 'recipes',\n  'refund', 'regclass', 'renv', 'report', 'reprex', 'reticulate', 'Rfast',\n  'rfishbase', 'rgbif', 'rgenoud', 'rgl', 'rio', 'riskRegression', 'rjson',\n  'RJSONIO', 'rlang', 'rle', 'RMariaDB', 'rmarkdown', 'rminer', 'rms',\n  'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpf', 'rpostgis', 'RPostgres',\n  'RPostgreSQL', 'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample',\n  'rsconnect', 'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite',\n  'Rssa', 'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rtkore',\n  'rugarch', 'Rvcg', 'rvg', 's2', 'sandwich', 'sass', 'scales', 'SciViews',\n  'SCtools', 'sctransform', 'sda', 'sdm', 'seas', 'seasonal', 'seasonalview',\n  'see', 'seededlda', 'segmented', 'sem', 'semTools', 'sensemakr',\n  'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat', 'SeuratObject',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinydashboard', 'shinyWidgets',\n  'SimDesign', 'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot',\n  'skmeans', 'slam', 'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2',\n  'sna', 'sodium', 'softImpute', 'soilDB', 'sommer', 'sp', 'spacefillr',\n  'spacetime', 'spam', 'sparklyr', 'sparseFLMM', 'SparseM', 'spatialreg',\n  'spatialrisk', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'stringmagic', 'strucchange',\n  'summarytools', 'SuppDists', 'survival', 'survminer', 'svd', 'svglite',\n  'svgtools', 'sys', 'systemfonts', 'tables', 'tabnet', 'tau', 'taxize',\n  'tcltk2', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract', 'testthat',\n  'texreg', 'text', 'textcat', 'textrecipes', 'textshaping', 'textutils',\n  'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis', 'threejs', 'tibble',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'tigris', 'timeDate', 'timeplyr', 'timereg', 'timeSeries',\n  'tinytable', 'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui',\n  'topicmodels', 'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal',\n  'tsbox', 'TSclust', 'tsDyn', 'tseries', 'tsibble', 'tsna', 'TSP', 'tswge',\n  'tuber', 'tune', 'tzdb', 'Unicode', 'units', 'univariateML', 'unpivotr',\n  'usethis', 'usmap', 'usmapdata', 'utf8', 'uuid', 'uwot', 'V8', 'vcd',\n  'vegan', 'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR',\n  'WaveletComp', 'wbacon', 'WDI', 'webshot2', 'websocket', 'webutils',\n  'WeightSVM', 'winch', 'withr', 'wk', 'wooldridge', 'workflows',\n  'workflowsets', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick', 'yfR',\n  'yuima', 'yulab.utils', 'zeallot', 'zip', 'zoo', 'spatial'\n\n","output_type":"stream"}],"execution_count":2},{"cell_type":"code","source":"library(AnnotationHub)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:42:41.096237Z","iopub.execute_input":"2025-06-09T11:42:41.097781Z","iopub.status.idle":"2025-06-09T11:42:44.965228Z","shell.execute_reply":"2025-06-09T11:42:44.963357Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"Loading required package: BiocGenerics\n\n\nAttaching package: ‘BiocGenerics’\n\n\nThe following objects are masked from ‘package:stats’:\n\n    IQR, mad, sd, var, xtabs\n\n\nThe following objects are masked from ‘package:base’:\n\n    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n    Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,\n    tapply, union, unique, unsplit, which.max, which.min\n\n\nLoading required package: BiocFileCache\n\nLoading required package: dbplyr\n\n","output_type":"stream"}],"execution_count":3},{"cell_type":"code","source":"ah=AnnotationHub()\nah","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:42:44.968845Z","iopub.execute_input":"2025-06-09T11:42:44.971709Z","iopub.status.idle":"2025-06-09T11:42:57.352498Z","shell.execute_reply":"2025-06-09T11:42:57.350725Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"AnnotationHub with 71634 records\n# snapshotDate(): 2024-04-30\n# $dataprovider: Ensembl, BroadInstitute, UCSC, ftp://ftp.ncbi.nlm.nih.gov/g...\n# $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Rattus norv...\n# $rdataclass: GRanges, TwoBitFile, BigWigFile, EnsDb, Rle, OrgDb, SQLiteFil...\n# additional mcols(): taxonomyid, genome, description,\n#   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\n#   rdatapath, sourceurl, sourcetype \n# retrieve records with, e.g., 'object[[\"AH5012\"]]' \n\n             title                                             \n  AH5012   | Chromosome Band                                   \n  AH5013   | STS Markers                                       \n  AH5014   | FISH Clones                                       \n  AH5015   | Recomb Rate                                       \n  AH5016   | ENCODE Pilot                                      \n  ...        ...                                               \n  AH117051 | Ensembl 112 EnsDb for Xiphophorus maculatus       \n  AH117052 | Ensembl 112 EnsDb for Xenopus tropicalis          \n  AH117053 | Ensembl 112 EnsDb for Zonotrichia albicollis      \n  AH117054 | Ensembl 112 EnsDb for Zalophus californianus      \n  AH117055 | Ensembl 112 EnsDb for Zosterops lateralis melanops"},"metadata":{}}],"execution_count":4},{"cell_type":"code","source":"ah= subset(ah,species=='Homo sapiens')\nah","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:42:57.355308Z","iopub.execute_input":"2025-06-09T11:42:57.356733Z","iopub.status.idle":"2025-06-09T11:43:02.658882Z","shell.execute_reply":"2025-06-09T11:43:02.657146Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"AnnotationHub with 26651 records\n# snapshotDate(): 2024-04-30\n# $dataprovider: BroadInstitute, UCSC, Ensembl, GENCODE, Google DeepMind, UW...\n# $species: Homo sapiens\n# $rdataclass: GRanges, BigWigFile, Rle, ChainFile, TwoBitFile, TxDb, list, ...\n# additional mcols(): taxonomyid, genome, description,\n#   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\n#   rdatapath, sourceurl, sourcetype \n# retrieve records with, e.g., 'object[[\"AH5012\"]]' \n\n             title                             \n  AH5012   | Chromosome Band                   \n  AH5013   | STS Markers                       \n  AH5014   | FISH Clones                       \n  AH5015   | Recomb Rate                       \n  AH5016   | ENCODE Pilot                      \n  ...        ...                               \n  AH116728 | ENCODE_pELS_regions               \n  AH116729 | ENCODE_PLS_regions                \n  AH116730 | GENCODE basic gene annotation v40 \n  AH116731 | ENCODE Registry of cCREs V3       \n  AH116860 | Ensembl 112 EnsDb for Homo sapiens"},"metadata":{}}],"execution_count":5},{"cell_type":"code","source":"qhs=query(ah,c('H3K4me3','Gm12878'))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:43:02.661497Z","iopub.execute_input":"2025-06-09T11:43:02.662883Z","iopub.status.idle":"2025-06-09T11:43:03.583216Z","shell.execute_reply":"2025-06-09T11:43:03.581419Z"}},"outputs":[],"execution_count":6},{"cell_type":"code","source":"if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n    install.packages(\"BiocManager\")\nBiocManager::install(\"rtracklayer\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:43:03.585969Z","iopub.execute_input":"2025-06-09T11:43:03.587359Z","iopub.status.idle":"2025-06-09T11:51:18.973636Z","shell.execute_reply":"2025-06-09T11:51:18.971366Z"},"collapsed":true,"jupyter":{"outputs_hidden":true}},"outputs":[{"name":"stderr","text":"'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n'help(\"repositories\", package = \"BiocManager\")' for details.\nReplacement repositories:\n    CRAN: http://cran.rstudio.com/\n\nBioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)\n\nInstalling package(s) 'rtracklayer'\n\nalso installing the dependencies ‘SparseArray’, ‘MatrixGenerics’, ‘S4Arrays’, ‘DelayedArray’, ‘BiocParallel’, ‘Rhtslib’, ‘SummarizedExperiment’, ‘GenomicRanges’, ‘Rsamtools’, ‘GenomicAlignments’, ‘BiocIO’, ‘restfulr’\n\n\nOld packages: 'abind', 'actuar', 'ade4', 'adegraphics', 'adehabitatHR',\n  'adehabitatLT', 'adehabitatMA', 'admisc', 'AER', 'afex', 'aglm', 'AID',\n  'akc', 'ale', 'AlgDesign', 'AlphaSimR', 'amap', 'Amelia', 'anacor',\n  'antiword', 'anytime', 'ape', 'aplot', 'apollo', 'aqp', 'arrow', 'ars',\n  'ARTool', 'AsioHeaders', 'askpass', 'ASMap', 'astsa', 'automap', 'baguette',\n  'BAS', 'base64', 'bayesplot', 'bayestestR', 'bbotk', 'BDgraph', 'bedr',\n  'beepr', 'Benchmarking', 'betareg', 'BH', 'bibliometrix', 'bigD',\n  'bigstatsr', 'bigstep', 'billboarder', 'bindr', 'BiocManager',\n  'BiodiversityR', 'biomod2', 'biotools', 'bipartite', 'bit', 'bit64',\n  'bitops', 'blme', 'blorr', 'BMA', 'BMisc', 'bnlearn', 'bonsai', 'bookdown',\n  'boot', 'brms', 'brnn', 'broom', 'broom.helpers', 'broom.mixed', 'brulee',\n  'bslib', 'bundle', 'butcher', 'C50', 'car', 'caret', 'caretEnsemble',\n  'caTools', 'cba', 'censored', 'changepoint', 'cheapr', 'checkmate', 'chk',\n  'CholWishart', 'choroplethr', 'chromote', 'chron', 'circular', 'class',\n  'classInt', 'cld2', 'cleaner', 'cli', 'climatol', 'clock', 'clue', 'cluster',\n  'clusterSim', 'ClustOfVar', 'cmsafops', 'coarseDataTools', 'cobalt', 'cobs',\n  'collapse', 'collections', 'colorspace', 'commonmark', 'compareGroups',\n  'Compind', 'condMVNorm', 'ConsRank', 'contentid', 'copula', 'CORElearn',\n  'coro', 'correlation', 'corrplot', 'COUNT', 'countrycode', 'cpp11',\n  'cppdoubles', 'credentials', 'cricketdata', 'crul', 'crypto2', 'cubature',\n  'Cubist', 'curl', 'cutpointr', 'cvms', 'data.table', 'dataMaid', 'datamods',\n  'datasauRus', 'datawizard', 'datetimeutils', 'dbarts', 'dbscan',\n  'dendextend', 'dendroTools', 'DEoptimR', 'Deriv', 'descriptio', 'DescTools',\n  'devEMF', 'dfidx', 'dgof', 'DHARMa', 'dials', 'diffobj', 'digest', 'dimRed',\n  'directlabels', 'dismo', 'distances', 'distillery', 'distr', 'distrEx',\n  'distributional', 'DistributionUtils', 'dlm', 'dlnm', 'dm', 'doBy', 'docopt',\n  'DoE.base', 'doFuture', 'doRNG', 'DoseFinding', 'dotCall64', 'dotwhisker',\n  'DoubleML', 'downloader', 'dplR', 'drat', 'DRDID', 'dreamerr', 'dtwclust',\n  'duckdb', 'e1071', 'earth', 'Ecfun', 'ecospat', 'ecp', 'effectsize', 'eha',\n  'embed', 'emmeans', 'energy', 'entropy', 'EnvStats', 'Epi', 'epiR', 'erer',\n  'ergm', 'ergm.multi', 'esquisse', 'estimatr', 'etm', 'evaluate', 'evd',\n  'Exact', 'explore', 'expm', 'ExPosition', 'extRemes', 'fable', 'fabletools',\n  'faraway', 'fastcluster', 'fastDummies', 'fastICA', 'fastmatch',\n  'fastmatrix', 'fastrmodels', 'fastverse', 'fBasics', 'fda', 'fda.usc',\n  'fdapace', 'fdrtool', 'feasts', 'ff', 'fftw', 'FielDHub', 'fields',\n  'filehashSQLite', 'finalfit', 'finetune', 'fitdistrplus', 'FKF', 'flare',\n  'flexclust', 'flexmix', 'flexsurv', 'flextable', 'float', 'FNN',\n  'fontawesome', 'forecast', 'foreign', 'forestplot', 'forestploter', 'fpc',\n  'fpp3', 'FrF2', 'fs', 'FSA', 'FSelectorRcpp', 'fstcore', 'ftsa', 'future',\n  'future.apply', 'gam', 'gamboostLSS', 'gamm4', 'gap', 'garma', 'GAS',\n  'gclus', 'gcookbook', 'gdata', 'GDAtools', 'gdtools', 'geepack', 'GeneCycle',\n  'GeneralizedHyperbolic', 'generics', 'genridge', 'GenSA', 'geobr', 'geodist',\n  'geojsonR', 'geometry', 'geomorph', 'geomtextpath', 'geoR', 'geosphere',\n  'gert', 'GetBCBData', 'ggdag', 'ggdist', 'ggeasy', 'ggeffects', 'ggfun',\n  'ggh4x', 'ggiraph', 'gglasso', 'ggmap', 'GGMridge', 'ggnewscale', 'ggpath',\n  'ggpattern', 'ggplate', 'ggplot2', 'ggpmisc', 'ggprism', 'ggrepel',\n  'ggstats', 'gh', 'ghyp', 'gifski', 'giscoR', 'git2r', 'glarma', 'gld',\n  'GLMMadaptive', 'glmmTMB', 'glmnet', 'glmtoolbox', 'globals', 'glue',\n  'gmapsdistance', 'gmp', 'GoFKernel', 'golem', 'googlePolylines', 'gower',\n  'GPArotation', 'GPfit', 'gplots', 'grafify', 'gRain', 'graphlayouts',\n  'gRbase', 'greybox', 'grf', 'gridpattern', 'grImport2', 'groupdata2',\n  'grpreg', 'gsheet', 'gss', 'gstat', 'gsw', 'gt', 'gtable', 'gtExtras',\n  'gtsummary', 'gutenbergr', 'GWalkR', 'GWmodel', 'h3lib', 'h3r', 'hardhat',\n  'harmony', 'haven', 'hdf5r', 'HelpersMG', 'heplots', 'hexbin', 'HistData',\n  'HMDHFDplus', 'Hmisc', 'HSAUR', 'htmlTable', 'hts', 'httpuv', 'httr2',\n  'hunspell', 'huxtable', 'hydroGOF', 'hydroTSM', 'hypergeo', 'igraph',\n  'ijtiff', 'imager', 'iml', 'infer', 'inline', 'insight', 'inspectdf',\n  'interactions', 'intervals', 'ipred', 'ISOcodes', 'isotree', 'ISwR', 'itsmr',\n  'ivreg', 'janitor', 'JMbayes2', 'jmvconnect', 'jmvcore', 'jmvReadWrite',\n  'jpeg', 'jqr', 'jsonlite', 'jtools', 'kdensity', 'keras3', 'kernelshap',\n  'kernlab', 'KernSmooth', 'keyring', 'KFAS', 'khroma', 'kit', 'kknn',\n  'KMsurv', 'knitr', 'ks', 'kyotil', 'L1pack', 'labelled', 'LaF', 'Lahman',\n  'lares', 'later', 'lattice', 'LatticeKrig', 'lava', 'lavaan', 'lazyWeave',\n  'lcmm', 'leafem', 'leafgl', 'lehdr', 'lemon', 'lessR', 'lfe', 'LiblineaR',\n  'lightgbm', 'limma', 'linelist', 'lintr', 'littler', 'lme4', 'lmodel2',\n  'lmom', 'locfit', 'logger', 'logistf', 'longitudinalData', 'loo', 'lordif',\n  'LPCM', 'lpSolve', 'lpSolveAPI', 'lsmeans', 'ltsa', 'lubridate',\n  'MachineShop', 'maditr', 'magick', 'manymome', 'maplegend', 'mapproj',\n  'maps', 'mapsf', 'marginaleffects', 'margins', 'markdown', 'markovchain',\n  'MASS', 'Matching', 'MatchIt', 'mathjaxr', 'matlib', 'Matrix',\n  'MatrixModels', 'matrixStats', 'maxstat', 'MazamaCoreUtils',\n  'MazamaSpatialUtils', 'MBA', 'mboost', 'MCMCpack', 'mda', 'mdsr',\n  'mediation', 'memisc', 'meta', 'metacoder', 'metadat', 'metafor',\n  'meteospain', 'MetricsWeighted', 'mets', 'mfx', 'mgcv', 'mgm', 'mgwrsar',\n  'mice', 'microbenchmark', 'midasr', 'mime', 'miniUI', 'minqa', 'minty',\n  'mirt', 'mirtCAT', 'missRanger', 'misty', 'mixlm', 'mixtools', 'mlbench',\n  'mlflow', 'mlogit', 'mlr3', 'mlr3benchmark', 'mlr3cluster', 'mlr3data',\n  'mlr3filters', 'mlr3fselect', 'mlr3learners', 'mlr3mbo', 'mlr3measures',\n  'mlr3misc', 'mlr3pipelines', 'mlr3tuning', 'mlr3tuningspaces', 'mlr3verse',\n  'mlr3viz', 'modelbased', 'modelenv', 'modelsummary', 'modeltime',\n  'modeltools', 'moderndive', 'mongolite', 'monitoR', 'msigdbr', 'msm', 'msme',\n  'mstate', 'mtsdi', 'multcomp', 'MuMIn', 'MVN', 'mvnormalTest', 'mvtnorm',\n  'NatureSounds', 'ncdf4', 'ncdump', 'ncvreg', 'ndjson', 'ndtv', 'network',\n  'networkD3', 'networkDynamic', 'networkLite', 'nflfastR', 'nflplotR',\n  'nflreadr', 'nflseedR', 'nhanesA', 'nimble', 'nlme', 'nloptr', 'NLP', 'NMOF',\n  'nnet', 'nnls', 'NNS', 'np', 'nullabor', 'oce', 'odbc', 'officer', 'olsrr',\n  'oompaBase', 'openalexR', 'opencv', 'opendatatoronto', 'openeo', 'openssl',\n  'openxlsx', 'optimx', 'ordinal', 'ordinalForest', 'ordinalNet', 'ore',\n  'osmextract', 'overlapping', 'packcircles', 'packHV', 'padr', 'pagedown',\n  'pammtools', 'pander', 'parabar', 'paradox', 'parallelly', 'parameters',\n  'ParamHelpers', 'paran', 'parsnip', 'partitions', 'party', 'partykit',\n  'patchwork', 'pbdZMQ', 'pbkrtest', 'pcaPP', 'pdftools', 'pdp', 'pdR',\n  'PearsonDS', 'performance', 'PerformanceAnalytics', 'pgmm', 'phangorn',\n  'pheatmap', 'phia', 'philentropy', 'phytools', 'pillar', 'pingr', 'pins',\n  'pivottabler', 'pkgbuild', 'pkgdown', 'pliman', 'plm', 'plotmo', 'PlotTools',\n  'pls', 'plsVarSel', 'plumber', 'PMA', 'PMwR', 'poibin', 'polyclip',\n  'popdemo', 'portfolio', 'PortfolioAnalytics', 'posterior', 'POT', 'poweRlaw',\n  'prabclus', 'precrec', 'PreProcess', 'prettyGraphs', 'priceR', 'probably',\n  'processx', 'prodlim', 'profvis', 'progressr', 'PROJ', 'proj4', 'promises',\n  'prospectr', 'protolite', 'proxyC', 'PRROC', 'ps', 'pspline', 'PSweight',\n  'psych', 'purrr', 'qdapRegex', 'qgam', 'qpdf', 'qs', 'qtl', 'qtlcharts',\n  'quanteda', 'quantmod', 'quantreg', 'quarto', 'questionr', 'QuickJSR',\n  'qvcalc', 'R.cache', 'R.oo', 'R.utils', 'r2rtf', 'R6', 'radiant.data',\n  'radiant.model', 'radiant.multivariate', 'ragg', 'rAmCharts', 'randomForest',\n  'randomForestSRC', 'randtoolbox', 'ranger', 'RANN', 'RApiSerialize',\n  'rapportools', 'raster', 'rayimage', 'rayrender', 'rayvertex', 'rbibutils',\n  'Rborist', 'Rcmdr', 'Rcompadre', 'rcompanion', 'Rcpp', 'RcppArmadillo',\n  'RcppDE', 'RcppEigen', 'RcppParallel', 'RcppRoll', 'RcppSimdJson',\n  'RcppThread', 'RcppTOML', 'RcppZiggurat', 'RCurl', 'Rdpack', 'Rdsdp',\n  'reactR', 'readODS', 'readstata13', 'readxl', 'rearrr', 'rebmix', 'recipes',\n  'refund', 'regclass', 'renv', 'report', 'reprex', 'reticulate', 'Rfast',\n  'rfishbase', 'rgbif', 'rgenoud', 'rgl', 'rio', 'riskRegression', 'rjson',\n  'RJSONIO', 'rlang', 'rle', 'RMariaDB', 'rmarkdown', 'rminer', 'rms',\n  'RMySQL', 'rngWELL', 'RNifti', 'RNiftyReg', 'robotstxt', 'robust',\n  'robustbase', 'RODBC', 'rpart', 'rpf', 'rpostgis', 'RPostgres',\n  'RPostgreSQL', 'rrcov', 'rrcovHD', 'rredlist', 'RRPP', 'RSAGA', 'rsample',\n  'rsconnect', 'rscopus', 'Rserve', 'rsm', 'rsparse', 'RSpectra', 'RSQLite',\n  'Rssa', 'rstac', 'rstan', 'rstpm2', 'rstudioapi', 'rsvg', 'rtkore',\n  'rugarch', 'Rvcg', 'rvg', 's2', 'sandwich', 'sass', 'scales', 'SciViews',\n  'SCtools', 'sctransform', 'sda', 'sdm', 'seas', 'seasonal', 'seasonalview',\n  'see', 'seededlda', 'segmented', 'sem', 'semTools', 'sensemakr',\n  'sensitivity', 'seriation', 'servr', 'sessioninfo', 'Seurat', 'SeuratObject',\n  'sf', 'sfnetworks', 'sfsmisc', 'sftime', 'shadowtext', 'shapr', 'shapviz',\n  'shiny', 'shinyAce', 'shinycssloaders', 'shinydashboard', 'shinyWidgets',\n  'SimDesign', 'simmer', 'simpleboot', 'simputation', 'sits', 'sjPlot',\n  'skmeans', 'slam', 'slider', 'SLOPE', 'smaa', 'smacof', 'smooth', 'smplot2',\n  'sna', 'sodium', 'softImpute', 'soilDB', 'sommer', 'sp', 'spacefillr',\n  'spacetime', 'spam', 'sparklyr', 'sparseFLMM', 'SparseM', 'spatialreg',\n  'spatialrisk', 'spatstat', 'spatstat.data', 'spatstat.explore',\n  'spatstat.geom', 'spatstat.linnet', 'spatstat.model', 'spatstat.random',\n  'spatstat.utils', 'spData', 'spdep', 'spmodel', 'spmoran', 'spotifyr',\n  'SSN2', 'stabledist', 'StableEstim', 'stacks', 'StanHeaders', 'stars',\n  'startupmsg', 'statgenGWAS', 'statnet.common', 'stdmod', 'steadyICA',\n  'StepReg', 'strex', 'stringdist', 'stringi', 'stringmagic', 'strucchange',\n  'summarytools', 'SuppDists', 'survival', 'survminer', 'svd', 'svglite',\n  'svgtools', 'sys', 'systemfonts', 'tables', 'tabnet', 'tau', 'taxize',\n  'tcltk2', 'TDboost', 'tergm', 'Ternary', 'terra', 'tesseract', 'testthat',\n  'texreg', 'text', 'textcat', 'textrecipes', 'textshaping', 'textutils',\n  'tfdatasets', 'tgp', 'TH.data', 'thematic', 'themis', 'threejs', 'tibble',\n  'tibbletime', 'tidybayes', 'tidycensus', 'tidycmprsk', 'tidyfst',\n  'tidygeocoder', 'tidymodels', 'tidypredict', 'tidyquant', 'tidytable',\n  'tidytuesdayR', 'tigris', 'timeDate', 'timeplyr', 'timereg', 'timeSeries',\n  'tinytable', 'tinytex', 'tm', 'tmap', 'tmaptools', 'TMB', 'toastui',\n  'topicmodels', 'torch', 'tourr', 'TraMineR', 'tree', 'TruncatedNormal',\n  'tsbox', 'TSclust', 'tsDyn', 'tseries', 'tsibble', 'tsna', 'TSP', 'tswge',\n  'tuber', 'tune', 'tzdb', 'Unicode', 'units', 'univariateML', 'unpivotr',\n  'usethis', 'usmap', 'usmapdata', 'utf8', 'uuid', 'uwot', 'V8', 'vcd',\n  'vegan', 'verification', 'VGAM', 'vioplot', 'vtable', 'waldo', 'warbleR',\n  'WaveletComp', 'wbacon', 'WDI', 'webshot2', 'websocket', 'webutils',\n  'WeightSVM', 'winch', 'withr', 'wk', 'wooldridge', 'workflows',\n  'workflowsets', 'writexl', 'WriteXLS', 'xfun', 'xgboost', 'XML', 'xml2',\n  'xplorerr', 'xts', 'yahoofinancer', 'yaImpute', 'yaml', 'yardstick', 'yfR',\n  'yuima', 'yulab.utils', 'zeallot', 'zip', 'zoo', 'spatial'\n\n","output_type":"stream"}],"execution_count":7},{"cell_type":"code","source":"gr1=qhs[[1]]\ngr2=qhs[[4]]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:18.976695Z","iopub.execute_input":"2025-06-09T11:51:18.978283Z","iopub.status.idle":"2025-06-09T11:51:32.724326Z","shell.execute_reply":"2025-06-09T11:51:32.722414Z"}},"outputs":[{"name":"stderr","text":"downloading 1 resources\n\nretrieving 1 resource\n\nloading from cache\n\nrequire(“rtracklayer”)\n\ndownloading 1 resources\n\nretrieving 1 resource\n\nloading from cache\n\n","output_type":"stream"}],"execution_count":8},{"cell_type":"code","source":"gr2","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:32.728511Z","iopub.execute_input":"2025-06-09T11:51:32.730845Z","iopub.status.idle":"2025-06-09T11:51:32.779452Z","shell.execute_reply":"2025-06-09T11:51:32.777591Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"GRanges object with 74470 ranges and 6 metadata columns:\n          seqnames              ranges strand |        name     score\n             <Rle>           <IRanges>  <Rle> | <character> <numeric>\n      [1]     chr1       713301-713450      * |        <NA>         0\n      [2]     chr1       713501-713650      * |        <NA>         0\n      [3]     chr1       713881-714030      * |        <NA>         0\n      [4]     chr1       714181-714330      * |        <NA>         0\n      [5]     chr1       714481-714630      * |        <NA>         0\n      ...      ...                 ...    ... .         ...       ...\n  [74466]     chrX 154492741-154492890      * |        <NA>         0\n  [74467]     chrX 154493401-154493550      * |        <NA>         0\n  [74468]     chrX 154560441-154560590      * |        <NA>         0\n  [74469]     chrX 154562121-154562270      * |        <NA>         0\n  [74470]     chrX 154842061-154842210      * |        <NA>         0\n          signalValue    pValue    qValue      peak\n            <numeric> <numeric> <numeric> <numeric>\n      [1]          91  112.7680        -1        -1\n      [2]          25   26.7181        -1        -1\n      [3]          81   77.4798        -1        -1\n      [4]          32  106.5650        -1        -1\n      [5]         122  153.8320        -1        -1\n      ...         ...       ...       ...       ...\n  [74466]          43  52.20930        -1        -1\n  [74467]         122 203.16800        -1        -1\n  [74468]           8   4.49236        -1        -1\n  [74469]           8   4.41978        -1        -1\n  [74470]         125 170.20100        -1        -1\n  -------\n  seqinfo: 298 sequences (2 circular) from hg19 genome"},"metadata":{}}],"execution_count":9},{"cell_type":"code","source":"summary(width(gr1))\nsummary(width(gr2))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:32.782419Z","iopub.execute_input":"2025-06-09T11:51:32.783926Z","iopub.status.idle":"2025-06-09T11:51:32.811405Z","shell.execute_reply":"2025-06-09T11:51:32.809684Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n    100     252     996    3222    3110  938261 "},"metadata":{}},{"output_type":"display_data","data":{"text/plain":"   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n  150.0   150.0   150.0   150.3   150.0   410.0 "},"metadata":{}}],"execution_count":10},{"cell_type":"code","source":"table(width(gr2))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:32.814085Z","iopub.execute_input":"2025-06-09T11:51:32.815460Z","iopub.status.idle":"2025-06-09T11:51:32.836476Z","shell.execute_reply":"2025-06-09T11:51:32.834657Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"\n  150   210   230   250   270   290   390   410 \n74313     1    13    24    37    77     4     1 "},"metadata":{}}],"execution_count":11},{"cell_type":"code","source":"qhs= query(ah,\"RefSeq\")\nqhs","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:32.839237Z","iopub.execute_input":"2025-06-09T11:51:32.840746Z","iopub.status.idle":"2025-06-09T11:51:33.557792Z","shell.execute_reply":"2025-06-09T11:51:33.555950Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"AnnotationHub with 8 records\n# snapshotDate(): 2024-04-30\n# $dataprovider: UCSC\n# $species: Homo sapiens\n# $rdataclass: GRanges\n# additional mcols(): taxonomyid, genome, description,\n#   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,\n#   rdatapath, sourceurl, sourcetype \n# retrieve records with, e.g., 'object[[\"AH5040\"]]' \n\n           title       \n  AH5040 | RefSeq Genes\n  AH5041 | Other RefSeq\n  AH5155 | RefSeq Genes\n  AH5156 | Other RefSeq\n  AH5306 | RefSeq Genes\n  AH5307 | Other RefSeq\n  AH5431 | RefSeq Genes\n  AH5432 | Other RefSeq"},"metadata":{}}],"execution_count":12},{"cell_type":"code","source":"genes=qhs[[1]]\ngenes","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:33.560446Z","iopub.execute_input":"2025-06-09T11:51:33.561876Z","iopub.status.idle":"2025-06-09T11:51:35.277122Z","shell.execute_reply":"2025-06-09T11:51:35.274631Z"}},"outputs":[{"name":"stderr","text":"downloading 1 resources\n\nretrieving 1 resource\n\nloading from cache\n\n","output_type":"stream"},{"output_type":"display_data","data":{"text/plain":"UCSC track 'refGene'\nUCSCData object with 50066 ranges and 5 metadata columns:\n                       seqnames            ranges strand |         name\n                          <Rle>         <IRanges>  <Rle> |  <character>\n      [1]                  chr1 66999825-67210768      + |    NM_032291\n      [2]                  chr1   8378145-8404227      + | NM_001080397\n      [3]                  chr1 48998527-50489626      - |    NM_032785\n      [4]                  chr1 16767167-16786584      + | NM_001145277\n      [5]                  chr1 16767167-16786584      + | NM_001145278\n      ...                   ...               ...    ... .          ...\n  [50062] chr19_gl000209_random       57209-68123      + |    NM_002255\n  [50063] chr19_gl000209_random       46646-68123      + | NM_001258383\n  [50064] chr19_gl000209_random      98135-112667      + |    NM_012313\n  [50065] chr19_gl000209_random       70071-84658      + | NM_001083539\n  [50066] chr19_gl000209_random     131433-145745      + |    NM_012312\n              score     itemRgb             thick\n          <numeric> <character>         <IRanges>\n      [1]         0        <NA> 67000042-67208778\n      [2]         0        <NA>   8378169-8404073\n      [3]         0        <NA> 48999845-50489468\n      [4]         0        <NA> 16767257-16785491\n      [5]         0        <NA> 16767257-16785385\n      ...       ...         ...               ...\n  [50062]         0        <NA>       57249-67717\n  [50063]         0        <NA>       57132-67717\n  [50064]         0        <NA>      98146-112480\n  [50065]         0        <NA>       70108-83979\n  [50066]         0        <NA>     131468-145120\n                                     blocks\n                              <IRangesList>\n      [1] 1-227,91706-91769,98929-98953,...\n      [2]     1-102,6222-6642,7214-7306,...\n      [3]    1-1439,2036-2062,6788-6884,...\n      [4]     1-182,2961-3061,7199-7303,...\n      [5]     1-104,2961-3061,7199-7303,...\n      ...                               ...\n  [50062]        1-80,280-315,1182-1466,...\n  [50063]  1-86,10414-10643,10843-10878,...\n  [50064]      1-46,1523-1557,4002-4301,...\n  [50065]      1-71,1071-1106,1851-2135,...\n  [50066]        1-69,862-897,3334-3633,...\n  -------\n  seqinfo: 93 sequences (1 circular) from hg19 genome"},"metadata":{}}],"execution_count":13},{"cell_type":"code","source":"table(table(genes$name))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:35.281208Z","iopub.execute_input":"2025-06-09T11:51:35.282799Z","iopub.status.idle":"2025-06-09T11:51:35.657220Z","shell.execute_reply":"2025-06-09T11:51:35.654808Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"\n    1     2     3     4     5     6     7     8     9    10    12    13    19 \n45178   568   102    27    74    52   171   129    13    19     1     1     5 "},"metadata":{}}],"execution_count":14},{"cell_type":"code","source":"prom= promoters(genes)\ntable(width(prom))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-06-09T11:51:35.661292Z","iopub.execute_input":"2025-06-09T11:51:35.662868Z","iopub.status.idle":"2025-06-09T11:51:35.729031Z","shell.execute_reply":"2025-06-09T11:51:35.727101Z"}},"outputs":[{"output_type":"display_data","data":{"text/plain":"\n 2200 \n50066 "},"metadata":{}}],"execution_count":15},{"cell_type":"code","source":"ov=findOverlaps(prom,peaks)\nlength(unique(queryHits(ov))\nsubsetByOverlaps(peaks,prom,ignore.strand=TRUE) / length(peaks) \nsum(width(reduce(peaks,ignore.strand=TRUE))) / 10^6\nsum(width(reduce(prom,ignore.strand=TRUE))) / 10^6  \nsum(width(intersect(peaks,prom,ignore.strand=TRUE))) / 10^6       \ninOut = matrix(0,ncol=2,nrow=2)\ncolnames(inOut)=c(\"in\",\"out\")\nrownames(inOut)=c(\"in\",\"out\")\ninOut[1,1] = sum(width(intersect(peaks,prom,ignore.strand=TRUE)))\ninOut[1,2] = setdiff(peak,prom,ignore.strand=TRUE)  \ninOut[2,1] = setdiff(prom,peak,ignore.strand=TRUE) \ncolsums(inOut)\ninOut[2,2]= 3*10^9 - sum(inOut)\noddstrain = inOut[1,1] * inOut[2,2] / (inOut[2,1] * inOut[1,2])\nodd strain \n       ","metadata":{"trusted":true},"outputs":[],"execution_count":null}]}